<!DOCTYPE html>
<html>









  <head>
    <meta charset="utf-8">
    <title>Introduction to Genome Assembly</title>
    
    <link rel="stylesheet" href="/training-material/assets/css/slides.css">
    <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css" id="theme">
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

    
    
    
    
    <meta name="description" content="Bioinformatics Introduction" />
    <meta property="og:title" content="VIB Bioinformatics Training: Introduction to Genome Assembly" />
    <meta property="og:description" content="Bioinformatics Introduction" />
    <meta property="og:image" content="/training-material/assets/images/bioinformatics_core_rgb_neg.png" />
    <script type="application/ld+json">
       


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "headline": "Introduction to Genome Assembly",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "provider": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "courseCode": "basic-bioinformatics / general-introduction / slides",
  "learningResourceType": "slides",
  "name": "Slides for 'Introduction to Genome Assembly' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "Introduction to Genome Assembly",
    "description": "Introduction to Genome Assembly",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "bits@vib.be",
      "name": "VIB Bioinformatics Core",
      "url": "https://www.bits.vib.be"
    }
  },
  "url": "https://vibbits.github.io//training-material/topics/basic-bioinformatics/tutorials/general-introduction/slides.html",
  "timeRequired": "PT30M",
  "description": "Introduction to Genome Assembly\\nThe questions this  addresses are:\n - How do we perform a very basic genome assembly from short read data?\n\n\\nThe objectives are:\n - assemble some paired end reads using Velvet\n - examine the output of the assembly.\n\n",
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.582600",
      "name": "Training data for Introduction to Genome Assembly tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Alexander Botzki"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Alexander Botzki"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Basic Bioinformatics Concepts, Databases and Tools",
      "description": "Bioinformatics Introduction",
      "url": "https://vibbits.github.io//training-material/topics/basic-bioinformatics/"
    }
  ]
}
    </script>
  </head>
  <body>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-header"><span>
<a href="/training-material/topics/basic-bioinformatics" title="Return to topic page" ><i class="fa fa-level-up" aria-hidden="true"></i></a>

<a class="nav-link" href="https://github.com/vibbits/training-material/edit/master/topics/basic-bioinformatics/tutorials/general-introduction/slides.html"><i class="fa fa-pencil" aria-hidden="true"></i></a>

</span></div>

<div class="my-footer"><span>

<img src="/training-material/assets/images/bioinformatics_core_rgb_neg.png" alt="Galaxy Training Network" style="height: 40px;"/>

</span></div>

---

# Introduction to Genome Assembly



---





### &lt;i class=&quot;fa fa-question-circle&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;question&lt;/span&gt; Questions


- How do we perform a very basic genome assembly from short read data?


---


### &lt;i class=&quot;fa fa-bullseye&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;objectives&lt;/span&gt; Objectives


- assemble some paired end reads using Velvet

- examine the output of the assembly.


---


.enlarge120[

# ***De novo* Genome Assembly**

]

#### With thanks to T Seemann, D Bulach, I Cooke and Simon Gladman
---
.enlarge120[

# ***De novo* assembly**

]

.pull-left[

**The process of reconstructing the original DNA sequence from the fragment reads alone.**

* Instinctively like a jigsaw puzzle

  * Find reads which &quot;fit together&quot; (overlap)
  * Could be missing pieces (sequencing bias)
  * Some pieces will be dirty (sequencing errors)

]

.pull-right[ ![](../../images/Humpty.jpg) ]

---

# **Another View**

![](../../images/newspaper.png)

---

# **Assembly: An Example**

---

# **A small &quot;genome&quot;**

![](../../images/shakespear1.png)

---

# **Shakespearomics**

![](../../images/shakespear2.png)

---

# **Shakespearomics**

![](../../images/shakespear3.png)

---

# **Shakespearomics**

![](../../images/shakespear4.png)

---

# **So far, so good!**

---

# **The Awful Truth**

![](../../images/notsimply.png)

## &quot;Genome assembly is impossible.&quot; - A/Prof. Mihai Pop

---
.enlarge120[

# **Why is it so hard?**

]

.pull-left[
* Millions of pieces
  * Much, much shorter than the genome
  * Lots of them look similar
* Missing pieces
  * Some parts can't be sequenced easily
* Dirty Pieces
  * Lots of errors in reads
]

.pull-right[ ![](../../images/worlds_hardest.png) ]

---

# **Assembly recipe**

* Find all overlaps between reads
  * Hmm, sounds like a lot of work..
* Build a graph
  * A picture of the read connections
* Simplify the graph
  * Sequencing errors will mess it up a lot
* Traverse the graph
  * Trace a sensible path to produce a consensus

---

![](../../images/olc_pic.png)

---

# **A more realistic graph**

![](../../images/real_graph.png)

---

# .image-15[![](../../images/nofun.png)] **What ruins the graph?**

* Read errors
  * Introduces false edges and nodes

* Non haploid organisms
  * Heterozygosity causes lots of detours

* Repeats
  * If they are longer than the read length
  * Causes nodes to be shared, locality confusion.

---

# **Repeats**

---
.enlarge120[
# **What is a repeat?**
]

.pull-left[

#### ***A segment of DNA which occurs more than once in the genome sequence***

* Very common
  * Transposons (self replicating genes)
  * Satellites (repetitive adjacent patterns)
  * Gene duplications (paralogs)

]

.pull-right[

![](../../images/triplets.png)

]

---

# **Effect on Assembly**

![](../../images/repeat_effect.png)

---
.enlarge120[
# **The law of repeats** .image-15[![](../../images/repeatafterme.png)]
]

## **It is impossible to resolve repeats of length S unless you have reads longer than S**

## **It is impossible to resolve repeats of length S unless you have reads longer than S**

---

# **Scaffolding**

---
.enlarge120[
# **Beyond contigs**
]

.pull-left[

Contig sizes are limited by:

* the length of the repeats in your genome
  * Can't change this


* the length (or &quot;span&quot;) of the reads
  * Use long read technology
  * Use tricks with other technology

]

---
.enlarge120[
# **Types of reads**
]

.pull-left[.enlarge120[**Example fragment**]]


.remark-code[.enlarge120[atcgtatgatcttgagattctctcttcccttatagctgctata]]

.pull-left[.enlarge120[**&quot;Single-end&quot; read**]]


.remark-code[.enlarge120[**atcgtatg**atcttgagattctctcttcccttatagctgctata]]

sequence *one* end of the fragment

.pull-left[.enlarge120[**&quot;Paired-end&quot; read**]]


.remark-code[.enlarge120[**atcgtatg**atcttgagattctctcttcccttatag**ctgctata**]]

sequence both ends of the same fragment

**We can exploit this information!**
---

.enlarge120[# **Scaffolding**]

* **Paired end reads**
  * Known sequences at each end of fragment
  * Roughly known fragment length

* **Most ends will occur in same contig**

* **Some will occur in different contigs**
  * ***evidence that these contigs are linked***
---

.enlarge120[# **Contigs to Scaffolds**]

![](../../images/scaffolding.png)

---

.enlarge120[# **Assessing assemblies**]

* We desire
  * Total length similar to genome size
  * Fewer, larger contigs
  * Correct contigs

* Metrics
  * No generally useful measure. (No real prior information)
  * Longest contigs, total base pairs in contigs, **N50**, ...

---

.enlarge120[# **The &quot;N50&quot;**]

.enlarge120[***The length of that contig from which 50% of the bases are in it and shorter contigs***]

* Imagine we have 7 contigs with lengths:
  * 1, 1, 3, 5, 8, 12, 20

* Total
  * 1+1+3+5+8+12+20 = 50

* N50 is the &quot;halfway sum&quot; = 25
  * 1+1+3+5+8+**12** = 30 (&gt;25) so **N50 is 12**

---

.enlarge120[# **2 levels of assembly**]

* Draft assembly
  * Will contain a number of non-linked scaffolds with gaps of unknown sequence
  * Fairly easy to get to

* Closed (finished) assembly
  * One sequence for each chromosome
  * Takes a **lot** more work
  * Small genomes are becoming easier with long read tech
  * Large genomes are the province of big consortia (e.g. Human Genome Consortium)

---
.enlarge120[# **How do I do it?**]
---
.enlarge120[
# **Example**

* Culture your bacterium


* Extract your genomic DNA


* Send it to your sequencing centre for Illumina sequencing
  * 250bp paired end


* Get back 2 files
  * .remark-code[MRSA_R1.fastq.gz]
  * .remark-code[MRSA_R2.fastq.gz]


* ***Now what?***
]

---
.enlarge120[# **Assembly tools**

* **Genome**
  * **Velvet, Velvet Optimizer, Spades,** Abyss, MIRA, Newbler, SGA, AllPaths, Ray, SOAPdenovo, ...


* Meta-genome
  * Meta Velvet, SGA, custom scripts + above


* Transcriptome
  * Trinity, Oases, Trans-abyss

***And many, many others...***

]

---
.enlarge120[
# **Assembly Exercise #1**

* We will do a simple assembly using **Velvet** in **Galaxy**
* We can do a number of different assemblies and compare some assembly metrics.

]




---
### &lt;i class=&quot;fa fa-key&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;keypoints&lt;/span&gt; Key points


- We assembled some Illumina fastq reads into contigs using a short read assembler called Velvet

- We showed what effect one of the key assembly parameters, the k-mer size, has on the assembly

- It looks as though there are some exploitable patterns in the metric data vs the k-mer size.







---

## Thank you!

This material is the result of a collaborative work. Thanks the colleagues of the VIB Bioinformatics Core and all the contributors!


<div class="contributors-line">


<a href="/training-material/hall-of-fame#abotzki" class="contributor-badge"><img src="https://avatars.githubusercontent.com/abotzki" alt="Alexander Botzki">Alexander Botzki</a>


</div>



<img src="/training-material/assets/images/bioinformatics_core_rgb_neg.png" alt="Galaxy Training Network" style="height: 100px;"/>



    </textarea>
    <script src="/training-material/assets/js/remark-latest.min.js" type="text/javascript">
    </script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false,}});
      var hljs = remark.highlighter.engine;
    </script>
  </body>
</html>
