<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <meta name="google-site-verification" content="dw_1zBYRNxF7-vYNDXh3Ght_JuqgJt3BD1N_Pynqmv0" />
        <title>05 Selecting reference genes: exercises</title>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/VIB.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="QPCR analysis using qbase+" />
        <meta property="og:title" content="VIB Bioinformatics Training: 05 Selecting reference genes: exercises" />
        <meta property="og:description" content="QPCR analysis using qbase+" />
        <meta property="og:image" content="/training-material/assets/images/bioinformatics_core_rgb_neg.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <div class="container main-content">













<header class="well row">
           <div id="logo" class="col-md-4 col-lg-2 text-center">
              <a href="/training-material/">
                <img src="/training-material/assets/images/vib_notag_neg_rgb.jpg" height="50" alt="VIB logo" class="margin20">
              </a>
           </div> 
           <div class="col-md-4 col-lg-8" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/qbase-plus" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> QPCR analysis using qbase+
                        </a>
                    </li>
            </div>
            <div class="col-md-4 col-lg-2 text-right form-group label-floating" id="help">
                       
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="VIB Bioinformatics Core" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="https://www.bits.vib.be" title="Contact us">
            Contact us 
        </a>
    </div>


            </div>
</header>

    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "headline": "05 Selecting reference genes: exercises",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "provider": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "courseCode": "qbase-plus / reference-genes / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for '05 Selecting reference genes: exercises' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "05 Selecting reference genes: exercises",
    "description": "05 Selecting reference genes: exercises",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "bits@vib.be",
      "name": "VIB Bioinformatics Core",
      "url": "https://www.bits.vib.be"
    }
  },
  "url": "https://vibbits.github.io//training-material/topics/qbase-plus/tutorials/reference-genes/tutorial.html",
  "timeRequired": "PT5M",
  "description": "05 Selecting reference genes: exercises\\nThe questions this  addresses are:\n - How do you perform a qPCR analysis using the qbase+ software?\n\n\\nThe objectives are:\n - Use the qbase+ software to analyze the qPCR experiment results.\n\n",
  "hasPart": [

  ],
  "mentions": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Christof De Bo"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Christof De Bo"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "QPCR analysis using qbase+",
      "description": "QPCR analysis using qbase+",
      "url": "https://vibbits.github.io//training-material/topics/qbase-plus/"
    }
  ]
}
    </script>

    <div class="row tutorial">
      <div class="col-md-12" role="main">
        <h1 data-toc-skip>05 Selecting reference genes: exercises</h1>
        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#chdeb" class="contributor-badge"><img src="https://avatars.githubusercontent.com/chdeb" alt="Christof De Bo">Christof De Bo</a>

</div>
        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>How do you perform a qPCR analysis using the qbase+ software?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>Use the qbase+ software to analyze the qPCR experiment results.</li>
            
            </ul>

            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 5 minutes</p>
            

            



            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul>
                        <li>
                            <div>
                    

                    

                    
                    


                    
                            </div>
                        </li>
                    </ul>
                </div>

            </div>

        </blockquote>

        <div class="container">
            <div class="row">
                <div class="col-sm-10" role="main">
                    <p>Since normalization of qPCR data is based on the assumption that the reference targets have the same expression level in all samples it is crucial that the expression of the chosen reference genes is stable.
However, none of the so-called <strong>housekeeping</strong> genes is universally stably expressed.</p>

<p><a href="https://www.genevestigator.com/gv/">Genevestigator</a>, both the commercial and the free version, contains a tool, called RefGenes, that allows to identify candidate reference genes that display very stable expression in the context that you are working in, typically a certain tissue of a certain organism.</p>

<p>Genevestigator is a platform that contains curated public microarray data from thousands of experiments/conditions.</p>

<p>RefGenes allows you to select the conditions that are relevant for you, e.g. mouse liver, human fibroblasts, or Arabidopsis thaliana leaves. In a next step, RefGenes identifies the genes with the most stable expression in the selected conditions.</p>

<h2 id="starting-the-refgenes-tool">Starting the RefGenes tool</h2>

<p>| How to start the RefGenes tool ? |
| :——————————– |
| - Open the <a href="http://www.refgenes.org/rg/">RefGenes page</a>.</p>
<ul>
  <li>Click <strong>start GENEVESTIGATOR</strong></li>
  <li>Click the <strong>Install/Start</strong> button</li>
  <li>This will automatically open a Genevestigator startup page. Keep this page open during the analysis. Closing this page will close Genevestigator.</li>
  <li>Login. Also for the free version you need to create an account (use your academic email for this since you will need your vib email to get access to the commercial version).</li>
  <li>Genevestigator is opened automatically</li>
</ul>

<h2 id="the-genevestigator-user-interface">The Genevestigator user interface</h2>

<p>The Genevestigator consists of the following components:</p>
<ul>
  <li><strong>Sample Selection</strong> panel: to choose the experimental conditions you’re interested in (green)</li>
  <li><strong>Gene Selection</strong> panel: to choose the genes you’re interested in (blue)</li>
  <li>Center panel shows an overview of all available tools (purple). Once you have selected a tool, the panel will show the results of the analysis that is done by the tool.</li>
  <li><strong>Home</strong> button (red) allows to return to the overview of the tools at any time. The text next to the home button indicates the toolset that you have selected.</li>
</ul>

<p>Click the <strong>RefGenes</strong> tool at the bottom.</p>

<h2 id="using-the-refgenes-tool-to-find-reference-genes">Using the RefGenes tool to find reference genes</h2>

<h3 id="step-1-choose-samples-from-a-biological-context-similar-to-those-in-your-qpcr-expriment">STEP 1: Choose samples from a biological context similar to those in your qPCR expriment</h3>

<p>| How to choose the samples you want to analyze ? |
| :——————————– |
|</p>
<ul>
  <li>Click the <strong>New</strong> button in the <strong>Sample Selection</strong> panel. The selection of samples defines which data are used for the analysis.</li>
  <li>Select the organism you’re interested in (in this example: human)</li>
  <li>Select the array type you want to analyze (in this example: human 133_2).
For most organisms Genevestigator contains expression data from multiple types of microarrays, e.g. different generations of Affymetrix GeneChips®. On these arrays, genes are sometimes represented by different sets of probes. To keep the analysis results easily interpretable, data from different array types are not mixed.</li>
  <li>Click the <strong>Select particular conditions</strong> button to select all samples with a certain annotation, e.g. all data from a certain tissue type.</li>
  <li>Select the type of conditions (red) you want to base your selection on (in this example: Anatomy). For each type (anatomy, neoplasms, perturbations, development…) you can browse the corresponding ontologies and select the desired condition(s) (green) (in this example: cardiac muscle).</li>
  <li>Click <strong>OK</strong></li>
</ul>

<p>Note that you can select multiple tissues.
When you select samples for use in the RefGenes tool, you have to focus on microarrays from samples that were collected in conditions similar to those in your qPCR experiment. Don’t make a too general selection, e.g. all human samples: you might end up with genes that are stable in most conditions but not in yours. Don’t make a very specific selection either, e.g. human heart samples from patients taking the same medication as yours. If you want to broaden your study later on with samples from other patients, your reference genes might not be valid anymore. It is recommended to select reference genes in the same organism and the same / a similar tissue type as the one that you used in your experiments.</p>

<h3 id="step-2-select-the-genes-you-want-to-measure-in-your-qpcr-experiment">STEP 2: Select the gene(s) you want to measure in your qPCR experiment</h3>

<p>This step is not essential, but it helps you to see whether your target gene(s) is (are) strongly or weakly expressed in the conditions of interest selected in STEP1. This allows you to search for candidate reference genes in a similar range of expression.</p>

<p>| How to choose the genes you want to analyze ? |
| :——————————– |
|</p>
<ul>
  <li>Click the <strong>New</strong> button in the <strong>Gene Selection</strong> panel.</li>
  <li>Enter the name of your target gene in the text area (in this example: GOT1) and click <strong>OK</strong></li>
  <li>Open the RefGenes tool (if you haven’t done that already). A red box plot representing the distribution of the expression levels of GOT1 in the 68 selected human heart samples appears in the center panel. As you can see, this gene is highly expressed in heart.</li>
</ul>

<h3 id="step-3-find-candidate-reference-genes">STEP 3: Find candidate reference genes</h3>

<p>The reference genes that are suggested by GeneVestigator have the
following characteristics:</p>

<ul>
  <li>They have the most stable expression levels across all selected samples (a small boxplot)</li>
  <li>Their overall expression level is similar to that of the target gene(s) of your qPCR experiment
| How to find the candidate reference genes ? |
| :——————————– |
|Click the <strong>Run</strong> button in the RefGenes tool. RefGenes will show the top 20 most stable genes with similar expression levels:</li>
</ul>

<h2 id="exercises">Exercises</h2>

<h3 id="finding-candidate-reference-genes-in-the-free-version-of-genevestigator">Finding candidate reference genes in the free version of Genevestigator</h3>

<p>Now we will make a more elaborate exercise on finding candidate reference genes. We will do the analysis in the free version of RefGenes but the analysis in the commercial version is very similar.
Suppose we want to compare the expression stability of the 4 commonly used reference genes for qPCR on mouse liver samples (ACTB, GAPDH, HPRT and TUBB4B) to that of 4 reference genes that are suggested by Genevestigator.
To this end we open the RefGenes tool and select the liver samples of the mouse 430_2 arrays.</p>

<p>| Check the expression stability of the 4 commonly used reference genes ? |
| :——————————– |
|</p>
<ul>
  <li>Click the <strong>New</strong> button in the <strong>Gene Selection</strong> panel to create a new selection. The selection of samples defines which data are used for the analysis.</li>
  <li>Enter the name of your target gene in the text area (for example: ACTB) and click <strong>OK</strong></li>
</ul>

<p>When you are using the commercial version, you may enter multiple genes at the same time, in the free version you have to enter them one by one. This means that you have to add the first gene as described above and then add the next gene by clicking the <strong>Add</strong> button and so on…</p>

<p>Finally you end up with an expandable list of the genes you asked for and you can tick or untick them to control the display of their expression data in the main window. When you tick the 4 commonly used reference genes you can see how stable they are expressed in the 651 mouse liver samples that are stored in Genevestigator:</p>

<p>As you can see, the expression levels of the commonly used reference genes in the selected mouse liver samples is pretty variable which is also confirmed by their relatively high SD values.
Often there are multiple probe sets for the same gene. When you use the free version you may only choose one probe set per gene so you have to make a choice. How to make that choice ?
Affymetrix probe set IDs have a certain meaning: what comes after the underscore tells you something about the quality of the probes:</p>

<ul>
  <li><strong>_at</strong> means that all the probes of the probe set hit one known transcript. This is what you want: probes specifically targeting one transcript of one gene</li>
  <li><strong>_a_at</strong> means that all the probes in the probe set hit alternate transcripts from the same gene. This is still ok the probes bind to multiple transcripts but at least the transcripts come from the same gene (splice variants)</li>
  <li><strong>_x_at</strong> means that some of the probes hit transcripts from different genes. This is still not what you want: the expression level is based on a combination of signals of all the probes in a probe set so also probes that cross-hybridize</li>
  <li><strong>_s_at</strong> means that all the probes in the probe set hit transcripts from different genes. This is definitely not what you want: if the probes bind to multiple genes you have no idea whose expression you have measured on the array</li>
</ul>

<p>So I always ignore probe sets with s or x. If you have two specific probe sets for a gene, they should more or less give similar signals. If this is not the case, I base my choice upon the expression level that I expect for that gene based on previous qPCR results.</p>

<p>As you can see, each of these 4 commonly used reference genes has a high expression level. Most genes do not have such high expression levels. In most qPCR experiments your genes of interest will have low or medium expression levels, so these reference genes will not be representative for the genes of interest.</p>

<p>Reference genes should ideally have similar expression levels as the genes of interest. Therefore, we will select the four most stably expressed genes with a medium expression level (between 8 and 12) according to the RefGenes tool.</p>

<p>| Select the 4 most stably expressed candidate reference gene with medium expression levels. |
| :——————————– |
|</p>
<ul>
  <li>Untick all target genes.</li>
  <li>Click the <strong>Run</strong> button at the top of the main window and check if the range is set correctly</li>
</ul>

<p>Select the 4 candidates with the lowest SD: Then, we performed qPCR on a representative set of 16 of our liver samples to measure the expression of these 8 candidate reference genes and analyzed the data (<a href="http://wiki.bits.vib.be/index.php/Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase%2B">See how to select the best reference genes using geNorm in qbase+</a>).</p>

<h3 id="finding-candidate-reference-genes-in-the-commercial-version-of-genevestigator">Finding candidate reference genes in the commercial version of Genevestigator</h3>

<p>We will do the same exercise as above in the commercial version of Genevestigator. The difference between the free and commercial version of RefGenes is the number of target genes you can select. In the free version you have to select one gene and then gradually add all other genes one at a time. The commercial version allows you to load as many target genes as you want simultaneously. As a consequence, you can select multiple probe sets for the same gene.
All VIB scientists have free access to the commercial version of Genevestigator via their VIB email address. If you don’t know your VIB email address, check <a href="http://www.vib.be/en/whoiswho/Pages/default.aspx">the Who’s Who of VIB</a>.</p>

<ul>
  <li>Open a browser and go to the <a href="https://www.genevestigator.com/">Genevestigator website</a></li>
  <li>If it’s your <strong>first time to access Genevestigator</strong>, create an account by clicking <strong>join now</strong> button. You will be redirected to a new window in which you will give some personal information including a valid VIB email address. Click <strong>Register</strong> and check your email to activate your new account. Go back to the <a href="https://www.genevestigator.com/">GeneVestigator website</a></li>
  <li>Choose the research field you want to investigate: <strong>pharma/biomediacal</strong> or <strong>plant biology</strong> by clicking the corresponding button</li>
  <li>Click <strong>Start</strong></li>
  <li>Use your VIB email address and password to login to Genevestigator.</li>
  <li>This will automatically open a Genevestigator startup page in your browser. Keep this page open during the analysis. Closing this page will close Genevestigator.</li>
  <li>Genevestigator is opened automatically</li>
</ul>

<p>Open the RefGenes tool by clicking its icon in the <strong>Further tools</strong> secion and select the liver samples of the mouse 430_2 arrays <a href="http://wiki.bits.vib.be/index.php/Using_GeneVestigator_to_select_candidate_reference_genes#STEP_1:_Choose_samples_from_a_biological_context_similar_to_those_in_your_qPCR_expriment">as explained in the previous exercise</a>.
| Check the expression stability of the 4 commonly used reference genes ? |
| :——————————– |
| - Click the <strong>New</strong> button in the <strong>Gene Selection</strong> panel to create a new selection. The selection of samples defines which data are used for the analysis.</p>
<ul>
  <li>Enter the names of the 4 commercial reference genes in the text area and click <strong>OK</strong></li>
</ul>

<p>I still remove probe sets with an _s or _x since they do not specifically bind to one single gene:
Finally you end up with an expandable list of the genes you asked for and you can tick or untick them to control the display of their expression data in the main window. By default all probe sets are ticked so you can see how stable the commonly used reference genes are expressed in the 651 mouse liver samples that are stored in Genevestigator:
As you can see, the expression levels of the commonly used reference genes in the selected mouse liver samples is pretty variable which is also confirmed by their relatively high SD values.</p>

<p>The next step of selecting the 4 most stable candidate reference genes with medium expression levels is exactly the same as described above for the free version of RefGenes.</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Create a new gene selection with 20 found candidate reference genes and call it mouse_references.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">Click the <strong>New</strong> button at the top of the main window to create a new selection.</td>
    </tr>
  </tbody>
</table>

<p>To change the name of the selection right click the name in the <strong>Gene selection</strong> panel and select <strong>Rename</strong></p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Identify perturbations where the mouse_references genes show more than 1,5 fold differential expression using the Condition perturbations tool.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">Click the <strong>Home</strong> button at the top to go back to the tools overview page.</td>
    </tr>
  </tbody>
</table>

<p>Click the <strong>Perturbations</strong> tool in the <strong>Condition Search tools</strong> section</p>

<p>Make a <strong>New Sample selection</strong> including all mouse 430_2 arrays.
Untick all genes except for the first one and filter the long heatmap for at least 1.5 fold change differential expression:</p>

<p>You now get a list of mouse samples in which the gene is not stably expressed so you can check if any of these samples is related to the samples in your study. Hover your mouse over the name of a sample to see more details about the sample.
You can do this for each of the candidate reference genes and select the ones that best fit your needs</p>

<p><a href="http://wiki.bits.vib.be/index.php/GV_Exercise.1">Exercise on selecting reference genes for metacaspases in Arabidopsis thaliana</a>.</p>

<p>In a geNorm pilot experiment you analyze a set of candidate reference genes in a representative set of samples that you want to test in your final experiment. Based on the M-values and CVs that are calculated by qbase+, you can choose the genes that most satisfy the criteria for a good reference gene.</p>

<h3 id="exercise-1-reference-genes-for-mouse-liver">Exercise 1: reference genes for mouse liver</h3>

<p>We come back on the 8 candidate reference genes that we selected for mouse liver:</p>

<ul>
  <li>4 commonly used reference genes: ACTB, TUBB4B, GAPDH and HPRT</li>
  <li>4 candidate reference genes with very stable medium expression levels selected based on expression data coming from more than 600 microarrays of mouse liver samples using Genevestigator: Gm16845, MUSK, OTOP3, EDN3</li>
</ul>

<p>We have measured their expression in a represetative set of 16 of our mouse liver samples, each in triplicate. We will now analyze the stability of these candidate reference genes in our samples.</p>

<h4 id="creating-a-new-experiment">Creating a new Experiment</h4>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Create a new Experiment called GeNormMouse in Project1</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">Open qbase+ or, if the software is already open, click the Launch Wizard button.</td>
    </tr>
  </tbody>
</table>

<p>You can find the details on how to create a new experiment in Creating a project and an experiment</p>

<h4 id="loading-the-data-into-qbase">Loading the data into qbase+</h4>

<p>The data is stored in <a href="http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/RefGenes.zip">the RefGenes folder</a>. It consists of 8 Excel files, one file for each candidate reference gene. If you are not working on a BITS laptop, download and unzip the folder.</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Import the data. This files are in qBase format.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to start the data import in Loading data into qbase+</td>
    </tr>
  </tbody>
</table>

<p>Unlike the previous exercise, qbase+ does not allow you to do a quick import this time. In the Import Run window Manual import is selected:
Make sure that Upload file to Biogazelle support for further analysis is NOT selected and click Next
Make sure the correct File type is selected (qBase) and click Finish.
This file contains the data of the geNorm pilot experiment. In the pilot experiment, 8 candidate reference genes were measured in 16 representative mouse liver samples.</p>
<h4 id="analyzing-the-genorm-pilot-data">Analyzing the geNorm pilot data</h4>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Specify the aim of the experiment.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">In this experiment we want to select the ideal reference genes for our next experiments so we choose selection of reference genes (geNorm)</td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Check the quality of the replicates (use default parameter settings).</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to check the quality of the replicates in the Checking the quality of technical replicates and controls section of Analyzing gene expression data in qbase+</td>
    </tr>
  </tbody>
</table>

<p>We haven’t included any positive or negative controls so you don’t need to show their details.</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Select the Amplification efficiencies strategy you want to use.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to select the Amplification effciencies strategy in the Taking into account amplification efficiencies section of Analyzing gene expression data in qbase+</td>
    </tr>
  </tbody>
</table>

<p>We haven’t included dilution series nor do we have data from previous qPCR experiments regarding the amplification efficiencies so we choose to use the same efficiency for all genes.
It is of course better to include a dilution series for each gene to have an idea of the amplification efficiencies of each primer pair.</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Convert all genes to Reference genes.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can convert all the genes simultaneously by selecting Use all targets as candidate reference genes</td>
    </tr>
  </tbody>
</table>

<p>Click Finish.</p>

<p>| Which genes are you going to use as reference targets in further experiments ? |
| :——————————————- |
| Upon clicking Finish, the geNorm window containing the analysis results is automatically opened. The geNorm window consists of three tabs. The tabs are located at the bottom of the window: geNorm M, geNorm V and Interpretation.
The first tab, geNorm M, shows a ranking of candidate genes according to their stability, expressed in M values, from the most unstable genes at the left (highest M value) to the best reference genes at the right (lowest M value):
The second tab, geNorm V, shows a bar chart that helps determining the optimal number of reference genes to be used in subsequent analyses:</p>

<p>The number of reference genes is a trade-off between practical considerations and accuracy. It is a waste of resources to quantify more genes than necessary if all candidate reference genes are relatively stably expressed and if normalization factors do not significantly change when more genes are included. However, Biogazelle recommends the minimal use of 3 reference genes and stepwise inclusion of more reference genes until the next gene has no significant contribution to the normalization factors.
To determine the need of including more than 3 genes for normalization, pairwise variations Vn/n+1 are calculated between two sequential normalization factors. Simply stated: V is measure of the added value of adding a next reference gene to the analysis. A large variation means that the added gene has a significant effect and should be included.
In normal experiments like the Gene expression experiment (see Analyzing gene expression data in qbase+), we only have 3 reference genes so we will see only 1 bar here. But in this geNorm pilot experiment, we analyzed 8 candidate reference genes, so we see 6 bars.
All pairwise variations are very low, so even the inclusion of a third gene has no significant effect. Based on a preliminary experiment that was done by Biogazelle, 0.15 is taken as a cut-off value for V, below which the inclusion of an additional reference gene is not required. Normally this threshold is indicated by a green line on the geNorm V bar chart. However since all V-values fall below the threshold in this geNorm pilot experiment, you don’t see this line on the bar chart.
So, these results mean that for all subsequent experiments on these samples, two reference genes, EDN3 and MUSK, would be sufficient. However, as stated before, Biogazelle recommends to always include at least three reference genes in case something goes wrong with one of the reference genes (so also include Gm16845). |
These are artificial data. But when you read <a href="http://www.biomedcentral.com/1471-2164/12/156/abstract">the paper by Hruz et al., 2011</a> you see that the genes that are selected by Genevestigator are often outperforming the commonly used reference genes.</p>

<h3 id="exercise-2-reference-genes-for-human-heart">Exercise 2: reference genes for human heart</h3>

<h4 id="creating-a-new-experiment-1">Creating a new Experiment</h4>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Create a new Experiment called GeNormHuman in Project1</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to create a new experiment in Creating a project and an experiment</td>
    </tr>
  </tbody>
</table>

<h4 id="loading-the-data-into-qbase-1">Loading the data into qbase+</h4>
<p>| Import <a href="http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run6.xls">Run6</a> . This file is in qBase format. |
| :——————————————- |
| You can find the details on how to start the data import in Loading data into qbase+. Unlike the previous exercise, qbase+ does not allow you to do a quick import this time. In the Import Run window Manual import is selected:</p>

<p>Make sure that Upload file to Biogazelle support for further analysis is NOT selected and click Next. Select the correct File type (qBase) and click Finish. This file contains the data of the geNorm pilot experiment. In the pilot experiment, 10 candidate reference genes were measured in 20 representative samples.</p>

<h4 id="analyzing-the-genorm-pilot-data-1">Analyzing the geNorm pilot data</h4>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Specify the aim of the experiment.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">In this experiment we want to select the ideal reference genes for our next experiments so we choose selection of reference genes (geNorm)</td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Check the quality of the replicates and the controls (use default parameter settings).</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to check the quality of the replicates in the Checking the quality of technical replicates and controls section of Analyzing gene expression data in qbase+</td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th style="text-align: left">All replicates and controls have met the quality criteria so there’s no need to inspect them further.</th>
      <th> </th>
    </tr>
    <tr>
      <th style="text-align: left"> </th>
      <th>Select the Amplification efficiencies strategy you want to use.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left"> </td>
      <td>You can find the details on how to select the Amplification effciencies strategy in the Taking into account amplification efficiencies section of Analyzing gene expression data in qbase+. We haven’t included dilution series nor do we have data from previous qPCR experiments regarding the amplification efficiencies so we choose to use the same efficiency (E=2) for all genes.</td>
    </tr>
  </tbody>
</table>

<p>It is of course better to include a dilution series for each gene to have an idea of the amplification efficiencies of each primer pair.
| Convert all genes to Reference genes.                                                                         |
| :———————————————————————————————————— |
| You can convert all the genes simultaneously by selecting Use all targets as candidate reference genes |</p>

<p>Click Finish.</p>

<p>| Which genes are you going to use as reference targets in further experiments ? |
| :——————————————- |
| Upon clicking Finish, the geNorm window containing the analysis results is automatically opened. The geNorm window consists of three tabs. The tabs are located at the bottom of the window: geNorm M, geNorm V and Interpretation.
The first tab, geNorm M, shows a ranking of candidate genes according to their stability, expressed in M values, from the most unstable genes at the left (highest M value) to the best reference genes at the right (lowest M value):
The second tab, geNorm V, shows a bar chart that helps determining the optimal number of reference genes to be used in subsequent analyses:</p>

<p>The number of reference genes is a trade-off between practical considerations and accuracy. It is a waste of resources to quantify more genes than necessary if all candidate reference genes are relatively stably expressed and if normalization factors do not significantly change when more genes are included. However, Biogazelle recommends the minimal use of the 3 most stable candidate reference genes and stepwise inclusion of more reference genes until the next gene has no significant contribution to the normalization factors.
To determine the need of including more than 3 genes for normalization, pairwise variations Vn/n+1 are calculated between two sequential normalization factors. Simply stated: V is measure of the added value of adding a next reference gene to the analysis. A large variation means that the added gene has a significant effect and should be included.
In normal experiments like the Gene expression experiment, see Analyzing_gene_expression_data_in_qbase+, we only have 3 reference genes so we will see only 1 bar here. But in this geNorm pilot experiment, we analyzed 10 candidate reference genes, so we see 8 bars.
All pairwise variations are very low, so even the inclusion of a third gene has no significant effect. Based on a preliminary experiment that was done by Biogazelle, 0.15 is taken as a cut-off value for V, below which the inclusion of an additional reference gene is not required. Normally this threshold is indicated by a green line on the geNorm V bar chart. However since all V-values fall below the threshold in this geNorm pilot experiment, you don’t see this line on the bar chart.
So, these results mean that for all subsequent experiments on these samples, two reference genes, HPRT1 and GADP, would be sufficient. However, as stated before, Biogazelle recommends to always include at least three reference genes in case something goes wrong with one of the reference genes (so also include YHWAZ).</p>

<p>In this example we will analyze data from an artificial expression study containing the following samples:</p>
<ul>
  <li>6 treated samples: treated1, treated2, … treated6</li>
  <li>6 control samples: control1, control2, … control6</li>
</ul>

<p>In this study, the expression of the following genes was measured:</p>
<ul>
  <li>4 commonly used reference genes: ACTB, HPRT, GAPDH, and TUBB4. We have seen in <a href="http://wiki.bits.vib.be/index.php/Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase%2B#Exercise_1:_reference_genes_for_mouse_liver">the previous exercise</a> that the expression of these reference genes in mouse liver samples is not as stable as generally thought.</li>
  <li>3 genes of interest:
    <ul>
      <li>Low: a gene with low expression levels</li>
      <li>Medium: a gene with moderate expression levels</li>
      <li>HighVar: a gene with low and very noisy expression</li>
    </ul>
  </li>
</ul>

<p>In general, the lower the expression level, the more noisy the qPCR results will become. For each of the genes of interest we have included a run in which a 2-fold difference in expression between control and treated samples was created (Low1, Medium1 and HighVar1) and a run with a 4-fold difference in expression (Low2, Medium2 and HighVar2).
There are three technical replicates per reaction. In a second experiment we used <a href="http://wiki.bits.vib.be/index.php/Using_GeneVestigator_to_select_candidate_reference_genes#Finding_candidate_reference_genes_in_the_free_version_of_Genevestigator">the reference genes that were obtained via Genevestigator</a> and that proved to be <a href="http://wiki.bits.vib.be/index.php/Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase%2B#Exercise_1:_reference_genes_for_mouse_liver">more stably expressed in mouse liver samples than the commonly used references</a>.
The data can be found in the NormGenes folder on the BITS laptops or can be downloaded: <a href="http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/NormGenes.zip">from our website</a>.</p>

<h4 id="creating-a-new-experiment-2">Creating a new experiment</h4>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Create a new Experiment called NormGenes1 in Project1</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to create a new experiment in Creating a project and an experiment</td>
    </tr>
  </tbody>
</table>

<h4 id="loading-the-data">Loading the data</h4>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Import Run1 to Run5. These files are in qBase format.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to import the data file in the Loading the data into qbase+ section of Analyzing data from a geNorm pilot experiment in qbase+</td>
    </tr>
  </tbody>
</table>

<p>We are going to compare expression in treated versus untreated samples so we need to tell qbase+ which samples are treated and which not. To this end, we have constructed <a href="http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Sample_Properties_Norm.xlsx">a sample properties file</a> in Excel containing the grouping annotation as a custom property called Treatment.</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Import the Sample Properties file.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to import the data file in the Adding annotation to the data section of Loading data into qbase+.</td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Select to import the custom property.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">So as you can see we have 6 treated and 6 untreated samples and we have measured the expression of the 4 commonly used reference genes and 6 genes of interest:</td>
    </tr>
  </tbody>
</table>

<h4 id="analyzing-the-data">Analyzing the data</h4>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Which amplification efficiencies strategy are you going to use ?</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You don’t have data of serial dilutions of representative template to build standard curves so the only choice you have is to use the default amplification efficiency (E = 2) for all the genes.</td>
    </tr>
  </tbody>
</table>

<table>
  <tbody>
    <tr>
      <td>Appoint the reference genes.</td>
      <td>ACTB, GAPDH, HPRT and TUBB4B are the reference genes:</td>
    </tr>
    <tr>
      <td>You can find the details on how to appoint reference targets in the Normalization section of Analyzing gene expression data in qbase+</td>
      <td> </td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Is the stability of the reference genes ok ?</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">The M and CV values of the reference genes are shown in green so the stability of the reference genes is ok.</td>
    </tr>
  </tbody>
</table>

<table>
  <tbody>
    <tr>
      <td>Which scaling strategy are you going to use ?</td>
      <td>Since you have a treated and a control group, it seems logical to use the average of the control group for scaling.</td>
    </tr>
  </tbody>
</table>

<p>You can find the details on how to specify the scaling strategy in the Scaling section of Analyzing gene expression data in qbase+
Look at the target bar charts.</p>

<table>
  <tbody>
    <tr>
      <td>In the target bar charts plot the average expression level of each group.</td>
      <td>In the Grouping section at the bottom of the chart you can select Plot group average: Now do exactly the same for the second experiment with the same genes of interest but with other reference genes. This means that you have to return to the Analysis wizard. To this end, click the Launch wizard button a the top of the page:</td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Create a new Experiment called NormGenes2 in Project1</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to create a new experiment in Creating a project and an experiment</td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Import Run5 to Run9. These files are in qBase format.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to import the data file in the Loading the data into qbase+ section of Analyzing data from a geNorm pilot experiment in qbase+</td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Import the Sample Properties file.</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find the details on how to import the data file in the Adding annotation to the data section of Loading data into qbase+. Select to import the custom property.</td>
    </tr>
  </tbody>
</table>

<p>So as you can see we have 6 treated and 6 untreated samples and we have measured the expression of the 4 new reference genes and 6 genes of interest:
| Appoint the reference genes. |EDN3, Gm16835, MUSK and OTOP3 are the reference genes:
| :—————————————————- |
| You can find the details on how to appoint reference targets in the Normalization section of Analyzing gene expression data in qbase+ |</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Is the stability of the reference genes ok ?</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">The M and CV values of the reference genes are shown in green so the stability of the reference genes is ok.</td>
    </tr>
  </tbody>
</table>

<p>As you can see the M and CV values of these reference genes is much lower than these of the 4 commonly used reference genes pointing to the fact that genes are more stably expressed. It’s not that the commonly used reference genes are bad references. Then qbase+ would not display them in green. It’s just that the other reference genes are more stable. But this can have a big impact on the results of your analysis…</p>

<table>
  <tbody>
    <tr>
      <td>Use the average of the control group for scaling</td>
      <td>You can find the details on how to specify the scaling strategy in the Scaling section of Analyzing gene expression data in qbase+</td>
    </tr>
  </tbody>
</table>

<p>Plot the average expression level of each group. Now we will compare the target bar charts of the second and the first experiment to assess the influence of the stability of the reference targets on the analysis results.</p>

<table>
  <tbody>
    <tr>
      <td>How to display the target bar charts of the second and the first experiment next to each other ?</td>
      <td>You can display the bar charts next to each other by clicking the tab of the bar chart of the second experiment. Drag the tab to the right while you hold down the mouse button until you see and arrow at the right side of the qbase+ window and a dark grey box in the right half of qbase+ window. Release the mouse button when you see the arrow and the box. Now the two bar charts should be next to each other. Some laptop screens are too small to nicely display the two bar charts next to other. If this is the case switch to full screen mode by double clicking the tab of the first experiment.</td>
    </tr>
  </tbody>
</table>

<p>Now you can compare the expression of each gene in the first and in the second experiment.</p>

<p>When we do this for HighVar1 for instance, you see that the average expression levels of both groups are the same in the first and the second experiment (check the scales of the Y—axis!). Both experiments detect the two-fold difference in expression level between the groups. However, the error bars are much larger in the first experiment than in the second. The variability of the reference genes does have a strong influence on the errors and the size of the error bars will influence the outcome of the statistical test to determine if a gene is differentially expressed or not. The larger the error bars the smaller the less likely it is that the test will say that the groups differ.</p>

<p>Remember that the error bars represent 95% confidence intervals:</p>
<ul>
  <li>if the error bars of the two groups do not overlap: you are certain that the difference between the means of the two groups is significant</li>
  <li>if they do not overlap: you know nothing with certainty: the means can be different or they can be the same. Of course the more they overlap the smaller the chance that there is a significant difference between the groups.</li>
</ul>

<p>Check out the results of HighVar2. Here, you clearly see the influence of the reference genes. Again, the fourfold difference in expression is detected by both experiments but:</p>

<ul>
  <li>the least stable reference genes (experiment 1) give large overlapping error bars</li>
  <li>the most stable reference (experiment 2) give smaller, barely overlapping error bars</li>
</ul>

<p>This means that in experiment 2, a statistical test will probably declare that HighVar2 is differentially expressed while in experiment 1 this will not be the case. We will test this assumption by performing a statistical test.</p>

<h4 id="statistical-analysis-of-differential-expression">Statistical analysis of differential expression</h4>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Use a non-parametric test to identify DE genes in experiment 1 ?</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+. In brief, you need to perform the following steps:</td>
    </tr>
  </tbody>
</table>

<p>Open the Statistical wizard</p>

<p>The goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples</p>

<p>Use the Treatment property to identify treated and untreated samples</p>

<p>Analyze all genes of interest</p>

<p>Use the default settings to perform the non-parametric Mann-Whitney test</p>

<p>As you can see, none of the genes is considered DE by the very conservative non-parametric test. Additionally most genes have the same p-value. That’s normal when you don’t have many replicates. In our case, we have 6 replicates. Non-parametric tests are based on a ranking of the data values and there are not so many ways to rank 6 data points. This is why you see the same p-values for many genes.
As said before, the non-parametric test is very stringent. If the data do come from a normal distribution, the test will generate false positives. Some of the genes might have have been labeled not DE while in fact they are DE so you might have missed some differential expression. The choice of statistical test with 6 biological replicates depends on what you prefer: false negatives or false positives. Most people will choose false negatives since they don’t want to invest time and money in research on a genes that was labeled DE while in fact it is not DE.</p>

<p>Suppose I don’t mind false positives but I don’t want to miss any potential DE genes. In that case, it’s better to go for a t-test. Let’s repeat the test n ow choosing a parametric t-test.
| Use a t-test to identify DE genes in experiment 1 ? |
| :—————————————————- |
| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+.
In brief, you need to perform the following steps:
Open the Statistical wizard
The goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples
Use the Treatment property to identify treated and untreated samples
Analyze all genes of interest
Describe the data set as log-normally distributed</p>

<p>Still none of the genes is considered DE but you do see that the p-values of the t-test are lower than these of the Mann-Whitney test.</p>

<p>| Use a non parametric test to identify DE genes in experiment 2 ? |
| :—————————————————- |
| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+.
In brief, you need to perform the following steps:</p>

<p>Open the Statistical wizard
The goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples
Use the Treatment property to identify treated and untreated samples
Analyze all genes of interest
Use default settings</p>

<p>Now you see that 4 out of the 6 genes are considered DE. This is also what we expected since 3 of our genes of interst have a 4-fold difference in expression level between the two groups. It’s understandable that it’s hard to detect 2-fold differences in expression especially when the expression of the gene is somewhat variable as is the case for Low1 and HighVar1 but a 4-fold difference is a difference that you would like to detect.
| Use a t-test to identify DE genes in experiment 2 ? |
| :—————————————————- |
| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+.
In brief, you need to perform the following steps:</p>

<p>Open the Statistical wizard
The goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples
Use the Treatment property to identify treated and untreated samples
Analyze all genes of interest
Describe the data as log normally distributed</p>

<p>Again the t-test generates lower p-values than the Mann-Whitney test but realize that choosing the t-test when the data is not normally distributed will generate false positives !</p>



                    

                    

                    

                </div>
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2" role="supplementary">
                    <nav id="toc" data-toggle="toc" class="sticky-top bs-docs-sidebar hidden-print hidden-xs hidden-sm"></nav>
                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

</div>
</div>


<footer>
        <div class="well row">
         <div class="col-md-3 text-center">             
            <span class="slogan">SCIENCE MEETS LIFE</span>
         </div>
         <div class="col-md-6">
            This material is the result of a collaborative work. Thanks to the
            <a href="https://www.bits.vib.be">VIB Bioinformatics Core</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a>.
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/vibbits/training-material/tree/master/topics/qbase-plus/tutorials/reference-genes/tutorial.md">GitHub</a>.
          </div>
          <div class="col-md-3 text-center">
                <a href="http://www.bits.vib.be"><img src="/training-material/assets/images/bioinformatics_core_rgb_neg.png"></a>
       </div>
   </div>
   </footer>
</div>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript" src="https://app.cloudsh.com/js/cloudsh.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
</html>
