<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta name="google-site-verification" content="dw_1zBYRNxF7-vYNDXh3Ght_JuqgJt3BD1N_Pynqmv0" />
    <title>02 Improving the quality of NGS data</title>
    <script async="" src="https://www.google-analytics.com/analytics.js"></script>
    <script>
        (function(i, s, o, g, r, a, m) {
            i['GoogleAnalyticsObject'] = r;
            i[r] = i[r] || function() {
                (i[r].q = i[r].q || []).push(arguments)
            }, i[r].l = 1 * new Date();
            a = s.createElement(o),
                m = s.getElementsByTagName(o)[0];
            a.async = 1;
            a.src = g;
            m.parentNode.insertBefore(a, m)
        })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
        ga('create', 'UA-61194136-18', 'auto');
        ga('send', 'pageview');
    </script>
    <link rel="stylesheet" href="/assets/css/bootstrap.min.css?v=3">
    <link rel="stylesheet" href="/assets/css/bootstrap-toc.min.css">
    <link rel="stylesheet" href="/assets/css/font-awesome.css">
    <link rel="stylesheet" href="/assets/css/main.scss">
    <link rel="stylesheet" href="/assets/css/VIB.css">
    <link rel="stylesheet" href="/assets/css/academicons.css">
    <link rel="stylesheet" href="/assets/css/syntax_highlighting.css">
    <link rel="shortcut icon" href="/favicon.ico " type="image/x-icon" />     
    <meta name="description" content="Collection of tutorials developed and maintained by the V..." />
    <meta property="og:title" content="VIB Bioinformatics Training: 02 Improving the quality of NGS data" />
    <meta property="og:description" content="Collection of tutorials developed and maintained by the V..." />
    <meta property="og:image" content="/assets/images/bioinformatics_core_rgb_neg.png" />
</head>

<body data-spy="scroll" data-target="#toc">
    <div class="top-bar-vib">
 <a href="http://vib.be/en" target="_blank" style="font-size:14px">VIB Home </a>
</div>

<div class="container main-content">

<div class="container page-top__container">
          <div class="region region-top">
  <div id="logo" class="col-md-3 col-lg-3 text-left" style="padding-left:0px;margin-top:20px">
       <a href="/">
           <img src="/assets/images/logo.svg" height="60" alt="VIB logo" class="margin20">
       </a>
    </div>
    <div class="col-md-9 col-lg-9 text-right" style="margin-top:50px;">
    <nav role="navigation" aria-labelledby="block-mainnavigation-menu" id="block-mainnavigation" class="block block-menu navigation menu--main">
  


        <a href="/" class="topmenu">Home</a>


        <a href="/about.html" class="topmenu" >About</a>


        <a href="/hall-of-fame.html" class="topmenu">Contribute</a>


        <a href="/faq.html" class="topmenu">Faq</a>


        <a href="/search/index.html" class="topmenu">Search</a>


  </nav>

    </div>
  </div>
</div>














    <script type="application/ld+json">
        


{}
    </script>

    <div class="row tutorial">
      <div class="col-md-12" role="main">
        <h1 data-toc-skip>02 Improving the quality of NGS data</h1>
        <div class="contributors-line">By: 

<a href="/hall-of-fame#janick-bits" class="contributor-badge">Janick Mathys</a>


</div>
        <blockquote class="overview">

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>Download the data from GEO or ENA</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>Basic knowledge of downloading data from GEO or ENA</li>
            
            </ul>

            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 15 minutes</p>
            

            


            

        </blockquote>

        <div class="container" data-field="body">
            <div class="row">
                <div class="col-md-10" role="main">
                    <p><a href="http://data.bits.vib.be/pub/trainingen/NGSIntro/GroupExNoSol.pdf">Group exercises</a></p>

<p>You can solve most quality issues found by FASTQC e.g. trimming contaminating adapters, low quality bases at the end of your reads, filtering low quality reads…
There’s is a lot of debate on whether it is required to do this. Reads that are contaminated with adapter sequences will not map but if these reads make up a large fraction of the total number of reads they might slow down the mapping a lot. While it is true that mappers can use noisy info (still containing adapters, low quality bases…), the mapping results will be negatively affected by this noise.
Cleaning is in my opinion worthwhile especially when working with small reads and in case of extensive adapter contamination (almost always).</p>

<h2 id="quality-control-in-galaxy">Quality control in Galaxy</h2>

<p>Links:</p>
<ul>
  <li><a href="http://usegalaxy.eu">European Galaxy</a></li>
  <li><a href="https://usegalaxy.eu/u/janick/h/rawdata">Raw Arabidopsis data in European Galaxy</a></li>
  <li><a href="https://usegalaxy.eu/u/janick/h/groomeddata">Groomed Arabidopsis data in European Galaxy</a></li>
  <li><a href="https://usegalaxy.eu/u/janick/h/cleandata">Clean Arabidopsis data in European Galaxy</a></li>
  <li><a href="https://usegalaxy.eu/u/janick/h/ecoliraw">Raw E. coli data in European Galaxy</a></li>
  <li><a href="https://usegalaxy.eu/u/janick/h/ecoligroomed">Groomed E. coli data in European Galaxy</a></li>
  <li>
    <p><a href="https://usegalaxy.eu/u/janick/h/ecolifiltered">Filtered E. coli data in European Galaxy</a></p>
  </li>
  <li>[Main Galaxy](http://usegalaxy.org</li>
  <li><a href="https://usegalaxy.org/u/janick/h/ngsdata">Raw Arabidopsis data in main Galaxy</a></li>
  <li><a href="https://usegalaxy.org:/u/janick/h/ngsgroomed-1">Groomed Arabidopsis data in main Galaxy</a></li>
</ul>

<p>Galaxy is a bioinformatics server that contains many tools, data and analysis results. Before you can upload your data to Galaxy, you have to register or log in to Galaxy (see slides).</p>

<h3 id="upload-data-to-galaxy">Upload data to Galaxy</h3>

<p>If you want to work on your data in Galaxy, you have to first get the data into Galaxy. To accomplish this you can use the <strong>Upload file</strong> tool in the <strong>Get data</strong> section.
<strong>Instead I shared the file on Galaxy so you can import it using <a href="https://usegalaxy.eu/u/janick/h/rawdata">this link</a>.</strong> Make sure that you are logged on to Galaxy before you do this. When you click this link you are redirected to a web page where you can import the file:</p>

<p><img src="../../images/Galaxy6a.png" alt="" /></p>

<h3 id="the-history">The history</h3>

<p>Data sets that are uploaded or created by running a tool appear in the history in the right Galaxy pane.
To give a history a new name, click the history’s current name, type a new one and hit enter.
Clicking the name of a data set unfolds a preview, a short description and tools to manipulate the data.</p>

<h4 id="icons-in-the-history">Icons in the History</h4>

<ul>
  <li>Clicking the floppy (<strong>Download</strong>) icon will download the file to your computer</li>
  <li>To visualize a file in the middle pane, click the eye (<strong>View data</strong>) icon next to the name of the file.</li>
</ul>

<h4 id="colors-of-files-in-the-historydata-sets-in-the-history-have-different-colors-representing-different-states">Colors of files in the HistoryData sets in the history have different colors representing different states.</h4>

<ul>
  <li><strong>Grey</strong>: The job is placed in the waiting queue. You can check the status of queued jobs by refreshing the History pane.</li>
  <li><strong>Yellow</strong>: The job is running.</li>
  <li><strong>Green</strong>: When the job has been run the status will change from yellow to green if completed successfully.</li>
  <li><strong>Red</strong>: When the job has been run the status will change from yellow to red if problems were encountered.</li>
</ul>

<h3 id="running-groomer-in-galaxy">Running Groomer in Galaxy</h3>

<p>If you select a tool in Galaxy it will automatically detect all data sets in your history that it can use as input. In the case shown below the tool does not recognize the fastq file in the history:</p>

<p><img src="../../images/Galaxy10b.png" alt="" /></p>

<p>The fact that the tool does not recognize the fastq file means that the fastq file is so messy that the tool can’t read it. Remember that there is a tool to clean messy fastq files: <strong>FASTQ Groomer</strong></p>

<p>Check the quality encoding in your fastq file (e.g. in FASTQC), and click the <strong>Execute</strong> button to start the tool:</p>

<p><img src="../../images/Galaxy12a.png" alt="" /></p>

<p>Grooming takes long (30 min when Galaxy traffic is low). You can choose to wait but if it takes too long you can click the <strong>Delete</strong> button in the History (see slides) to stop the tool. I have provided the groomed file: import it in Galaxy using [https://usegalaxy.eu/u/janick/h/groomeddata this link].</p>

<h3 id="using-trimmomatic-in-galaxy">Using Trimmomatic in Galaxy</h3>

<p>To clean your data use the <strong>Trimmomatic</strong> tool in the <strong>Quality Control</strong> section of tools. Click the name of the tool to display its parameters in the middle pane.</p>

<p>See <a href="http://wiki.bits.vib.be/index.php/Parameters_of_Trimmomatic">this page</a> for an overview of the Trimmomatic parameters.</p>

<p>A bit more explanation:</p>

<ul>
  <li><strong>The input file with the reads</strong>: Galaxy will automatically suggest a file from your History that has the right format, in this case: a fastq file. If Galaxy doesn’t make a suggestion it means it cannot find any files in your History with the right format.</li>
  <li><strong>The sequence of the adapter</strong>: provide a custom sequence. If you analyze your own data you know which adapter sequences were used. Since this is public data we don’t really know the name of the adapter. However, remember that FASTQC gives you a list of contaminating adapter sequences so you have the sequence of the adapter. Choose <strong>custom adapter sequence</strong> and paste the adapter sequence from FASTQC. You can only enter one sequence.</li>
</ul>

<p>Click <strong>Execute</strong> to run the tool.</p>

<p>In the history you see a new item, colored in yellow as long as the tool is running. Regularly hit the <strong>Refresh</strong> button in the History to check if the tool has finished. Clipping should go fast, after a few minutes you should have the result.</p>

<h3 id="running-fastqc-in-galaxy">Running FASTQC in Galaxy</h3>

<p>Search for <strong>FASTQC</strong> in the tools pane and click the resulting <strong>FastQC</strong> link to open the parameter settings in the middle pane:</p>

<p><img src="../../images/Galaxy18b.png" alt="" /></p>

<p>FASTQC automatically recognizes all files it can use as an input. Select the file you want to use.
The FASTQC implementation in Galaxy can take an optional file containing a list of contaminants. If you don’t specify one, FASTQC will look for standardly used Illumina adapters. <!--As another option the tool takes a custom limits .txt file that allows setting the warning thresholds for the different modules and to specify which modules to include in the output.--></p>

<p>In most cases you keep the default settings and click <strong>Execute</strong>.</p>

<h2 id="quality-control-in-genepattern">Quality control in GenePattern</h2>

<p><a href="http://www.broadinstitute.org/cancer/software/genepattern/">Genepattern</a> is very similar to Galaxy. It’s as user-friendly as Galaxy, allows analysis of NGS data just like Galaxy…</p>

<p>It provides easy access to hundreds of tools for different kinds of analyses (e.g. RNA-seq, microarray, proteomics and flow cytometry, sequence variation, copy number and network analysis) via a web browser.</p>

<p><strong>Links</strong></p>

<ul>
  <li><a href="https://dev.bits.vib.be:8686/gp/pages/index.jsf">BITS Genepattern server</a></li>
  <li><a href="https://data.bits.vib.be/pub/trainingen/NGSIntro/adapter.fa">fasta file containing Arabidopsis adapter sequence</a></li>
  <li><a href="https://data.bits.vib.be/pub/trainingen/NGSIntro/adapterEcoli.fa">fasta file containing E. coli adapter sequence</a></li>
  <li><a href="http://wiki.bits.vib.be/index.php/Parameters_of_Trimmomatic">Overview of Trimmomatic parameters</a></li>
</ul>

<p>Consult the <a href="https://wiki.bits.vib.be/index.php/GenePattern_tutorial">GenePattern tutorial</a> for more info.</p>

<h3 id="running-groomer-in-genepattern">Running Groomer in GenePattern</h3>

<p>The Broad Genepattern server does not contain the Groomer tool, but we have added the tool to our BITS Genepattern server.</p>

<ul>
  <li>Search for the Groomer tool in GenePattern.</li>
  <li>Define the parameters: one of the parameters you need to define is <strong>Input format</strong>: the encoding of the fastq file you want to clean. The encoding is important because it determines the offset of the quality scores (ASCII offset 33 or ASCII offset 64). If you’re not sure you can check the encoding of your file in the FastQC report (take into account that FastQC sporadically makes the wrong guess).
 <img src="../../images/GP9.png" alt="" /></li>
  <li>Run the Groomer tool.</li>
</ul>

<h3 id="running-fastqc-in-genepattern">Running FastQC in GenePattern</h3>

<ul>
  <li>Search for the FASTQC tool</li>
  <li>Fill in the parameters</li>
  <li>Run the FASTQC tool</li>
</ul>

<p>You can open the resulting HTML report in your browser:</p>

<ul>
  <li>Click the name of the output file at the bottom of the page</li>
  <li>Select <strong>Open Link</strong>
 <img src="../../images/GP18.png" alt="" /></li>
</ul>

<h3 id="running-trimmomatic-in-genepattern">Running Trimmomatic in GenePattern</h3>

<p>In GenePattern you can improve the quality of your NGS data using the Trimmomatic tool.</p>
<ul>
  <li>Search for the Trimmomatic tool</li>
  <li>Fill in the parameters: See <a href="http://wiki.bits.vib.be/index.php/Parameters_of_Trimmomatic">this page</a> for an overview of the Trimmomatic parameters.</li>
  <li>Run Trimmomatic</li>
</ul>

<h2 id="removing-adapters-using-command-line-tools">Removing adapters using command line tools</h2>

<p>See <a href="http://wiki.bits.vib.be/index.php/Linux_command_line#Improving_the_quality_of_the_data">exercise on using cutadapt to trim adapter sequences</a></p>



                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>

                        <ul>
                            
                            <li>SRA ID</li>
                            
                            <li>GEO</li>
                            
                            <li>ENA</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    

                </div>
                <!-- sidebar, which will be hidden on a small screen -->
                <nav class="col-md-2 sticky-toc bs-docs-sidebar hidden-print hidden-xs hidden-sm" role="complementary" id="toc" data-toggle="toc"></nav>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

</div>
</div>
</div>
<footer>
   <div class="well row">
       <div class="col-md-2 text-center" style="padding: 25px 20px 25px 20px;">				
         <a href="https://www.bits.vib.be" target="_blank"><img src="/assets/images/bioinformatics_core_rgb_neg.png" width="200"></a>
       </div>
       <div class="col-md-8 text-center" style="padding-top: 20px;">	
        <a href="https://github.com/vibbits/training-material" title="View on GitHub" class="margin20" target="_blank">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View on GitHub
                        </a>
       </div>
       <div class="col-md-2 text-center" style="padding-top: 30px;">
         <span class="slogan"><br>SCIENCE MEETS LIFE</span>
       </div>
   </div>
</footer>


</body>
<script type="text/javascript" src="/assets/js/jquery.slim.min.js"></script>
<script type="text/javascript" src="/assets/js/popper.min.js"></script>
<script type="text/javascript" src="/assets/js/bootstrap.min.js?v=3"></script>
<script type="text/javascript" src="/assets/js/details-element-polyfill.js"></script>
<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<script type="text/javascript" src="/assets/js/bootstrap-toc.min.js"></script>
<script type="text/javascript" src="/assets/js/main.js"></script>
<script type="text/javascript" src="/assets/js/clipboard.min.js"></script>
<script type="text/javascript">
    var snippets = document.querySelectorAll('div.highlight');
    [].forEach.call(snippets, function(snippet) {
        snippet.firstChild.insertAdjacentHTML('beforebegin', '<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets = new ClipboardJS('[data-clipboard-snippet]', {
        target: function(trigger) {
            return trigger.nextElementSibling;
        }
    });
</script>

</html>
