<!DOCTYPE html>
<html>









  <head>
    <meta charset="utf-8">
    <title>Introduction to Basic Bioinformatics Concepts, Databases and Tools</title>
    
    <link rel="stylesheet" href="/training-material/assets/css/slides.css">
    <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css" id="theme">
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

    
    
    
    
    <meta name="description" content="Bioinformatics Introduction" />
    <meta property="og:title" content="VIB Bioinformatics Training: Introduction to Basic Bioinformatics Concepts, Databases ..." />
    <meta property="og:description" content="Bioinformatics Introduction" />
    <meta property="og:image" content="/training-material/assets/images/bioinformatics_core_rgb_neg.png" />
    <script type="application/ld+json">
       


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "headline": "Introduction to Basic Bioinformatics Concepts, Databases and Tools",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "provider": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "courseCode": "basic-bioinformatics / introduction / introduction.html",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Basic Bioinformatics Concepts, Databases and Tools",
    "description": "Bioinformatics Introduction",
    "url": "https://vibbits.github.io//training-material/topics/basic-bioinformatics/"
  },
  "learningResourceType": "slides",
  "name": "Introduction to 'Basic Bioinformatics Concepts, Databases and Tools'",
  "url": "https://vibbits.github.io//training-material/topics/basic-bioinformatics/slides/introduction.html",
  "description": "Introduction to Basic Bioinformatics Concepts, Databases and Tools",
  "hasPart": [

  ],
  "mentions": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Alexander Botzki"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Alexander Botzki"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Basic Bioinformatics Concepts, Databases and Tools",
      "description": "Bioinformatics Introduction",
      "url": "https://vibbits.github.io//training-material/topics/basic-bioinformatics/"
    }
  ]
}
    </script>
  </head>
  <body>
         <pre>
              site: {&quot;config&quot;:null,&quot;data&quot;:{&quot;basic-bioinformatics&quot;:{&quot;name&quot;:&quot;basic-bioinformatics&quot;,&quot;type&quot;:&quot;basics&quot;,&quot;category&quot;:&quot;basics&quot;,&quot;title&quot;:&quot;Basic Bioinformatics Concepts, Databases and Tools&quot;,&quot;summary&quot;:&quot;Bioinformatics Introduction&quot;,&quot;requirements&quot;:null,&quot;maintainers&quot;:[&quot;abotzki&quot;]},&quot;contributors&quot;:{&quot;chdeb&quot;:{&quot;name&quot;:&quot;Christof De Bo&quot;,&quot;email&quot;:&quot;christof.debo@vib.be&quot;},&quot;tmuylder&quot;:{&quot;name&quot;:&quot;Tuur Muyldermans&quot;,&quot;email&quot;:&quot;tuur.muyldermans@vib.be&quot;},&quot;janick-bits&quot;:{&quot;name&quot;:&quot;Janick Mathys&quot;,&quot;email&quot;:&quot;janick.mathys@vib.be&quot;},&quot;abotzki&quot;:{&quot;name&quot;:&quot;Alexander Botzki&quot;,&quot;email&quot;:&quot;alexander.botzki@vib.be&quot;},&quot;jvdurme&quot;:{&quot;name&quot;:&quot;Joost Van Durme&quot;,&quot;email&quot;:&quot;bits@vib.be&quot;}},&quot;protein-structure-analysis&quot;:{&quot;name&quot;:&quot;protein-structure-analysis&quot;,&quot;type&quot;:&quot;basics&quot;,&quot;category&quot;:&quot;basics&quot;,&quot;title&quot;:&quot;Protein Structure Analysis&quot;,&quot;summary&quot;:&quot;Analyzing the protein structure of your protein-of-interest can be advantageous in multiple ways. It can help you discover regions which are good candidates to interact with other proteins. It can help you discover new domains. It can help with identifying differences with homologuous proteins and a lot more.&quot;,&quot;extra&quot;:&quot;protein_analysis&quot;,&quot;requirements&quot;:null,&quot;maintainers&quot;:[&quot;abotzki&quot;],&quot;references&quot;:[{&quot;authors&quot;:&quot;Switchlab&quot;,&quot;title&quot;:&quot;Home of FoldX plugin&quot;,&quot;link&quot;:&quot;http://foldxyasara.switchlab.org/&quot;,&quot;summary&quot;:&quot;More information about FoldX plugin and troubleshooting&quot;},{&quot;authors&quot;:&quot;Wikipedia&quot;,&quot;title&quot;:&quot;Wiki page about PDB&quot;,&quot;link&quot;:&quot;https://en.wikipedia.org/wiki/Protein_Data_Bank&quot;,&quot;summary&quot;:&quot;wiki page explaining information on PDB&quot;},{&quot;authors&quot;:&quot;YASARA developers&quot;,&quot;title&quot;:&quot;Working with YASARA&quot;,&quot;link&quot;:&quot;http://www.yasara.org/movies.htm&quot;,&quot;summary&quot;:&quot;Movie tutorials on YASARA&quot;}]},&quot;qbase-plus&quot;:{&quot;name&quot;:&quot;qbase-plus&quot;,&quot;type&quot;:&quot;basics&quot;,&quot;category&quot;:&quot;software&quot;,&quot;title&quot;:&quot;QPCR analysis using qbase+&quot;,&quot;summary&quot;:&quot;Bioinformatics Introduction&quot;,&quot;requirements&quot;:null,&quot;maintainers&quot;:[&quot;chdeb&quot;]}},&quot;time&quot;:&quot;2019-09-14 09:10:14 +0000&quot;,&quot;tags&quot;:{},&quot;pages&quot;:[&quot;\n&quot;,&quot;---\nname: galaxy_training_material\nchannels:\n  - conda-forge\n  - bioconda\n  - defaults\ndependencies:\n  - gmp=6.1.2\n  - jemalloc=5.0.1\n  - libiconv=1.15\n  - nodejs=9.11.1\n  - openssl=1.0.2o\n  - pandas=0.22.0\n  - pip\n  - pip:\n    - pathspec==0.5.6\n    - oyaml\n  - planemo&gt;=0.55.0\n  - readline=7.0\n  - requests=2.18.4\n  - ruby=2.4.4\n  - yaml=0.1.7\n  - yamllint=1.11.0\n  - zlib=1.2.11\n  - libxml2\n  - ephemeris\n  - pkg-config\n&quot;,&quot;&lt;ol id=\&quot;markdown-toc\&quot;&gt;\n  &lt;li&gt;&lt;a href=\&quot;#overview-questions\&quot; id=\&quot;markdown-toc-overview-questions\&quot;&gt;Overview Questions&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-is-this-website\&quot; id=\&quot;markdown-toc-what-is-this-website\&quot;&gt;What is this website?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-is-galaxy\&quot; id=\&quot;markdown-toc-what-is-galaxy\&quot;&gt;What is Galaxy?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-are-the-tutorials-for\&quot; id=\&quot;markdown-toc-what-are-the-tutorials-for\&quot;&gt;What are the tutorials for?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-audiences-are-the-tutorials-for\&quot; id=\&quot;markdown-toc-what-audiences-are-the-tutorials-for\&quot;&gt;What audiences are the tutorials for?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-is-the-content-licensed\&quot; id=\&quot;markdown-toc-how-is-the-content-licensed\&quot;&gt;How is the content licensed?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-cite-this-project\&quot; id=\&quot;markdown-toc-how-can-i-cite-this-project\&quot;&gt;How can I cite this project?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-advertise-the-training-materials-on-my-posters\&quot; id=\&quot;markdown-toc-how-can-i-advertise-the-training-materials-on-my-posters\&quot;&gt;How can I advertise the training materials on my posters?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#for-individual-learners\&quot; id=\&quot;markdown-toc-for-individual-learners\&quot;&gt;For Individual Learners&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-do-i-start\&quot; id=\&quot;markdown-toc-where-do-i-start\&quot;&gt;Where do I start?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-do-i-use-this-material\&quot; id=\&quot;markdown-toc-how-do-i-use-this-material\&quot;&gt;How do I use this material?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-can-i-run-the-hands-on-tutorials\&quot; id=\&quot;markdown-toc-where-can-i-run-the-hands-on-tutorials\&quot;&gt;Where can I run the hands-on tutorials?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-get-help\&quot; id=\&quot;markdown-toc-how-can-i-get-help\&quot;&gt;How can I get help?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#for-instructors\&quot; id=\&quot;markdown-toc-for-instructors\&quot;&gt;For Instructors&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-do-i-start-1\&quot; id=\&quot;markdown-toc-where-do-i-start-1\&quot;&gt;Where do I start?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-galaxy-instance-should-i-use-for-my-training\&quot; id=\&quot;markdown-toc-what-galaxy-instance-should-i-use-for-my-training\&quot;&gt;What Galaxy instance should I use for my training?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-do-i-get-help\&quot; id=\&quot;markdown-toc-how-do-i-get-help\&quot;&gt;How do I get help?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#contributing\&quot; id=\&quot;markdown-toc-contributing\&quot;&gt;Contributing&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-get-started-with-contributing\&quot; id=\&quot;markdown-toc-how-can-i-get-started-with-contributing\&quot;&gt;How can I get started with contributing?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-give-feedback\&quot; id=\&quot;markdown-toc-how-can-i-give-feedback\&quot;&gt;How can I give feedback?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-test-an-interactive-tour\&quot; id=\&quot;markdown-toc-how-can-i-test-an-interactive-tour\&quot;&gt;How can I test an Interactive Tour?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot; id=\&quot;markdown-toc-how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot;&gt;How can I fix mistakes or expand an existing tutorial using the GitHub interface?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-contribute-in-advanced-mode\&quot; id=\&quot;markdown-toc-how-can-i-contribute-in-advanced-mode\&quot;&gt;How can I contribute in “advanced” mode?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#other-questions\&quot; id=\&quot;markdown-toc-other-questions\&quot;&gt;Other Questions&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#sustainability-of-the-training-material-and-metadata\&quot; id=\&quot;markdown-toc-sustainability-of-the-training-material-and-metadata\&quot;&gt;Sustainability of the training-material and metadata&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h1 id=\&quot;overview-questions\&quot;&gt;Overview Questions&lt;/h1&gt;\n\n&lt;h2 id=\&quot;what-is-this-website\&quot;&gt;What is this website?&lt;/h2&gt;\n\n&lt;p&gt;This website is a collection of hands-on tutorials that are designed to be interactive and are built around Galaxy:&lt;/p&gt;\n\n&lt;figure id=\&quot;figure-1\&quot;&gt;&lt;img src=\&quot;shared/images/interactive_training.png\&quot; alt=\&quot;Interactive training\&quot; /&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 1:&lt;/span&gt; The hands-on nature of our training material. It can be used with two web browser windows open side-by-side, one pointed at the current tutorial and the other at a Galaxy instance&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;p&gt;This material is developed and maintained by the &lt;a href=\&quot;https://galaxyproject.org/\&quot;&gt;worldwide Galaxy community&lt;/a&gt;. You can learn more about this effort by reading our &lt;a href=\&quot;https://doi.org/10.1016/j.cels.2018.05.012\&quot;&gt;article&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-is-galaxy\&quot;&gt;What is Galaxy?&lt;/h2&gt;\n\n&lt;p&gt;&lt;a href=\&quot;https://galaxyproject.org/\&quot;&gt;Galaxy&lt;/a&gt; is an open data integration and analysis platform for the life sciences, and it is particularly well-suited for data analysis training in life science research.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-are-the-tutorials-for\&quot;&gt;What are the tutorials for?&lt;/h2&gt;\n\n&lt;p&gt;These tutorials can be used for learning and teaching how to use Galaxy for general data analysis, and for learning/teaching specific domains such as assembly and differential gene expression analysis with RNA-Seq data.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-audiences-are-the-tutorials-for\&quot;&gt;What audiences are the tutorials for?&lt;/h2&gt;\n\n&lt;p&gt;There are two distinct audiences for these materials.&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;&lt;strong&gt;Self-paced individual learners.&lt;/strong&gt; These tutorials provide everything you need to learn a topic, from explanations of concepts to detailed hands-on exercises.&lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Instructors.&lt;/strong&gt; They are also designed to be used by instructors in teaching/training settings. Slides, and detailed tutorials are provided. Most tutorials also include computational support with the needed tools, data as well as Docker images that can be used to scale the lessons up to many participants.&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h2 id=\&quot;how-is-the-content-licensed\&quot;&gt;How is the content licensed?&lt;/h2&gt;\n\n&lt;p&gt;The content of this website is licensed under the &lt;a href=\&quot;https://creativecommons.org/licenses/by/4.0/\&quot;&gt;Creative Commons Attribution 4.0 License&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-cite-this-project\&quot;&gt;How can I cite this project?&lt;/h2&gt;\n\n&lt;p&gt;We wrote an &lt;a href=\&quot;https://doi.org/10.1016/j.cels.2018.05.012\&quot;&gt;article&lt;/a&gt; about our efforts.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-advertise-the-training-materials-on-my-posters\&quot;&gt;How can I advertise the training materials on my posters?&lt;/h2&gt;\n\n&lt;p&gt;We provide some QR codes and logos in the &lt;a href=\&quot;https://github.com/vibbits/training-material/tree/master/assets/images\&quot;&gt;images folder&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;for-individual-learners\&quot;&gt;For Individual Learners&lt;/h1&gt;\n\n&lt;p&gt;Learning Galaxy and data analysis on your own and at your own pace?  This material is for you.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-do-i-start\&quot;&gt;Where do I start?&lt;/h2&gt;\n\n&lt;p&gt;If you are new to Galaxy then start with one of the introductory topics.  These introduce you to concepts that are useful in Galaxy, no matter what domain you are doing analysis in.&lt;/p&gt;\n\n&lt;p&gt;If you are already familiar with Galaxy basics and want to learn how to use it in a particular domain (for example, ChIP-Seq), then start with one of those topics.&lt;/p&gt;\n\n&lt;p&gt;If you are already well informed about bioinformatics data analysis and you just want to get a feel for how it works in Galaxy, then many tutorials include &lt;em&gt;Instructions for the impatient&lt;/em&gt; sections.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-do-i-use-this-material\&quot;&gt;How do I use this material?&lt;/h2&gt;\n\n&lt;p&gt;Many topics include slide decks and if the topic you are interested in has slides then start there.  These will introduce the topic and important concepts.&lt;/p&gt;\n\n&lt;p&gt;Most of your learning will happen in the next step - the hands-on tutorials.  This is where you’ll become familiar with using the Galaxy interface and experiment with different ways to use Galaxy and the tools in Galaxy.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-can-i-run-the-hands-on-tutorials\&quot;&gt;Where can I run the hands-on tutorials?&lt;/h2&gt;\n\n&lt;p&gt;To run the hands-on tutorials you need a Galaxy server to run them on.&lt;/p&gt;\n\n&lt;p&gt;Each tutorial is annotated with information about which &lt;a href=\&quot;https://galaxyproject.org/public-galaxy-servers/\&quot;&gt;public Galaxy servers&lt;/a&gt; it can be run on. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial.&lt;/p&gt;\n\n&lt;p&gt;Some topics have a &lt;a href=\&quot;https://www.docker.com/\&quot;&gt;Docker&lt;/a&gt; image that can be installed and run on participants’ laptops.  These Docker images contain Galaxy instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant. You will need to &lt;a href=\&quot;https://docs.docker.com/install/\&quot;&gt;install Docker&lt;/a&gt;.&lt;/p&gt;\n\n&lt;p&gt;Finally, you can also run your tutorials on cloud-based infrastructures.  Galaxy is &lt;a href=\&quot;https://galaxyproject.org/galaxy-services/\&quot;&gt;available on many national research infrastructures&lt;/a&gt; such as &lt;a href=\&quot;https://galaxyproject.org/cloud/jetstream/\&quot;&gt;Jetstream&lt;/a&gt; (United States), &lt;a href=\&quot;https://www.genap.ca/\&quot;&gt;GenAP&lt;/a&gt; (Canada), &lt;a href=\&quot;https://www.gvl.org.au/\&quot;&gt;GVL&lt;/a&gt; (Australia), &lt;a href=\&quot;http://www.climb.ac.uk/\&quot;&gt;CLIMB&lt;/a&gt; (United Kingdom), and more.  These instances are typically easy to launch, and easy to shut down when you are done.&lt;/p&gt;\n\n&lt;p&gt;If you are already familiar with, and have an account on &lt;a href=\&quot;https://aws.amazon.com/\&quot;&gt;Amazon Web Services&lt;/a&gt; then you can also launch a Galaxy server there using &lt;a href=\&quot;https://launch.usegalaxy.org/\&quot;&gt;CloudLaunch&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-get-help\&quot;&gt;How can I get help?&lt;/h2&gt;\n\n&lt;p&gt;If you have questions about this training material, you can reach us using the &lt;a href=\&quot;\&quot;&gt;Gitter chat&lt;/a&gt;.  You’ll need a &lt;a href=\&quot;https://github.com/\&quot;&gt;GitHub&lt;/a&gt; or &lt;a href=\&quot;https://twitter.com/\&quot;&gt;Twitter&lt;/a&gt; account to post questions.  If you have questions about Galaxy outside the context of training, see the &lt;a href=\&quot;https://galaxyproject.org/support/\&quot;&gt;Galaxy Support page&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;for-instructors\&quot;&gt;For Instructors&lt;/h1&gt;\n\n&lt;p&gt;This material can also be used to teach Galaxy and data analysis in a group setting to students and researchers.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-do-i-start-1\&quot;&gt;Where do I start?&lt;/h2&gt;\n\n&lt;p&gt;Spend some time exploring the different tutorials and the different resources that are available. Become familiar with the structure of the tutorials and think about how you might use them in your teaching.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-galaxy-instance-should-i-use-for-my-training\&quot;&gt;What Galaxy instance should I use for my training?&lt;/h2&gt;\n\n&lt;p&gt;To teach the hands-on tutorials you need a Galaxy server to run the examples on.&lt;/p&gt;\n\n&lt;p&gt;Each tutorial is annotated with the information on which &lt;a href=\&quot;https://galaxyproject.org/public-galaxy-servers/\&quot;&gt;public Galaxy servers&lt;/a&gt; it can be run. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial. If you choose this option then you should work with that server’s admins to confirm that think the server can handle the workload for a workshop. For example, the &lt;a href=\&quot;https://usegalaxy.eu/\&quot;&gt;usegalaxy.eu&lt;/a&gt;&lt;/p&gt;\n\n&lt;p&gt;If your organization/consortia/community has its own Galaxy server, then you may  want to run tutorials on that. This can be ideal because then the instance you are teaching on is the same you your participants will be using after the training. They’ll also be able to revisit any analysis they did during the training. If you pursue this option you’ll need to work with your organization’s Galaxy Admins to confirm that&lt;/p&gt;\n\n&lt;ul&gt;\n  &lt;li&gt;the server can support a room full of people all doing the same analysis at the same time.&lt;/li&gt;\n  &lt;li&gt;all participants will be able to create/use accounts on the system.&lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;Some training topics have a Docker image that can be installed and run on all participants’ laptops.  These images contain Galaxy instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant.&lt;/p&gt;\n\n&lt;p&gt;Finally, you can also run your tutorials on cloud-based infrastructures.  Galaxy is &lt;a href=\&quot;https://galaxyproject.org/galaxy-services/\&quot;&gt;available on many national research infrastructures&lt;/a&gt; such as &lt;a href=\&quot;https://galaxyproject.org/cloud/jetstream/\&quot;&gt;Jetstream&lt;/a&gt; (United States), &lt;a href=\&quot;https://www.genap.ca/\&quot;&gt;GenAP&lt;/a&gt; (Canada), &lt;a href=\&quot;https://launch.genome.edu.au/launch\&quot;&gt;GVL&lt;/a&gt; (Australia), &lt;a href=\&quot;http://www.climb.ac.uk/\&quot;&gt;CLIMB&lt;/a&gt; (United Kingdom), and more.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-do-i-get-help\&quot;&gt;How do I get help?&lt;/h2&gt;\n\n&lt;p&gt;The support channel for instructors is the same as for individual learners.  We suggest you start by posting a question to the Galaxy Training Network &lt;a href=\&quot;\&quot;&gt;Gitter chat&lt;/a&gt;. Anyone can view the discussion, but you’ll need to login (using your GitHub or Twitter account) to add to the discussion.&lt;/p&gt;\n\n&lt;p&gt;If you have questions about Galaxy in general (that are not training-centric) then there are &lt;a href=\&quot;https://galaxyproject.org/support/\&quot;&gt;several support options&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;contributing\&quot;&gt;Contributing&lt;/h1&gt;\n\n&lt;p&gt;First off, thanks for your interest in contributing to the Galaxy training materials!&lt;/p&gt;\n\n&lt;p&gt;Individual learners and instructors can make these training more effective by contributing back to them. You can report mistakes and errors, create more content, etc. Whatever is your background, there is a way to contribute: via the GitHub website, via command-line or even without dealing with GitHub.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-get-started-with-contributing\&quot;&gt;How can I get started with contributing?&lt;/h2&gt;\n\n&lt;p&gt;If you would like to get involved in the project but are unsure where to start, there are some easy ways to contribute which will also help you familiarize yourself with the project.&lt;/p&gt;\n\n&lt;h3 id=\&quot;1-checking-existing-tutorials\&quot;&gt;1. Checking existing tutorials&lt;/h3&gt;\n\n&lt;p&gt;A great way to help out the project is to test/edit existing tutorials. Pick a tutorial and check the contents. Does everything work as expected? Are there things that could be improved?&lt;/p&gt;\n\n&lt;p&gt;Below is a checklist of things to look out for to help you get started. If you feel confident in making changes yourself, please open a pull request, otherwise please file an issue with any problems you run into or suggestions for improvements.&lt;/p&gt;\n\n&lt;p&gt;&lt;em&gt;Basic&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Test&lt;/strong&gt; the tutorial on a running Galaxy instance\n    &lt;ul&gt;\n      &lt;li&gt;For example &lt;a href=\&quot;https://usegalaxy.org\&quot;&gt;Galaxy Main&lt;/a&gt;, &lt;a href=\&quot;https://usegalaxy.eu\&quot;&gt;Galaxy Europe&lt;/a&gt;, or &lt;a href=\&quot;https://usegalaxy.org.au\&quot;&gt;Galaxy Australia&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;Report any issues you run into&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Language&lt;/strong&gt; editing\n    &lt;ul&gt;\n      &lt;li&gt;Fix spelling and grammar mistakes&lt;/li&gt;\n      &lt;li&gt;Simplify the English (to make it more accessible)&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;&lt;em&gt;Intermediate&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Metadata&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Are the objectives, keypoints and time estimate filled in?&lt;/li&gt;\n      &lt;li&gt;Do they fit with the contents of the tutorial?&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Content&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Is there enough background information provided in the introduction section and throughout the tutorial?&lt;/li&gt;\n      &lt;li&gt;&lt;strong&gt;Question boxes&lt;/strong&gt;\n        &lt;ul&gt;\n          &lt;li&gt;Add questions or question boxes where you think they might be useful (make people think about results they got, test their understanding, etc)&lt;/li&gt;\n          &lt;li&gt;Check that answers are still up-to-date&lt;/li&gt;\n        &lt;/ul&gt;\n      &lt;/li&gt;\n      &lt;li&gt;&lt;strong&gt;Screenshots and Videos&lt;/strong&gt;\n        &lt;ul&gt;\n          &lt;li&gt;Make sure there is also a textual description of the image/video contents&lt;/li&gt;\n          &lt;li&gt;Does the screenshot add value to the tutorial or can it be removed?&lt;/li&gt;\n        &lt;/ul&gt;\n      &lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;&lt;em&gt;Advanced&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Workflows&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Add a workflow definition file &lt;code class=\&quot;highlighter-rouge\&quot;&gt;.ga&lt;/code&gt; if none is present&lt;/li&gt;\n      &lt;li&gt;Check that the existing workflow is up-to-date with the tutorial contents&lt;/li&gt;\n      &lt;li&gt;Enable &lt;a href=\&quot;https://github.com/usegalaxy-eu/workflow-testing\&quot;&gt;workflow testing&lt;/a&gt;&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Tours&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Add a tour if none exists&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-test-an-interactive-tour\&quot;&gt;Run the existing tour&lt;/a&gt; and check that it is up-to-date with the tutorial contents&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Datasets&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Check that all datasets used in the tutorial are present in Zenodo&lt;/li&gt;\n      &lt;li&gt;Add a data-library.yaml file if none exists&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;h2 id=\&quot;how-can-i-give-feedback\&quot;&gt;How can I give feedback?&lt;/h2&gt;\n\n&lt;p&gt;At the end of each tutorial, there is a link to &lt;a href=\&quot;https://docs.google.com/forms/d/1BILuBp8bPv42odpZ55Kcf2rqzR-k0Czx46agVVSRgm4/edit\&quot;&gt;a feedback form&lt;/a&gt;. We use this information to improve our tutorials.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-test-an-interactive-tour\&quot;&gt;How can I test an Interactive Tour?&lt;/h2&gt;\n\n&lt;p&gt;Perhaps you’ve been asked to review an interactive tour, or maybe you just want to try one out. The easiest way to run an interactive tour is to use the &lt;a href=\&quot;https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/\&quot;&gt;Tour builder&lt;/a&gt; browser extension.&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;Install the Tour Builder extension to your browser (&lt;a href=\&quot;https://chrome.google.com/webstore/detail/galaxy-tour-builder/mdfbapknmcpnbmggahhaegehbbbmhmgg\&quot;&gt;Chrome Web Store&lt;/a&gt;, &lt;a href=\&quot;https://addons.mozilla.org/en-US/firefox/addon/galaxy-tour-builder/\&quot;&gt;Firefox add-on&lt;/a&gt;, &lt;a href=\&quot;https://addons.opera.com/en/extensions/details/galaxy-tour-builder/\&quot;&gt;Opera add-on&lt;/a&gt;).&lt;/li&gt;\n  &lt;li&gt;Navigate to a Galaxy instance supporting the tutorial. To find which Galaxy instances support each tutorial, please see the dropdown menu next to the tutorial on the training website. Using one of the usegalaxy.* instances (&lt;a href=\&quot;https://usegalaxy.org\&quot;&gt;Galaxy Main&lt;/a&gt;, &lt;a href=\&quot;https://usegalaxy.eu\&quot;&gt;Galaxy Europe&lt;/a&gt;, or &lt;a href=\&quot;https://usegalaxy.org.au\&quot;&gt;Galaxy Australia&lt;/a&gt;\n) is usually a good bet.&lt;/li&gt;\n  &lt;li&gt;Start the Tour Builder plugin by clicking on the icon in your browser menu bar&lt;/li&gt;\n  &lt;li&gt;Copy the contents of the &lt;code class=\&quot;highlighter-rouge\&quot;&gt;tour.yaml&lt;/code&gt; file into the Tour builder editor window&lt;/li&gt;\n  &lt;li&gt;Click &lt;code class=\&quot;highlighter-rouge\&quot;&gt;Save&lt;/code&gt; and then &lt;code class=\&quot;highlighter-rouge\&quot;&gt;Run&lt;/code&gt;&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h2 id=\&quot;how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot;&gt;How can I fix mistakes or expand an existing tutorial using the GitHub interface?&lt;/h2&gt;\n\n&lt;p&gt;Check our tutorial to learn how to use the GitHub interface (soon…)&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-contribute-in-advanced-mode\&quot;&gt;How can I contribute in “advanced” mode?&lt;/h2&gt;\n\n&lt;p&gt;Most of the content is written in &lt;a href=\&quot;https://guides.github.com/features/mastering-markdown/\&quot;&gt;GitHub Flavored Markdown&lt;/a&gt; with some metadata (or variables) found in &lt;a href=\&quot;http://yaml.org/\&quot;&gt;YAML&lt;/a&gt; files. Everything is stored on our &lt;a href=\&quot;https://github.com/vibbits/training-material\&quot;&gt;GitHub repository&lt;/a&gt;. Each training material is related to a topic. All training materials (slides, tutorials, etc) related to a topic are found in a dedicated directory (&lt;em&gt;e.g.&lt;/em&gt; &lt;code class=\&quot;highlighter-rouge\&quot;&gt;transcriptomics&lt;/code&gt; directory contains the material related to transcriptomic analysis). Each topic has the following &lt;a name=\&quot;structure\&quot;&gt;structure&lt;/a&gt;:&lt;/p&gt;\n\n&lt;p&gt;&lt;img src=\&quot;shared/images/repo_organization.png\&quot; alt=\&quot;Structure of the repository\&quot; width=\&quot;400px\&quot; /&gt;&lt;/p&gt;\n\n&lt;ul&gt;\n  &lt;li&gt;a metadata file in YAML format&lt;/li&gt;\n  &lt;li&gt;a directory with the topic introduction slide deck in Markdown with introductions to the topic&lt;/li&gt;\n  &lt;li&gt;\n    &lt;p&gt;a directory with the tutorials:&lt;/p&gt;\n\n    &lt;p&gt;Inside the tutorials directory, each tutorial related to the topic has its own subdirectory with several files:&lt;/p&gt;\n    &lt;ul&gt;\n      &lt;li&gt;a tutorial file written in Markdown with hands-on&lt;/li&gt;\n      &lt;li&gt;an optional slides file in Markdown with slides to support the tutorial&lt;/li&gt;\n      &lt;li&gt;a directory with Galaxy Interactive Tours to reproduce the tutorial&lt;/li&gt;\n      &lt;li&gt;a directory with workflows extracted from the tutoria&lt;/li&gt;\n      &lt;li&gt;a YAML file with the links to the input data needed for the tutorial&lt;/li&gt;\n      &lt;li&gt;a YAML file with the description of needed tools to run the tutorial&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;a directory with the Dockerfile describing the details to build a container for the topic (self-study environments).&lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;To manage changes, we use &lt;a href=\&quot;https://guides.github.com/introduction/flow/\&quot;&gt;GitHub flow&lt;/a&gt; based on Pull Requests:&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;&lt;a href=\&quot;https://help.github.com/articles/fork-a-repo/\&quot;&gt;Create a fork&lt;/a&gt; of this repository on GitHub&lt;/li&gt;\n  &lt;li&gt;Clone your fork of this repository to create a local copy on your computer and initialize the required submodules (&lt;code class=\&quot;highlighter-rouge\&quot;&gt;git submodule init&lt;/code&gt; and &lt;code class=\&quot;highlighter-rouge\&quot;&gt;git submodule update&lt;/code&gt;)&lt;/li&gt;\n  &lt;li&gt;Create a new branch in your local copy for each significant change&lt;/li&gt;\n  &lt;li&gt;Commit the changes in that branch&lt;/li&gt;\n  &lt;li&gt;Push that branch to your fork on GitHub&lt;/li&gt;\n  &lt;li&gt;Submit a pull request from that branch to the &lt;a href=\&quot;https://github.com/vibbits/training-material\&quot;&gt;original repository&lt;/a&gt;&lt;/li&gt;\n  &lt;li&gt;If you receive feedback, make changes in your local clone and push them to your branch on GitHub: the pull request will update automatically&lt;/li&gt;\n  &lt;li&gt;Pull requests will be merged by the training team members after at least one other person has reviewed the Pull request and approved it.&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;p&gt;Globally, the process of development of new content is open and transparent:&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;Creation of a branch derived from the master branch of the GitHub repository&lt;/li&gt;\n  &lt;li&gt;Initialization of a new directory for the tutorial&lt;/li&gt;\n  &lt;li&gt;Filling of the metadata with title, questions, learning objectives, etc&lt;/li&gt;\n  &lt;li&gt;Generation of the input dataset for the tutorial&lt;/li&gt;\n  &lt;li&gt;Filling of the tutorial content&lt;/li&gt;\n  &lt;li&gt;Extraction of the workflows of the tutorial&lt;/li&gt;\n  &lt;li&gt;Automatic extraction of the required tools to populate the tool file&lt;/li&gt;\n  &lt;li&gt;Automatic annotation of the public Galaxy servers&lt;/li&gt;\n  &lt;li&gt;Generation of an interactive tour for the tutorial with the &lt;a href=\&quot;https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/\&quot;&gt;Tourbuilder&lt;/a&gt; web-browser extension&lt;/li&gt;\n  &lt;li&gt;Upload of the datasets to Zenodo and addition of the links in the data library file.&lt;/li&gt;\n  &lt;li&gt;Once ready, opening a Pull Request&lt;/li&gt;\n  &lt;li&gt;Automatic checks of the changes are automatically checked for the right format and working links using continuous integration testing on Travis CI&lt;/li&gt;\n  &lt;li&gt;Review of the content by several other instructors via discussions&lt;/li&gt;\n  &lt;li&gt;After the review process, merge of the content into the main branch, starting a series of automatic steps triggered by Travis CI&lt;/li&gt;\n  &lt;li&gt;Regeneration of the website and publication on &lt;a href=\&quot;/training-material/\&quot;&gt;https://vibbits.github.io//training-material/&lt;/a&gt;&lt;/li&gt;\n  &lt;li&gt;Generation of PDF artifacts of the tutorials and slides and upload on the FTP server&lt;/li&gt;\n  &lt;li&gt;Population of &lt;a href=\&quot;https://tess.elixir-europe.org/\&quot;&gt;TeSS&lt;/a&gt;, the ELIXIR’s Training Portal, via the metadata&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;p&gt;&lt;img src=\&quot;shared/images/development_process.png\&quot; alt=\&quot;Development process\&quot; /&gt;&lt;/p&gt;\n\n&lt;ol&gt;\n\n\n &lt;/ol&gt;\n\n&lt;p&gt;We also strongly recommend you read and follow &lt;a href=\&quot;https://carpentries.org/\&quot;&gt;The Carpentries&lt;/a&gt; recommendations on &lt;a href=\&quot;https://carpentries.github.io/lesson-example/01-design/\&quot;&gt;lesson design&lt;/a&gt; and &lt;a href=\&quot;https://carpentries.github.io/instructor-training/15-lesson-study/\&quot;&gt;lesson writing&lt;/a&gt; if you plan to add or change some training materials, and also to check the &lt;a href=\&quot;#structure\&quot;&gt;structure of the training material&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;other-questions\&quot;&gt;Other Questions&lt;/h1&gt;\n\n&lt;p&gt;Yes, always! Have a look at the &lt;a href=\&quot;https://galaxyproject.org/events/\&quot;&gt;Galaxy Community Events Calendar&lt;/a&gt; for what coming up right now.&lt;/p&gt;\n\n&lt;h2 id=\&quot;sustainability-of-the-training-material-and-metadata\&quot;&gt;Sustainability of the training-material and metadata&lt;/h2&gt;\n\n&lt;p&gt;This repository is hosted on &lt;a href=\&quot;https://github.com/\&quot;&gt;GitHub&lt;/a&gt; using git as a &lt;a href=\&quot;https://en.wikipedia.org/wiki/Distributed_version_control\&quot;&gt;DVCS&lt;/a&gt;. Therefore the community is hosting backups of this\nrepository in a decentralised way. The repository is self-contained and contains all needed content and all metadata.\nIn addition we mirror snapshops of this repo on &lt;a href=\&quot;http://zenodo.org\&quot;&gt;Zenodo&lt;/a&gt;.&lt;/p&gt;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;### How to fill the slide decks?\n\nPlease follow our\n[tutorial to learn how to fill the slides](/training-material/topics/contributing/tutorials/create-new-tutorial-slides/slides.html)\n&quot;,&quot;## Protein Structure Analysis ##\n\n- Sequences, structures and databases\n- Experimental methods (X-rays, electrons and NMR)\n- Finding and visualising structures from the  Protein Data Bank\n- Comparing structures\n- Modelling mutations\n- Creating homology models\n\n---\n\n## The Structure-Function Connection ##\n\n.left[![Left-aligned image](/training-material/topics/protein-structure-analysis/images/hemoglobin.png)]\n.right[Folded proteins provide a well-  defined 3D  arrangement of functional groups, creating microenvironments and active sites.\nStructural changes  are often involved in functional  mechanisms  (motor proteins, ...)\n]\n\n---\n\n### Databases\n\n.pull-left[\n**Uniprot** approx. 1100,000 sequences\n\n- mainly determined by large-scale DNA sequencing of individual genes or whole genomes\n- increasingly automated annotation\n\n**Protein Data Bank** approx. 141,000 \nexperimentally determined structures\n\n- Mainly determined by X-ray crystallography and high-resolution NMR spectroscopy\n- Automation is increasing, but there are still significant limitations in the rate of solving new structures\n\n\n]\n.pull-right[ ![](/training-material/assets/images/GTN-60px.png) ]\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### The Protein Databank ###\n\n.pull-left[ ![](/training-material/assets/images/GTN-60px.png) ] \n\n\n.pull-right[\n[http://www.wwpdb.org](http://www.wwpdb.org)\n- contains structures of  proteins, nucleic acids  and complexes,  determined by X-ray  crystallography, NMR  spectroscopy\n- No purely theoretical  or ab initio models  (since 2006)\n- Also stores supporting  experimental data\n- Full deposition now  required by all peer-reviewed journals\n\n---\n\n### Exercise 1: Search the PDB ###\n\n- Use the UniProt site to search for “dnak”.  \n- Use the PDB site to search for “dnak”.\n- Compare the UniProt and PDB result lists.\n- Use the sequence search function to see if there are structures with sequences similar to that of the  DnaK C-terminal domain.\n- Look at the summary pages of a number of  structures and note some interesting properties.\n- Select a number of structures and create a report  page.\n\n---\n\n### PDB File Organisation ###\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- Nucleic Acid Database: DNA and RNA structures\n\n[http://ndbserver.rutgers.edu/](http://ndbserver.rutgers.edu/)\n\n- PDB-REDO: automatically re-refined deposited  structures, using the latest methods\n\n[http://www.cmbi.ru.nl/pdb_redo/](http://www.cmbi.ru.nl/pdb_redo)\n\n- EBI: many online tools for structure analysis\n\n[http://www.ebi.ac.uk/Tools/structure/](http://www.ebi.ac.uk/Tools/structure/)\n\n- Replaced Electron Density Server: convenient overview of  quality parameters for crystal structures\n\n[http://www.ebi.ac.uk/pdbe/litemol](http://www.ebi.ac.uk/pdbe/litemol)\n\n---\n\n### High-Resolution NMR Spectrometry ###\n\nMany atomic nuclei, including the ubiquitous hydrogen nuclei,  resonate at specific radio frequencies when placed in a  strong, uniform magnetic field. The chemical environment of each individual atom slightly modulates its exact resonance  frequency.\n\n![]()\n\nIn macromolecules with thousands of atoms, many different  effects combine to generate an extremely complicated pattern of chemical shifts, which therefore more or less uniquely  identify each atom. Multidimensional spectra allow these  frequencies to be assigned to specific atoms.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nWhen two atoms are near each other in 3D space, they can exchange magnetization, giving rise to crosspeaks at the  intersection of their respective frequencies.\n\n![]()\n\nThis nuclear Overhauser effect (NOE) is used to effectively measure the distances between pairs of atoms, at least  qualitatively.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nAfter identification of the atoms by means of their unique chemical shifts, distance restraints are derived from the  Overhauser crosspeaks. An extended model of the protein  is  generated,  and  then  condensed  into  a  shape  that  is consistent with as many of distance restraints as possible.\n\n![]()\n\n---\n\n### Other methods ###\n\nElectron microscopy (and especially Cryo-electron microscopy): Electron crystallography and single particle reconstruction\r\n\nSmall-angle X-ray and neutron scattering (SAXS and SANS)\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- BioMagResBank: experimental data for NMR-  derived structures (lists of distance restraints  and other experimentally derived properties)\n\n[http://www.bmrb.wisc.edu/](http://www.bmrb.wisc.edu/)\n\n- BioIsis: database of SAXS-derived structures\n\n[http://www.bioisis.net](http://www.bioisis.net)\n\n- EMBL database of SAXS-derived structures\n\n[http://www.sasbdb.org](http://www.sasbdb.org)\n\n- EM Databank for cryo-EM structures\n\n[http://www.emdatabank.org](http://www.emdatabank.org)\n\n---\n\n### Assessing Structure Quality ###\n\nGeneral geometric properties (bond lengths and angles,  Ramachandran distribution, …): MolProbity  [Link](http://molprobity.biochem.duke.edu/)\n\nCrystal Structures\n- Diffraction data resolution and completeness (PDB)\n- Final Rcryst and Rfree factors (PDB)\n- Local correspondence to electron density (EDS)\n\nNMR Structures\n- Number of distance restraints and additional experimental  data sources (BMRB)\n- Local restraint density (on-line NMR constraint analyser)\n[link](http://molsim.sci.univr.it/bioinfo/tools/constraint/index.html)\n\nOther techniques\n- Difficult to generalise: carefully read and consider the  published protocol\n\n---\n\n### Molecular Graphics Software ###\n\n- [PyMOL](http://www.pymol.org/): high-quality output,  good examples on [Wiki](http://www.pymolwiki.org/)\n- [Chimera](http://www.cgl.ucsf.edu/chimera/): good  documentation on website\n- [VMD](http://www.ks.uiuc.edu/Research/vmd/):  excellent for the analysis of MD trajectories\n- [Yasara](http://www.yasara.org)  \n- [SwissPDBViewer](http://spdbv.vital-it.ch/)\n\n---\n\n### YASARA ###\n\nYasara View is freely available and provides  basic visualisation functions.\n\nYasara Model, Dynamics and Structure provide  additional functionality.\n\nAn add-on module for NMR structures is  available.\n\nThe program can be combined with the WHAT-  IF program for structure validation, and with  FoldX for energy calculations.\n\n---\n\n### Exercise 2: show a structure ###\n\nLoad PDB entry 1TRZ using the File menu.\n\nCompare the default representations (F1-F8)  and use the various controls to change the view  of the protein.\n\nExplore the Edit and View menus to change  various aspects of the graphical representation  of the structure.\n\nExamine the hydrogen bonding patterns.  Display a molecular surface.\n\nCreate an interesting view of the molecule and  save it as a high-resolution image.\n\n---\n\n### Structure Superposition ###\n\nStructures can be superimposed to achieve the best  possible match between corresponding atoms. It is  possible to consider all atoms, or only a subset  (such as the Cα atoms).\n\nWhen the structures are very similar, determining  which atoms should be matched to each other is  trivial. When there are larger differences, it takes  more preparation.\n\nDifferent algorithms use different combinations of  sequence- and secondary of tertiary structure-based  information.\n\n---\n\n### Exercise 3: Compare Structures ###\n\nDownload the five provided PDB files and open  them in Yasara.\n\nUse the `Analyze|Align|Objects` with MUSTANG  function to align the four last objects with the first one.\n\nUse the space bar to open the text console and  see the reported root mean square deviations\n\n\nas well as the number of residues matched.\n\nColor all structures by B factor and compare the  distribution to the local variability of the structures.\n\n---\n\n### PDB File Surprises ###\n\n- Missing atoms, residues and loops  \n- Multiple molecules in the asymmetric unit\n- Incomplete oligomers due to coincident crystal and  oligomer symmetry\n- Cα-only models  lternate conformations\n- Multiple models, especially for NMR ensembles\n- Use of B factors to represent other properties  ther non-standard extensions (PDBQT, ...)\n\n---\n\n### Force Fields ###\n\n- Energy terms representing physical interactions\n\nCovalent bond lengths and angles\nDihedral angles and van der Waals forces (steric effects)\nElectrostatic interactions and hydrogen bonds\n…\n\n- Energy can be minimized, and forces can literally be derived from the potential function.\n\n- Consistency and careful consideration of the  properties to be simulated are essential.\n\n---\n\n### Force Field Terms ###\n\nEach energy term has a functional form, which includes one or more parameters:\n\n- Covalent bond energy term\n To do: add formula\n- Van der Waals contact energy term\n\nThe parameters are collectively optimized to  reproduce a chosen set of experimentally  observed parameters.\n\nA given force field should be used as a  consistent system, and can only be used to  predict properties that are covered by the  training set.\n\n---\n\n### FoldX ###\n\n\ns designed for quantitative modelling of the  contribution of structural interactions to the stability of  proteins and protein complexes. It also supports  protein/DNA complexes.\n\nThe force field describes the different interactions in a  protein structure or complex in analytical terms. It has  been calibrated using a set of experimentally determined  stabilities.\n\nApplications include the optimisation of structures, the calculation of the stability of complexes, and predicting  the effect of amino-acid or base-pair mutations on these  properties.\n\n---\n\n### The FoldX Plugin for Yasara\t###\n\nIn order to make FoldX more accessible and  integrate its functions into Yasara, dr. Joost van  Durme (SWITCH laboratory) made a Yasara plugin  module that can apply FoldX functions to structures  that are loaded as Yasara objects.\nThis greatly simplifies the use of FoldX, and allows  for a quick visual analysis of the resulting changes  in the structures.\nMore information can be found at [wiki](http://foldxyasara.switchlab.org/index.php/) [FoldX](http://foldxsuite.crg.eu/)\n\n---\n\n### Exercise 4a: Repair a PDB File ###\n\n- Load the 1CRN PDB file.\n- Use the “Repair object” option in the Analysis |  FoldX menu to activate the corresponding FoldX  function.\n- Select the object to repair.\n\nThis exports the object as a temporary PDB file,  starts FoldX with the appropriate options, and loads  the repaired PDB file as a new object in Yasara.\n\n- Compare the original and repaired objects.  \n- Describe the changes that were introduced.\n\n---\n\n### Exercise 4b: Model a Mutation ###\n\n- Load the 2AC0.sce Yasara scene file.\n- Set an appropriate structure representation.\n- Locate residue Ala159 using the sequence view,  and right-click to access the `FoldX|Mutate` residue function. Change it to a Trp residue.\n- Look at the effect of the substitution on the  structure, and use the space bar to open the text  console and read the output of the FoldX  calculation.\n- Mutate Arg273 to an alanine side chain. Discuss  the effects of this substitution.\n\n---\n\n### Homology Modelling ###\n\n- When a structure is available for a protein with a  similar sequences, it is possible to predict the  structure of a new sequence with varying degrees of  confidence.\n- All homology modelling procedures start from an  alignment of the template and target sequences.  The quality of this alignment will have a major  impact on the resulting model.\n- Available applications include stand-alone programs  (Modeller, FoldX, …) and web-based services (such as SwissModel).\n\n---\n\n### Exercise 5: Make a Homology  Model using Swiss Model ###\n\nsee []()\n\n---\n\n### Exercise 6: Study Protein-Ligand Interactions ###\n\nsee []()\n\n---\n\n### On-Line References ###\n\n- Crystallography 101 (Bernhard Rupp):  (http://www.ruppweb.org/Xray/101index.html)\n- Protein NMR, A Practical Guide (Vicky Higman) ( http://www.protein-nmr.org.uk/)\n- Model validation course  (http://xray.bmc.uu.se/gerard/embo2001/modval/index.html)\n- Assessing model quality (http://spdbv.vital-it.ch/TheMolecularLevel/ModQual/)\n\n- Lectures by Burkhard Rost on protein structure  prediction (https://www.youtube.com/channel/UCU6j8BG4RbEtTgyIZJ6Vpow)\n\n---\n&quot;,&quot;$font-size: 15px;\n$tutorial-box-spacing: 1rem;\n\n// palette\n$white: #fff;\n$gray: #777;\n$light-gray: #ddd;\n$text-color: #292b2c;\n\n$red: #f00000;\n$orange: #f06d00;\n$yellow: #be9900;\n$green: #339e3a;\n$blue: #009090;\n$dark-blue: #1b2944;\n$light-blue: #5daadd;\n$purple: #882d60;\n$mid-gray: #9d9d9d;\n\n// colors general\n$bg-color: $dark-blue;\n$menu-color: $dark-blue;\n\n// colors tutorial boxes\n$overview-color: #8A9AD0;\n$agenda-color: #86D486;\n$keypoints-color: #FFA1A1;\n$tip-color: #FFE19E;\n$warning-color: #de8875;\n$comment-color: #ffecc1;\n$handson-color: #dfe5f9;\n$question-color: #8A9AD0;\n$solution-color: #B8C3EA;\n$details-color: $light-gray;\n$feedback-color: #86D486;\n\n\n@mixin matrix-table (){\n  table {\n      width:unset;\n      margin-left:auto;\n      margin-right:auto;\n  }\n\n  table thead th {\n      border-bottom: 2px solid #777;\n      text-align: left;\n  }\n\n  table td:first-child, table th:first-child  {\n      border-right: 2px solid #777;\n      font-weight: bold;\n      text-align: left;\n  }\n\n  table td:not(first-child), table th:not(first-child) {\n      text-align: center;\n      border-right: 1px solid #ddd;\n      border-left: 1px solid #ddd;\n  }\n}\n\n@mixin tutorial-box ($bg-color, $color: rgba(255,255,255,.75)) {\n    margin-top: 3 * $tutorial-box-spacing;\n    padding: $tutorial-box-spacing $tutorial-box-spacing - 0.11rem $tutorial-box-spacing;\n    border: 1px solid $bg-color;\n    box-shadow: 3px 3px $bg-color;\n    &amp; &gt; h3 {\n        font-size: $font-size + 3;\n        margin-left: -$tutorial-box-spacing;\n        margin-right: -$tutorial-box-spacing;\n        margin-top: -3.90rem;\n        padding-top: $tutorial-box-spacing / 2;\n        padding-bottom: $tutorial-box-spacing / 2;\n        padding-left: $tutorial-box-spacing;\n        padding-right: $tutorial-box-spacing;\n        background-color: $bg-color;\n        color: $color;\n        vertical-align: 1em;\n        float: left;\n        border-top-left-radius: 8px;\n        border-top-right-radius: 8px;\n\n        .fa {\n            padding-right: $tutorial-box-spacing / 5;\n        }\n    }\n}\n\n@mixin matrix-table (){\n  table {\n      width:unset;\n      margin-left:auto;\n      margin-right:auto;\n      margin-top: 20pt;\n  }\n\n  table thead th {\n      border-bottom: 2px solid #777;\n      text-align: left;\n  }\n\n  table td:first-child, table th:first-child  {\n      border-right: 2px solid #777;\n      font-weight: bold;\n      text-align: left;\n  }\n\n  table td:not(first-child), table th:not(first-child) {\n      text-align: center;\n      border-right: 1px solid #ddd;\n      border-left: 1px solid #ddd;\n  }\n}\n\n\nbody {\n    font-size: $font-size;\n    word-wrap: break-word;\n    position: relative;\n}\n\nh1,\nh2,\nh3,\nh4 {\n    margin-top: 2rem;\n}\n\nimg {\n    max-width: 95%;\n    margin: 2.5%;\n}\n\npre {\n    padding: 1rem;\n    background-color: lighten($gray, 40%);\n}\n\nfigure {\n  text-align: center;\n  margin: 1rem 2rem;\n\n  &amp; &gt; img {\n      margin-bottom: 1rem;\n  }\n\n  .figcaption-prefix {\n      font-weight: 600;\n  }\n\n  figcaption {\n    text-align: justify;\n  }\n}\n\nfooter {\n    margin-top: 3em;\n    text-align: center;\n    color: $gray;\n    font-size: $font-size - 2;\n}\n\n.follow-up {\n    p {\n        font-size: 18px;\n    }\n}\n\n.navbar {\n    background-color: $bg-color!important;\n\n    .navbar-collapse {\n        justify-content: flex-end;\n        align-items: flex-end;\n        min-width: 0;\n    }\n}\n\n.main-content {\n    padding-top: 1rem;\n}\n\n.table {\n    td {\n        vertical-align: middle;\n    }\n\n    .fa {\n        font-size: $font-size + 5;\n    }\n\n}\n\n.topic-icon {\n    padding: .5rem .75rem;\n}\n\n.training-network-map {\n    width: 90%;\n    margin: 0 auto;\n}\n\n.tutorial {\n    &amp; &gt; h1:first-child {\n        margin-bottom: 1rem;\n    }\n\n    table {\n        display: block;\n        width: 100%;\n        overflow-x: auto;\n        max-width: 100%;\n        margin-bottom: 1rem;\n\n        th,\n        td {\n            padding: .75rem;\n            vertical-align: top;\n            border-top: 1px solid #eceeef;\n        }\n\n        thead th {\n            vertical-align: bottom;\n            border-bottom: 2px solid #eceeef;\n        }\n    }\n\n    // tutorial boxes\n    blockquote {\n        ul,\n        ol,\n        &amp; &gt; blockquote,\n        p {\n            margin-bottom: 0px;\n        }\n\n        &amp;.agenda {\n            @include tutorial-box($agenda-color, $text-color);\n        }\n\n        &amp;.tip {\n            @include tutorial-box($tip-color, $text-color);\n        }\n\n        &amp;.question {\n            @include tutorial-box($question-color, $text-color);\n        }\n\n        &amp;.solution {\n            @include tutorial-box($solution-color, $text-color);\n        }\n\n        &amp;.comment {\n            @include tutorial-box($comment-color, $text-color);\n        }\n\n        &amp;.hands_on {\n            @include tutorial-box($handson-color, $text-color);\n        }\n\n        &amp;.key_points {\n            @include tutorial-box($keypoints-color, $text-color);\n        }\n\n        &amp;.overview {\n            @include tutorial-box($overview-color, $text-color);\n        }\n\n        &amp;.warning {\n            @include tutorial-box($warning-color, $text-color);\n        }\n\n        &amp;.details {\n            @include tutorial-box($details-color, $text-color);\n        }\n\n        &amp;.quote {\n            border-left: 5px solid #eee;\n            padding-left: 1em;\n        }\n\n        &amp;.matrix {\n            @include matrix-table();\n        }\n\n        &amp;.feedback{\n            @include tutorial-box($feedback-color, $text-color);\n        }\n    }\n\n    .fold-unfold {\n        margin-left: $tutorial-box-spacing;\n    }\n\n    .google-form {\n        width: 100%;\n        height: 1100px;\n        border: 0;\n    }\n}\n\n.contributors {\n    padding: 0;\n\n    .carousel-item {\n        align-items: flex-start;\n        flex-wrap: wrap;\n        flex-direction: row;\n        justify-content: space-around;\n    }\n\n    .carousel-item.active,\n    .carousel-item-next,\n    .carousel-item-prev {\n        display: flex;\n    }\n\n    .hall-of-fame-hero {\n        margin: 0;\n        width: 20%;\n\n        .name {\n            font-size: 0.7em;\n        }\n    }\n}\n\n.contributors-line {\n    color: $gray;\n    font-size: 1.2em;\n    margin-bottom: 5rem;\n}\n\n.hall-of-fame {\n    display: flex;\n    flex-wrap: wrap;\n    flex-direction: row;\n    justify-content: space-around;\n}\n\n.hall-of-fame-hero {\n    width: 140px;\n    margin: 1rem;\n    text-align: center;\n\n    .name {\n        margin-top: 0.5rem;\n    }\n\n    .contact-items {\n        margin-top: 0.2rem;\n    }\n\n    .thumbnail img {\n        background-color: white;\n        border-radius: 20px;\n        border: 1px solid $light-gray;\n        padding: 5px;\n        width: 100%;\n        -webkit-transition: border 0.2s ease-in-out;\n        -o-transition: border 0.2s ease-in-out;\n        transition: border 0.2s ease-in-out;\n    }\n\n    a.thumbnail {\n        &amp;:hover,\n        &amp;:focus,\n        &amp;:active {\n            img {\n                border-color: $menu-color;\n            }\n        }\n    }\n}\n\n// This is a fix to get the hamburger button (menu/navbar) correctly placed on\n// small screens.\n@media (max-width: 575px) {\n    .navbar &gt; .container {\n        position: initial;\n        margin-left: initial;\n        margin-right: initial;\n    }\n}\n\n// This applies a \&quot;dark\&quot; style to the dropdown menus on small devices since\n// Bootstrap v4 does not do that by default.\n@media (max-width: 768px) {\n    .navbar {\n        .dropdown-menu {\n            background-color: transparent;\n            border-color: transparent;\n        }\n\n        .dropdown-item {\n            color: rgba($white, .5);\n\n            &amp;:hover,\n            &amp;:focus,\n            &amp;:active {\n                background-color: transparent;\n                color: rgba($white, .75);\n            }\n        }\n\n        .nav-item,\n        .dropdown-item {\n            overflow: hidden;\n            white-space: nowrap;\n            text-overflow: ellipsis;\n        }\n    }\n}\n\n@media (min-width: 768px) {\n    .tutorial table,\n    .table-responsive {\n        display: table;\n    }\n\n    .citation {\n        text-align: justify;\n    }\n}\n\n@media (max-width: 1200px) {\n    .navbar-nav i {\n        display: none;\n    }\n}\n\n.label {\n    display: inline;\n    padding: .2em .6em .3em;\n    font-size: 75%;\n    font-weight: bold;\n    line-height: 1;\n    color: #fff;\n    text-align: center;\n    white-space: nowrap;\n    vertical-align: baseline;\n    border-radius: .25em;\n}\n\n.label-default {\n    background-color: #999999;\n}\n\n.level {\n    color: lightgray;\n    margin-right: 0.5em;\n\n    .fa {\n        font-size: 0.9em;\n    }\n    &amp;.introductory i:first-of-type {\n        color: green;\n    }\n    &amp;.intermediate i:not(:last-of-type) {\n        color: orange;\n    }\n    &amp;.advanced {\n        color: red;\n    }\n}\n\n\n\n.visually-hidden {\n    border: 0;\n    clip: rect(0 0 0 0);\n    height: 1px;\n    margin: -1px;\n    overflow: hidden;\n    padding: 0;\n    position: absolute;\n    width: 1px;\n}\n\n\n/* WCAG Contrast. These aim to pass AA level for normal sized text. */\na {\n    color: #0070e8;\n\n    &amp;:hover,\n    &amp;:focus,\n    &amp;:active {\n        color: #004590;\n    }\n}\n\ncode {\n    color: #8b3035;\n}\n\n\nnav[data-toggle=&#39;toc&#39;] {\n    margin-top: 30px;\n    margin-bottom: 30px;\n    .nav {\n        li {\n            a {\n                padding-left: 0px;\n                &amp;:hover,\n                &amp;:focus,\n                &amp;:active{\n                    padding-left: 0px;\n                    color: #004590;\n                    border-left: none;\n                }\n            }\n\n            .active {\n                padding-left: 0px;\n                color: #004590;\n                border-left: none;\n            }\n        }\n\n        .nav li {\n            a,\n            .active {\n                padding-left: 10px;\n                &amp;:hover,\n                &amp;:focus,\n                &amp;:active{\n                    padding-left: 10px;\n                }\n            }\n\n            &amp;:hover,\n            &amp;:focus,\n            &amp;:active{\n                padding-left: 10px;\n            }\n        }\n    }\n\n    .nav-link.active:hover {\n        padding-left: 0px;\n        color: #004590;\n        border-left: none;\n    }\n}\n\n.col-sm-2 {\n    padding-left: 0px;\n}\n\n/* small screens */\n@media screen and (max-width: 768px) {\n    nav[data-toggle=&#39;toc&#39;] {\n        display: none !important;\n    }\n\n    .col-sm-10 {\n        max-width: 100%;\n    }\n}\n\n.contributor-badge {\n    /* prevent breaking across lines */\n    white-space: nowrap;\n\n    img {\n        height: 1.25em;\n        border-radius: 50%;\n        margin: 0.25em;\n    }\n}\n\n\ndiv.highlight {\n\tposition: relative;\n}\ndiv.highlight .btn{\n\t-webkit-transition:opacity .3s ease-in-out;\n\t-o-transition:opacity .3s ease-in-out;\n\ttransition:opacity .3s ease-in-out;\n\topacity:0;\n\tpadding:2px 6px;\n\tposition:absolute;\n\tright:4px;\n\ttop:4px;\n\tfont-family: -apple-system,BlinkMacSystemFont,\&quot;Segoe UI\&quot;,Roboto,\&quot;Helvetica Neue\&quot;,Arial,sans-serif;\n}\n\ndiv.highlight:hover .btn,div.highlight .btn:focus{\n\topacity:1\n}\n\n#supporting_materials ul li {\n    div {\n        display: flex;\n        flex-direction: row;\n        align-items: center;\n\n        div {\n            a.topic-icon {\n                padding: 0 .5rem 0 .5rem;\n            }\n\n            a.topic-icon.btn {\n                padding-top: 0;\n                padding-bottom: 0;\n            }\n        }\n\n        div:first-child {\n            a.topic-icon {\n                padding-left: 0;\n            }\n        }\n\n        div:not(:first-child):before {\n            content: \&quot; - \&quot;;\n        }\n    }\n}\n&quot;,&quot;.enlarge120[\n\n# ***De novo* Genome Assembly**\n\n]\n\n#### With thanks to T Seemann, D Bulach, I Cooke and Simon Gladman\n---\n.enlarge120[\n\n# ***De novo* assembly**\n\n]\n\n.pull-left[\n\n**The process of reconstructing the original DNA sequence from the fragment reads alone.**\n\n* Instinctively like a jigsaw puzzle\n\n  * Find reads which \&quot;fit together\&quot; (overlap)\n  * Could be missing pieces (sequencing bias)\n  * Some pieces will be dirty (sequencing errors)\n\n]\n\n.pull-right[ ![](../../images/Humpty.jpg) ]\n\n---\n\n# **Another View**\n\n![](../../images/newspaper.png)\n\n---\n\n# **Assembly: An Example**\n\n---\n\n# **A small \&quot;genome\&quot;**\n\n![](../../images/shakespear1.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear2.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear3.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear4.png)\n\n---\n\n# **So far, so good!**\n\n---\n\n# **The Awful Truth**\n\n![](../../images/notsimply.png)\n\n## \&quot;Genome assembly is impossible.\&quot; - A/Prof. Mihai Pop\n\n---\n.enlarge120[\n\n# **Why is it so hard?**\n\n]\n\n.pull-left[\n* Millions of pieces\n  * Much, much shorter than the genome\n  * Lots of them look similar\n* Missing pieces\n  * Some parts can&#39;t be sequenced easily\n* Dirty Pieces\n  * Lots of errors in reads\n]\n\n.pull-right[ ![](../../images/worlds_hardest.png) ]\n\n---\n\n# **Assembly recipe**\n\n* Find all overlaps between reads\n  * Hmm, sounds like a lot of work..\n* Build a graph\n  * A picture of the read connections\n* Simplify the graph\n  * Sequencing errors will mess it up a lot\n* Traverse the graph\n  * Trace a sensible path to produce a consensus\n\n---\n\n![](../../images/olc_pic.png)\n\n---\n\n# **A more realistic graph**\n\n![](../../images/real_graph.png)\n\n---\n\n# .image-15[![](../../images/nofun.png)] **What ruins the graph?**\n\n* Read errors\n  * Introduces false edges and nodes\n\n* Non haploid organisms\n  * Heterozygosity causes lots of detours\n\n* Repeats\n  * If they are longer than the read length\n  * Causes nodes to be shared, locality confusion.\n\n---\n\n# **Repeats**\n\n---\n.enlarge120[\n# **What is a repeat?**\n]\n\n.pull-left[\n\n#### ***A segment of DNA which occurs more than once in the genome sequence***\n\n* Very common\n  * Transposons (self replicating genes)\n  * Satellites (repetitive adjacent patterns)\n  * Gene duplications (paralogs)\n\n]\n\n.pull-right[\n\n![](../../images/triplets.png)\n\n]\n\n---\n\n# **Effect on Assembly**\n\n![](../../images/repeat_effect.png)\n\n---\n.enlarge120[\n# **The law of repeats** .image-15[![](../../images/repeatafterme.png)]\n]\n\n## **It is impossible to resolve repeats of length S unless you have reads longer than S**\n\n## **It is impossible to resolve repeats of length S unless you have reads longer than S**\n\n---\n\n# **Scaffolding**\n\n---\n.enlarge120[\n# **Beyond contigs**\n]\n\n.pull-left[\n\nContig sizes are limited by:\n\n* the length of the repeats in your genome\n  * Can&#39;t change this\n\n\n* the length (or \&quot;span\&quot;) of the reads\n  * Use long read technology\n  * Use tricks with other technology\n\n]\n\n---\n.enlarge120[\n# **Types of reads**\n]\n\n.pull-left[.enlarge120[**Example fragment**]]\n\n\n.remark-code[.enlarge120[atcgtatgatcttgagattctctcttcccttatagctgctata]]\n\n.pull-left[.enlarge120[**\&quot;Single-end\&quot; read**]]\n\n\n.remark-code[.enlarge120[**atcgtatg**atcttgagattctctcttcccttatagctgctata]]\n\nsequence *one* end of the fragment\n\n.pull-left[.enlarge120[**\&quot;Paired-end\&quot; read**]]\n\n\n.remark-code[.enlarge120[**atcgtatg**atcttgagattctctcttcccttatag**ctgctata**]]\n\nsequence both ends of the same fragment\n\n**We can exploit this information!**\n---\n\n.enlarge120[# **Scaffolding**]\n\n* **Paired end reads**\n  * Known sequences at each end of fragment\n  * Roughly known fragment length\n\n* **Most ends will occur in same contig**\n\n* **Some will occur in different contigs**\n  * ***evidence that these contigs are linked***\n---\n\n.enlarge120[# **Contigs to Scaffolds**]\n\n![](../../images/scaffolding.png)\n\n---\n\n.enlarge120[# **Assessing assemblies**]\n\n* We desire\n  * Total length similar to genome size\n  * Fewer, larger contigs\n  * Correct contigs\n\n* Metrics\n  * No generally useful measure. (No real prior information)\n  * Longest contigs, total base pairs in contigs, **N50**, ...\n\n---\n\n.enlarge120[# **The \&quot;N50\&quot;**]\n\n.enlarge120[***The length of that contig from which 50% of the bases are in it and shorter contigs***]\n\n* Imagine we have 7 contigs with lengths:\n  * 1, 1, 3, 5, 8, 12, 20\n\n* Total\n  * 1+1+3+5+8+12+20 = 50\n\n* N50 is the \&quot;halfway sum\&quot; = 25\n  * 1+1+3+5+8+**12** = 30 (&gt;25) so **N50 is 12**\n\n---\n\n.enlarge120[# **2 levels of assembly**]\n\n* Draft assembly\n  * Will contain a number of non-linked scaffolds with gaps of unknown sequence\n  * Fairly easy to get to\n\n* Closed (finished) assembly\n  * One sequence for each chromosome\n  * Takes a **lot** more work\n  * Small genomes are becoming easier with long read tech\n  * Large genomes are the province of big consortia (e.g. Human Genome Consortium)\n\n---\n.enlarge120[# **How do I do it?**]\n---\n.enlarge120[\n# **Example**\n\n* Culture your bacterium\n\n\n* Extract your genomic DNA\n\n\n* Send it to your sequencing centre for Illumina sequencing\n  * 250bp paired end\n\n\n* Get back 2 files\n  * .remark-code[MRSA_R1.fastq.gz]\n  * .remark-code[MRSA_R2.fastq.gz]\n\n\n* ***Now what?***\n]\n\n---\n.enlarge120[# **Assembly tools**\n\n* **Genome**\n  * **Velvet, Velvet Optimizer, Spades,** Abyss, MIRA, Newbler, SGA, AllPaths, Ray, SOAPdenovo, ...\n\n\n* Meta-genome\n  * Meta Velvet, SGA, custom scripts + above\n\n\n* Transcriptome\n  * Trinity, Oases, Trans-abyss\n\n***And many, many others...***\n\n]\n\n---\n.enlarge120[\n# **Assembly Exercise #1**\n\n* We will do a simple assembly using **Velvet** in **Galaxy**\n* We can do a number of different assemblies and compare some assembly metrics.\n\n]\n&quot;,&quot;# Genome assembly with Velvet: Background\n{:.no_toc}\n\nVelvet is one of a number of *de novo* assemblers that use short read sets as input (*e.g.* Illumina Reads). The assembly method is based on the manipulation of de Bruijn graphs, via the removal of errors and the simplication of repeated regions.\n\n&gt; ### {% icon comment %} Comment\n&gt;\n&gt; For information about Velvet, you can check its (nice) [Wikipedia page](https://en.wikipedia.org/wiki/Velvet_assembler).\n{: .comment}\n\nFor this tutorial, we have a set of reads from an imaginary *Staphylococcus aureus* bacterium with a miniature genome (197,394 bp). Our mutant strain read set was sequenced with the whole genome shotgun method, using an Illumina DNA sequencing instrument. From these reads, we would like to rebuild our imaginary *Staphylococcus aureus* bacterium via a *de novo* assembly of a short read set using the Velvet assembler.\n\n&gt; ### Agenda\n&gt;\n&gt; In this tutorial, we will deal with:\n&gt;\n&gt; 1. TOC\n&gt; {:toc}\n&gt;\n{: .agenda}\n\n# Get the data\n\nWe will now import the data that we will use for the tutorial.\n\n&gt; ### {% icon hands_on %} Hands-on: Getting the data\n&gt;\n&gt; 1. Create and name a new history for this tutorial.\n&gt; 2. Import from [Zenodo](https://doi.org/10.5281/zenodo.582600) or from the data library the files:\n&gt;    - [`mutant_R1.fastq`](https://zenodo.org/record/582600/files/mutant_R1.fastq)\n&gt;    - [`mutant_R2.fastq`](https://zenodo.org/record/582600/files/mutant_R2.fastq)\n&gt;\n&gt;    &gt; ### {% icon tip %} Tip: Importing data via links\n&gt;    &gt;\n&gt;    &gt; * Copy the link location (Right-click on the filename then \&quot;Copy Link Address\&quot;)\n&gt;    &gt; * Open the Galaxy Upload Manager\n&gt;    &gt; * Select **Paste/Fetch Data**\n&gt;    &gt; * Paste the link into the text field\n&gt;    &gt; * Change the data-type to **fastqsanger**\n&gt;    &gt; * Press **Start**\n&gt;    {: .tip}\n&gt;\n&gt; 3. Change the name of the files to `mutant_R1` and `mutant_R2`.\n&gt;\n&gt;    As a default, Galaxy uses the link as the name of the new dataset. It also does not link the dataset to a database or a reference genome.\n&gt;\n&gt;    {% include snippets/rename_dataset.md %}\n&gt;\n&gt; 4. Inspect the content of a file.\n&gt;\n&gt;    &gt; ### {% icon tip %} Tip: Inspecting the content of a dataset\n&gt;    &gt;\n&gt;    &gt; * Click on the {% icon galaxy-eye %} (eye) icon next to the relevant history entry\n&gt;    &gt; * View the content of the file in the central panel\n&gt;    {: .tip}\n&gt;\n&gt;    &gt; ### {% icon question %} Questions\n&gt;    &gt;\n&gt;    &gt; 1. What are four key features of a FASTQ file?\n&gt;    &gt; 2. What is the main difference between a FASTQ and a FASTA file?\n&gt;    &gt;\n&gt;    &gt; &gt; ### {% icon solution %} Solution\n&gt;    &gt; &gt; 1. Each sequence in a FASTQ file is represented by 4 lines: 1st line is the id, 2nd line is the sequence, 3rd line is not used, and 4th line is the quality of sequencing per nucleotide\n&gt;    &gt; &gt; 2. In a FASTQ file, not only are the sequences present, but information about the quality of sequencing is also included.\n&gt;    &gt; {: .solution }\n&gt;    {: .question}\n&gt;\n{: .hands_on}\n\nThe reads have been sequenced from an imaginary *Staphylococcus aureus* bacterium using an Illumina DNA sequencing instrument. We obtained the 2 files we imported (`mutant_R1` and `mutant_R2`)\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; Why do we have 2 files here if we only sequenced the bacteria once?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. The bacteria has been sequenced using paired-end sequencing. The first file corresponds to forward reads and the second file to reverse reads.\n&gt; {: .solution }\n{: .question}\n\n# Evaluate the input reads\n\nBefore doing any assembly, the first questions you should ask about your input reads include:\n\n- What is the coverage of my genome?\n- How good is my read set?\n- Do I need to ask for a new sequencing run?\n- Is it suitable for the analysis I need to do?\n\nWe will evaluate the input reads using the FastQC tool. This tool runs a standard series of tests on your read set and returns a relatively easy-to-interpret report. We will use it to evaluate the quality of our FASTQ files and combine the results with MultiQC.\n\n&gt; ### {% icon hands_on %} Hands-on: FastQC on a fastq file\n&gt;\n&gt; 1. **FastQC** {% icon tool %} with the following parameters\n&gt;    - \&quot;Short read data from your current history\&quot; to (**Multiple datasets**) `mutant_R1.fastq` and `mutant_R2.fastq`\n&gt;\n&gt; 2. **MultiQC** {% icon tool %} with the following parameters\n&gt;    - \&quot;Software name\&quot; to `FastQC`\n&gt;    - \&quot;Result file\&quot; to the raw data files generated by FastQC\n&gt;\n{: .hands_on}\n\nMultiQC generates a webpage combining reports for FastQC on both datasets. It includes these graphs and tables:\n\n- General statistics\n\n    This is important in setting maximum k-mer size for an assembly.\n\n    &gt; ### {% icon comment %} Getting the length of sequences\n    &gt;\n    &gt; * Click on **Configure Columns**\n    &gt; * Check **Length**\n    &gt; * Close the window\n    {: .comment}\n\n    &gt; ### {% icon question %} Questions\n    &gt;\n    &gt; 1. How long are the sequences?\n    &gt; 2. What is the average coverage of the genome, given our imaginary *Staphylococcus aureus* bacterium has a genome of 197,394 bp?\n    &gt;\n    &gt; &gt; ### {% icon solution %} Solution\n    &gt; &gt; 1. The sequences are 150 bp long\n    &gt; &gt; 2. We have 2 x 12,480 sequences of 150 bp, so the average genome coverage is: 2 * 12480 * 150 / 197394, or approximately 19 X coverage.\n    &gt; {: .solution }\n    {: .question}\n\n- Sequence Quality Histograms\n\n    Dips in quality near the beginning, middle or end of the reads may determine the trimming/cleanup methods and parameters to be used, or may indicate technical problems with the sequencing process/machine run.\n\n    &lt;figure id=\&quot;figure-1\&quot;&gt;&lt;img src=\&quot;../../images/fastqc_per_base_sequence_quality_plot.png\&quot; alt=\&quot;Sequence Quality Histograms with the mean quality value across each base position in the read\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 1:&lt;/span&gt; The mean quality value across each base position in the read&lt;/figcaption&gt;&lt;/figure&gt;\n\n    &gt; ### {% icon question %} Questions\n    &gt;\n    &gt; 1. What does the y-axis represent?\n    &gt; 2. Why is the quality score decreasing across the length of the reads?\n    &gt;\n    &gt; &gt; ### {% icon solution %} Solution\n    &gt; &gt; 1. The y-axis represents the quality score for each base (an estimate of the error during sequencing).\n    &gt; &gt; 2. The quality score is decreasing accross the length of the reads because the sequencing become less and less reliable at the end of the reads.\n    &gt; {: .solution }\n    {: .question}\n\n- Per Sequence GC Content\n\n    High GC organisms tend not to assemble well and may have an uneven read coverage distribution.\n\n- Per Base N Content\n\n    The presence of large numbers of Ns in reads may point to a poor quality sequencing run. You will need to trim these reads to remove Ns.\n\n- k-mer content\n\n    The presence of highly recurring k-mers may point to contamination of reads with barcodes or adapter sequences.\n\n\n&gt; ### {% icon comment %} Comment\n&gt;\n&gt; For a fuller discussion of FastQC outputs and warnings, see the [FastQC website link](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/), including the section on each of the output [reports](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/), and examples of [\&quot;good\&quot;](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html) and [\&quot;bad\&quot;](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html) Illumina data.\n{: .comment}\n\nWe won&#39;t be doing anything to these data to clean it up as there isn&#39;t much need. Therefore we will get on with the assembly!\n\n\n# Assemble reads with Velvet\n\nNow, we want to assemble our reads to find the sequence of our imaginary *Staphylococcus aureus* bacterium. We will perform a *de novo* assembly of the reads into long contiguous sequences using the Velvet short read assembler.\n\nThe first step of the assembler is to build a de Bruijn graph. For that, it will break our reads into k-mers, *i.e.* fragments of length *k*. Velvet requires the user to input a value of *k* (k-mer size) for the assembly process. Small k-mers will give greater connectivity, but large k-mers will give better specificity.\n\n&gt; ### {% icon hands_on %} Hands-on: Assemble the reads\n&gt;\n&gt; 1. **FASTQ interlacer** {% icon tool %} with the following parameters\n&gt;    - \&quot;Type of paired-end datasets\&quot; to `2 separate datasets`\n&gt;    - \&quot;Left-hand mates\&quot; to `mutant_R1.fastq`\n&gt;    - \&quot;Right-hand mates\&quot; to `mutant_R2.fastq`\n&gt;\n&gt;    Currently our paired-end reads are in 2 files (one with the forward reads and one with the reverse reads), but Velvet requires only one file, where each read is next to its mate read. In other words, if the reads are indexed from 0, then reads 0 and 1 are paired, 2 and 3, 4 and 5, etc. Before doing the assembly *per se*, we need to prepare the files by combining them.\n&gt;\n&gt; 2. **velveth** {% icon tool %} with the following parameters\n&gt;    - \&quot;Hash Length\&quot; to `29`\n&gt;    - \&quot;Input Files\&quot;: click on `Insert Input Files`\n&gt;    - \&quot;file format\&quot; to `fastq`\n&gt;    - \&quot;read type\&quot; to `shortPaired reads`\n&gt;    - \&quot;Dataset\&quot; to the pairs output of **FASTQ interlacer**\n&gt;\n&gt;    The tool takes our reads and break them into k-mers.\n&gt;\n&gt; 3. **velvetg** {% icon tool %} with the following parameters\n&gt;    - \&quot;Velvet Dataset\&quot; to the output of **velveth**\n&gt;    - \&quot;Using Paired Reads\&quot; to `Yes`\n&gt;\n&gt;    This last tool actually does the assembly.\n{: .hands_on}\n\nTwo files are generated:\n\n- A \&quot;Contigs\&quot; file\n\n    This file contains the sequences of the contigs longer than 2k. In the header of each contig, a bit of information is added:\n    - the k-mer length (called \&quot;length\&quot;): For the value of k chosen in the assembly, a measure of how many k-mers overlap (by 1 bp each overlap) to give this length\n    - the k-mer coverage (called \&quot;coverage\&quot;): For the value of k chosen in the assembly, a measure of how many k-mers overlap each base position (in the assembly).\n\n    ![Contigs output](../../images/image10.png)\n\n- A \&quot;Stats\&quot; file\n\n    This is a tabular file giving for each contig the k-mer lengths, k-mer coverages and other measures.\n\n    ![Contigs stats output](../../images/image11.png)\n\n# Collect some statistics on the contigs\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. How many contigs have been built?\n&gt; 2. What is the mean, min and max length of the contigs?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. 190\n&gt; &gt; 2. To compute this information, we can use the Datamash tool on the 2nd columns (length). Be careful with the first line, the header. As a result, we obtain: 597.82 as mean, 1 as min and 12904 as max. It would mean that the smallest contig has a length of 1 bp, even smaller than k. The length on the 2nd column corresponds to length of the contig in k-mers. This means that the smallest contig has a length of 1k = 29. So to obtain the real length, we need to add k-1 to the length. We then obtain a mean contig length of 625.82 bp, a min contig of 29 bp and a max contig of 12,932 bp.\n&gt; {: .solution }\n{: .question}\n\nThis table is limitted, but we will now collect more basic statistics on our assembly.\n\n&gt; ### {% icon hands_on %} Hands-on: Collect fasta statistics on our contigs\n&gt;\n&gt; 1. **Quast** {% icon tool %} with\n&gt;    - \&quot;Contigs/scaffolds output file\&quot; to the output of **velvetg**\n&gt;    - \&quot;Type of data\&quot; to `contig`\n&gt;    - \&quot;Reference File\&quot; to `wildtype.fna`\n&gt;    - \&quot;Type of organism\&quot; to `Prokaryotes`\n&gt;    - \&quot;Lower Threshold\&quot; to `500`\n&gt;    - \&quot;Thresholds\&quot; to `0,1000`\n{: .hands_on}\n\nThis tool generates 5 output files, but we will focus on the HTML report and the Icarus viewer.\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. What is represented in the Icarus viewer?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. Icarus is a novel genome visualizer for accurate assessment and analysis of genomic draft assemblies. It draws contigs ordered from longest to shortest, highlights N50, N75 (NG50, NG75) and long contigs larger than a user-specified threshold\n&gt; {: .solution }\n{: .question}\n\nThe HTML report reports many statistics computed by QUAST to assess the quality of the assembly:\n\n- Statistics about the quality of the assembly when compared to the reference (fraction of the genome, duplication ratio, etc)\n- Misassembly statistics, including the number of misassemblies\n\n    A misassembly is a position in the contigs (breakpoints) that satisfy one of the following criteria:\n    - the left flanking sequence aligns over 1 kbp away from the right flanking sequence on the reference;\n    - flanking sequences overlap on more than 1 kbp\n    - flanking sequences align to different strands or different chromosomes\n\n- Unaligned regions in the assembly\n- Mismatches compared to the reference genomes\n- Statistics about the assembly *per se*, such as the number of contigs and the length of the largest contig\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. How many contigs have been constructed?\n&gt; 2. Which proportion of the reference genome do they represent?\n&gt; 3. How many misassemblies have been found?\n&gt; 4. Has the assembly introduced mismatches and indels?\n&gt; 5. What are N50 and L50?\n&gt; 6. Is there a bias in GC percentage induced by the assembly?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. 190 contigs have been constructed, but only 47 have a length &gt; 500 bp.\n&gt; &gt; 2. The contigs represents 87.965% of the reference genome.\n&gt; &gt; 3. 1 misassembly has been found: it corresponds to a relocation, *i.e.* a misassembly event (breakpoint) where the left flanking sequence aligns over 1 kbp away from the right flanking sequence on the reference genome.\n&gt; &gt; 4. 8.06 mismatches per 100 kbp and 4.03 indels per 100 kbp are found.\n&gt; &gt; 5. N50 is the length for which the collection of all contigs of that length or longer covers at least half an assembly. In other words, if contigs were ordered from small to large, half of all the nucleotides will be in contigs this size or larger. And L50 is the number of contigs equal to or longer than N50: L50 is the minimal number of contigs that cover half the assembly.\n&gt; &gt; 6. The GC % in the assembly is 33.64%, really similar to the one of the reference genome (33.43%).\n&gt; {: .solution }\n{: .question}\n\n# Discussion\n\n&gt; ### {% icon hands_on %} (Optional) Hands-on: Rerun for values *k* ranging from 31 to 101\n&gt;\n&gt; 1. **velveth** {% icon tool %} with the same parameters as before except\n&gt;    - \&quot;Hash Length\&quot; to a value between 31 and 101\n&gt; 2. **velvetg** {% icon tool %} with the same parameters as before\n&gt; 3. **Quast** {% icon tool %} with the same parameters as before\n{: .hands_on}\n\nWe have completed an assembly on this data set for a number of k values ranging from 29 to 101. A few of the assembly metrics appear below.\n\n&lt;figure id=\&quot;figure-2\&quot;&gt;&lt;img src=\&quot;../../images/number_of_contigs.png\&quot; alt=\&quot;contigs\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 2:&lt;/span&gt; Number of contigs in the assembly for various k-mer sizes&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-3\&quot;&gt;&lt;img src=\&quot;../../images/largest_contig.png\&quot; alt=\&quot;largest_contig\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 3:&lt;/span&gt; Largest contig in each of the assemblies by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-4\&quot;&gt;&lt;img src=\&quot;../../images/total_bp.png\&quot; alt=\&quot;total_bp\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 4:&lt;/span&gt; Total number of base pairs in all the contigs for each assembly by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-5\&quot;&gt;&lt;img src=\&quot;../../images/n50.png\&quot; alt=\&quot;n50\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 5:&lt;/span&gt; N50 metric for each of the assemblies by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&gt; ### {% icon question %} Questions\n&gt;\n&gt; 1. Are there any distinct features in the charts?\n&gt; 2. Does it look like one assembly might be better than some of the others?\n&gt;\n{: .question}\n\nThe reasons for these patterns will be discussed in detail in the De Bruijn graph assembly slides and tutorial.\n&quot;,&quot;#### Create a project\n\nWhen you use qbase+ for the first time, you can&#39;t do anything unless you\ncreate a project to store your experiments in.\n\n| Create a new Project |\n| :----------------------------------- |\n| When you double click the qbase+ icon, the software starts up automatically opens the Start page where you can create a new project by clicking the Create new project button : This will create a new project with a default name like Project 1 . |\n\n#### Create an experiment\n\nTo open actual data (one/more runs) in qbase+, creating a project is not sufficient. You need to create an experiment in this project to hold the run data.\n\n| Create a new Experiment called GeneExpression in the new project. |\n| :----------------------------------- |\n| Select the Create a new qbase+ experiment option  in the Start page. Type a name for th new experiment . Click the Next button at the bottom of the page . This will create the experiment.\n\nWhen you leave the **Start page**, the **Import run** page is automatically opened allowing you to import the actual qPCR data into qbase+.\n\n#### Loading the data\n\nFirst a few quick words about the data set. We’ll be working with data coming from 3 runs (plates in the qPCR instrument): [Run1](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run1.xls), [Run2](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run2.xls) and [Run3](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run3.xls)\n\nThe data consist of Cq values for:\n\n  - 3 reference target genes: Stable, Nonregulated, and Flexible\n  - 3 target genes of interest: Duvel, Leffe, and Palm\n\neach measured twice (= technical replicates) in 16 different samples. Half of the samples have undergone a treatment, half of them are untreated control samples.\n\nThe data set also contains a series of standard samples consisting of a four-fold dilution series of cDNA for each target gene. These measurements allow to generate a standard curve from which target-specific amplification efficiencies can be calculated. Finally, negative controls (No Template Controls) have been measured. The goal of the analysis is to identify target genes of interest that have different expression levels in the treated samples compared to the untreated control samples.\n\n| In GeneExpression load CFX run files [Run1](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run1.xls), [Run2](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run2.xls) and [Run3](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run3.xls).\n| :---------------------------- |\n|  \n* Click the **Import runs** button  to open the **Import Run** window\n* Click the **Browse** button  to go to the directory that stores the files containing the qPCR data\n\nSelect the 3 run files simultaneously by holding the **Ctrl** key on your keyboard during the selection in Windows or the command button in MacOSX.\n\nClick the **Open** button \n\nNow you go back to the **Import Run** window, click the **Next** button (purple)\n\n* \nqbase+ tries to recognize the format of the selected import files. If only one format matches the files (as in our case CFX), it is selected and the quick import option is enabled. Click the **Finish** button.\n\nIn the Imported run names area on the **Import run** page you should now see the names of the 3 run files. If these are the correct files, click the **Next** button at the bottom of the page.\n\n#### Adding annotation to the data\n\nWhen you leave the **Import run** page, you are redirected to the **Sample target list** page, which gives you an overview of the targets (= genes) and samples qbase+ detected when reading in the datafiles.\nTake a look at the data. You see that the list of samples and targets matches the description of the qPCR experiment at the top of this page. The samples in this experiment are divided into two groups: samples that received some kind of treatment and untreated control samples. This information was not included in the run files so qbase+ does not know which sample belongs to which group. However, this is relevant information: in our analysis we are going to compare the expression of our genes of interest between treated and untreated samples. This means that qbase+ needs the grouping annotation to be able to perform the analysis we want to do. So we have to give qbase+ this annotation: we can do this by adding a custom sample property. To do this we need to create a sample properties file with a specific format that is described in [the tutorial](http://data.bits.vib.be/pub/trainingen/qbasePLUS/TutorialIII.pdf). You can find the file in the qbase+ folder on the BITS laptops or you can [download the file here](http://data.bits.vib.be/pub/trainingen/qbasePLUS/Sample_Properties_file.xlsx).\n\n| How to add the grouping annotation ?\n| :---------------------------- |\n|  To import the file containing to grouping annotation:\n\n* select **Add samples and targets** \n* click **Import sample list** \n* browse to the folder that contains the samples file\n* select the file and click **Open**\n* click **Next**\n\nIn the **Importing samples** window, you have to tell qbase+ which sample annotation you want to import from the sample properties file\n\nIn our case we could import Quantities (this annnotation is available in the sample properties file) but the quantities of the standard samples were included in the run files so qbase+ has already imported this annotation from the run files during data import.\nWe definitely need to import the Custom properties since they were not a part of the run files. The Treatment property will tell qbase+ which samples belong to the group of control samples and which samples belong to the group of treated samples.\nClick the **Next** button at the bottom of the page to finish the import.\n\nAt this point you don&#39;t see the custom annotation that you have imported, you will see it later in the analysis during scaling\nLeaving the **Sample target list** page takes you to the **Run annotation** page, where you have to confirm again that the sample and gene names are ok. If this is not the case you can adjust the annotation here.\n\nClick the **Next** button at the bottom of the page\n\nOur data file contains all required annotation:\n\n  - Cq values\n  - sample and target names\n  - sample types\n  - quantities for the standard samples\n  - grouping of the samples\n\n\n\nOnce runs are imported, you can start analyzing the data. Data consist\nof Cq values for all the wells.\n\n#### Specifying the aim of the experiment\n\nOn the **Aim** page you tell the software what type of analysis you want to do. Different types of analyses require different parameters, parameter settings and different calculations. By selecting the proper analysis type, qbase+ will only show the relevant parameters and parameter settings.\n\nSince we are doing a **gene expression analysis** in this exercise, this the option we should select. Click the **Next** button on the bottom of the page to go to the **Technical quality control** page.\n\n#### Checking the quality of technical replicates and controls\n\nThe **Technical quality control** page handles the settings of the requirements that the data have to meet to be considered high quality. For instance the maximum difference between technical replicates is defined on this page. If there are technical replicates in the data set, qbase+ will detect them automatically (they have the same sample and target name) and calculate the average Cq value. In theory, technical replicates should generate more or less identical signals.\n\n| How to set the maximum difference in Cq values for technical replicates ?\n| :---------------------------- |\n|  The quality criterium that the replicates must meet to be included for further analysis is one of the parameters in qbase+. You can set it on the **Technical quality control** page:\n\nThe default maximum allowed difference in Cq values between technical replicates is 0.5\n\nAdditionally, you can do quality checks based on the data of the positive and negative controls.\n| How to set quality requirements for the control samples ?\n| :---------------------------- |\n|  On the same **Technical quality control** page you can define the minimum requirements for a well to be included in the calculations:\n\n* **Negative control threshold** : minimum allowed difference in Cq value between the sample with the highest Cq value and the negative control with the lowest Cq value: the default is 5 which means that negative controls should be more than 5 cycles away from the sample of interest.\n* **Lower and upper boundary** : allowed range of Cq values for positive controls.\n\nExcluded means that the data are ignored in the calculations.\n\n| How to check if there are wells that do not meet these criteria ?\n| :---------------------------- |\n|  You can see flagged and excluded data by ticking the **Show details…** options  on the **Technical quality control** page and clicking the **Next** button (purple) at the bottom of the page.\n\nQbase+ will open the results of the quality checks for the replicates  and the controls  on two different tabs. These tabs show lists of samples that failed the quality control criteria. When you open the replicates tab  you can get an overview of the flagged  or the excluded (purple) wells. Select the **failing**  wells.\n\nWhen the difference in Cq between technical replicates exceeds 0.5, the wells end up in the flagged or failing list. They are included in calculations unless you exclude them by unticking them. You see that the two replicates of Palm in Sample05 have very different Cq values. All other bad replicates are coming from standard samples.\nIf you are finished checking the data quality, click **Next** to go to the **Amplification efficiencies** page.\n\n#### Taking into account amplification efficiencies\n\nQbase+ calculates an amplification efficiency (E) for each primer pair (= gene). Genes have different amplification efficiencies because:\n\n  - some primer pairs anneal better than others\n  - the presence of inhibitors in the reaction mix (salts, detergents…) decreases the amplification efficiency\n  - inaccurate pipetting\n\nQbase+ has a parameter that allows you to specify how you want to handle amplification efficiencies on the **Amplification efficiencies** page.\n\n| How to specify the amplification efficiencies strategy you want to use ?\n| :---------------------------- |\n|  Since we have included a dilution series for creating a standard curve in our qPCR experiment, we will select\n\n* **Use assay specific amplification efficiencies**\n* **Calculate efficiencies from included standard curves**\n\nAmplification efficiencies are calculated based on the Cq values of a serial dilution of representative template, preferably a mixture of cDNAs from all your samples. Since you know the quantity of the template in each dilution, you can plot Cq values against template quantities for each primer pair. Linear regression will fit a standard curve to the data of each gene, and the slope of this curve is used to calculate the amplification efficiency.\n\n| How to check the amplification efficiencies of the genes ?\n| :---------------------------- |\n|  Once you have made this selection, qbase+ starts calculating the efficiencies and the results are immediately shown in the **calculation efficiencies** table.\n\nIn this way, one amplification efficiency (E) for each gene is calculated and used to calculate **Relative Quantities (RQ)**:\n∆Cq is calculated for each well by subtracting the Cq of that well from the average Cq across all samples for the gene that is measured in the well. So ∆Cq is the difference between the Cq value of a gene in a given sample and the average Cq value of that gene across all samples. Cq is subtracted from the average because in this way high expression will result in a positive ∆Cq and low expression in a negative ∆Cq. \n**So at this point the data set contains one RQ value for each gene in each sample.**\n\nClick **Next** to go to the **Normalization** page.\n\n#### Normalization\n\nDifferences in amplification efficiency are not the only source of variability in a qPCR experiment. Several factors are responsible for noise in qPCR experiments e.g. differences in:\n\n  - amount of template cDNA between wells\n  - RNA integrity of samples\n  - efficiency of enzymes used in the PCR or in the reverse\n    transcription\n\nNormalization will eliminate this noise as much as possible. In this way it is possible to make a distinction between genes that are really upregulated and genes with high expression levels in one group of samples simply because higher cDNA concentrations were used in these samples.\nIn qPCR analysis, normalization is done based on housekeeping genes.\n\nHousekeeping genes are measured in all samples along with the genes of interest. In theory, a housekeeping gene should have identical RQ values in all samples. In reality, noise generates variation in the expression levels of the housekeeping genes. This variation is a direct measure of the noise and is used to calculate a normalization factor for each sample.\nThese normalization factors are used to adjust the RQ values of the genes of interest accordingly so that the variability is eliminated.\n\nThese adjusted RQ values are called **Normalized Relative Quantities (NRQs)**. In qbase+ housekeeping genes are called reference genes. In our data set there are three reference genes: Stable, Non-regulated and Flexible. On the **Normalization page** we can define the normalization strategy we are going to use, appoint the reference genes and check their stability of expression.\n\n| How to specify the normalization strategy you want to use ?\n| :---------------------------- |\n|  You can specify the normalization strategy you want to use on the Normalization method page:\n\n* **Reference genes** normalization is based on the RQ values of the housekeeping genes\n* **Global mean** normalization calculates normalization factors based on the RQ values of all genes instead of only using the reference genes. This strategy is recommended for experiments with more than 50 random genes. Random means that the genes are randomly distributed over all biological pathways.\n* **Custom value** normalization is used for specific study types. This strategy allows users to provide custom normalization factors such as for example the cell count.\n* **None** means that you choose to do no normalization at all. This option should only be used for single cell qPCR.\n\nWe have incorporated 3 housekeeping genes in our experiment so we select the **Reference genes** strategy.\n\n| How to appoint reference targets ?\n| :---------------------------- |\n|  You have to indicate which targets should be used as reference genes since qbase+ treats all genes as targets of interest unless you explicitly mark them as reference genes on the Normalization method page:\n\nWe have measured 3 housekeeping genes: Stable, Flexible and Non-regulated so we tick the boxes in front of their names.\n\nIt&#39;s not because you have appointed genes as reference genes that they necessarily are **good** reference genes. They should have stable expression values over all samples in your study. Fortunately, qbase+ checks the quality of the reference genes. For each appointed reference gene, qbase+ calculates two indicators of expression stability\n\n  - **M** (geNorm expression stability value): calculated based on the pairwise variations of the reference genes.\n  - **CV** (coefficient of variation): the ratio of the standard deviation of the NRQs of a reference gene over all samples to the mean NRQ of that reference gene.\n\nIt is considered that the higher these indicators the less stable the reference gene.\n\n| Are Flexible, Stable and Nonregulated good reference targets ?\n| :---------------------------- |\n|  M and CV values of the appointed reference genes are automatically calculated by qbase+ and shown on the Normalization method page:\n\nThe default limits for M and CV were determined by checking M-values and CVs for established reference genes in a pilot experiment that was done by Biogazelle. Based on the results of this pilot experiment, the threshold for CV and M was set to 0.2 and 0.5 respectively.\nIf a reference gene does not meet these criteria it is displayed in red. As you can see the M and CV values of all our reference exceed the limits and are displayed in red.\n\nIf the quality of the reference genes is not good enough, it is advised to remove the reference gene with the worst M and CV values and re-evaluate the remaining reference genes.\n\n| Which reference target are you going to remove ?                                                                                                                                    |\n| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |\n| Both the M-value and the CV are measures of variability. The higher these values the more variable the expression values are. So we will remove the gene with the highest M and CV. |\n\nYou can remove a reference gene simply by unticking the box in front of its name.\n\n| Are the two remaining reference genes good references ?\n| :---------------------------- |\n|  After removing Flexible as a reference gene the M and CV values of the two remaining reference genes decrease drastically to values that do meet the quality criteria. M and CV values that meet the criteria are displayed in green.\n\nThis exercise shows the importance of using a minimum of three reference genes. If one of the reference genes does not produce stable expression values as is the case for Flexible, you always have two remaining reference genes to do the normalization.\n\n[See how to select reference genes for your qPCR experiment](http://wiki.bits.vib.be/index.php/Using_GeneVestigator_to_select_candidate_reference_genes).\n\n**So after normalization you have one NRQ value for each gene in each sample.**\n\nClick **Next** to go to the **Scaling** page.\n\n#### Scaling\n\nRescaling means that you calculate NRQ values relative to a specified reference level.\n\nQbase+ allows you to rescale the NRQ values using one of the following as a reference:\n\n  - the sample with the minimal expression\n  - the average expression level of a gene across all samples\n  - the sample with the maximal expression\n  - a specific sample (e.g. untreated control)\n  - the average of a certain group (e.g. all control samples): this is\n    often how people want to visualize their results\n  - positive control: only to be used for copy number analysis\n\nAfter scaling, the expression values of the choice you make here will be set to 1 e.g. when you choose **average** the average expression level across all samples will be set to 1 and the expression levels of the individual samples will be scaled accordingly.\n\n| How to scale to the average of the untreated samples ?\n| :---------------------------- |\n|  You can specify the scaling strategy on the **Scaling** page. Select **Scale to group** and set the **Scaling group** to the **untreated** samples . This is one of the reasons why you need the grouping annotation.\n\nRescaling to the average of a group is typically used to compare results between 2 groups, e.g. treated samples against untreated controls. After rescaling, the average of the NRQs across all untreated samples is 1 and the NRQs of the treated samples are scaled accordingly.\n\nClick **Next** to go to the **Analysis** page.\n\n#### Visualization of the results\n\nOne of the things you can select to do on the **Analysis** page is viewing the relative expression levels (= scaled NRQs) of each of the genes in a bar chart per gene. It is recommended to visualize your results like this.\n\nIt is possible to view the relative expression levels of all genes of interest on the same bar chart. You can use this view to see if these genes show the same expression pattern but you cannot directly compare the heights of the different genes because each gene is independently rescaled\\!\n\n| How to visualize single gene expression bar charts ?\n| :---------------------------- |\n|  Select **Visually inspect results For individual targets** on the **Analysis** page and click **Finish**\n\n| How to visualize the expression levels of Palm in each sample ?\n| :---------------------------- |\n|  Select **Visually inspect results For individual targets** on the **Analysis** page and click **Finish**\n\nThe **Target** select box allows you to select the gene you want to view the expression levels of. Relative expression levels are shown for each sample. Error bars are shown and represent the technical variation in your experiment (variation generated by differences in amounts pipetted, efficiency of enzymes, purity of the samples...).\n\nYou see that Palm has a low expression level and a very large error bar in Sample05 because the two replicates of this sample had very different Cq values. You can group and colour the bars according to a property.\n\n| How to group the bars of Palm according to treatment (so treated at one side and untreated at the other side)\n| :---------------------------- |\n|  In the **Grouping** section you can specify the property you want to group by.\n\n| How to view average expression levels in each group ?\n| :---------------------------- |\n|  In the **Grouping** section you can choose to plot individual samples as shown above but you can also choose to **plot group average** expression levels.\n\nThe error bars that you see here represent biological variation and will be used later on in the statistical analysis. The error bars are 95% confidence intervals which means that they represent the range that will contain with 95% certainty the real average expression level in that group of samples.\nThe nice characteristic of 95% confidence intervals is the following:\n\n  - if they do not overlap you are sure that the expression levels in the two groups are significantly different, in other words the gene is differentially expressed\n  - if they do overlap you cannot say that you are sure that the expression levels are the same. You simply don’t know if the gene is differentially expressed or not.\n\n| Assess the effect of switching the Y-axis to a logarithmic scale for Palm.\n| :---------------------------- |\n|  In the **Y axis** section you can specify if you want a linear or logarithmic axis.\nAs you can see you do not change the expression values, you just change the scale of the Y axis. Switching the Y-axis to a logarithmic scale can be helpful if you have large differences in NRQs between different samples\n\n| Assess the effect of switching the Y-axis to a logarithmic scale for Flexible.\n| :---------------------------- |\n|  Switch to the bar charts of Flexible. By switching the Y-axis to logarithmic you can now see more clearly the differences between samples with small NRQs.\n\n#### Statistical analysis\n\nOnce you generate target bar charts you leave the **Analysis wizard** and you go to the regular qbase+ interface. Suppose that you want to perform a statistical test to prove that the difference in expression that you see in the target chart is significant.\nAt some point, qbase+ will ask you if your data is coming from a normal distribution. If you don&#39;t know, you can select **I don&#39;t know** and qbase+ will assume the data are not coming from a normal distribution and perform a stringent non-parametric test.\nHowever, when you have **7 or more replicates per group**, you can check if the data is normally distributed using a statistical test. If it is, qbase+ will perform a regular t-test. The upside is that the t-test is less stringent than the non-parametric tests and will find more DE genes. However, you may only perform it on normally distributed data. If you perform the t-test on data that is not normally distributed you will generate false positives i.e. qbase+ will say that genes are DE while in fact they are not. Performing a non-parametric test on normally distributed data will generate false negatives i.e. you will miss DE genes.\n\nChecking if the data is normally distributed can be easily done in GraphPad Prism. To this end you have to export the data.\n| How to export the data ?\n| :---------------------------- |\n|  To export the results click **the upward pointing arrow** in the qbase+ toolbar:\nYou want to export the normalized data so select **Export Result Table (CNRQ)**:\nYou will be given the choice to export results only (CNRQs) or to include the errors (standard error of the mean) as well . We don&#39;t need the errors in Prism so we do not select this option.\nThe scale of the Result table can be linear or logarithmic (base 10) . Without user intervention, qbase+ will automatically log10 transform the CNRQs prior to doing statistics. So we need to check in Prism if the log transformed data are normally distributed.\nAdditionally, you need to tell qbase+ where to store the file containing the exported data. Click the **Browse** button for this .\n\nExporting will generate an Excel file in the location that you specified. However, the file contains the results for all samples and we need to check the two groups (treated and untreated) separately. The sample properties show that the even samples belong to the treated group and the odd samples to the untreated group.\nThis means we have to generate two files:\n\n  - [a file containing the data of the untreated samples](http://data.bits.vib.be/pub/trainingen/qbasePLUS/resultslog.csv)\n  - [a file containing the data of the treated samples](http://data.bits.vib.be/pub/trainingen/qbasePLUS/resultslogTreated.csv)\n\nNow we can open these files in Prism to check if the data is normally distributed.\n\n| How to import the data of the untreated samples in Prism ?\n| :---------------------------- |\n|  \n* Open Prism\n* Expand **File** in the top menu\n* Select **New**\n* Click **New Project File**\n* In the left menu select to create a **Column** table. Data representing different groups (in our case measurements for different genes) should always be loaded into a column table.\n* Select **Enter replicate values, stacked into columns** (this is normally the default selection) since the replicates (measurements for the same gene) are stacked in the columns.\n* Click **Create**\n\nPrism has now created a table to hold the data of the untreated samples but at this point the table is still empty. To load the data:\n\n* Expand **File** in the top menu\n* Select **Import**\n* Browse to the resultslog.csv file, select it and click **Open**\n* In the **Source** tab select **Insert data only**\n* Since this is a European csv file commas are used as decimal separators so in contrast to what its name might imply, semicolons and not commas are used to separate the columns in the csv file (you can open the file in a text editor to take a look). In American csv files dots are used as decimal separator and the comma is used to separate the columns. Prism doesn&#39;t know the format of your csv file so you have to tell him the role of the comma in your file. Select **Separate decimals**\n* Go to the **Filter** tab and specify the rows you want to import (the last rows are these of the standard and the water samples, you don&#39;t want to include them)\n* Click **Import**\n\nAs the file is opened in Prism you see that the first column containing the sample names is treated as a data column. Right click the header of the first column and select **Delete**\n\n| How to check if the data of the untreated samples comes from a normal distribution ?\n| :---------------------------- |\n|  \n* Click the **Analyze** button in the top menu\n* Select to do the **Column statistics** analysis in the **Column analyses** section of the left menu\n* In the right menu, deselect **Flexible**. It&#39;s a bad reference gene so you will not include it in the qbase+ analysis so there&#39;s no point checking its normality (it is probably not normally distributed). In that respect you could also deselect the other two reference genes since you will do the DE test on the target genes and not on the reference genes.\n* Click **OK**\n* In the **Descriptive statistics** and the **Confidence intervals** section deselect everything except **Mean, SD, SEM**. These statistics is not what we are interested in: we want to know if the data comes from a normal distribution. The only reason we select Mean, SD, SEM is because if we make no selection here Prism throws an error.\n* In the **Test if the values come from a Gaussian distribution** section select the **D&#39;agostino-Pearson omnibus test** to test if the data are drawn from a normal distribution. Although Prism offers three tests for this, the D&#39;Agostino-Pearson test is the safest option.\n* Click **OK**\n\nPrism now generates a table to hold the results of the statistical analysis: As you can see, the data for Palm are not normally distributed.\n\nSince we found that there&#39;s one group of data that does not follow a normal distribution, it&#39;s no longer necessary to check if the treated data are normally distributed but you can do it if you want to. We will now proceed with the statistical analysis in qbase+. Statistical analyses can be performed via the **Statistics wizard**.\n\n| How to open the Statistics wizard ?\n| :---------------------------- |\n|  You can open it in the **Project Explorer** (window at the left):\n\n* expand **Project1** if it&#39;s not yet expanded\n* expand the **Experiments** folder in the project if it&#39;s not yet expanded\n* expand the **GeneExpression** experiment if it&#39;s not yet expanded\n* expand the **Analysis** section if it&#39;s not yet expanded\n* expand the **Statistics** section\n* double click **Stat wizard**\n\nThis opens the **Statistics wizard** that allows you to perform various kinds of statistical analyses.\n\n| Which kind of analysis are you going to do ?                                                                                                                                                                                                                                                      |\n| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |\n| On the **Goal** page: Select **Mean comparison** since you want to compare expression between two groups of samples so what you want to do is comparing the mean expression of each gene in the treated samples with its mean expression level in the untreated samples. Click **Next**. |\n\n| How to define the groups that you are going to compare ?\n| :---------------------------- |\n|  On the **Groups** page: specify how to define the two groups of samples that you want to compare. Select **Treatment** as the grouping variable to compare treated and untreated samples. Click **Next**.\n\n| How to define the genes that you want to analyze ?\n| :---------------------------- |\n|  On the **Targets** page: specify for which targets of interest you want to do the test. Deselect **Flexible** since you do not want to include it in the analysis. It&#39;s just a bad reference gene. Click **Next**.\n\nOn the **Settings** page you have to describe the characteristics of your data set, allowing qbase+ to choose the appropriate test for your data. \n\nThe first thing you need to tell qbase+ is whether the data was drawn from a normal or a non-normal distribution. Since we have 8 biological replicates per group we can do a test in Prism to check if the data are normally distributed.\n\n| Which gene(s) is/are differentially expressed ?\n| :---------------------------- |\n|  On the **Settings** page you describe the characteristics of your data set so that qbase+ can choose the ideal test for your data. For our data set we can use the default settings. Click **Next**. In the results **Table** you can see that the p-value for Palm is below 0.05 so Palm is differentially expressed.\n\n\n\nIn this example we will analyze data from another expression study with the following characteristics:\n\nAll samples fit in a single run: [Run7](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run7.xls)\nWe have the following samples:\n\n  - 5 control samples: control1, control2…\n  - 5 treated samples: treated1, treated2…\n  - 1 no template control: NTC\n\nThe expression of the following genes was measured:\n\n  - 2 reference genes: refgene1 and refgene2\n  - 2 genes of interest: gene1 and gene2\n\nThere are two technical replicates per reaction\n\n#### Creating a new experiment\n\n| Create a new Experiment called GeneExpression2 in Project1\n| :---------------------------- |\n| You can find the details on how to create a new experiment in Creating a project and an experiment\n\n#### Loading the data\n\n| Import [Run7](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run7.xls). This file is in qBase format.                    |\n| :-------------------------------------------------------------------------------------------------------------------------------------- |\n| You can find the details on how to import the data file in the **Loading the data into qbase+** section of Analyzing data from a geNorm pilot experiment in qbase+ |\n\n#### Adding sample annotation\n\nDownload the [the sample properties file](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Sample_Properties_GE2.xlsx).\n| Add a custom sample property called Treatment.\n| :---------------------------- |\n| You can find the details on how to add a custom sample property in the **Adding annotation to the data** section of Loading data into qbase+\n\n#### Analyzing the data\n\n| Choose the type of analysis you want to perform.\n| :---------------------------- |\n| \n\n| Check controls and replicates.\n| :---------------------------- |\n| First set the minimum requirements for controls and replicates  You see that 6 replicates do not meet these requirements . Select to **Show details and manually exclude bad replicates**\nAll negative controls pass the test . Positive controls were not included in this analysis. Qbase+ will now open the results for the failing replicates: as you can see the difference in Cq values between these replicates is not that big. They fail to meet the requirement just slightly.\n\n| Which amplification efficiencies strategy are you going to use ?\n| :---------------------------- |\n| You don&#39;t have data of serial dilutions of representative template to build standard curves so the only choice you have is to use the default amplification efficiency (E = 2) for all the genes.\n\n| Appoint the reference genes as reference targets.\n| :---------------------------- |\n| You can find the details on how to appoint reference targets in the **Normalization** section of Analyzing gene expression data in qbase+\n\n| Is the stability of the reference genes ok ?\n| :---------------------------- |\n| In the **Reference target stability window** the M and CV values of the reference genes are shown in green so the stability of the reference genes is ok. You can find the details on how to check reference target stability in the **Normalization** section of [](Analyzing_gene_expression_data_in_qbase+\&quot; title=\&quot;wikilink)Analyzing gene expression data in qbase+\n\n| Which scaling strategy are you going to use ?\n| :---------------------------- |\n| Since you have a treated and a **control** group, it seems logical to use the average of the control group for scaling. You can find the details on how to specify the scaling strategy in the **Scaling** section of Analyzing gene expression data in qbase+\n\nLook at the target bar charts.\n\n| In the target bar charts group the samples according to treatment.\n| :---------------------------- |\n| You can find the details on how to group the samples in the **Visualization of the results** section of Analyzing gene expression data in qbase+\n\nThe samples of each group are biological replicates so you might want to generate a plot that compares the average expression of the treated samples with the average expression of the untreated samples.\n\n| In the target bar charts plot the group averages instead of the individual samples.\n| :---------------------------- |\n| In the **Grouping** section at the bottom of the chart you can select **Plot group average**:\n\n| Are there any genes for which you see a clear difference in expression between the two groups ?\n| :---------------------------- |\n| For gene 1, the mean expression levels in the two groups are almost the same and the error bars completely overlap.\n\nWhen you look at the title of the Y-axis, you see that 95% confidence levels are used as error bars. In case of 95% confidence intervakls you can use the following rules:\n\n* if they do not overlap: you are certain that the difference between the means of the two groups is significant\n* if they do not overlap: you know nothing with certainty: the means can be different or they can be the same\n\nSo for gene 1 the means are very close but just based on the plot we may not make any conclusions with certainty. For gene 2, the mean expression levels in the two groups are very different and the error bars do not overlap. So the 95% confidence intervals do not overlap meaning that we can be certain that the difference between the means of the two groups is significant.\n\n| Use a statistical test to compare the expression levels between the two groups of samples ?\n| :---------------------------- |\n| You only have 5 replicates per group so you cannot test if the data comes from a normal distribution. Qbase+ will assume they&#39;re not normally distributed and perform a non-parametric Mann-Whitney test.\n\nThe p-value of gene2 is smaller than 0.05 so it has a statistically significant difference in expression levels in treated samples compared to untreated samples. For gene1 the p-value is 1 so we have no evidence to conclude that the expression of gene1 is different in treated compared to untreated samples. You can find the details on how to compare the means of the two groups in the **Statistical analysis** section of Analyzing gene expression data in qbase+\n&quot;,&quot;### Exercise 1: simple gene expression study\n\nIn my qPCR experiment I want to study the expression of 12 genes of interest in 8 samples of interest. I want to use 2 PCR replicates for each reaction.\n\n&gt; How many 96 well plates do I need for this experiment ?\n&gt; &gt; I have 12 genes in 8 samples which gives a total of 96 reactions (one plate). I want to perform each reaction twice (2 PCR replicates) so I need two plates. However, I need to include reference genes in my experiment, preferably more than one. I can put these reference genes on a separate plate, I do not have to include them on each plate.\nIdeally, you need to include 3 reference genes so having 8 samples and 2 replicates this gives an additional 48 reactions. Thus, I need three 96 well plates to perform this experiment.\n\n| Do I need to include IRCs (inter-run calibrators) ?                      |\n| :--------------------------------------------------- |\n| No, I can easily fit all samples of the same gene on the same plate so I don&#39;t need to include IRCs. |\n\n### Exercise 2: a large study\n\nIn my qPCR experiment I want to study the pattern of expression of 96 genes (genes of interest and reference genes) in 96 samples of interest, divided into a few groups. I want to use 2 PCR replicates for each reaction.\n\n| Do I need to include IRCs (inter-run calibrators) ?                    |\n| :------------------------------------------------------- |\n| No, I can fit all samples of the same gene on the same plate so I don&#39;t need to include IRCs. |\n\nI want to include PCR replicates.\n\n| Do I need to include IRCs when I work on a 96 well plate ?                                                                                    |\n| :-------------------------------------------------------------------------------------------------------------------------------------------- |\n| Yes, I have 192 reactions per gene so I cannot place them on the same plate. Remember that replicates have to be located on the same plate \\! |\n\n| Do I need to include IRCs when I work on a 384 well plate ?                        |\n| :--------------------------------------------------------------------------------- |\n| No, I have 192 reactions per gene so I can even place two genes on the same plate. |\n\nI want to include no template controls but I don&#39;t want to increase the\nnumber of plates.\n\n| What is the most elegant strategy to make room for including negative controls ?      |                                                      \n| :------------------------------------------------------------------------------------ |\n| This kind of study screen for expression patterns and requires statistical analysis. Since you have many samples divided over a few groups it means you have many biological replicates so you could easily do without the PCR replicates. By doing so you preserve the biological variability which is often far greater than the technical variation. |\n\n### Exercise 3: how to fill plates ?\n\nIn my qPCR experiment I want to study the pattern of expression of 5 genes (genes of interest and reference genes) in 38 samples (samples of interest and control samples). I want to use 2 PCR replicates for each reaction.\n\n| What is the minimum number of 96 well plates I need for this experiment ? |\n| :------------------------------ |\n| 5 genes * 38 samples * 2 replicates = 380 reactions.\nI need a minimum of 4 plates for this experiment.\n\n| If I use the minimum number of 96 well plates do I need to include IRCs ?                                                      |\n| :----------------------------------------------------------------------------------------------------------------------------- |\n| Yes, 5 genes spread over 4 plates with 72 reactions per gene means that at least one gene will be spread over multiple plates. |\n\n| What can I do to avoid inter-run variability ?                                                                                                                                                                                         |\n| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |\n| I can use 5 plates and fill them with one gene each. They will not be completely filled (72 reactions) but at least I do not have to use IRCs (which are additional reactions that also cost money) and I have no inter-run variation. |\n\nSuppose there&#39;s only one 96-well plate left in your lab. You have 10 samples (samples of interest + control samples) and you want to make the most of what you have.\n\n| How many genes of interest would you measure ? |\n| :------------------------------ |\n| Since you want to make most of what you have, let&#39;s assume you are omitting PCR replicates.\nTheoretically, you could fit 9 genes on your 96-well plate. However, to avoid pipetting mistakes I would measure only 8 genes so I can work with one row / gene. This is very handy for multichannel pipets.\n\n### Exercise 4: a growing study\n\nIn my qPCR experiment I want to study the pattern of expression of 24 genes (genes of interest and reference genes) in 48 samples (samples of interest and control samples). I want to use 2 PCR replicates for each reaction.\n\n| How many genes can I analyze on one 384 well plate ? |\n| :------------------------------ |\n| 48 samples * 2 replicates = 96 reactions per gene.\nI can analyze 4 genes on each 384 well plate.\n\nEach week I receive 2 additional samples to analyze.\n\n| Do I analyze them immediately after I get them ? |\n| :------------------------------ |\n| No. Since the samples are placed on different plates as in the previous experiment, you have to use IRCs. You typically need 3 IRCs and a no template control sample. It means that if you want to analyze these 2 samples you have to include 4 additional samples for each gene. This is a lot of overhead for just 2 samples !\nTry to avoid this: it&#39;s better to wait a few weeks until you have 6 or 8 or even more samples.\n\n### Exercise 5: a diagnostic copy number screen\n\nIn diagnostic screens all samples are important: you cannot leave out samples and all measurements need to be of the highest quality possible. In my qPCR experiment I want to study copy number variation of 16 genes\n(genes of interest and reference genes) and 2 calibrator samples (samples with known copy number). Since we need high quality data we will use 4 technical replicates.\n\n| Are we going to use sample maximization ?                                   |                   \n| :------------------------------------------------------------------------- |\n| No. In contrast to gene expression studies, where we want to compare expression levels of a gene between different groups of samples, copy number analyses do compare genes. It means that in this case the sample maximization approach (placing all samples of the same gene on the same plate) is not valid. Instead we use a gene maximization approach here (placing same sample for different genes on the same plate). |\n\n| How many samples can I fit on a 384 well plate ? |\n| :------------------------------ |\n| We have 16 (genes) * 4 (replicates) = 64 reactions per sample.\nThis means that we can fit 6 samples on a 384 well plate: 4 unknowns and 2 calibrators.\n\n### Exercise 6: fix experiments with bad or missing data\n\nIn my qPCR experiment I want to study gene expression of 6 genes (3 genes of interest and 3 reference genes) in 20 samples (samples of interest and control samples). I want to use 2 technical replicates. One of my genes of interest failed completely and I want to repeat the measurements for this gene in a new run.\n\n| Do I need to include IRCs ?                                                                                |\n| :--------------------------------------------------------------------------------------------------------- |\n| No. We can put the 20 samples of the gene that failed on a single plate so we do not have to include IRCs. |\n\n| Do I need to include reference genes ?                                                                 |\n| :----------------------------------------------------------------------------------------------------- |\n| No. We just repeat all samples for the gene that failed and replace the old data with the new results. |\n\nOne of the reference genes failed completely.\n\n| What should I do ?                                                                                                                                                                                                                                                                                     |\n| :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |\n| Depending on the quality of the two remaining reference genes, you should either do nothing or do the same as in the previous example where one of your genes of interest failed. If the two remaining reference genes are stable you can do the normalization with the two remaining reference genes. |\n\nThree samples failed completely.\n\n| What&#39;s the first thing I need to do ?                                                                                                                             |\n| :---------------------------------------------------------------------------------------------------------------------------------------------------------------- |\n| Since they failed completely, they are probably of low quality. Therefore, you have to prepare the samples again, check their quality and then use them for qPCR. |\n\n| Do I need to include IRCs ?                                                                                             |\n| :---------------------------------------------------------------------------------------------------------------------- |\n| Yes. If you want to compare these samples with the samples that didn&#39;t fail, you have to perform inter-run calibration. |\n\nThree samples failed for one of the genes of interest\n\n| What is the first question I need to ask ? |\n| :----------------------------------------- |\n| Is the gene expressed in these samples ?   |\n\n| Is it possible the RNA of these three samples was of low quality ?        |\n| :------------------------------------------------------------------------ |\n| Not likely, the measurements for the other genes in these samples are ok. |\n\nThree samples failed for one of the reference genes\n\n| Can I use the measurements of that reference gene in the non-failing samples for normalization ?                                         |\n| :--------------------------------------------------------------------------------------------------------------------------------------- |\n| No, qbasePLUS requires that you use the same reference genes for all samples so you have to discard all samples for that reference gene. |\n\n### Exercise 7: dilution series for calculating amplification efficiencies\n\nIn my qPCR experiment I want to study 8 new genes for which I had to design new primer pairs in 12 samples (samples of interest and control samples). I want to use 2 technical replicates and 96 well plates.\n\n| What is the first thing I need to do ?                                                        |\n| :-------------------------------------------------------------------------------------------- |\n| Perform a pilot experiment to determine the amplification efficiencies of these primer pairs. |\n\nFor this I need a dilution series of representative cDNA template.\n\n| How many dilutions would you include ?                                           |\n| :------------------------------------------------------------------------------- |\n| A dilution series with 6 dilutions for 8 genes nicely fits into a 96 well plate. |\n\nA few weeks after my initial qPCR experiment I want to test these 8 genes in a new set of samples.\n\n| Do I have to repeat the pilot experiment ?      |\n| :---------------------------------------------- |\n| No, dilution series do not need to be repeated. |&quot;,&quot;You need to do inter-run calibration if you want to compare samples from different runs e.g.:\n\n  - when it is not possible to get all samples for the same gene on the same plate\n  - when you do additional runs weeks or months after your initial experiment\n\nOf course there is a lot of variability between runs on a qPCR instrument:\n\n  - thermal block is not always heating uniformously\n  - quality of the lamp, the filters and the detector decreases over time\n  - data analysis settings on the qPCR instrument (baseline correction and threshold) can be slightly different\n  - efficiency of reagents (polymerase, fluorophores) is variable\n  - optical properties of the plastic plates vary\n\nFortunately, inter-run calibration allows you to eliminate most of this variability.\n\nIn this experiment we will analyze the data from the gene expression experiment (see Analyzing gene expression data in qbase+) together with data from 2 runs (Run4 and Run5) that were done weeks after the initial gene expression experiment.\n\nBecause the data comes from two different experiments spread over time, we have included three inter-run calibrators on the plates: Sample01, Sample02 and Sample03.\n\nThe principle of the IRCs is very similar to that of the reference genes:\nIn theory, the IRCs should have the same NRQ in each run. In practice, the difference in NRQ between two runs is a measure of the inter-run variation and can be used to adjust the NRQs to remove the inter-run variation.\n\n#### Creating a new Experiment\n| Import [Run1](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run1.xls), [Run2](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run2.xls), [Run3(all three in CFX format)](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run3.xls), [Run4](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run4.xls) and [Run5 (the latter two are in qBase format)](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run5.xls).\n| :----------------------------- |\n| Since the data is in files of two different format, you have to do a separate import for each format. So first import Run1, Run2 and Run3, then import Run4 and Run5. You can find the details on how to import CFX files in [](Loading_data_into_qbase+\&quot; title=\&quot;wikilink)Loading data into qbase+.\nThe details of importing qBase files are in [](Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase+\&quot; title=\&quot;wikilink)Analyzing data from a geNorm pilot experiment in qbase+\n\n#### Analyzing the data\n\n| Use assay specific amplification efficiencies.\n| :----------------------------- |\n| You can find the details on how to convert the targets in the **Taking into account amplification efficiencies** section of Analyzing gene expression data in qbase+\n\nIn Analyzing gene expression data in qbase+ we have already checked the stability of the reference genes (see **Normalization** section). We determined that Flexible did not show stable expression.\n\n| Convert Stable and Nonregulated to Reference targets.\n| :----------------------------- |\n| You can find the details on how to convert the targets in the **Normalization** section of Analyzing gene expression data in qbase+\n| Appoint Sample01, Sample02 and Sample03 as IRCs.\n| :----------------------------- |\n| Leave the Analysis wizard by clicking the **Close wizard** button in the top menu.\n\n - Expand **Intermediate results** (red) in the **Project Explorer**\n - Double click **Interrun calibration** (green)\n\nThis opens the **Interrun calibration window**:\n\n - Click the **New** button (blue) to create a IRC\n - Once the IRC is created you have to appoint samples to it: select **Sample01** in the list of **Other samples**\n - Click the **Add Sample** button (purple)\n - Remember that you cannot give IRCs the same name in different runs: the software would think that they are technical replicates spread over different plates (which is not allowed). Therefore, in Run4 and Run5 we have given Sample01 another name: Sample01_2. Select **Sample01_2** in the list of **Other samples**\n - Click the **Add Sample** button (purple)\n\nYou have appointed the first IRC (grey), now do the same for the other two IRCs.\n\nRemember that for each target the variability of the normalized\nexpression levels of the IRCs between different runs will be used to\nadjust the other normalized expression levels of that target gene. The\nadjustment is done by amplifying the normalized expression levels with a\ncalibration factor that is calculated based on the normalized expression\nlevels of the IRCs.\nSince variability between runs is the same for each IRC, you expect that\nall IRCs measure the variability between the runs to the same extent,\nhence leading to similar calibration factors.\n\n| Do these IRCs generate similar calibration factors ?\n| :----------------------------- |\n| Open the **Calibration Factors** tab (red) of the **Interrun calibration window** and look at the result for Duvel:\n\nYou see that IRC2 returns a substantially different calibration factor in Run5 (green) so the validity of this IRC should be interpreted with care.\nFor Leffe the IRCs also gives inconsistent results in Run5. Switch to the results for Leffe by selecting **Leffe** in the **Targets** list (blue)\n| Do you still see the same expression pattern for Palm as you did in the first three runs ?\n| :----------------------------- |\n| Open the target bar chart for Palm.\n\nYou see that the pattern Palm showed in the first three runs (sample01 to sample16): high expression in the odd and low expression in the even samples is reversed in the samples from Run4 and Run5 (sample17 to sample25). In the latter runs you see high expression in the even and low expression in the odd samples. However, without annotation for Run4 and Run5 (which samples are treated and which not) it&#39;s impossible to interpret the bar chart.\n\n1. [Link](http://youtu.be/OJFsuZqNUHs)&quot;,&quot;Since normalization of qPCR data is based on the assumption that the reference targets have the same expression level in all samples it is crucial that the expression of the chosen reference genes is stable.\nHowever, none of the so-called **housekeeping** genes is universally stably expressed.\n\n[Genevestigator](https://www.genevestigator.com/gv/), both the commercial and the free version, contains a tool, called RefGenes, that allows to identify candidate reference genes that display very stable expression in the context that you are working in, typically a certain tissue of a certain organism.\n\nGenevestigator is a platform that contains curated public microarray data from thousands of experiments/conditions.\n\nRefGenes allows you to select the conditions that are relevant for you, e.g. mouse liver, human fibroblasts, or Arabidopsis thaliana leaves. In a next step, RefGenes identifies the genes with the most stable expression in the selected conditions.\n\n## Starting the RefGenes tool\n\n| How to start the RefGenes tool ? |\n| :-------------------------------- |\n| - Open the [RefGenes page](http://www.refgenes.org/rg/).\n - Click **start GENEVESTIGATOR**\n - Click the **Install/Start** button\n - This will automatically open a Genevestigator startup page. Keep this page open during the analysis. Closing this page will close Genevestigator.\n - Login. Also for the free version you need to create an account (use your academic email for this since you will need your vib email to get access to the commercial version).\n - Genevestigator is opened automatically\n\n## The Genevestigator user interface\n\nThe Genevestigator consists of the following components:\n  - **Sample Selection** panel: to choose the experimental conditions you&#39;re interested in (green)\n  - **Gene Selection** panel: to choose the genes you&#39;re interested in (blue)\n  - Center panel shows an overview of all available tools (purple). Once you have selected a tool, the panel will show the results of the analysis that is done by the tool.\n  - **Home** button (red) allows to return to the overview of the tools at any time. The text next to the home button indicates the toolset that you have selected.\n\nClick the **RefGenes** tool at the bottom.\n\n## Using the RefGenes tool to find reference genes\n\n### STEP 1: Choose samples from a biological context similar to those in your qPCR expriment\n\n| How to choose the samples you want to analyze ? |\n| :-------------------------------- |\n|\n - Click the **New** button in the **Sample Selection** panel. The selection of samples defines which data are used for the analysis.\n - Select the organism you&#39;re interested in (in this example: human)\n - Select the array type you want to analyze (in this example: human 133_2).\nFor most organisms Genevestigator contains expression data from multiple types of microarrays, e.g. different generations of Affymetrix GeneChips®. On these arrays, genes are sometimes represented by different sets of probes. To keep the analysis results easily interpretable, data from different array types are not mixed.\n - Click the **Select particular conditions** button to select all samples with a certain annotation, e.g. all data from a certain tissue type.\n - Select the type of conditions (red) you want to base your selection on (in this example: Anatomy). For each type (anatomy, neoplasms, perturbations, development...) you can browse the corresponding ontologies and select the desired condition(s) (green) (in this example: cardiac muscle).\n - Click **OK**\n\nNote that you can select multiple tissues.\nWhen you select samples for use in the RefGenes tool, you have to focus on microarrays from samples that were collected in conditions similar to those in your qPCR experiment. Don&#39;t make a too general selection, e.g. all human samples: you might end up with genes that are stable in most conditions but not in yours. Don&#39;t make a very specific selection either, e.g. human heart samples from patients taking the same medication as yours. If you want to broaden your study later on with samples from other patients, your reference genes might not be valid anymore. It is recommended to select reference genes in the same organism and the same / a similar tissue type as the one that you used in your experiments.\n\n### STEP 2: Select the gene(s) you want to measure in your qPCR experiment\n\nThis step is not essential, but it helps you to see whether your target gene(s) is (are) strongly or weakly expressed in the conditions of interest selected in STEP1. This allows you to search for candidate reference genes in a similar range of expression.\n\n| How to choose the genes you want to analyze ? |\n| :-------------------------------- |\n|\n - Click the **New** button in the **Gene Selection** panel.\n - Enter the name of your target gene in the text area (in this example: GOT1) and click **OK**\n - Open the RefGenes tool (if you haven&#39;t done that already). A red box plot representing the distribution of the expression levels of GOT1 in the 68 selected human heart samples appears in the center panel. As you can see, this gene is highly expressed in heart.\n\n\n\n\n### STEP 3: Find candidate reference genes\n\nThe reference genes that are suggested by GeneVestigator have the\nfollowing characteristics:\n\n  - They have the most stable expression levels across all selected samples (a small boxplot)\n  - Their overall expression level is similar to that of the target gene(s) of your qPCR experiment\n| How to find the candidate reference genes ? |\n| :-------------------------------- |\n|Click the **Run** button in the RefGenes tool. RefGenes will show the top 20 most stable genes with similar expression levels:\n\n\n\n## Exercises\n\n### Finding candidate reference genes in the free version of Genevestigator\n\nNow we will make a more elaborate exercise on finding candidate reference genes. We will do the analysis in the free version of RefGenes but the analysis in the commercial version is very similar.\nSuppose we want to compare the expression stability of the 4 commonly used reference genes for qPCR on mouse liver samples (ACTB, GAPDH, HPRT and TUBB4B) to that of 4 reference genes that are suggested by Genevestigator.\nTo this end we open the RefGenes tool and select the liver samples of the mouse 430_2 arrays.\n\n| Check the expression stability of the 4 commonly used reference genes ? |\n| :-------------------------------- |\n|\n - Click the **New** button in the **Gene Selection** panel to create a new selection. The selection of samples defines which data are used for the analysis.\n - Enter the name of your target gene in the text area (for example: ACTB) and click **OK**\n\nWhen you are using the commercial version, you may enter multiple genes at the same time, in the free version you have to enter them one by one. This means that you have to add the first gene as described above and then add the next gene by clicking the **Add** button and so on...\n\nFinally you end up with an expandable list of the genes you asked for and you can tick or untick them to control the display of their expression data in the main window. When you tick the 4 commonly used reference genes you can see how stable they are expressed in the 651 mouse liver samples that are stored in Genevestigator:\n\nAs you can see, the expression levels of the commonly used reference genes in the selected mouse liver samples is pretty variable which is also confirmed by their relatively high SD values.\nOften there are multiple probe sets for the same gene. When you use the free version you may only choose one probe set per gene so you have to make a choice. How to make that choice ?\nAffymetrix probe set IDs have a certain meaning: what comes after the underscore tells you something about the quality of the probes:\n\n  - **_at** means that all the probes of the probe set hit one known transcript. This is what you want: probes specifically targeting one transcript of one gene\n  - **_a_at** means that all the probes in the probe set hit alternate transcripts from the same gene. This is still ok the probes bind to multiple transcripts but at least the transcripts come from the same gene (splice variants)\n  - **_x_at** means that some of the probes hit transcripts from different genes. This is still not what you want: the expression level is based on a combination of signals of all the probes in a probe set so also probes that cross-hybridize\n  - **_s_at** means that all the probes in the probe set hit transcripts from different genes. This is definitely not what you want: if the probes bind to multiple genes you have no idea whose expression you have measured on the array\n\nSo I always ignore probe sets with s or x. If you have two specific probe sets for a gene, they should more or less give similar signals. If this is not the case, I base my choice upon the expression level that I expect for that gene based on previous qPCR results.\n\nAs you can see, each of these 4 commonly used reference genes has a high expression level. Most genes do not have such high expression levels. In most qPCR experiments your genes of interest will have low or medium expression levels, so these reference genes will not be representative for the genes of interest.\n\nReference genes should ideally have similar expression levels as the genes of interest. Therefore, we will select the four most stably expressed genes with a medium expression level (between 8 and 12) according to the RefGenes tool.\n\n| Select the 4 most stably expressed candidate reference gene with medium expression levels. |\n| :-------------------------------- |\n|\n - Untick all target genes.\n - Click the **Run** button at the top of the main window and check if the range is set correctly\n\nSelect the 4 candidates with the lowest SD: Then, we performed qPCR on a representative set of 16 of our liver samples to measure the expression of these 8 candidate reference genes and analyzed the data ([See how to select the best reference genes using geNorm in qbase+](http://wiki.bits.vib.be/index.php/Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase%2B)).\n\n\n### Finding candidate reference genes in the commercial version of Genevestigator\n\nWe will do the same exercise as above in the commercial version of Genevestigator. The difference between the free and commercial version of RefGenes is the number of target genes you can select. In the free version you have to select one gene and then gradually add all other genes one at a time. The commercial version allows you to load as many target genes as you want simultaneously. As a consequence, you can select multiple probe sets for the same gene.\nAll VIB scientists have free access to the commercial version of Genevestigator via their VIB email address. If you don&#39;t know your VIB email address, check [the Who&#39;s Who of VIB](http://www.vib.be/en/whoiswho/Pages/default.aspx).\n\n  - Open a browser and go to the [Genevestigator website](https://www.genevestigator.com/)\n  - If it&#39;s your **first time to access Genevestigator**, create an account by clicking **join now** button. You will be redirected to a new window in which you will give some personal information including a valid VIB email address. Click **Register** and check your email to activate your new account. Go back to the [GeneVestigator website](https://www.genevestigator.com/)\n  - Choose the research field you want to investigate: **pharma/biomediacal** or **plant biology** by clicking the corresponding button\n  - Click **Start**\n  - Use your VIB email address and password to login to Genevestigator.\n  - This will automatically open a Genevestigator startup page in your browser. Keep this page open during the analysis. Closing this page will close Genevestigator.\n  - Genevestigator is opened automatically\n\nOpen the RefGenes tool by clicking its icon in the **Further tools** secion and select the liver samples of the mouse 430_2 arrays [as explained in the previous exercise](http://wiki.bits.vib.be/index.php/Using_GeneVestigator_to_select_candidate_reference_genes#STEP_1:_Choose_samples_from_a_biological_context_similar_to_those_in_your_qPCR_expriment).\n| Check the expression stability of the 4 commonly used reference genes ? |\n| :-------------------------------- |\n| - Click the **New** button in the **Gene Selection** panel to create a new selection. The selection of samples defines which data are used for the analysis.\n - Enter the names of the 4 commercial reference genes in the text area and click **OK**\n\nI still remove probe sets with an _s or _x since they do not specifically bind to one single gene:\nFinally you end up with an expandable list of the genes you asked for and you can tick or untick them to control the display of their expression data in the main window. By default all probe sets are ticked so you can see how stable the commonly used reference genes are expressed in the 651 mouse liver samples that are stored in Genevestigator:\nAs you can see, the expression levels of the commonly used reference genes in the selected mouse liver samples is pretty variable which is also confirmed by their relatively high SD values.\n\nThe next step of selecting the 4 most stable candidate reference genes with medium expression levels is exactly the same as described above for the free version of RefGenes.\n\n| Create a new gene selection with 20 found candidate reference genes and call it mouse_references. |\n| :-------------------------------- |\n|Click the **New** button at the top of the main window to create a new selection.\n\nTo change the name of the selection right click the name in the **Gene selection** panel and select **Rename**\n\n| Identify perturbations where the mouse_references genes show more than 1,5 fold differential expression using the Condition perturbations tool. |\n| :-------------------------------- |\n|Click the **Home** button at the top to go back to the tools overview page.\n\nClick the **Perturbations** tool in the **Condition Search tools** section\n\n\nMake a **New Sample selection** including all mouse 430_2 arrays.\nUntick all genes except for the first one and filter the long heatmap for at least 1.5 fold change differential expression:\n\n\nYou now get a list of mouse samples in which the gene is not stably expressed so you can check if any of these samples is related to the samples in your study. Hover your mouse over the name of a sample to see more details about the sample.\nYou can do this for each of the candidate reference genes and select the ones that best fit your needs\n\n[Exercise on selecting reference genes for metacaspases in Arabidopsis thaliana](http://wiki.bits.vib.be/index.php/GV_Exercise.1).\n\n\nIn a geNorm pilot experiment you analyze a set of candidate reference genes in a representative set of samples that you want to test in your final experiment. Based on the M-values and CVs that are calculated by qbase+, you can choose the genes that most satisfy the criteria for a good reference gene.\n\n### Exercise 1: reference genes for mouse liver\n\nWe come back on the 8 candidate reference genes that we selected for mouse liver:\n\n  - 4 commonly used reference genes: ACTB, TUBB4B, GAPDH and HPRT\n  - 4 candidate reference genes with very stable medium expression levels selected based on expression data coming from more than 600 microarrays of mouse liver samples using Genevestigator: Gm16845, MUSK, OTOP3, EDN3\n\nWe have measured their expression in a represetative set of 16 of our mouse liver samples, each in triplicate. We will now analyze the stability of these candidate reference genes in our samples.\n\n#### Creating a new Experiment\n\n| Create a new Experiment called GeNormMouse in Project1 |\n| :------------------------------------------- |\n| Open qbase+ or, if the software is already open, click the Launch Wizard button.\n\nYou can find the details on how to create a new experiment in Creating a project and an experiment\n\n#### Loading the data into qbase+\n\nThe data is stored in [the RefGenes folder](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/RefGenes.zip). It consists of 8 Excel files, one file for each candidate reference gene. If you are not working on a BITS laptop, download and unzip the folder.\n\n| Import the data. This files are in qBase format. |\n| :------------------------------------------- |\n| You can find the details on how to start the data import in Loading data into qbase+\n\nUnlike the previous exercise, qbase+ does not allow you to do a quick import this time. In the Import Run window Manual import is selected:\nMake sure that Upload file to Biogazelle support for further analysis is NOT selected and click Next\nMake sure the correct File type is selected (qBase) and click Finish.\nThis file contains the data of the geNorm pilot experiment. In the pilot experiment, 8 candidate reference genes were measured in 16 representative mouse liver samples.\n#### Analyzing the geNorm pilot data\n\n| Specify the aim of the experiment. |\n| :------------------------------------------- |\n| In this experiment we want to select the ideal reference genes for our next experiments so we choose selection of reference genes (geNorm)\n\n| Check the quality of the replicates (use default parameter settings). |\n| :------------------------------------------- |\n| You can find the details on how to check the quality of the replicates in the Checking the quality of technical replicates and controls section of Analyzing gene expression data in qbase+\n\nWe haven&#39;t included any positive or negative controls so you don&#39;t need to show their details.\n\n| Select the Amplification efficiencies strategy you want to use. |\n| :------------------------------------------- |\n| You can find the details on how to select the Amplification effciencies strategy in the Taking into account amplification efficiencies section of Analyzing gene expression data in qbase+\n\nWe haven&#39;t included dilution series nor do we have data from previous qPCR experiments regarding the amplification efficiencies so we choose to use the same efficiency for all genes.\nIt is of course better to include a dilution series for each gene to have an idea of the amplification efficiencies of each primer pair.\n\n| Convert all genes to Reference genes. |\n| :------------------------------------------- |\n| You can convert all the genes simultaneously by selecting Use all targets as candidate reference genes\n\nClick Finish.\n\n| Which genes are you going to use as reference targets in further experiments ? |\n| :------------------------------------------- |\n| Upon clicking Finish, the geNorm window containing the analysis results is automatically opened. The geNorm window consists of three tabs. The tabs are located at the bottom of the window: geNorm M, geNorm V and Interpretation.\nThe first tab, geNorm M, shows a ranking of candidate genes according to their stability, expressed in M values, from the most unstable genes at the left (highest M value) to the best reference genes at the right (lowest M value):\nThe second tab, geNorm V, shows a bar chart that helps determining the optimal number of reference genes to be used in subsequent analyses:\n\nThe number of reference genes is a trade-off between practical considerations and accuracy. It is a waste of resources to quantify more genes than necessary if all candidate reference genes are relatively stably expressed and if normalization factors do not significantly change when more genes are included. However, Biogazelle recommends the minimal use of 3 reference genes and stepwise inclusion of more reference genes until the next gene has no significant contribution to the normalization factors.\nTo determine the need of including more than 3 genes for normalization, pairwise variations Vn/n+1 are calculated between two sequential normalization factors. Simply stated: V is measure of the added value of adding a next reference gene to the analysis. A large variation means that the added gene has a significant effect and should be included.\nIn normal experiments like the Gene expression experiment (see Analyzing gene expression data in qbase+), we only have 3 reference genes so we will see only 1 bar here. But in this geNorm pilot experiment, we analyzed 8 candidate reference genes, so we see 6 bars.\nAll pairwise variations are very low, so even the inclusion of a third gene has no significant effect. Based on a preliminary experiment that was done by Biogazelle, 0.15 is taken as a cut-off value for V, below which the inclusion of an additional reference gene is not required. Normally this threshold is indicated by a green line on the geNorm V bar chart. However since all V-values fall below the threshold in this geNorm pilot experiment, you don’t see this line on the bar chart.\nSo, these results mean that for all subsequent experiments on these samples, two reference genes, EDN3 and MUSK, would be sufficient. However, as stated before, Biogazelle recommends to always include at least three reference genes in case something goes wrong with one of the reference genes (so also include Gm16845). |\nThese are artificial data. But when you read [the paper by Hruz et al., 2011](http://www.biomedcentral.com/1471-2164/12/156/abstract) you see that the genes that are selected by Genevestigator are often outperforming the commonly used reference genes.\n\n### Exercise 2: reference genes for human heart\n\n#### Creating a new Experiment\n\n| Create a new Experiment called GeNormHuman in Project1        |\n| :------------------------------------------------------------ |\n| You can find the details on how to create a new experiment in Creating a project and an experiment |\n\n#### Loading the data into qbase+\n| Import [Run6](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run6.xls) . This file is in qBase format. |\n| :------------------------------------------- |\n| You can find the details on how to start the data import in Loading data into qbase+. Unlike the previous exercise, qbase+ does not allow you to do a quick import this time. In the Import Run window Manual import is selected:\n\nMake sure that Upload file to Biogazelle support for further analysis is NOT selected and click Next. Select the correct File type (qBase) and click Finish. This file contains the data of the geNorm pilot experiment. In the pilot experiment, 10 candidate reference genes were measured in 20 representative samples.\n\n#### Analyzing the geNorm pilot data\n\n| Specify the aim of the experiment.        |\n| :---------------------------------------- |\n| In this experiment we want to select the ideal reference genes for our next experiments so we choose selection of reference genes (geNorm) |\n\n| Check the quality of the replicates and the controls (use default parameter settings). |\n| :------------------------------------------- |\n| You can find the details on how to check the quality of the replicates in the Checking the quality of technical replicates and controls section of Analyzing gene expression data in qbase+\n\nAll replicates and controls have met the quality criteria so there&#39;s no need to inspect them further. |\n| Select the Amplification efficiencies strategy you want to use. |\n| :------------------------------------------- |\n| You can find the details on how to select the Amplification effciencies strategy in the Taking into account amplification efficiencies section of Analyzing gene expression data in qbase+. We haven&#39;t included dilution series nor do we have data from previous qPCR experiments regarding the amplification efficiencies so we choose to use the same efficiency (E=2) for all genes. |\n\nIt is of course better to include a dilution series for each gene to have an idea of the amplification efficiencies of each primer pair.\n| Convert all genes to Reference genes.                                                                         |\n| :------------------------------------------------------------------------------------------------------------ |\n| You can convert all the genes simultaneously by selecting Use all targets as candidate reference genes |\n\nClick Finish.\n\n| Which genes are you going to use as reference targets in further experiments ? |\n| :------------------------------------------- |\n| Upon clicking Finish, the geNorm window containing the analysis results is automatically opened. The geNorm window consists of three tabs. The tabs are located at the bottom of the window: geNorm M, geNorm V and Interpretation.\nThe first tab, geNorm M, shows a ranking of candidate genes according to their stability, expressed in M values, from the most unstable genes at the left (highest M value) to the best reference genes at the right (lowest M value):\nThe second tab, geNorm V, shows a bar chart that helps determining the optimal number of reference genes to be used in subsequent analyses:\n\nThe number of reference genes is a trade-off between practical considerations and accuracy. It is a waste of resources to quantify more genes than necessary if all candidate reference genes are relatively stably expressed and if normalization factors do not significantly change when more genes are included. However, Biogazelle recommends the minimal use of the 3 most stable candidate reference genes and stepwise inclusion of more reference genes until the next gene has no significant contribution to the normalization factors.\nTo determine the need of including more than 3 genes for normalization, pairwise variations Vn/n+1 are calculated between two sequential normalization factors. Simply stated: V is measure of the added value of adding a next reference gene to the analysis. A large variation means that the added gene has a significant effect and should be included.\nIn normal experiments like the Gene expression experiment, see Analyzing_gene_expression_data_in_qbase+, we only have 3 reference genes so we will see only 1 bar here. But in this geNorm pilot experiment, we analyzed 10 candidate reference genes, so we see 8 bars.\nAll pairwise variations are very low, so even the inclusion of a third gene has no significant effect. Based on a preliminary experiment that was done by Biogazelle, 0.15 is taken as a cut-off value for V, below which the inclusion of an additional reference gene is not required. Normally this threshold is indicated by a green line on the geNorm V bar chart. However since all V-values fall below the threshold in this geNorm pilot experiment, you don’t see this line on the bar chart.\nSo, these results mean that for all subsequent experiments on these samples, two reference genes, HPRT1 and GADP, would be sufficient. However, as stated before, Biogazelle recommends to always include at least three reference genes in case something goes wrong with one of the reference genes (so also include YHWAZ). \n\n\n\nIn this example we will analyze data from an artificial expression study containing the following samples:\n  - 6 treated samples: treated1, treated2, ... treated6\n  - 6 control samples: control1, control2, ... control6\n\nIn this study, the expression of the following genes was measured:\n  - 4 commonly used reference genes: ACTB, HPRT, GAPDH, and TUBB4. We have seen in [the previous exercise](http://wiki.bits.vib.be/index.php/Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase%2B#Exercise_1:_reference_genes_for_mouse_liver) that the expression of these reference genes in mouse liver samples is not as stable as generally thought.\n  - 3 genes of interest:\n      - Low: a gene with low expression levels\n      - Medium: a gene with moderate expression levels\n      - HighVar: a gene with low and very noisy expression\n\nIn general, the lower the expression level, the more noisy the qPCR results will become. For each of the genes of interest we have included a run in which a 2-fold difference in expression between control and treated samples was created (Low1, Medium1 and HighVar1) and a run with a 4-fold difference in expression (Low2, Medium2 and HighVar2).\nThere are three technical replicates per reaction. In a second experiment we used [the reference genes that were obtained via Genevestigator](http://wiki.bits.vib.be/index.php/Using_GeneVestigator_to_select_candidate_reference_genes#Finding_candidate_reference_genes_in_the_free_version_of_Genevestigator) and that proved to be [more stably expressed in mouse liver samples than the commonly used references](http://wiki.bits.vib.be/index.php/Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase%2B#Exercise_1:_reference_genes_for_mouse_liver).\nThe data can be found in the NormGenes folder on the BITS laptops or can be downloaded: [from our website](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/NormGenes.zip).\n\n#### Creating a new experiment\n\n| Create a new Experiment called NormGenes1 in Project1 |\n| :---------------------------------------------------- |\n| You can find the details on how to create a new experiment in Creating a project and an experiment |\n\n#### Loading the data\n\n| Import Run1 to Run5. These files are in qBase format. |\n| :---------------------------------------------------- |\n| You can find the details on how to import the data file in the Loading the data into qbase+ section of Analyzing data from a geNorm pilot experiment in qbase+ |\n\nWe are going to compare expression in treated versus untreated samples so we need to tell qbase+ which samples are treated and which not. To this end, we have constructed [a sample properties file](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Sample_Properties_Norm.xlsx) in Excel containing the grouping annotation as a custom property called Treatment.\n\n| Import the Sample Properties file. |\n| :---------------------------------------------------- |\n| You can find the details on how to import the data file in the Adding annotation to the data section of Loading data into qbase+.\n\n| Select to import the custom property. |\n| :---------------------------------------------------- |\n| So as you can see we have 6 treated and 6 untreated samples and we have measured the expression of the 4 commonly used reference genes and 6 genes of interest:\n\n#### Analyzing the data\n\n| Which amplification efficiencies strategy are you going to use ?     |\n| :------------------------------------------------------------------- |\n| You don&#39;t have data of serial dilutions of representative template to build standard curves so the only choice you have is to use the default amplification efficiency (E = 2) for all the genes. |\n\n| Appoint the reference genes. |ACTB, GAPDH, HPRT and TUBB4B are the reference genes:\nYou can find the details on how to appoint reference targets in the Normalization section of Analyzing gene expression data in qbase+ |\n\n| Is the stability of the reference genes ok ?                                                                 |\n| :----------------------------------------------------------------------------------------------------------- |\n| The M and CV values of the reference genes are shown in green so the stability of the reference genes is ok. |\n\n| Which scaling strategy are you going to use ? |Since you have a treated and a control group, it seems logical to use the average of the control group for scaling.\n\nYou can find the details on how to specify the scaling strategy in the Scaling section of Analyzing gene expression data in qbase+\nLook at the target bar charts.\n\n| In the target bar charts plot the average expression level of each group. |In the Grouping section at the bottom of the chart you can select Plot group average: Now do exactly the same for the second experiment with the same genes of interest but with other reference genes. This means that you have to return to the Analysis wizard. To this end, click the Launch wizard button a the top of the page:\n\n| Create a new Experiment called NormGenes2 in Project1 |\n| :---------------------------------------------------- |\n| You can find the details on how to create a new experiment in Creating a project and an experiment |\n\n| Import Run5 to Run9. These files are in qBase format. |\n| :---------------------------------------------------- |\n| You can find the details on how to import the data file in the Loading the data into qbase+ section of Analyzing data from a geNorm pilot experiment in qbase+ |\n\n| Import the Sample Properties file.                    |\n| :---------------------------------------------------- |\n| You can find the details on how to import the data file in the Adding annotation to the data section of Loading data into qbase+. Select to import the custom property. |\n\nSo as you can see we have 6 treated and 6 untreated samples and we have measured the expression of the 4 new reference genes and 6 genes of interest:\n| Appoint the reference genes. |EDN3, Gm16835, MUSK and OTOP3 are the reference genes:\n| :---------------------------------------------------- |\n| You can find the details on how to appoint reference targets in the Normalization section of Analyzing gene expression data in qbase+ |\n\n| Is the stability of the reference genes ok ?                                                                 |\n| :----------------------------------------------------------------------------------------------------------- |\n| The M and CV values of the reference genes are shown in green so the stability of the reference genes is ok. |\n\nAs you can see the M and CV values of these reference genes is much lower than these of the 4 commonly used reference genes pointing to the fact that genes are more stably expressed. It&#39;s not that the commonly used reference genes are bad references. Then qbase+ would not display them in green. It&#39;s just that the other reference genes are more stable. But this can have a big impact on the results of your analysis...\n\n| Use the average of the control group for scaling |You can find the details on how to specify the scaling strategy in the Scaling section of Analyzing gene expression data in qbase+\n\nPlot the average expression level of each group. Now we will compare the target bar charts of the second and the first experiment to assess the influence of the stability of the reference targets on the analysis results.\n\n| How to display the target bar charts of the second and the first experiment next to each other ? |You can display the bar charts next to each other by clicking the tab of the bar chart of the second experiment. Drag the tab to the right while you hold down the mouse button until you see and arrow at the right side of the qbase+ window and a dark grey box in the right half of qbase+ window. Release the mouse button when you see the arrow and the box. Now the two bar charts should be next to each other. Some laptop screens are too small to nicely display the two bar charts next to other. If this is the case switch to full screen mode by double clicking the tab of the first experiment. |\n\nNow you can compare the expression of each gene in the first and in the second experiment.\n\nWhen we do this for HighVar1 for instance, you see that the average expression levels of both groups are the same in the first and the second experiment (check the scales of the Y—axis\\!). Both experiments detect the two-fold difference in expression level between the groups. However, the error bars are much larger in the first experiment than in the second. The variability of the reference genes does have a strong influence on the errors and the size of the error bars will influence the outcome of the statistical test to determine if a gene is differentially expressed or not. The larger the error bars the smaller the less likely it is that the test will say that the groups differ.\n\nRemember that the error bars represent 95% confidence intervals:\n  - if the error bars of the two groups do not overlap: you are certain that the difference between the means of the two groups is significant\n  - if they do not overlap: you know nothing with certainty: the means can be different or they can be the same. Of course the more they overlap the smaller the chance that there is a significant difference between the groups.\n\nCheck out the results of HighVar2. Here, you clearly see the influence of the reference genes. Again, the fourfold difference in expression is detected by both experiments but:\n\n  - the least stable reference genes (experiment 1) give large overlapping error bars\n  - the most stable reference (experiment 2) give smaller, barely overlapping error bars\n\nThis means that in experiment 2, a statistical test will probably declare that HighVar2 is differentially expressed while in experiment 1 this will not be the case. We will test this assumption by performing a statistical test.\n\n#### Statistical analysis of differential expression\n\n| Use a non-parametric test to identify DE genes in experiment 1 ? |\n| :---------------------------------------------------- |\n| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+. In brief, you need to perform the following steps:\n\nOpen the Statistical wizard\n\nThe goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples\n\nUse the Treatment property to identify treated and untreated samples\n\nAnalyze all genes of interest\n\nUse the default settings to perform the non-parametric Mann-Whitney test\n\nAs you can see, none of the genes is considered DE by the very conservative non-parametric test. Additionally most genes have the same p-value. That&#39;s normal when you don&#39;t have many replicates. In our case, we have 6 replicates. Non-parametric tests are based on a ranking of the data values and there are not so many ways to rank 6 data points. This is why you see the same p-values for many genes.\nAs said before, the non-parametric test is very stringent. If the data do come from a normal distribution, the test will generate false positives. Some of the genes might have have been labeled not DE while in fact they are DE so you might have missed some differential expression. The choice of statistical test with 6 biological replicates depends on what you prefer: false negatives or false positives. Most people will choose false negatives since they don&#39;t want to invest time and money in research on a genes that was labeled DE while in fact it is not DE.\n\nSuppose I don&#39;t mind false positives but I don&#39;t want to miss any potential DE genes. In that case, it&#39;s better to go for a t-test. Let&#39;s repeat the test n ow choosing a parametric t-test.\n| Use a t-test to identify DE genes in experiment 1 ? |\n| :---------------------------------------------------- |\n| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+.\nIn brief, you need to perform the following steps:\nOpen the Statistical wizard\nThe goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples\nUse the Treatment property to identify treated and untreated samples\nAnalyze all genes of interest\nDescribe the data set as log-normally distributed\n\nStill none of the genes is considered DE but you do see that the p-values of the t-test are lower than these of the Mann-Whitney test.\n\n| Use a non parametric test to identify DE genes in experiment 2 ? |\n| :---------------------------------------------------- |\n| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+.\nIn brief, you need to perform the following steps:\n\nOpen the Statistical wizard\nThe goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples\nUse the Treatment property to identify treated and untreated samples\nAnalyze all genes of interest\nUse default settings\n\nNow you see that 4 out of the 6 genes are considered DE. This is also what we expected since 3 of our genes of interst have a 4-fold difference in expression level between the two groups. It&#39;s understandable that it&#39;s hard to detect 2-fold differences in expression especially when the expression of the gene is somewhat variable as is the case for Low1 and HighVar1 but a 4-fold difference is a difference that you would like to detect.\n| Use a t-test to identify DE genes in experiment 2 ? |\n| :---------------------------------------------------- |\n| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+.\nIn brief, you need to perform the following steps:\n\nOpen the Statistical wizard\nThe goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples\nUse the Treatment property to identify treated and untreated samples\nAnalyze all genes of interest\nDescribe the data as log normally distributed\n\nAgain the t-test generates lower p-values than the Mann-Whitney test but realize that choosing the t-test when the data is not normally distributed will generate false positives \\!&quot;,&quot;The following exercise will make you familiar with the Primer3Plus software for designing primers for PCR. Primer3Plus is the user-friendly version of Primer3, the standard software for primer design.\n\n### Criteria for qPCR primers\n\nPrimers for qPCR have to follow all the gudelines for regular primers is and an additional set of rules specific for qPCR primers:\n\n  - qPCR products are small: 80-160 bp\n  - use intron or exon-exon junction spanning primers to detect genomic DNA contamination in the RNA samples. Primers of intron spanning primer pairs are located at both sides of an intron and will therefore generate a larger product on genomic DNA (containing the intron). Primer pairs containing an exon-exon junction spanning primer will not generate a PCR product on genomic DNA since the exon-exon junction only exist in the cDNA.\n  - primer length between 9 and 30 bp with an optimum at 20 bp\n  - melting temperature (Tm) of the primers between 58 and 60°C with an optimum at 59°C\n  - maximum Tm difference between the primers of a pair: 2°C\n  - GC content of the primers between 30 and 80% with an optimum at 50%\n  - the 5 nucleotides at the 3&#39; end of the primers should have no more than 2 G or C bases\n  - avoid runs of 4 or more identical nucleotides (especially Gs)\n  - primers must specifically target the region you want to amplify\n\nThere are many programs for designing primers, the most important ones:\n\n  - [Primer3](http://frodo.wi.mit.edu/) \\[1\\] or use it&#39;s user-friendly version: [Primer3Plus](http://primer3plus.com/cgi-bin/dev/primer3plus.cgi)\\[2\\]\n  - [PrimerBLAST](http://www.ncbi.nlm.nih.gov/tools/primer-blast/index.cgi?LINK_LOC=BlastHome)\\[3\\]\n\nThe major downside of Primer3 and Primer3Plus is the fact that you have to check the specificity of the primers yourself. Primer3 will suggest a number of primer pairs that fulfill all of the above requirements, but Primer3 will not check the specificity of the primers. So you have use BLAST to check the specificity of the suggested primer pairs. Very often, the selected primers are not specific and you have to repeat the entire Primer3 analysis.\nIf you use Primer3 and do the BLAST yourself, BLAST against Refseq sequences unless they are not available for the organism you work with or you have reasons to believe that they are not complete (i.e. they do not represent the full genome). For model organisms, you can BLASTagainst the Refseq database. Limit the database to sequences from the organism you work with.\nAdditionally, it is especially important to check that the primers are specific at the 3&#39; end because that&#39;s the site where the polymerase will attach nucleotides. So it is recommended to not use primers that contain long identical stretches (\\&gt; 15nt for primers of 20nt long) to other regions in the genome, and certainly not if these stretches comprise the last nucleotide at the 3&#39; end of the primer.\nFor these exercises we will use PrimerBLAST since [it uses the same algorithm to pick primers as Primer3](http://www.ncbi.nlm.nih.gov/tools/primer-blast/primerinfo.html) \\[4\\] and does the specificity check for you\\!\n\n## Designing qPCR primers for the fruit fly tap gene\n\n### Designing qPCR primers using PrimerBLAST\n\nThe RefSeq entry NM_079400 contains the sequence of the D. melanogaster mRNA coding for tap, the target of Poxn. Tap encodes a bHLH protein expressed in larval chemosensory organs and involved in the response to sugar and salt. We wish to amplify the region encoding the Helix-loop-helix domain. In the sequence of the RefSeq record, the domain is located between position +577 and +745.\nWe want to design qPCR primers for measuring the expression level of the hlh domain using SYBR green. Remember that it is advised to design intron/exon-exon junction spanning primers for qPCR experiments that are based on fluorescent labels to detect/avoid amplification of contaminating genomic DNA.\n\n| Check in NCBIs Gene database if the hlh domain contains any introns ? |\n| :------------------------------ |\n|To know the location of the introns, you need the genomic sequence instead of the mRNA sequence.\n\n - Go to [the NCBI RefSeq record](https://www.ncbi.nlm.nih.gov/nuccore/NM_079400).\n - In the right menu click the link to the **Gene** record\n - In the **Genomic regions, transcripts and products** secton you can see that the gene contains no introns: the transcript is not chopped up into pieces when aligned to the genome. Click [here](https://www.ncbi.nlm.nih.gov/gene/39934) for an example of a gene with introns.\n\nNext, we will design primers to measure the expression of the hlh domain.\n\n| Go to Primer BLAST by using the link in the Refseq record |\n| :------------------------------ |\n|Go back to the RefSeq mRNA record. There, you can go directly to PrimerBLAST by clicking the **Pick Primers** link in the **Analyze this sequence** section of the right menu.\n\nSince you want to measure the expression of the hlh domain you want\nprimers that are located inside the domain.\n\n| Define the range of the sequence in which you want to design primers. |\n| :------------------------------ |\n|You have to specify the range as follows:\n\n| Define the primer parameters to comply with the rules of qPCR primer design: product size and Tm. |\n| :------------------------------ |\n|To comply with the rules for qPCR primer design, you have to change the settings for PCR product size and melting temperature:\n\n| The PrimerBLAST automatically decides to check primer specificity in the Drosophila (organism ID: 7227) RefSeq mRNA database which is exactly what you want. For the qPCR you are going to use RNA samples from fruitfly. This means that the primers will only come into contact with Drosophila mRNAs so you only have to check their specifity in this database. Make sure the last 2 nucleotides are completely specific. |\n| :------------------------------ |\n|You want to ensure that the 3&#39; end of the primers really is specific:\n\nThe PrimerBLAST gives you a set of 9 primer pairs that are specific (according to the criteria that you have specified) and that fulfill all other requirements that you have defined. Look at the detailed report of the first primer pair:\nAll parameters are quite self-explanatory except for the Self complementary and Self 3&#39;complementarity scores.\n\n  - The first score represents the local alignment score when aligning a primer to itself. The scoring system gives 1.00 for a match, -1.00 for a mismatch. This means that the lower the score (the more mismatches), the less likely that the primer binds to itself.\n  - The second score represents the global alignment score when aligning a primer to itself. Here again, the lower the score, the better.\n\nThe scores are followed by information on the specificity of the primer: alignments of the two primers to all target sequences from the database that match the criteria that you specified. In these alignments dots represent matching nucleotides while letters represent mismatches. A specific primer pair will have two alignments (one for each primer): both perfect alignments (all dots) to the sequence you want to amplify.\n\n### Analyzing primer characteristics using OligoAnalyzer\n\n[OligoAnalyzer](https://eu.idtdna.com/calc/analyzer) is a tool implemented by ID\\&amp;T (who sell primers) to check the characteristics of your primers. Take the first primer that is suggested by Primer-BLAST, the pair resulting in a product of 100bp.\n\n| What&#39;s the Tm of the first primer ? |\n| :------------------------------ |\n|Copy the sequence of the first primer in the **Sequence** box, adjust the concentrations to these that are typically used in PCR (see slides) and click **Analyze**:\nAs you can see the predicted melting temperature is 63.9 ºC, which is slightly different from the prediction made by BLAST. There are many different methods to predict Tm and each method will give a different result. Assumed concentrations of primers and ions have an enormous impact on the Tm prediction. So don&#39;t worry about these differences: these are theoretical calculations anyway, the only way to determine Tm values is by doing actual PCR. As long as the difference in Tm between the two primers is not too large, everything is fine.\n\n| What&#39;s the Tm of the second primer ?                                                                                                                                             |\n| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |\n| Copy the sequence of the second primer in the **Sequence** box and click **Analyze**. The predicted melting temperature is also 63.9 ºC , the same Tm as the first primer. |\n\nRemember that the second primer had a large Self complementarity score according to PrimerBLAST.\n\n| Check the self-complementarity of the second primer in OligoAnalyzer ? |\n| :------------------------------ |\n|Click **Self-Dimer**:\n\nYou see that the highest scoring alignment indeed has 6 matches, giving a score of 6 as predicted by PrimerBLAST.\n\n| Do you expect this self-complementarity will give problems in the PCR ? |\n| :------------------------------ |\n|No, the complementarity is concentrated at the center of the primer, not at the 3&#39; end. Since polymerases add bases at the 3’ end of the primer, the primer duplex cannot be extended so it will not give rise to aspecific products. [ID&amp;amp;T](https://eu.idtdna.com/pages/docs/default-source/default-document-library/idt_self-dimer_and_hetero-dimer_results_guide.pdf) recommends to avoid complementary stretches of more than 2 bp at the 3’ end.\nHowever, even if the primer dimer cannot be extended, it could interfere when its formation competes with the annealing of primer and target. This is only the case when the stability of the dimer is similar to the stability of a perfectly matched primer-target duplex. The stability of the perfectly matched duplex is shown as a Maximum Delta G at the top of results. So non-extendable dimer structures that are much shorter than the intended duplex, as we have here, are not going to disrupt the PCR reaction.\nIt is advised to review all possible interactions between primers so both Self-Dimer (primers binding to themselves) and Hetero-Dimer (primers binding to each other) interactions between primers are examined.\n\n| Is it likely that the primers bind to each other ? |\n| :------------------------------ |\n|Click **Hetero-Dimer**:\n\nThis opens a text box to enter the second primer. Click **Analyze**. There is one structure (the fourth one) that looks problematic because there is a stretch of 3 matching nucleotides at the 3&#39;end of one of the primers.\n\nSo you might consider taking a look at the second pair of primers that PrimerBLAST suggests. On the other hand, this structure is has relatively high free energy (delta G). The structure with the lowest total free energy, the target-primer duplex, is most important because it will dominate in solution. Structures with higher free energy are less stable and will be present in smaller amounts in the reaction mixture.\n\nTake a look at the second primer pair that was suggested by PrimerBLAST.\n\n| Is it likely that these primers bind to each other ?                                 |\n| :----------------------------------------------------------------------------------- |\n| No these primers do not form duplex structures that could pose a problem during PCR. |\n\n## Designing qPCR primers for the human F9 gene\n\n### Designing qPCR primers using PrimerBLAST\n\nThe RefSeq entry NM_000133.3 contains the sequence of the human mRNA coding for coagulation factor F9. The gene contains 8 coding exons and gives rise to a transcript of 2780 bp encoding a protein of 461 amino acids.\nNext, we want to design primers to measure the expression of the F9 gene.\n\nGo to [the RefSeq record of this transcript](http://www.ncbi.nlm.nih.gov/nuccore/NM_000133.3) to study its structure. When you scroll down to the **features** section you see that the CDS is located from position 40 to position 1415. Since RNA degradation starts at the 5&#39;end of transcripts, we don&#39;t want to pick primers at the 5&#39;end. On the other hand, we don&#39;t want to pick primers in the long 3&#39;UTR either because it doesn&#39;t contain any introns (the exons are all coding) and we want to design exon-exon junction or intron spanning primers.\nLet&#39;s try to find exon-exon junction spanning primers between position 400 and 1600, with optimal anneal temperature = 60.\n\n| Find primers that fulfill the above defined criteria |\n| :------------------------------ |\n|Go to [PrimerBLAST](http://www.ncbi.nlm.nih.gov/tools/primer-blast/index.cgi?LINK_LOC=BlastHome) and fill in the form as follows:\n\nExclude predicted sequences in the database to search in .\n\n| Find primers that fulfill the above defined criteria |\n| :------------------------------ |\n|Go to [PrimerBLAST](http://www.ncbi.nlm.nih.gov/tools/primer-blast/index.cgi?LINK_LOC=BlastHome) and fill in the remainder of the form as follows:\n\nThe PrimerBLAST gives you a set of 10 primer pairs. Look at the detailed\nreport of the first primer pair:\n\nAs you can see the primers are not specific: they can bind to various other targets albeit with lower affinity because of the mismatches . The best option seems to be primer pair 7, which binds to both F9 transcript variants and potentially to one unintended target, but as you can see the last nucleotide at the 3&#39; end of both primers are specific.\n\n### In silico PCR in the UCSC Browser\n\nWe will proceed using the third primer pair Primer-BLAST suggests. You can visualize the PCR product (and additional annotation) in the UCSC Genome Browser using [UCSC&#39;s In Silico PCR tool](http://genome.ucsc.edu/cgi-bin/hgPcr).\nSelect the most recent version of the human genome and paste the sequences of forward and reverse primers in their respective boxes. Click **submit**\nNormally, this returns the location and the sequence of the PCR product but our primer pair doesn&#39;t return a match. When you think about this was to be expected since we are working with exon-exon junction spanning primers that are not able to match the genome sequence. So checking SNPs is not so straight-forward in the case of exon-exon junction spanning primers.\nWe will repeat the primer search now searching for intron-spanning primers to show you how to use the in silico PCR tool. Taking into account the fact that the results for the exon-exon junction spanning primers were so messy we will make the search more stringent this time:\n\n  - We will the minimum number of mismatches to 4\n  - and at least 3 mismatches in the last 3 bps at the 3&#39;end\n\n| Find intron spanning primers that fulfill the above defined criteria |\n| :------------------------------ |\n|Go back to the Primer-BLAST and fill in the form like in the previous exercise except that they should span an intron:\n\nPrimer-BLAST returns 10 primer pairs. Again the seventh primer pair is\nthe specific one.\n\n| Take the seventh suggested primer pair and check for SNPs in the UCSC Browser |\n| :------------------------------ |\n|Go to [PrimerBLAST](http://www.ncbi.nlm.nih.gov/tools/primer-blast/index.cgi?LINK_LOC=BlastHome) and paste the sequences of forward and reverse primers in their respective boxes.\nThis time the search finds a PCR product:\n\nClicking the location visualizes the PCR product in the UCSC genome browser. Remove unnecessary trancks by right clicking the box in front of them and selecting **hide**\n\nAdd tracks showing relevant annotation like position of SNPs...\n\nSetting the SNPs track from **hide** to **full** shows the SNPs in the browser. Center the forward primer by grabbing and dragging it to the center.\n\nZoom in to **base** display to see if the forward primer is matching any SNPs.\n\nAs you can see the forward primer does match two SNPs but none of them are located near the 3&#39;end of the primer.\n\n1.  &lt;http://frodo.wi.mit.edu/&gt;\n2.  &lt;http://primer3plus.com/cgi-bin/dev/primer3plus.cgi&gt;\n3.  &lt;http://www.ncbi.nlm.nih.gov/tools/primer-blast/index.cgi?LINK_LOC=BlastHome&gt;\n4.  &lt;http://www.ncbi.nlm.nih.gov/tools/primer-blast/primerinfo.html&gt;&quot;,&quot;qbase+ is software to visualize and analyze qPCR data. It allows you to perform various types of analyses:\n  - statistical analysis of gene expression\n  - advanced copy number analysis\n  - miRNA profiling\n  - ChIP-qPCR analysis\n# Installation and licensing\nYou can find the installation instructions on [VIB qbase+ support page](https://www.bits.vib.be/index.php/software-overview/qbaseplus)\nVIB only offers qbase+ to VIB scientists, you need a valid VIB email address to run the software. Biogazelle (the company who has developed the software) have written a manual with instructions on how to use the software. Download [Biogazelle&#39;s user manual](https://www.biogazelle.com/system/files/manuals/qbaseplus_manual_0.pdf). Before you can download the manual you have to log on to [the qbase+ website](https://www.qbaseplus.com/) using your qbase+ account. Use your VIB email address for setting up this account.\n# Training material\n  - [slides](http://data.bits.vib.be/pub/trainingen/qbasePLUS/qbase_2018.pdf)\n  \n  **Extra**\n  - [clean log10 transformed CNRQs](http://data.bits.vib.be/pub/trainingen/qbasePLUS/Log10CNRQsClean.xlsx) for checking normality in Prism\n  - [clean untransformed CNRQs](http://data.bits.vib.be/pub/trainingen/qbasePLUS/CNRQsClean.xlsx) for visualization in Prism\n  - [R script](http://data.bits.vib.be/pub/trainingen/qbasePLUS/qPCR.R) for analysis and visualization\n  - [log10 transformed CNRQs of control samples](http://data.bits.vib.be/pub/trainingen/qbasePLUS/resultslog.csv) for analysis and visualization in R\n  - [log10 transformed CNRQs of treated samples](http://data.bits.vib.be/pub/trainingen/qbasePLUS/resultslogTreated.csv) for analysis and visualization in R\n&quot;,&quot;&lt;?xml version=\&quot;1.0\&quot; encoding=\&quot;utf-8\&quot;?&gt;{% if page.xsl %}&lt;?xml-stylesheet type=\&quot;text/xml\&quot; href=\&quot;{{ &#39;/feed.xslt.xml&#39; | absolute_url }}\&quot;?&gt;{% endif %}&lt;feed xmlns=\&quot;http://www.w3.org/2005/Atom\&quot; {% if site.lang %}xml:lang=\&quot;{{ site.lang }}\&quot;{% endif %}&gt;&lt;generator uri=\&quot;https://jekyllrb.com/\&quot; version=\&quot;{{ jekyll.version }}\&quot;&gt;Jekyll&lt;/generator&gt;&lt;link href=\&quot;{{ page.url | absolute_url }}\&quot; rel=\&quot;self\&quot; type=\&quot;application/atom+xml\&quot; /&gt;&lt;link href=\&quot;{{ &#39;/&#39; | absolute_url }}\&quot; rel=\&quot;alternate\&quot; type=\&quot;text/html\&quot; {% if site.lang %}hreflang=\&quot;{{ site.lang }}\&quot; {% endif %}/&gt;&lt;updated&gt;{{ site.time | date_to_xmlschema }}&lt;/updated&gt;&lt;id&gt;{{ page.url | absolute_url | xml_escape }}&lt;/id&gt;{% assign title = site.title | default: site.name %}{% if page.collection != \&quot;posts\&quot; %}{% assign collection = page.collection | capitalize %}{% assign title = title | append: \&quot; | \&quot; | append: collection %}{% endif %}{% if page.category %}{% assign category = page.category | capitalize %}{% assign title = title | append: \&quot; | \&quot; | append: category %}{% endif %}{% if title %}&lt;title type=\&quot;html\&quot;&gt;{{ title | smartify | xml_escape }}&lt;/title&gt;{% endif %}{% if site.description %}&lt;subtitle&gt;{{ site.description | xml_escape }}&lt;/subtitle&gt;{% endif %}{% if site.author %}&lt;author&gt;&lt;name&gt;{{ site.author.name | default: site.author | xml_escape }}&lt;/name&gt;{% if site.author.email %}&lt;email&gt;{{ site.author.email | xml_escape }}&lt;/email&gt;{% endif %}{% if site.author.uri %}&lt;uri&gt;{{ site.author.uri | xml_escape }}&lt;/uri&gt;{% endif %}&lt;/author&gt;{% endif %}{% assign posts = site[page.collection] | where_exp: \&quot;post\&quot;, \&quot;post.draft != true\&quot; | sort: \&quot;date\&quot; | reverse %}{% if page.category %}{% assign posts = posts | where: \&quot;category\&quot;,page.category %}{% endif %}{% for post in posts limit: 10 %}&lt;entry{% if post.lang %}{{\&quot; \&quot;}}xml:lang=\&quot;{{ post.lang }}\&quot;{% endif %}&gt;&lt;title type=\&quot;html\&quot;&gt;{{ post.title | smartify | strip_html | normalize_whitespace | xml_escape }}&lt;/title&gt;&lt;link href=\&quot;{{ post.url | absolute_url }}\&quot; rel=\&quot;alternate\&quot; type=\&quot;text/html\&quot; title=\&quot;{{ post.title | xml_escape }}\&quot; /&gt;&lt;published&gt;{{ post.date | date_to_xmlschema }}&lt;/published&gt;&lt;updated&gt;{{ post.last_modified_at | default: post.date | date_to_xmlschema }}&lt;/updated&gt;&lt;id&gt;{{ post.id | absolute_url | xml_escape }}&lt;/id&gt;&lt;content type=\&quot;html\&quot; xml:base=\&quot;{{ post.url | absolute_url | xml_escape }}\&quot;&gt;{{ post.content | strip | xml_escape }}&lt;/content&gt;{% assign post_author = post.author | default: post.authors[0] | default: site.author %}{% assign post_author = site.data.authors[post_author] | default: post_author %}{% assign post_author_email = post_author.email | default: nil %}{% assign post_author_uri = post_author.uri | default: nil %}{% assign post_author_name = post_author.name | default: post_author %}&lt;author&gt;&lt;name&gt;{{ post_author_name | default: \&quot;\&quot; | xml_escape }}&lt;/name&gt;{% if post_author_email %}&lt;email&gt;{{ post_author_email | xml_escape }}&lt;/email&gt;{% endif %}{% if post_author_uri %}&lt;uri&gt;{{ post_author_uri | xml_escape }}&lt;/uri&gt;{% endif %}&lt;/author&gt;{% if post.category %}&lt;category term=\&quot;{{ post.category | xml_escape }}\&quot; /&gt;{% endif %}{% for tag in post.tags %}&lt;category term=\&quot;{{ tag | xml_escape }}\&quot; /&gt;{% endfor %}{% if post.excerpt and post.excerpt != empty %}&lt;summary type=\&quot;html\&quot;&gt;{{ post.excerpt | strip_html | normalize_whitespace | xml_escape }}&lt;/summary&gt;{% endif %}{% assign post_image = post.image.path | default: post.image %}{% if post_image %}{% unless post_image contains \&quot;://\&quot; %}{% assign post_image = post_image | absolute_url %}{% endunless %}&lt;media:thumbnail xmlns:media=\&quot;http://search.yahoo.com/mrss/\&quot; url=\&quot;{{ post_image | xml_escape }}\&quot; /&gt;{% endif %}&lt;/entry&gt;{% endfor %}&lt;/feed&gt;&quot;],&quot;posts&quot;:[],&quot;static_files&quot;:[{&quot;name&quot;:&quot;CODEOWNERS&quot;,&quot;path&quot;:&quot;/CODEOWNERS&quot;,&quot;basename&quot;:&quot;CODEOWNERS&quot;,&quot;extname&quot;:&quot;&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;CODE_OF_CONDUCT.md&quot;,&quot;path&quot;:&quot;/CODE_OF_CONDUCT.md&quot;,&quot;basename&quot;:&quot;CODE_OF_CONDUCT&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Procfile&quot;,&quot;path&quot;:&quot;/Procfile&quot;,&quot;basename&quot;:&quot;Procfile&quot;,&quot;extname&quot;:&quot;&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;app.json&quot;,&quot;path&quot;:&quot;/app.json&quot;,&quot;basename&quot;:&quot;app&quot;,&quot;extname&quot;:&quot;.json&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;VIB.css&quot;,&quot;path&quot;:&quot;/assets/css/VIB.css&quot;,&quot;basename&quot;:&quot;VIB&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-08-09 12:47:05 +0000&quot;},{&quot;name&quot;:&quot;academicons.css&quot;,&quot;path&quot;:&quot;/assets/css/academicons.css&quot;,&quot;basename&quot;:&quot;academicons&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap-theme.css&quot;,&quot;path&quot;:&quot;/assets/css/bootstrap-theme.css&quot;,&quot;basename&quot;:&quot;bootstrap-theme&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap-theme.min.css&quot;,&quot;path&quot;:&quot;/assets/css/bootstrap-theme.min.css&quot;,&quot;basename&quot;:&quot;bootstrap-theme.min&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap-toc.min.css&quot;,&quot;path&quot;:&quot;/assets/css/bootstrap-toc.min.css&quot;,&quot;basename&quot;:&quot;bootstrap-toc.min&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap.css&quot;,&quot;path&quot;:&quot;/assets/css/bootstrap.css&quot;,&quot;basename&quot;:&quot;bootstrap&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap.min.css&quot;,&quot;path&quot;:&quot;/assets/css/bootstrap.min.css&quot;,&quot;basename&quot;:&quot;bootstrap.min&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;font-awesome.css&quot;,&quot;path&quot;:&quot;/assets/css/font-awesome.css&quot;,&quot;basename&quot;:&quot;font-awesome&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;slides.css&quot;,&quot;path&quot;:&quot;/assets/css/slides.css&quot;,&quot;basename&quot;:&quot;slides&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;syntax_highlighting.css&quot;,&quot;path&quot;:&quot;/assets/css/syntax_highlighting.css&quot;,&quot;basename&quot;:&quot;syntax_highlighting&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Dense-Regular.otf&quot;,&quot;path&quot;:&quot;/assets/fonts/Dense-Regular.otf&quot;,&quot;basename&quot;:&quot;Dense-Regular&quot;,&quot;extname&quot;:&quot;.otf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Dense-Regular.ttf&quot;,&quot;path&quot;:&quot;/assets/fonts/Dense-Regular.ttf&quot;,&quot;basename&quot;:&quot;Dense-Regular&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;FontAwesome.otf&quot;,&quot;path&quot;:&quot;/assets/fonts/FontAwesome.otf&quot;,&quot;basename&quot;:&quot;FontAwesome&quot;,&quot;extname&quot;:&quot;.otf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;MaterialIcons.woff&quot;,&quot;path&quot;:&quot;/assets/fonts/MaterialIcons.woff&quot;,&quot;basename&quot;:&quot;MaterialIcons&quot;,&quot;extname&quot;:&quot;.woff&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;MaterialIcons.woff2&quot;,&quot;path&quot;:&quot;/assets/fonts/MaterialIcons.woff2&quot;,&quot;basename&quot;:&quot;MaterialIcons&quot;,&quot;extname&quot;:&quot;.woff2&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;OpenSans.otf&quot;,&quot;path&quot;:&quot;/assets/fonts/OpenSans.otf&quot;,&quot;basename&quot;:&quot;OpenSans&quot;,&quot;extname&quot;:&quot;.otf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;OpenSans.woff&quot;,&quot;path&quot;:&quot;/assets/fonts/OpenSans.woff&quot;,&quot;basename&quot;:&quot;OpenSans&quot;,&quot;extname&quot;:&quot;.woff&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;OpenSans.woff2&quot;,&quot;path&quot;:&quot;/assets/fonts/OpenSans.woff2&quot;,&quot;basename&quot;:&quot;OpenSans&quot;,&quot;extname&quot;:&quot;.woff2&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;academicons.eot&quot;,&quot;path&quot;:&quot;/assets/fonts/academicons.eot&quot;,&quot;basename&quot;:&quot;academicons&quot;,&quot;extname&quot;:&quot;.eot&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;academicons.svg&quot;,&quot;path&quot;:&quot;/assets/fonts/academicons.svg&quot;,&quot;basename&quot;:&quot;academicons&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;academicons.ttf&quot;,&quot;path&quot;:&quot;/assets/fonts/academicons.ttf&quot;,&quot;basename&quot;:&quot;academicons&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;academicons.woff&quot;,&quot;path&quot;:&quot;/assets/fonts/academicons.woff&quot;,&quot;basename&quot;:&quot;academicons&quot;,&quot;extname&quot;:&quot;.woff&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;fontawesome-webfont.eot&quot;,&quot;path&quot;:&quot;/assets/fonts/fontawesome-webfont.eot&quot;,&quot;basename&quot;:&quot;fontawesome-webfont&quot;,&quot;extname&quot;:&quot;.eot&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;fontawesome-webfont.svg&quot;,&quot;path&quot;:&quot;/assets/fonts/fontawesome-webfont.svg&quot;,&quot;basename&quot;:&quot;fontawesome-webfont&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;fontawesome-webfont.ttf&quot;,&quot;path&quot;:&quot;/assets/fonts/fontawesome-webfont.ttf&quot;,&quot;basename&quot;:&quot;fontawesome-webfont&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;fontawesome-webfont.woff&quot;,&quot;path&quot;:&quot;/assets/fonts/fontawesome-webfont.woff&quot;,&quot;basename&quot;:&quot;fontawesome-webfont&quot;,&quot;extname&quot;:&quot;.woff&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;fontawesome-webfont.woff2&quot;,&quot;path&quot;:&quot;/assets/fonts/fontawesome-webfont.woff2&quot;,&quot;basename&quot;:&quot;fontawesome-webfont&quot;,&quot;extname&quot;:&quot;.woff2&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;glyphicons-halflings-regular.eot&quot;,&quot;path&quot;:&quot;/assets/fonts/glyphicons-halflings-regular.eot&quot;,&quot;basename&quot;:&quot;glyphicons-halflings-regular&quot;,&quot;extname&quot;:&quot;.eot&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;glyphicons-halflings-regular.svg&quot;,&quot;path&quot;:&quot;/assets/fonts/glyphicons-halflings-regular.svg&quot;,&quot;basename&quot;:&quot;glyphicons-halflings-regular&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;glyphicons-halflings-regular.ttf&quot;,&quot;path&quot;:&quot;/assets/fonts/glyphicons-halflings-regular.ttf&quot;,&quot;basename&quot;:&quot;glyphicons-halflings-regular&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;glyphicons-halflings-regular.woff&quot;,&quot;path&quot;:&quot;/assets/fonts/glyphicons-halflings-regular.woff&quot;,&quot;basename&quot;:&quot;glyphicons-halflings-regular&quot;,&quot;extname&quot;:&quot;.woff&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;glyphicons-halflings-regular.woff2&quot;,&quot;path&quot;:&quot;/assets/fonts/glyphicons-halflings-regular.woff2&quot;,&quot;basename&quot;:&quot;glyphicons-halflings-regular&quot;,&quot;extname&quot;:&quot;.woff2&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;GTN-60px.png&quot;,&quot;path&quot;:&quot;/assets/images/GTN-60px.png&quot;,&quot;basename&quot;:&quot;GTN-60px&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;GTN.png&quot;,&quot;path&quot;:&quot;/assets/images/GTN.png&quot;,&quot;basename&quot;:&quot;GTN&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;GTNLogo1000.png&quot;,&quot;path&quot;:&quot;/assets/images/GTNLogo1000.png&quot;,&quot;basename&quot;:&quot;GTNLogo1000&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bioinformatics_core_rgb_neg.png&quot;,&quot;path&quot;:&quot;/assets/images/bioinformatics_core_rgb_neg.png&quot;,&quot;basename&quot;:&quot;bioinformatics_core_rgb_neg&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-black-logo.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-black-logo.png&quot;,&quot;basename&quot;:&quot;qr-gtn-black-logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-black.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-black.png&quot;,&quot;basename&quot;:&quot;qr-gtn-black&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-darkblue-logo.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-darkblue-logo.png&quot;,&quot;basename&quot;:&quot;qr-gtn-darkblue-logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-darkblue.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-darkblue.png&quot;,&quot;basename&quot;:&quot;qr-gtn-darkblue&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-darkgrey-logo.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-darkgrey-logo.png&quot;,&quot;basename&quot;:&quot;qr-gtn-darkgrey-logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-darkgrey.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-darkgrey.png&quot;,&quot;basename&quot;:&quot;qr-gtn-darkgrey&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-logo.xcf&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-logo.xcf&quot;,&quot;basename&quot;:&quot;qr-gtn-logo&quot;,&quot;extname&quot;:&quot;.xcf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-white-logo.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-white-logo.png&quot;,&quot;basename&quot;:&quot;qr-gtn-white-logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-white.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-white.png&quot;,&quot;basename&quot;:&quot;qr-gtn-white&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn.xcf&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn.xcf&quot;,&quot;basename&quot;:&quot;qr-gtn&quot;,&quot;extname&quot;:&quot;.xcf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;vib_notag_neg_rgb.jpg&quot;,&quot;path&quot;:&quot;/assets/images/vib_notag_neg_rgb.jpg&quot;,&quot;basename&quot;:&quot;vib_notag_neg_rgb&quot;,&quot;extname&quot;:&quot;.jpg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap-toc.min.js&quot;,&quot;path&quot;:&quot;/assets/js/bootstrap-toc.min.js&quot;,&quot;basename&quot;:&quot;bootstrap-toc.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap.js&quot;,&quot;path&quot;:&quot;/assets/js/bootstrap.js&quot;,&quot;basename&quot;:&quot;bootstrap&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap.min.js&quot;,&quot;path&quot;:&quot;/assets/js/bootstrap.min.js&quot;,&quot;basename&quot;:&quot;bootstrap.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;clipboard.min.js&quot;,&quot;path&quot;:&quot;/assets/js/clipboard.min.js&quot;,&quot;basename&quot;:&quot;clipboard.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;details-element-polyfill.js&quot;,&quot;path&quot;:&quot;/assets/js/details-element-polyfill.js&quot;,&quot;basename&quot;:&quot;details-element-polyfill&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;jquery.slim.min.js&quot;,&quot;path&quot;:&quot;/assets/js/jquery.slim.min.js&quot;,&quot;basename&quot;:&quot;jquery.slim.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;main.js&quot;,&quot;path&quot;:&quot;/assets/js/main.js&quot;,&quot;basename&quot;:&quot;main&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;popper.min.js&quot;,&quot;path&quot;:&quot;/assets/js/popper.min.js&quot;,&quot;basename&quot;:&quot;popper.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;remark-latest.min.js&quot;,&quot;path&quot;:&quot;/assets/js/remark-latest.min.js&quot;,&quot;basename&quot;:&quot;remark-latest.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;favicon.ico&quot;,&quot;path&quot;:&quot;/favicon.ico&quot;,&quot;basename&quot;:&quot;favicon&quot;,&quot;extname&quot;:&quot;.ico&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;galaxy_instances.yaml&quot;,&quot;path&quot;:&quot;/galaxy_instances.yaml&quot;,&quot;basename&quot;:&quot;galaxy_instances&quot;,&quot;extname&quot;:&quot;.yaml&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;LICENSE.txt&quot;,&quot;path&quot;:&quot;/shared/fonts/NotoSans/LICENSE.txt&quot;,&quot;basename&quot;:&quot;LICENSE&quot;,&quot;extname&quot;:&quot;.txt&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;NotoSans-Bold.ttf&quot;,&quot;path&quot;:&quot;/shared/fonts/NotoSans/NotoSans-Bold.ttf&quot;,&quot;basename&quot;:&quot;NotoSans-Bold&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;NotoSans-BoldItalic.ttf&quot;,&quot;path&quot;:&quot;/shared/fonts/NotoSans/NotoSans-BoldItalic.ttf&quot;,&quot;basename&quot;:&quot;NotoSans-BoldItalic&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;NotoSans-Italic.ttf&quot;,&quot;path&quot;:&quot;/shared/fonts/NotoSans/NotoSans-Italic.ttf&quot;,&quot;basename&quot;:&quot;NotoSans-Italic&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;NotoSans-Regular.ttf&quot;,&quot;path&quot;:&quot;/shared/fonts/NotoSans/NotoSans-Regular.ttf&quot;,&quot;basename&quot;:&quot;NotoSans-Regular&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Excelerate_whitebackground.png&quot;,&quot;path&quot;:&quot;/shared/images/Excelerate_whitebackground.png&quot;,&quot;basename&quot;:&quot;Excelerate_whitebackground&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;FreeBayes_settings.png&quot;,&quot;path&quot;:&quot;/shared/images/FreeBayes_settings.png&quot;,&quot;basename&quot;:&quot;FreeBayes_settings&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;GTNLogo1000.png&quot;,&quot;path&quot;:&quot;/shared/images/GTNLogo1000.png&quot;,&quot;basename&quot;:&quot;GTNLogo1000&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;MA_plot.png&quot;,&quot;path&quot;:&quot;/shared/images/MA_plot.png&quot;,&quot;basename&quot;:&quot;MA_plot&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;RNAseq_histiry_image.key&quot;,&quot;path&quot;:&quot;/shared/images/RNAseq_histiry_image.key&quot;,&quot;basename&quot;:&quot;RNAseq_histiry_image&quot;,&quot;extname&quot;:&quot;.key&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Screen Shot 2016-04-21 at 3.55.19 PM.png&quot;,&quot;path&quot;:&quot;/shared/images/Screen Shot 2016-04-21 at 3.55.19 PM.png&quot;,&quot;basename&quot;:&quot;Screen Shot 2016-04-21 at 3.55.19 PM&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Screen Shot 2016-12-06 at 18.14.25.png&quot;,&quot;path&quot;:&quot;/shared/images/Screen Shot 2016-12-06 at 18.14.25.png&quot;,&quot;basename&quot;:&quot;Screen Shot 2016-12-06 at 18.14.25&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;accepted_hits_1.png&quot;,&quot;path&quot;:&quot;/shared/images/accepted_hits_1.png&quot;,&quot;basename&quot;:&quot;accepted_hits_1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;accepted_hits_2.png&quot;,&quot;path&quot;:&quot;/shared/images/accepted_hits_2.png&quot;,&quot;basename&quot;:&quot;accepted_hits_2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;ansible_logo.png&quot;,&quot;path&quot;:&quot;/shared/images/ansible_logo.png&quot;,&quot;basename&quot;:&quot;ansible_logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bioconda_logo.png&quot;,&quot;path&quot;:&quot;/shared/images/bioconda_logo.png&quot;,&quot;basename&quot;:&quot;bioconda_logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bw.png&quot;,&quot;path&quot;:&quot;/shared/images/bw.png&quot;,&quot;basename&quot;:&quot;bw&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bw_b_rank.png&quot;,&quot;path&quot;:&quot;/shared/images/bw_b_rank.png&quot;,&quot;basename&quot;:&quot;bw_b_rank&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_q1.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_q1.png&quot;,&quot;basename&quot;:&quot;bwt_q1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_q2.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_q2.png&quot;,&quot;basename&quot;:&quot;bwt_q2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_q3.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_q3.png&quot;,&quot;basename&quot;:&quot;bwt_q3&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_q4.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_q4.png&quot;,&quot;basename&quot;:&quot;bwt_q4&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_q5.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_q5.png&quot;,&quot;basename&quot;:&quot;bwt_q5&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_rev.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_rev.png&quot;,&quot;basename&quot;:&quot;bwt_rev&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_rev2.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_rev2.png&quot;,&quot;basename&quot;:&quot;bwt_rev2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;cloudman_based_server_landing_page.png&quot;,&quot;path&quot;:&quot;/shared/images/cloudman_based_server_landing_page.png&quot;,&quot;basename&quot;:&quot;cloudman_based_server_landing_page&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;cloudman_management_console_showing_autoscaled_w_1_worker.png&quot;,&quot;path&quot;:&quot;/shared/images/cloudman_management_console_showing_autoscaled_w_1_worker.png&quot;,&quot;basename&quot;:&quot;cloudman_management_console_showing_autoscaled_w_1_worker&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;collections_video.png&quot;,&quot;path&quot;:&quot;/shared/images/collections_video.png&quot;,&quot;basename&quot;:&quot;collections_video&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;conda_logo.png&quot;,&quot;path&quot;:&quot;/shared/images/conda_logo.png&quot;,&quot;basename&quot;:&quot;conda_logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;create_index.png&quot;,&quot;path&quot;:&quot;/shared/images/create_index.png&quot;,&quot;basename&quot;:&quot;create_index&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;dT_random.png&quot;,&quot;path&quot;:&quot;/shared/images/dT_random.png&quot;,&quot;basename&quot;:&quot;dT_random&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;deNBI.png&quot;,&quot;path&quot;:&quot;/shared/images/deNBI.png&quot;,&quot;basename&quot;:&quot;deNBI&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;deseq2_interface.png&quot;,&quot;path&quot;:&quot;/shared/images/deseq2_interface.png&quot;,&quot;basename&quot;:&quot;deseq2_interface&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;deseq2_output.png&quot;,&quot;path&quot;:&quot;/shared/images/deseq2_output.png&quot;,&quot;basename&quot;:&quot;deseq2_output&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;development_process.png&quot;,&quot;path&quot;:&quot;/shared/images/development_process.png&quot;,&quot;basename&quot;:&quot;development_process&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;diff.png&quot;,&quot;path&quot;:&quot;/shared/images/diff.png&quot;,&quot;basename&quot;:&quot;diff&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;dispersion.png&quot;,&quot;path&quot;:&quot;/shared/images/dispersion.png&quot;,&quot;basename&quot;:&quot;dispersion&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;dna_rna.png&quot;,&quot;path&quot;:&quot;/shared/images/dna_rna.png&quot;,&quot;basename&quot;:&quot;dna_rna&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;docker_logo.png&quot;,&quot;path&quot;:&quot;/shared/images/docker_logo.png&quot;,&quot;basename&quot;:&quot;docker_logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;dotplot.gif&quot;,&quot;path&quot;:&quot;/shared/images/dotplot.gif&quot;,&quot;basename&quot;:&quot;dotplot&quot;,&quot;extname&quot;:&quot;.gif&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;elixir.png&quot;,&quot;path&quot;:&quot;/shared/images/elixir.png&quot;,&quot;basename&quot;:&quot;elixir&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;em.png&quot;,&quot;path&quot;:&quot;/shared/images/em.png&quot;,&quot;basename&quot;:&quot;em&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;euc_dist.png&quot;,&quot;path&quot;:&quot;/shared/images/euc_dist.png&quot;,&quot;basename&quot;:&quot;euc_dist&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;everything_connected.png&quot;,&quot;path&quot;:&quot;/shared/images/everything_connected.png&quot;,&quot;basename&quot;:&quot;everything_connected&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;f_from_l.png&quot;,&quot;path&quot;:&quot;/shared/images/f_from_l.png&quot;,&quot;basename&quot;:&quot;f_from_l&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;filter_gtf.png&quot;,&quot;path&quot;:&quot;/shared/images/filter_gtf.png&quot;,&quot;basename&quot;:&quot;filter_gtf&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;forward_index.png&quot;,&quot;path&quot;:&quot;/shared/images/forward_index.png&quot;,&quot;basename&quot;:&quot;forward_index&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;freebayes.png&quot;,&quot;path&quot;:&quot;/shared/images/freebayes.png&quot;,&quot;basename&quot;:&quot;freebayes&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;freebayes_gq.png&quot;,&quot;path&quot;:&quot;/shared/images/freebayes_gq.png&quot;,&quot;basename&quot;:&quot;freebayes_gq&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;galaxy_command.png&quot;,&quot;path&quot;:&quot;/shared/images/galaxy_command.png&quot;,&quot;basename&quot;:&quot;galaxy_command&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;galaxy_logo.png&quot;,&quot;path&quot;:&quot;/shared/images/galaxy_logo.png&quot;,&quot;basename&quot;:&quot;galaxy_logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;galaxy_logo_25percent_transparent.png&quot;,&quot;path&quot;:&quot;/shared/images/galaxy_logo_25percent_transparent.png&quot;,&quot;basename&quot;:&quot;galaxy_logo_25percent_transparent&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;gemini_command.png&quot;,&quot;path&quot;:&quot;/shared/images/gemini_command.png&quot;,&quot;basename&quot;:&quot;gemini_command&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;gemini_db_info.png&quot;,&quot;path&quot;:&quot;/shared/images/gemini_db_info.png&quot;,&quot;basename&quot;:&quot;gemini_db_info&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;gemini_load.png&quot;,&quot;path&quot;:&quot;/shared/images/gemini_load.png&quot;,&quot;basename&quot;:&quot;gemini_load&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;gemini_query1.png&quot;,&quot;path&quot;:&quot;/shared/images/gemini_query1.png&quot;,&quot;basename&quot;:&quot;gemini_query1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;gemini_query2.png&quot;,&quot;path&quot;:&quot;/shared/images/gemini_query2.png&quot;,&quot;basename&quot;:&quot;gemini_query2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;goblet.png&quot;,&quot;path&quot;:&quot;/shared/images/goblet.png&quot;,&quot;basename&quot;:&quot;goblet&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;hash.png&quot;,&quot;path&quot;:&quot;/shared/images/hash.png&quot;,&quot;basename&quot;:&quot;hash&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;hisat.png&quot;,&quot;path&quot;:&quot;/shared/images/hisat.png&quot;,&quot;basename&quot;:&quot;hisat&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;history_menu_buttons2.png&quot;,&quot;path&quot;:&quot;/shared/images/history_menu_buttons2.png&quot;,&quot;basename&quot;:&quot;history_menu_buttons2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;history_menu_extract_workflow.png&quot;,&quot;path&quot;:&quot;/shared/images/history_menu_extract_workflow.png&quot;,&quot;basename&quot;:&quot;history_menu_extract_workflow&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;history_options_menu.png&quot;,&quot;path&quot;:&quot;/shared/images/history_options_menu.png&quot;,&quot;basename&quot;:&quot;history_options_menu&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;htseq_count.png&quot;,&quot;path&quot;:&quot;/shared/images/htseq_count.png&quot;,&quot;basename&quot;:&quot;htseq_count&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;htseq_count_interface.png&quot;,&quot;path&quot;:&quot;/shared/images/htseq_count_interface.png&quot;,&quot;basename&quot;:&quot;htseq_count_interface&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;igv_tophat.png&quot;,&quot;path&quot;:&quot;/shared/images/igv_tophat.png&quot;,&quot;basename&quot;:&quot;igv_tophat&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;import_history.png&quot;,&quot;path&quot;:&quot;/shared/images/import_history.png&quot;,&quot;basename&quot;:&quot;import_history&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;interactive_training.png&quot;,&quot;path&quot;:&quot;/shared/images/interactive_training.png&quot;,&quot;basename&quot;:&quot;interactive_training&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;inverted_index.png&quot;,&quot;path&quot;:&quot;/shared/images/inverted_index.png&quot;,&quot;basename&quot;:&quot;inverted_index&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;kallisto.png&quot;,&quot;path&quot;:&quot;/shared/images/kallisto.png&quot;,&quot;basename&quot;:&quot;kallisto&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;landing_page.png&quot;,&quot;path&quot;:&quot;/shared/images/landing_page.png&quot;,&quot;basename&quot;:&quot;landing_page&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;lcs.png&quot;,&quot;path&quot;:&quot;/shared/images/lcs.png&quot;,&quot;basename&quot;:&quot;lcs&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;lf_a.png&quot;,&quot;path&quot;:&quot;/shared/images/lf_a.png&quot;,&quot;basename&quot;:&quot;lf_a&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;lf_b.png&quot;,&quot;path&quot;:&quot;/shared/images/lf_b.png&quot;,&quot;basename&quot;:&quot;lf_b&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;lib_type.png&quot;,&quot;path&quot;:&quot;/shared/images/lib_type.png&quot;,&quot;basename&quot;:&quot;lib_type&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;library_import.png&quot;,&quot;path&quot;:&quot;/shared/images/library_import.png&quot;,&quot;basename&quot;:&quot;library_import&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;library_import_complete.png&quot;,&quot;path&quot;:&quot;/shared/images/library_import_complete.png&quot;,&quot;basename&quot;:&quot;library_import_complete&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;map.png&quot;,&quot;path&quot;:&quot;/shared/images/map.png&quot;,&quot;basename&quot;:&quot;map&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;mum.png&quot;,&quot;path&quot;:&quot;/shared/images/mum.png&quot;,&quot;basename&quot;:&quot;mum&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;nih.png&quot;,&quot;path&quot;:&quot;/shared/images/nih.png&quot;,&quot;basename&quot;:&quot;nih&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;nsf.gif&quot;,&quot;path&quot;:&quot;/shared/images/nsf.gif&quot;,&quot;basename&quot;:&quot;nsf&quot;,&quot;extname&quot;:&quot;.gif&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pA.png&quot;,&quot;path&quot;:&quot;/shared/images/pA.png&quot;,&quot;basename&quot;:&quot;pA&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pAB.png&quot;,&quot;path&quot;:&quot;/shared/images/pAB.png&quot;,&quot;basename&quot;:&quot;pAB&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pB.png&quot;,&quot;path&quot;:&quot;/shared/images/pB.png&quot;,&quot;basename&quot;:&quot;pB&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;p_val_hist.png&quot;,&quot;path&quot;:&quot;/shared/images/p_val_hist.png&quot;,&quot;basename&quot;:&quot;p_val_hist&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pca.png&quot;,&quot;path&quot;:&quot;/shared/images/pca.png&quot;,&quot;basename&quot;:&quot;pca&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pcr-duplicates.png&quot;,&quot;path&quot;:&quot;/shared/images/pcr-duplicates.png&quot;,&quot;basename&quot;:&quot;pcr-duplicates&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;phinch_overviewpage.png&quot;,&quot;path&quot;:&quot;/shared/images/phinch_overviewpage.png&quot;,&quot;basename&quot;:&quot;phinch_overviewpage&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;psu.png&quot;,&quot;path&quot;:&quot;/shared/images/psu.png&quot;,&quot;basename&quot;:&quot;psu&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;quasi_aln.png&quot;,&quot;path&quot;:&quot;/shared/images/quasi_aln.png&quot;,&quot;basename&quot;:&quot;quasi_aln&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rename_history.png&quot;,&quot;path&quot;:&quot;/shared/images/rename_history.png&quot;,&quot;basename&quot;:&quot;rename_history&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;repo_organization.png&quot;,&quot;path&quot;:&quot;/shared/images/repo_organization.png&quot;,&quot;basename&quot;:&quot;repo_organization&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;repo_organization.svg&quot;,&quot;path&quot;:&quot;/shared/images/repo_organization.svg&quot;,&quot;basename&quot;:&quot;repo_organization&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rnaseq_comparison.png&quot;,&quot;path&quot;:&quot;/shared/images/rnaseq_comparison.png&quot;,&quot;basename&quot;:&quot;rnaseq_comparison&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rnaseq_data_in_history.png&quot;,&quot;path&quot;:&quot;/shared/images/rnaseq_data_in_history.png&quot;,&quot;basename&quot;:&quot;rnaseq_data_in_history&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rnaseq_library.png&quot;,&quot;path&quot;:&quot;/shared/images/rnaseq_library.png&quot;,&quot;basename&quot;:&quot;rnaseq_library&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;sa.png&quot;,&quot;path&quot;:&quot;/shared/images/sa.png&quot;,&quot;basename&quot;:&quot;sa&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;sailfish.png&quot;,&quot;path&quot;:&quot;/shared/images/sailfish.png&quot;,&quot;basename&quot;:&quot;sailfish&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;salmon.png&quot;,&quot;path&quot;:&quot;/shared/images/salmon.png&quot;,&quot;basename&quot;:&quot;salmon&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;sampling-bias.png&quot;,&quot;path&quot;:&quot;/shared/images/sampling-bias.png&quot;,&quot;basename&quot;:&quot;sampling-bias&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;sashimi.png&quot;,&quot;path&quot;:&quot;/shared/images/sashimi.png&quot;,&quot;basename&quot;:&quot;sashimi&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;search_index.png&quot;,&quot;path&quot;:&quot;/shared/images/search_index.png&quot;,&quot;basename&quot;:&quot;search_index&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;side-by-side.png&quot;,&quot;path&quot;:&quot;/shared/images/side-by-side.png&quot;,&quot;basename&quot;:&quot;side-by-side&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;snpeff.png&quot;,&quot;path&quot;:&quot;/shared/images/snpeff.png&quot;,&quot;basename&quot;:&quot;snpeff&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;snpeff_chart.png&quot;,&quot;path&quot;:&quot;/shared/images/snpeff_chart.png&quot;,&quot;basename&quot;:&quot;snpeff_chart&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;snpeff_codons.png&quot;,&quot;path&quot;:&quot;/shared/images/snpeff_codons.png&quot;,&quot;basename&quot;:&quot;snpeff_codons&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;sorted_list.png&quot;,&quot;path&quot;:&quot;/shared/images/sorted_list.png&quot;,&quot;basename&quot;:&quot;sorted_list&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;specificity.png&quot;,&quot;path&quot;:&quot;/shared/images/specificity.png&quot;,&quot;basename&quot;:&quot;specificity&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;star.png&quot;,&quot;path&quot;:&quot;/shared/images/star.png&quot;,&quot;basename&quot;:&quot;star&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;stranded_protocols.png&quot;,&quot;path&quot;:&quot;/shared/images/stranded_protocols.png&quot;,&quot;basename&quot;:&quot;stranded_protocols&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;stranded_result.png&quot;,&quot;path&quot;:&quot;/shared/images/stranded_result.png&quot;,&quot;basename&quot;:&quot;stranded_result&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;stringtie1.png&quot;,&quot;path&quot;:&quot;/shared/images/stringtie1.png&quot;,&quot;basename&quot;:&quot;stringtie1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;stringtie2.png&quot;,&quot;path&quot;:&quot;/shared/images/stringtie2.png&quot;,&quot;basename&quot;:&quot;stringtie2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;substring_trie.png&quot;,&quot;path&quot;:&quot;/shared/images/substring_trie.png&quot;,&quot;basename&quot;:&quot;substring_trie&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_tree_1.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_tree_1.png&quot;,&quot;basename&quot;:&quot;suffix_tree_1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_tree_2.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_tree_2.png&quot;,&quot;basename&quot;:&quot;suffix_tree_2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_tree_3.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_tree_3.png&quot;,&quot;basename&quot;:&quot;suffix_tree_3&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_tree_4.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_tree_4.png&quot;,&quot;basename&quot;:&quot;suffix_tree_4&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_tree_5.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_tree_5.png&quot;,&quot;basename&quot;:&quot;suffix_tree_5&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_1.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_1.png&quot;,&quot;basename&quot;:&quot;suffix_trie_1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_2.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_2.png&quot;,&quot;basename&quot;:&quot;suffix_trie_2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_3.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_3.png&quot;,&quot;basename&quot;:&quot;suffix_trie_3&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_4.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_4.png&quot;,&quot;basename&quot;:&quot;suffix_trie_4&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_5.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_5.png&quot;,&quot;basename&quot;:&quot;suffix_trie_5&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_6.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_6.png&quot;,&quot;basename&quot;:&quot;suffix_trie_6&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_7.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_7.png&quot;,&quot;basename&quot;:&quot;suffix_trie_7&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_8.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_8.png&quot;,&quot;basename&quot;:&quot;suffix_trie_8&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_9.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_9.png&quot;,&quot;basename&quot;:&quot;suffix_trie_9&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tools_collection_input.png&quot;,&quot;path&quot;:&quot;/shared/images/tools_collection_input.png&quot;,&quot;basename&quot;:&quot;tools_collection_input&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tophat.png&quot;,&quot;path&quot;:&quot;/shared/images/tophat.png&quot;,&quot;basename&quot;:&quot;tophat&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tophat2.png&quot;,&quot;path&quot;:&quot;/shared/images/tophat2.png&quot;,&quot;basename&quot;:&quot;tophat2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tophat_interface.png&quot;,&quot;path&quot;:&quot;/shared/images/tophat_interface.png&quot;,&quot;basename&quot;:&quot;tophat_interface&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tophat_output.png&quot;,&quot;path&quot;:&quot;/shared/images/tophat_output.png&quot;,&quot;basename&quot;:&quot;tophat_output&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;training_infra.svg&quot;,&quot;path&quot;:&quot;/shared/images/training_infra.svg&quot;,&quot;basename&quot;:&quot;training_infra&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;trie.png&quot;,&quot;path&quot;:&quot;/shared/images/trie.png&quot;,&quot;basename&quot;:&quot;trie&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;trie_no_end.png&quot;,&quot;path&quot;:&quot;/shared/images/trie_no_end.png&quot;,&quot;basename&quot;:&quot;trie_no_end&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_footer.png&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_footer.png&quot;,&quot;basename&quot;:&quot;tutorial_footer&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_footer.svg&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_footer.svg&quot;,&quot;basename&quot;:&quot;tutorial_footer&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_header.png&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_header.png&quot;,&quot;basename&quot;:&quot;tutorial_header&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_introduction.png&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_introduction.png&quot;,&quot;basename&quot;:&quot;tutorial_introduction&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_introduction.svg&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_introduction.svg&quot;,&quot;basename&quot;:&quot;tutorial_introduction&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_part.png&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_part.png&quot;,&quot;basename&quot;:&quot;tutorial_part&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_part.svg&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_part.svg&quot;,&quot;basename&quot;:&quot;tutorial_part&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;ucsc_dm3.png&quot;,&quot;path&quot;:&quot;/shared/images/ucsc_dm3.png&quot;,&quot;basename&quot;:&quot;ucsc_dm3&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;unification_scheme.graphml&quot;,&quot;path&quot;:&quot;/shared/images/unification_scheme.graphml&quot;,&quot;basename&quot;:&quot;unification_scheme&quot;,&quot;extname&quot;:&quot;.graphml&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;unification_scheme.png&quot;,&quot;path&quot;:&quot;/shared/images/unification_scheme.png&quot;,&quot;basename&quot;:&quot;unification_scheme&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;vcfallelicprimitives.png&quot;,&quot;path&quot;:&quot;/shared/images/vcfallelicprimitives.png&quot;,&quot;basename&quot;:&quot;vcfallelicprimitives&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;viewatphinch.png&quot;,&quot;path&quot;:&quot;/shared/images/viewatphinch.png&quot;,&quot;basename&quot;:&quot;viewatphinch&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;we_need_you.jpg&quot;,&quot;path&quot;:&quot;/shared/images/we_need_you.jpg&quot;,&quot;basename&quot;:&quot;we_need_you&quot;,&quot;extname&quot;:&quot;.jpg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;within_norm.png&quot;,&quot;path&quot;:&quot;/shared/images/within_norm.png&quot;,&quot;basename&quot;:&quot;within_norm&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;workflow_editor_mark_output.png&quot;,&quot;path&quot;:&quot;/shared/images/workflow_editor_mark_output.png&quot;,&quot;basename&quot;:&quot;workflow_editor_mark_output&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;workflow_editor_save.png&quot;,&quot;path&quot;:&quot;/shared/images/workflow_editor_save.png&quot;,&quot;basename&quot;:&quot;workflow_editor_save&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;literature.md&quot;,&quot;path&quot;:&quot;/shared/literature.md&quot;,&quot;basename&quot;:&quot;literature&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;add_custom_build.md&quot;,&quot;path&quot;:&quot;/snippets/add_custom_build.md&quot;,&quot;basename&quot;:&quot;add_custom_build&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;add_tag.md&quot;,&quot;path&quot;:&quot;/snippets/add_tag.md&quot;,&quot;basename&quot;:&quot;add_tag&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;ansible_local.md&quot;,&quot;path&quot;:&quot;/snippets/ansible_local.md&quot;,&quot;basename&quot;:&quot;ansible_local&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;build_dataset_list.md&quot;,&quot;path&quot;:&quot;/snippets/build_dataset_list.md&quot;,&quot;basename&quot;:&quot;build_dataset_list&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;build_list_collection.md&quot;,&quot;path&quot;:&quot;/snippets/build_list_collection.md&quot;,&quot;basename&quot;:&quot;build_list_collection&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;change_datatype.md&quot;,&quot;path&quot;:&quot;/snippets/change_datatype.md&quot;,&quot;basename&quot;:&quot;change_datatype&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;change_dbkey.md&quot;,&quot;path&quot;:&quot;/snippets/change_dbkey.md&quot;,&quot;basename&quot;:&quot;change_dbkey&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;create_dataset_collection.md&quot;,&quot;path&quot;:&quot;/snippets/create_dataset_collection.md&quot;,&quot;basename&quot;:&quot;create_dataset_collection&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;create_new_file.md&quot;,&quot;path&quot;:&quot;/snippets/create_new_file.md&quot;,&quot;basename&quot;:&quot;create_new_file&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;create_new_history.md&quot;,&quot;path&quot;:&quot;/snippets/create_new_history.md&quot;,&quot;basename&quot;:&quot;create_new_history&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;display_extra_training.md&quot;,&quot;path&quot;:&quot;/snippets/display_extra_training.md&quot;,&quot;basename&quot;:&quot;display_extra_training&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;display_extra_training_slides.md&quot;,&quot;path&quot;:&quot;/snippets/display_extra_training_slides.md&quot;,&quot;basename&quot;:&quot;display_extra_training_slides&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;extra_protein.md&quot;,&quot;path&quot;:&quot;/snippets/extra_protein.md&quot;,&quot;basename&quot;:&quot;extra_protein&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;extract_workflow.md&quot;,&quot;path&quot;:&quot;/snippets/extract_workflow.md&quot;,&quot;basename&quot;:&quot;extract_workflow&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;history_create_new.md&quot;,&quot;path&quot;:&quot;/snippets/history_create_new.md&quot;,&quot;basename&quot;:&quot;history_create_new&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;import_from_data_library.md&quot;,&quot;path&quot;:&quot;/snippets/import_from_data_library.md&quot;,&quot;basename&quot;:&quot;import_from_data_library&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;import_via_link.md&quot;,&quot;path&quot;:&quot;/snippets/import_via_link.md&quot;,&quot;basename&quot;:&quot;import_via_link&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;import_workflow.md&quot;,&quot;path&quot;:&quot;/snippets/import_workflow.md&quot;,&quot;basename&quot;:&quot;import_workflow&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rename_dataset.md&quot;,&quot;path&quot;:&quot;/snippets/rename_dataset.md&quot;,&quot;basename&quot;:&quot;rename_dataset&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rename_history.md&quot;,&quot;path&quot;:&quot;/snippets/rename_history.md&quot;,&quot;basename&quot;:&quot;rename_history&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;run_workflow.md&quot;,&quot;path&quot;:&quot;/snippets/run_workflow.md&quot;,&quot;basename&quot;:&quot;run_workflow&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;select_collection.md&quot;,&quot;path&quot;:&quot;/snippets/select_collection.md&quot;,&quot;basename&quot;:&quot;select_collection&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;select_multiple_datasets.md&quot;,&quot;path&quot;:&quot;/snippets/select_multiple_datasets.md&quot;,&quot;basename&quot;:&quot;select_multiple_datasets&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;use_scratchbook.md&quot;,&quot;path&quot;:&quot;/snippets/use_scratchbook.md&quot;,&quot;basename&quot;:&quot;use_scratchbook&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;warning_results_may_vary.md&quot;,&quot;path&quot;:&quot;/snippets/warning_results_may_vary.md&quot;,&quot;basename&quot;:&quot;warning_results_may_vary&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Dockerfile&quot;,&quot;path&quot;:&quot;/topics/basic-bioinformatics/docker/Dockerfile&quot;,&quot;basename&quot;:&quot;Dockerfile&quot;,&quot;extname&quot;:&quot;&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;assembly-general-introduction.ga&quot;,&quot;path&quot;:&quot;/topics/basic-bioinformatics/tutorials/general-introduction/workflows/assembly-general-introduction.ga&quot;,&quot;basename&quot;:&quot;assembly-general-introduction&quot;,&quot;extname&quot;:&quot;.ga&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Dockerfile&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/docker/Dockerfile&quot;,&quot;basename&quot;:&quot;Dockerfile&quot;,&quot;extname&quot;:&quot;&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Blastpdb.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Blastpdb.png&quot;,&quot;basename&quot;:&quot;Blastpdb&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Mol_desc_1DKX.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Mol_desc_1DKX.png&quot;,&quot;basename&quot;:&quot;Mol_desc_1DKX&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdb_expdetails_1dkx.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdb_expdetails_1dkx.png&quot;,&quot;basename&quot;:&quot;Pdb_expdetails_1dkx&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdb_firstresiduesmissing_1dkx.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdb_firstresiduesmissing_1dkx.png&quot;,&quot;basename&quot;:&quot;Pdb_firstresiduesmissing_1dkx&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdb_seqtab.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdb_seqtab.png&quot;,&quot;basename&quot;:&quot;Pdb_seqtab&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdbdownloadfile1.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdbdownloadfile1.png&quot;,&quot;basename&quot;:&quot;Pdbdownloadfile1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdbsearchbox_RCSB.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdbsearchbox_RCSB.png&quot;,&quot;basename&quot;:&quot;Pdbsearchbox_RCSB&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdf_uniprotview_button.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdf_uniprotview_button.png&quot;,&quot;basename&quot;:&quot;Pdf_uniprotview_button&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;blastpdb.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/blastpdb.png&quot;,&quot;basename&quot;:&quot;blastpdb&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pdbdownloadfile1.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/pdbdownloadfile1.png&quot;,&quot;basename&quot;:&quot;pdbdownloadfile1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pdbsearchbox_RCSB.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/pdbsearchbox_RCSB.png&quot;,&quot;basename&quot;:&quot;pdbsearchbox_RCSB&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;uniprotsearchbox.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/uniprotsearchbox.png&quot;,&quot;basename&quot;:&quot;uniprotsearchbox&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;}],&quot;documents&quot;:[],&quot;categories&quot;:{},&quot;collections&quot;:[{&quot;label&quot;:&quot;posts&quot;,&quot;files&quot;:[],&quot;directory&quot;:&quot;/home/albot/alex/training-material/_posts&quot;,&quot;output&quot;:true,&quot;docs&quot;:[],&quot;relative_directory&quot;:&quot;_posts&quot;,&quot;permalink&quot;:&quot;/:categories/:year/:month/:day/:title:output_ext&quot;}],&quot;html_pages&quot;:[&quot;\n&quot;,&quot;&lt;ol id=\&quot;markdown-toc\&quot;&gt;\n  &lt;li&gt;&lt;a href=\&quot;#overview-questions\&quot; id=\&quot;markdown-toc-overview-questions\&quot;&gt;Overview Questions&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-is-this-website\&quot; id=\&quot;markdown-toc-what-is-this-website\&quot;&gt;What is this website?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-is-galaxy\&quot; id=\&quot;markdown-toc-what-is-galaxy\&quot;&gt;What is Galaxy?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-are-the-tutorials-for\&quot; id=\&quot;markdown-toc-what-are-the-tutorials-for\&quot;&gt;What are the tutorials for?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-audiences-are-the-tutorials-for\&quot; id=\&quot;markdown-toc-what-audiences-are-the-tutorials-for\&quot;&gt;What audiences are the tutorials for?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-is-the-content-licensed\&quot; id=\&quot;markdown-toc-how-is-the-content-licensed\&quot;&gt;How is the content licensed?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-cite-this-project\&quot; id=\&quot;markdown-toc-how-can-i-cite-this-project\&quot;&gt;How can I cite this project?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-advertise-the-training-materials-on-my-posters\&quot; id=\&quot;markdown-toc-how-can-i-advertise-the-training-materials-on-my-posters\&quot;&gt;How can I advertise the training materials on my posters?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#for-individual-learners\&quot; id=\&quot;markdown-toc-for-individual-learners\&quot;&gt;For Individual Learners&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-do-i-start\&quot; id=\&quot;markdown-toc-where-do-i-start\&quot;&gt;Where do I start?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-do-i-use-this-material\&quot; id=\&quot;markdown-toc-how-do-i-use-this-material\&quot;&gt;How do I use this material?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-can-i-run-the-hands-on-tutorials\&quot; id=\&quot;markdown-toc-where-can-i-run-the-hands-on-tutorials\&quot;&gt;Where can I run the hands-on tutorials?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-get-help\&quot; id=\&quot;markdown-toc-how-can-i-get-help\&quot;&gt;How can I get help?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#for-instructors\&quot; id=\&quot;markdown-toc-for-instructors\&quot;&gt;For Instructors&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-do-i-start-1\&quot; id=\&quot;markdown-toc-where-do-i-start-1\&quot;&gt;Where do I start?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-galaxy-instance-should-i-use-for-my-training\&quot; id=\&quot;markdown-toc-what-galaxy-instance-should-i-use-for-my-training\&quot;&gt;What Galaxy instance should I use for my training?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-do-i-get-help\&quot; id=\&quot;markdown-toc-how-do-i-get-help\&quot;&gt;How do I get help?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#contributing\&quot; id=\&quot;markdown-toc-contributing\&quot;&gt;Contributing&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-get-started-with-contributing\&quot; id=\&quot;markdown-toc-how-can-i-get-started-with-contributing\&quot;&gt;How can I get started with contributing?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-give-feedback\&quot; id=\&quot;markdown-toc-how-can-i-give-feedback\&quot;&gt;How can I give feedback?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-test-an-interactive-tour\&quot; id=\&quot;markdown-toc-how-can-i-test-an-interactive-tour\&quot;&gt;How can I test an Interactive Tour?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot; id=\&quot;markdown-toc-how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot;&gt;How can I fix mistakes or expand an existing tutorial using the GitHub interface?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-contribute-in-advanced-mode\&quot; id=\&quot;markdown-toc-how-can-i-contribute-in-advanced-mode\&quot;&gt;How can I contribute in “advanced” mode?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#other-questions\&quot; id=\&quot;markdown-toc-other-questions\&quot;&gt;Other Questions&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#sustainability-of-the-training-material-and-metadata\&quot; id=\&quot;markdown-toc-sustainability-of-the-training-material-and-metadata\&quot;&gt;Sustainability of the training-material and metadata&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h1 id=\&quot;overview-questions\&quot;&gt;Overview Questions&lt;/h1&gt;\n\n&lt;h2 id=\&quot;what-is-this-website\&quot;&gt;What is this website?&lt;/h2&gt;\n\n&lt;p&gt;This website is a collection of hands-on tutorials that are designed to be interactive and are built around Galaxy:&lt;/p&gt;\n\n&lt;figure id=\&quot;figure-1\&quot;&gt;&lt;img src=\&quot;shared/images/interactive_training.png\&quot; alt=\&quot;Interactive training\&quot; /&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 1:&lt;/span&gt; The hands-on nature of our training material. It can be used with two web browser windows open side-by-side, one pointed at the current tutorial and the other at a Galaxy instance&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;p&gt;This material is developed and maintained by the &lt;a href=\&quot;https://galaxyproject.org/\&quot;&gt;worldwide Galaxy community&lt;/a&gt;. You can learn more about this effort by reading our &lt;a href=\&quot;https://doi.org/10.1016/j.cels.2018.05.012\&quot;&gt;article&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-is-galaxy\&quot;&gt;What is Galaxy?&lt;/h2&gt;\n\n&lt;p&gt;&lt;a href=\&quot;https://galaxyproject.org/\&quot;&gt;Galaxy&lt;/a&gt; is an open data integration and analysis platform for the life sciences, and it is particularly well-suited for data analysis training in life science research.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-are-the-tutorials-for\&quot;&gt;What are the tutorials for?&lt;/h2&gt;\n\n&lt;p&gt;These tutorials can be used for learning and teaching how to use Galaxy for general data analysis, and for learning/teaching specific domains such as assembly and differential gene expression analysis with RNA-Seq data.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-audiences-are-the-tutorials-for\&quot;&gt;What audiences are the tutorials for?&lt;/h2&gt;\n\n&lt;p&gt;There are two distinct audiences for these materials.&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;&lt;strong&gt;Self-paced individual learners.&lt;/strong&gt; These tutorials provide everything you need to learn a topic, from explanations of concepts to detailed hands-on exercises.&lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Instructors.&lt;/strong&gt; They are also designed to be used by instructors in teaching/training settings. Slides, and detailed tutorials are provided. Most tutorials also include computational support with the needed tools, data as well as Docker images that can be used to scale the lessons up to many participants.&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h2 id=\&quot;how-is-the-content-licensed\&quot;&gt;How is the content licensed?&lt;/h2&gt;\n\n&lt;p&gt;The content of this website is licensed under the &lt;a href=\&quot;https://creativecommons.org/licenses/by/4.0/\&quot;&gt;Creative Commons Attribution 4.0 License&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-cite-this-project\&quot;&gt;How can I cite this project?&lt;/h2&gt;\n\n&lt;p&gt;We wrote an &lt;a href=\&quot;https://doi.org/10.1016/j.cels.2018.05.012\&quot;&gt;article&lt;/a&gt; about our efforts.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-advertise-the-training-materials-on-my-posters\&quot;&gt;How can I advertise the training materials on my posters?&lt;/h2&gt;\n\n&lt;p&gt;We provide some QR codes and logos in the &lt;a href=\&quot;https://github.com/vibbits/training-material/tree/master/assets/images\&quot;&gt;images folder&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;for-individual-learners\&quot;&gt;For Individual Learners&lt;/h1&gt;\n\n&lt;p&gt;Learning Galaxy and data analysis on your own and at your own pace?  This material is for you.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-do-i-start\&quot;&gt;Where do I start?&lt;/h2&gt;\n\n&lt;p&gt;If you are new to Galaxy then start with one of the introductory topics.  These introduce you to concepts that are useful in Galaxy, no matter what domain you are doing analysis in.&lt;/p&gt;\n\n&lt;p&gt;If you are already familiar with Galaxy basics and want to learn how to use it in a particular domain (for example, ChIP-Seq), then start with one of those topics.&lt;/p&gt;\n\n&lt;p&gt;If you are already well informed about bioinformatics data analysis and you just want to get a feel for how it works in Galaxy, then many tutorials include &lt;em&gt;Instructions for the impatient&lt;/em&gt; sections.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-do-i-use-this-material\&quot;&gt;How do I use this material?&lt;/h2&gt;\n\n&lt;p&gt;Many topics include slide decks and if the topic you are interested in has slides then start there.  These will introduce the topic and important concepts.&lt;/p&gt;\n\n&lt;p&gt;Most of your learning will happen in the next step - the hands-on tutorials.  This is where you’ll become familiar with using the Galaxy interface and experiment with different ways to use Galaxy and the tools in Galaxy.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-can-i-run-the-hands-on-tutorials\&quot;&gt;Where can I run the hands-on tutorials?&lt;/h2&gt;\n\n&lt;p&gt;To run the hands-on tutorials you need a Galaxy server to run them on.&lt;/p&gt;\n\n&lt;p&gt;Each tutorial is annotated with information about which &lt;a href=\&quot;https://galaxyproject.org/public-galaxy-servers/\&quot;&gt;public Galaxy servers&lt;/a&gt; it can be run on. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial.&lt;/p&gt;\n\n&lt;p&gt;Some topics have a &lt;a href=\&quot;https://www.docker.com/\&quot;&gt;Docker&lt;/a&gt; image that can be installed and run on participants’ laptops.  These Docker images contain Galaxy instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant. You will need to &lt;a href=\&quot;https://docs.docker.com/install/\&quot;&gt;install Docker&lt;/a&gt;.&lt;/p&gt;\n\n&lt;p&gt;Finally, you can also run your tutorials on cloud-based infrastructures.  Galaxy is &lt;a href=\&quot;https://galaxyproject.org/galaxy-services/\&quot;&gt;available on many national research infrastructures&lt;/a&gt; such as &lt;a href=\&quot;https://galaxyproject.org/cloud/jetstream/\&quot;&gt;Jetstream&lt;/a&gt; (United States), &lt;a href=\&quot;https://www.genap.ca/\&quot;&gt;GenAP&lt;/a&gt; (Canada), &lt;a href=\&quot;https://www.gvl.org.au/\&quot;&gt;GVL&lt;/a&gt; (Australia), &lt;a href=\&quot;http://www.climb.ac.uk/\&quot;&gt;CLIMB&lt;/a&gt; (United Kingdom), and more.  These instances are typically easy to launch, and easy to shut down when you are done.&lt;/p&gt;\n\n&lt;p&gt;If you are already familiar with, and have an account on &lt;a href=\&quot;https://aws.amazon.com/\&quot;&gt;Amazon Web Services&lt;/a&gt; then you can also launch a Galaxy server there using &lt;a href=\&quot;https://launch.usegalaxy.org/\&quot;&gt;CloudLaunch&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-get-help\&quot;&gt;How can I get help?&lt;/h2&gt;\n\n&lt;p&gt;If you have questions about this training material, you can reach us using the &lt;a href=\&quot;\&quot;&gt;Gitter chat&lt;/a&gt;.  You’ll need a &lt;a href=\&quot;https://github.com/\&quot;&gt;GitHub&lt;/a&gt; or &lt;a href=\&quot;https://twitter.com/\&quot;&gt;Twitter&lt;/a&gt; account to post questions.  If you have questions about Galaxy outside the context of training, see the &lt;a href=\&quot;https://galaxyproject.org/support/\&quot;&gt;Galaxy Support page&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;for-instructors\&quot;&gt;For Instructors&lt;/h1&gt;\n\n&lt;p&gt;This material can also be used to teach Galaxy and data analysis in a group setting to students and researchers.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-do-i-start-1\&quot;&gt;Where do I start?&lt;/h2&gt;\n\n&lt;p&gt;Spend some time exploring the different tutorials and the different resources that are available. Become familiar with the structure of the tutorials and think about how you might use them in your teaching.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-galaxy-instance-should-i-use-for-my-training\&quot;&gt;What Galaxy instance should I use for my training?&lt;/h2&gt;\n\n&lt;p&gt;To teach the hands-on tutorials you need a Galaxy server to run the examples on.&lt;/p&gt;\n\n&lt;p&gt;Each tutorial is annotated with the information on which &lt;a href=\&quot;https://galaxyproject.org/public-galaxy-servers/\&quot;&gt;public Galaxy servers&lt;/a&gt; it can be run. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial. If you choose this option then you should work with that server’s admins to confirm that think the server can handle the workload for a workshop. For example, the &lt;a href=\&quot;https://usegalaxy.eu/\&quot;&gt;usegalaxy.eu&lt;/a&gt;&lt;/p&gt;\n\n&lt;p&gt;If your organization/consortia/community has its own Galaxy server, then you may  want to run tutorials on that. This can be ideal because then the instance you are teaching on is the same you your participants will be using after the training. They’ll also be able to revisit any analysis they did during the training. If you pursue this option you’ll need to work with your organization’s Galaxy Admins to confirm that&lt;/p&gt;\n\n&lt;ul&gt;\n  &lt;li&gt;the server can support a room full of people all doing the same analysis at the same time.&lt;/li&gt;\n  &lt;li&gt;all participants will be able to create/use accounts on the system.&lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;Some training topics have a Docker image that can be installed and run on all participants’ laptops.  These images contain Galaxy instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant.&lt;/p&gt;\n\n&lt;p&gt;Finally, you can also run your tutorials on cloud-based infrastructures.  Galaxy is &lt;a href=\&quot;https://galaxyproject.org/galaxy-services/\&quot;&gt;available on many national research infrastructures&lt;/a&gt; such as &lt;a href=\&quot;https://galaxyproject.org/cloud/jetstream/\&quot;&gt;Jetstream&lt;/a&gt; (United States), &lt;a href=\&quot;https://www.genap.ca/\&quot;&gt;GenAP&lt;/a&gt; (Canada), &lt;a href=\&quot;https://launch.genome.edu.au/launch\&quot;&gt;GVL&lt;/a&gt; (Australia), &lt;a href=\&quot;http://www.climb.ac.uk/\&quot;&gt;CLIMB&lt;/a&gt; (United Kingdom), and more.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-do-i-get-help\&quot;&gt;How do I get help?&lt;/h2&gt;\n\n&lt;p&gt;The support channel for instructors is the same as for individual learners.  We suggest you start by posting a question to the Galaxy Training Network &lt;a href=\&quot;\&quot;&gt;Gitter chat&lt;/a&gt;. Anyone can view the discussion, but you’ll need to login (using your GitHub or Twitter account) to add to the discussion.&lt;/p&gt;\n\n&lt;p&gt;If you have questions about Galaxy in general (that are not training-centric) then there are &lt;a href=\&quot;https://galaxyproject.org/support/\&quot;&gt;several support options&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;contributing\&quot;&gt;Contributing&lt;/h1&gt;\n\n&lt;p&gt;First off, thanks for your interest in contributing to the Galaxy training materials!&lt;/p&gt;\n\n&lt;p&gt;Individual learners and instructors can make these training more effective by contributing back to them. You can report mistakes and errors, create more content, etc. Whatever is your background, there is a way to contribute: via the GitHub website, via command-line or even without dealing with GitHub.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-get-started-with-contributing\&quot;&gt;How can I get started with contributing?&lt;/h2&gt;\n\n&lt;p&gt;If you would like to get involved in the project but are unsure where to start, there are some easy ways to contribute which will also help you familiarize yourself with the project.&lt;/p&gt;\n\n&lt;h3 id=\&quot;1-checking-existing-tutorials\&quot;&gt;1. Checking existing tutorials&lt;/h3&gt;\n\n&lt;p&gt;A great way to help out the project is to test/edit existing tutorials. Pick a tutorial and check the contents. Does everything work as expected? Are there things that could be improved?&lt;/p&gt;\n\n&lt;p&gt;Below is a checklist of things to look out for to help you get started. If you feel confident in making changes yourself, please open a pull request, otherwise please file an issue with any problems you run into or suggestions for improvements.&lt;/p&gt;\n\n&lt;p&gt;&lt;em&gt;Basic&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Test&lt;/strong&gt; the tutorial on a running Galaxy instance\n    &lt;ul&gt;\n      &lt;li&gt;For example &lt;a href=\&quot;https://usegalaxy.org\&quot;&gt;Galaxy Main&lt;/a&gt;, &lt;a href=\&quot;https://usegalaxy.eu\&quot;&gt;Galaxy Europe&lt;/a&gt;, or &lt;a href=\&quot;https://usegalaxy.org.au\&quot;&gt;Galaxy Australia&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;Report any issues you run into&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Language&lt;/strong&gt; editing\n    &lt;ul&gt;\n      &lt;li&gt;Fix spelling and grammar mistakes&lt;/li&gt;\n      &lt;li&gt;Simplify the English (to make it more accessible)&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;&lt;em&gt;Intermediate&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Metadata&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Are the objectives, keypoints and time estimate filled in?&lt;/li&gt;\n      &lt;li&gt;Do they fit with the contents of the tutorial?&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Content&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Is there enough background information provided in the introduction section and throughout the tutorial?&lt;/li&gt;\n      &lt;li&gt;&lt;strong&gt;Question boxes&lt;/strong&gt;\n        &lt;ul&gt;\n          &lt;li&gt;Add questions or question boxes where you think they might be useful (make people think about results they got, test their understanding, etc)&lt;/li&gt;\n          &lt;li&gt;Check that answers are still up-to-date&lt;/li&gt;\n        &lt;/ul&gt;\n      &lt;/li&gt;\n      &lt;li&gt;&lt;strong&gt;Screenshots and Videos&lt;/strong&gt;\n        &lt;ul&gt;\n          &lt;li&gt;Make sure there is also a textual description of the image/video contents&lt;/li&gt;\n          &lt;li&gt;Does the screenshot add value to the tutorial or can it be removed?&lt;/li&gt;\n        &lt;/ul&gt;\n      &lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;&lt;em&gt;Advanced&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Workflows&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Add a workflow definition file &lt;code class=\&quot;highlighter-rouge\&quot;&gt;.ga&lt;/code&gt; if none is present&lt;/li&gt;\n      &lt;li&gt;Check that the existing workflow is up-to-date with the tutorial contents&lt;/li&gt;\n      &lt;li&gt;Enable &lt;a href=\&quot;https://github.com/usegalaxy-eu/workflow-testing\&quot;&gt;workflow testing&lt;/a&gt;&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Tours&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Add a tour if none exists&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-test-an-interactive-tour\&quot;&gt;Run the existing tour&lt;/a&gt; and check that it is up-to-date with the tutorial contents&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Datasets&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Check that all datasets used in the tutorial are present in Zenodo&lt;/li&gt;\n      &lt;li&gt;Add a data-library.yaml file if none exists&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;h2 id=\&quot;how-can-i-give-feedback\&quot;&gt;How can I give feedback?&lt;/h2&gt;\n\n&lt;p&gt;At the end of each tutorial, there is a link to &lt;a href=\&quot;https://docs.google.com/forms/d/1BILuBp8bPv42odpZ55Kcf2rqzR-k0Czx46agVVSRgm4/edit\&quot;&gt;a feedback form&lt;/a&gt;. We use this information to improve our tutorials.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-test-an-interactive-tour\&quot;&gt;How can I test an Interactive Tour?&lt;/h2&gt;\n\n&lt;p&gt;Perhaps you’ve been asked to review an interactive tour, or maybe you just want to try one out. The easiest way to run an interactive tour is to use the &lt;a href=\&quot;https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/\&quot;&gt;Tour builder&lt;/a&gt; browser extension.&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;Install the Tour Builder extension to your browser (&lt;a href=\&quot;https://chrome.google.com/webstore/detail/galaxy-tour-builder/mdfbapknmcpnbmggahhaegehbbbmhmgg\&quot;&gt;Chrome Web Store&lt;/a&gt;, &lt;a href=\&quot;https://addons.mozilla.org/en-US/firefox/addon/galaxy-tour-builder/\&quot;&gt;Firefox add-on&lt;/a&gt;, &lt;a href=\&quot;https://addons.opera.com/en/extensions/details/galaxy-tour-builder/\&quot;&gt;Opera add-on&lt;/a&gt;).&lt;/li&gt;\n  &lt;li&gt;Navigate to a Galaxy instance supporting the tutorial. To find which Galaxy instances support each tutorial, please see the dropdown menu next to the tutorial on the training website. Using one of the usegalaxy.* instances (&lt;a href=\&quot;https://usegalaxy.org\&quot;&gt;Galaxy Main&lt;/a&gt;, &lt;a href=\&quot;https://usegalaxy.eu\&quot;&gt;Galaxy Europe&lt;/a&gt;, or &lt;a href=\&quot;https://usegalaxy.org.au\&quot;&gt;Galaxy Australia&lt;/a&gt;\n) is usually a good bet.&lt;/li&gt;\n  &lt;li&gt;Start the Tour Builder plugin by clicking on the icon in your browser menu bar&lt;/li&gt;\n  &lt;li&gt;Copy the contents of the &lt;code class=\&quot;highlighter-rouge\&quot;&gt;tour.yaml&lt;/code&gt; file into the Tour builder editor window&lt;/li&gt;\n  &lt;li&gt;Click &lt;code class=\&quot;highlighter-rouge\&quot;&gt;Save&lt;/code&gt; and then &lt;code class=\&quot;highlighter-rouge\&quot;&gt;Run&lt;/code&gt;&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h2 id=\&quot;how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot;&gt;How can I fix mistakes or expand an existing tutorial using the GitHub interface?&lt;/h2&gt;\n\n&lt;p&gt;Check our tutorial to learn how to use the GitHub interface (soon…)&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-contribute-in-advanced-mode\&quot;&gt;How can I contribute in “advanced” mode?&lt;/h2&gt;\n\n&lt;p&gt;Most of the content is written in &lt;a href=\&quot;https://guides.github.com/features/mastering-markdown/\&quot;&gt;GitHub Flavored Markdown&lt;/a&gt; with some metadata (or variables) found in &lt;a href=\&quot;http://yaml.org/\&quot;&gt;YAML&lt;/a&gt; files. Everything is stored on our &lt;a href=\&quot;https://github.com/vibbits/training-material\&quot;&gt;GitHub repository&lt;/a&gt;. Each training material is related to a topic. All training materials (slides, tutorials, etc) related to a topic are found in a dedicated directory (&lt;em&gt;e.g.&lt;/em&gt; &lt;code class=\&quot;highlighter-rouge\&quot;&gt;transcriptomics&lt;/code&gt; directory contains the material related to transcriptomic analysis). Each topic has the following &lt;a name=\&quot;structure\&quot;&gt;structure&lt;/a&gt;:&lt;/p&gt;\n\n&lt;p&gt;&lt;img src=\&quot;shared/images/repo_organization.png\&quot; alt=\&quot;Structure of the repository\&quot; width=\&quot;400px\&quot; /&gt;&lt;/p&gt;\n\n&lt;ul&gt;\n  &lt;li&gt;a metadata file in YAML format&lt;/li&gt;\n  &lt;li&gt;a directory with the topic introduction slide deck in Markdown with introductions to the topic&lt;/li&gt;\n  &lt;li&gt;\n    &lt;p&gt;a directory with the tutorials:&lt;/p&gt;\n\n    &lt;p&gt;Inside the tutorials directory, each tutorial related to the topic has its own subdirectory with several files:&lt;/p&gt;\n    &lt;ul&gt;\n      &lt;li&gt;a tutorial file written in Markdown with hands-on&lt;/li&gt;\n      &lt;li&gt;an optional slides file in Markdown with slides to support the tutorial&lt;/li&gt;\n      &lt;li&gt;a directory with Galaxy Interactive Tours to reproduce the tutorial&lt;/li&gt;\n      &lt;li&gt;a directory with workflows extracted from the tutoria&lt;/li&gt;\n      &lt;li&gt;a YAML file with the links to the input data needed for the tutorial&lt;/li&gt;\n      &lt;li&gt;a YAML file with the description of needed tools to run the tutorial&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;a directory with the Dockerfile describing the details to build a container for the topic (self-study environments).&lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;To manage changes, we use &lt;a href=\&quot;https://guides.github.com/introduction/flow/\&quot;&gt;GitHub flow&lt;/a&gt; based on Pull Requests:&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;&lt;a href=\&quot;https://help.github.com/articles/fork-a-repo/\&quot;&gt;Create a fork&lt;/a&gt; of this repository on GitHub&lt;/li&gt;\n  &lt;li&gt;Clone your fork of this repository to create a local copy on your computer and initialize the required submodules (&lt;code class=\&quot;highlighter-rouge\&quot;&gt;git submodule init&lt;/code&gt; and &lt;code class=\&quot;highlighter-rouge\&quot;&gt;git submodule update&lt;/code&gt;)&lt;/li&gt;\n  &lt;li&gt;Create a new branch in your local copy for each significant change&lt;/li&gt;\n  &lt;li&gt;Commit the changes in that branch&lt;/li&gt;\n  &lt;li&gt;Push that branch to your fork on GitHub&lt;/li&gt;\n  &lt;li&gt;Submit a pull request from that branch to the &lt;a href=\&quot;https://github.com/vibbits/training-material\&quot;&gt;original repository&lt;/a&gt;&lt;/li&gt;\n  &lt;li&gt;If you receive feedback, make changes in your local clone and push them to your branch on GitHub: the pull request will update automatically&lt;/li&gt;\n  &lt;li&gt;Pull requests will be merged by the training team members after at least one other person has reviewed the Pull request and approved it.&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;p&gt;Globally, the process of development of new content is open and transparent:&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;Creation of a branch derived from the master branch of the GitHub repository&lt;/li&gt;\n  &lt;li&gt;Initialization of a new directory for the tutorial&lt;/li&gt;\n  &lt;li&gt;Filling of the metadata with title, questions, learning objectives, etc&lt;/li&gt;\n  &lt;li&gt;Generation of the input dataset for the tutorial&lt;/li&gt;\n  &lt;li&gt;Filling of the tutorial content&lt;/li&gt;\n  &lt;li&gt;Extraction of the workflows of the tutorial&lt;/li&gt;\n  &lt;li&gt;Automatic extraction of the required tools to populate the tool file&lt;/li&gt;\n  &lt;li&gt;Automatic annotation of the public Galaxy servers&lt;/li&gt;\n  &lt;li&gt;Generation of an interactive tour for the tutorial with the &lt;a href=\&quot;https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/\&quot;&gt;Tourbuilder&lt;/a&gt; web-browser extension&lt;/li&gt;\n  &lt;li&gt;Upload of the datasets to Zenodo and addition of the links in the data library file.&lt;/li&gt;\n  &lt;li&gt;Once ready, opening a Pull Request&lt;/li&gt;\n  &lt;li&gt;Automatic checks of the changes are automatically checked for the right format and working links using continuous integration testing on Travis CI&lt;/li&gt;\n  &lt;li&gt;Review of the content by several other instructors via discussions&lt;/li&gt;\n  &lt;li&gt;After the review process, merge of the content into the main branch, starting a series of automatic steps triggered by Travis CI&lt;/li&gt;\n  &lt;li&gt;Regeneration of the website and publication on &lt;a href=\&quot;/training-material/\&quot;&gt;https://vibbits.github.io//training-material/&lt;/a&gt;&lt;/li&gt;\n  &lt;li&gt;Generation of PDF artifacts of the tutorials and slides and upload on the FTP server&lt;/li&gt;\n  &lt;li&gt;Population of &lt;a href=\&quot;https://tess.elixir-europe.org/\&quot;&gt;TeSS&lt;/a&gt;, the ELIXIR’s Training Portal, via the metadata&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;p&gt;&lt;img src=\&quot;shared/images/development_process.png\&quot; alt=\&quot;Development process\&quot; /&gt;&lt;/p&gt;\n\n&lt;ol&gt;\n\n\n &lt;/ol&gt;\n\n&lt;p&gt;We also strongly recommend you read and follow &lt;a href=\&quot;https://carpentries.org/\&quot;&gt;The Carpentries&lt;/a&gt; recommendations on &lt;a href=\&quot;https://carpentries.github.io/lesson-example/01-design/\&quot;&gt;lesson design&lt;/a&gt; and &lt;a href=\&quot;https://carpentries.github.io/instructor-training/15-lesson-study/\&quot;&gt;lesson writing&lt;/a&gt; if you plan to add or change some training materials, and also to check the &lt;a href=\&quot;#structure\&quot;&gt;structure of the training material&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;other-questions\&quot;&gt;Other Questions&lt;/h1&gt;\n\n&lt;p&gt;Yes, always! Have a look at the &lt;a href=\&quot;https://galaxyproject.org/events/\&quot;&gt;Galaxy Community Events Calendar&lt;/a&gt; for what coming up right now.&lt;/p&gt;\n\n&lt;h2 id=\&quot;sustainability-of-the-training-material-and-metadata\&quot;&gt;Sustainability of the training-material and metadata&lt;/h2&gt;\n\n&lt;p&gt;This repository is hosted on &lt;a href=\&quot;https://github.com/\&quot;&gt;GitHub&lt;/a&gt; using git as a &lt;a href=\&quot;https://en.wikipedia.org/wiki/Distributed_version_control\&quot;&gt;DVCS&lt;/a&gt;. Therefore the community is hosting backups of this\nrepository in a decentralised way. The repository is self-contained and contains all needed content and all metadata.\nIn addition we mirror snapshops of this repo on &lt;a href=\&quot;http://zenodo.org\&quot;&gt;Zenodo&lt;/a&gt;.&lt;/p&gt;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;### How to fill the slide decks?\n\nPlease follow our\n[tutorial to learn how to fill the slides](/training-material/topics/contributing/tutorials/create-new-tutorial-slides/slides.html)\n&quot;,&quot;## Protein Structure Analysis ##\n\n- Sequences, structures and databases\n- Experimental methods (X-rays, electrons and NMR)\n- Finding and visualising structures from the  Protein Data Bank\n- Comparing structures\n- Modelling mutations\n- Creating homology models\n\n---\n\n## The Structure-Function Connection ##\n\n.left[![Left-aligned image](/training-material/topics/protein-structure-analysis/images/hemoglobin.png)]\n.right[Folded proteins provide a well-  defined 3D  arrangement of functional groups, creating microenvironments and active sites.\nStructural changes  are often involved in functional  mechanisms  (motor proteins, ...)\n]\n\n---\n\n### Databases\n\n.pull-left[\n**Uniprot** approx. 1100,000 sequences\n\n- mainly determined by large-scale DNA sequencing of individual genes or whole genomes\n- increasingly automated annotation\n\n**Protein Data Bank** approx. 141,000 \nexperimentally determined structures\n\n- Mainly determined by X-ray crystallography and high-resolution NMR spectroscopy\n- Automation is increasing, but there are still significant limitations in the rate of solving new structures\n\n\n]\n.pull-right[ ![](/training-material/assets/images/GTN-60px.png) ]\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### The Protein Databank ###\n\n.pull-left[ ![](/training-material/assets/images/GTN-60px.png) ] \n\n\n.pull-right[\n[http://www.wwpdb.org](http://www.wwpdb.org)\n- contains structures of  proteins, nucleic acids  and complexes,  determined by X-ray  crystallography, NMR  spectroscopy\n- No purely theoretical  or ab initio models  (since 2006)\n- Also stores supporting  experimental data\n- Full deposition now  required by all peer-reviewed journals\n\n---\n\n### Exercise 1: Search the PDB ###\n\n- Use the UniProt site to search for “dnak”.  \n- Use the PDB site to search for “dnak”.\n- Compare the UniProt and PDB result lists.\n- Use the sequence search function to see if there are structures with sequences similar to that of the  DnaK C-terminal domain.\n- Look at the summary pages of a number of  structures and note some interesting properties.\n- Select a number of structures and create a report  page.\n\n---\n\n### PDB File Organisation ###\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- Nucleic Acid Database: DNA and RNA structures\n\n[http://ndbserver.rutgers.edu/](http://ndbserver.rutgers.edu/)\n\n- PDB-REDO: automatically re-refined deposited  structures, using the latest methods\n\n[http://www.cmbi.ru.nl/pdb_redo/](http://www.cmbi.ru.nl/pdb_redo)\n\n- EBI: many online tools for structure analysis\n\n[http://www.ebi.ac.uk/Tools/structure/](http://www.ebi.ac.uk/Tools/structure/)\n\n- Replaced Electron Density Server: convenient overview of  quality parameters for crystal structures\n\n[http://www.ebi.ac.uk/pdbe/litemol](http://www.ebi.ac.uk/pdbe/litemol)\n\n---\n\n### High-Resolution NMR Spectrometry ###\n\nMany atomic nuclei, including the ubiquitous hydrogen nuclei,  resonate at specific radio frequencies when placed in a  strong, uniform magnetic field. The chemical environment of each individual atom slightly modulates its exact resonance  frequency.\n\n![]()\n\nIn macromolecules with thousands of atoms, many different  effects combine to generate an extremely complicated pattern of chemical shifts, which therefore more or less uniquely  identify each atom. Multidimensional spectra allow these  frequencies to be assigned to specific atoms.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nWhen two atoms are near each other in 3D space, they can exchange magnetization, giving rise to crosspeaks at the  intersection of their respective frequencies.\n\n![]()\n\nThis nuclear Overhauser effect (NOE) is used to effectively measure the distances between pairs of atoms, at least  qualitatively.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nAfter identification of the atoms by means of their unique chemical shifts, distance restraints are derived from the  Overhauser crosspeaks. An extended model of the protein  is  generated,  and  then  condensed  into  a  shape  that  is consistent with as many of distance restraints as possible.\n\n![]()\n\n---\n\n### Other methods ###\n\nElectron microscopy (and especially Cryo-electron microscopy): Electron crystallography and single particle reconstruction\r\n\nSmall-angle X-ray and neutron scattering (SAXS and SANS)\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- BioMagResBank: experimental data for NMR-  derived structures (lists of distance restraints  and other experimentally derived properties)\n\n[http://www.bmrb.wisc.edu/](http://www.bmrb.wisc.edu/)\n\n- BioIsis: database of SAXS-derived structures\n\n[http://www.bioisis.net](http://www.bioisis.net)\n\n- EMBL database of SAXS-derived structures\n\n[http://www.sasbdb.org](http://www.sasbdb.org)\n\n- EM Databank for cryo-EM structures\n\n[http://www.emdatabank.org](http://www.emdatabank.org)\n\n---\n\n### Assessing Structure Quality ###\n\nGeneral geometric properties (bond lengths and angles,  Ramachandran distribution, …): MolProbity  [Link](http://molprobity.biochem.duke.edu/)\n\nCrystal Structures\n- Diffraction data resolution and completeness (PDB)\n- Final Rcryst and Rfree factors (PDB)\n- Local correspondence to electron density (EDS)\n\nNMR Structures\n- Number of distance restraints and additional experimental  data sources (BMRB)\n- Local restraint density (on-line NMR constraint analyser)\n[link](http://molsim.sci.univr.it/bioinfo/tools/constraint/index.html)\n\nOther techniques\n- Difficult to generalise: carefully read and consider the  published protocol\n\n---\n\n### Molecular Graphics Software ###\n\n- [PyMOL](http://www.pymol.org/): high-quality output,  good examples on [Wiki](http://www.pymolwiki.org/)\n- [Chimera](http://www.cgl.ucsf.edu/chimera/): good  documentation on website\n- [VMD](http://www.ks.uiuc.edu/Research/vmd/):  excellent for the analysis of MD trajectories\n- [Yasara](http://www.yasara.org)  \n- [SwissPDBViewer](http://spdbv.vital-it.ch/)\n\n---\n\n### YASARA ###\n\nYasara View is freely available and provides  basic visualisation functions.\n\nYasara Model, Dynamics and Structure provide  additional functionality.\n\nAn add-on module for NMR structures is  available.\n\nThe program can be combined with the WHAT-  IF program for structure validation, and with  FoldX for energy calculations.\n\n---\n\n### Exercise 2: show a structure ###\n\nLoad PDB entry 1TRZ using the File menu.\n\nCompare the default representations (F1-F8)  and use the various controls to change the view  of the protein.\n\nExplore the Edit and View menus to change  various aspects of the graphical representation  of the structure.\n\nExamine the hydrogen bonding patterns.  Display a molecular surface.\n\nCreate an interesting view of the molecule and  save it as a high-resolution image.\n\n---\n\n### Structure Superposition ###\n\nStructures can be superimposed to achieve the best  possible match between corresponding atoms. It is  possible to consider all atoms, or only a subset  (such as the Cα atoms).\n\nWhen the structures are very similar, determining  which atoms should be matched to each other is  trivial. When there are larger differences, it takes  more preparation.\n\nDifferent algorithms use different combinations of  sequence- and secondary of tertiary structure-based  information.\n\n---\n\n### Exercise 3: Compare Structures ###\n\nDownload the five provided PDB files and open  them in Yasara.\n\nUse the `Analyze|Align|Objects` with MUSTANG  function to align the four last objects with the first one.\n\nUse the space bar to open the text console and  see the reported root mean square deviations\n\n\nas well as the number of residues matched.\n\nColor all structures by B factor and compare the  distribution to the local variability of the structures.\n\n---\n\n### PDB File Surprises ###\n\n- Missing atoms, residues and loops  \n- Multiple molecules in the asymmetric unit\n- Incomplete oligomers due to coincident crystal and  oligomer symmetry\n- Cα-only models  lternate conformations\n- Multiple models, especially for NMR ensembles\n- Use of B factors to represent other properties  ther non-standard extensions (PDBQT, ...)\n\n---\n\n### Force Fields ###\n\n- Energy terms representing physical interactions\n\nCovalent bond lengths and angles\nDihedral angles and van der Waals forces (steric effects)\nElectrostatic interactions and hydrogen bonds\n…\n\n- Energy can be minimized, and forces can literally be derived from the potential function.\n\n- Consistency and careful consideration of the  properties to be simulated are essential.\n\n---\n\n### Force Field Terms ###\n\nEach energy term has a functional form, which includes one or more parameters:\n\n- Covalent bond energy term\n To do: add formula\n- Van der Waals contact energy term\n\nThe parameters are collectively optimized to  reproduce a chosen set of experimentally  observed parameters.\n\nA given force field should be used as a  consistent system, and can only be used to  predict properties that are covered by the  training set.\n\n---\n\n### FoldX ###\n\n\ns designed for quantitative modelling of the  contribution of structural interactions to the stability of  proteins and protein complexes. It also supports  protein/DNA complexes.\n\nThe force field describes the different interactions in a  protein structure or complex in analytical terms. It has  been calibrated using a set of experimentally determined  stabilities.\n\nApplications include the optimisation of structures, the calculation of the stability of complexes, and predicting  the effect of amino-acid or base-pair mutations on these  properties.\n\n---\n\n### The FoldX Plugin for Yasara\t###\n\nIn order to make FoldX more accessible and  integrate its functions into Yasara, dr. Joost van  Durme (SWITCH laboratory) made a Yasara plugin  module that can apply FoldX functions to structures  that are loaded as Yasara objects.\nThis greatly simplifies the use of FoldX, and allows  for a quick visual analysis of the resulting changes  in the structures.\nMore information can be found at [wiki](http://foldxyasara.switchlab.org/index.php/) [FoldX](http://foldxsuite.crg.eu/)\n\n---\n\n### Exercise 4a: Repair a PDB File ###\n\n- Load the 1CRN PDB file.\n- Use the “Repair object” option in the Analysis |  FoldX menu to activate the corresponding FoldX  function.\n- Select the object to repair.\n\nThis exports the object as a temporary PDB file,  starts FoldX with the appropriate options, and loads  the repaired PDB file as a new object in Yasara.\n\n- Compare the original and repaired objects.  \n- Describe the changes that were introduced.\n\n---\n\n### Exercise 4b: Model a Mutation ###\n\n- Load the 2AC0.sce Yasara scene file.\n- Set an appropriate structure representation.\n- Locate residue Ala159 using the sequence view,  and right-click to access the `FoldX|Mutate` residue function. Change it to a Trp residue.\n- Look at the effect of the substitution on the  structure, and use the space bar to open the text  console and read the output of the FoldX  calculation.\n- Mutate Arg273 to an alanine side chain. Discuss  the effects of this substitution.\n\n---\n\n### Homology Modelling ###\n\n- When a structure is available for a protein with a  similar sequences, it is possible to predict the  structure of a new sequence with varying degrees of  confidence.\n- All homology modelling procedures start from an  alignment of the template and target sequences.  The quality of this alignment will have a major  impact on the resulting model.\n- Available applications include stand-alone programs  (Modeller, FoldX, …) and web-based services (such as SwissModel).\n\n---\n\n### Exercise 5: Make a Homology  Model using Swiss Model ###\n\nsee []()\n\n---\n\n### Exercise 6: Study Protein-Ligand Interactions ###\n\nsee []()\n\n---\n\n### On-Line References ###\n\n- Crystallography 101 (Bernhard Rupp):  (http://www.ruppweb.org/Xray/101index.html)\n- Protein NMR, A Practical Guide (Vicky Higman) ( http://www.protein-nmr.org.uk/)\n- Model validation course  (http://xray.bmc.uu.se/gerard/embo2001/modval/index.html)\n- Assessing model quality (http://spdbv.vital-it.ch/TheMolecularLevel/ModQual/)\n\n- Lectures by Burkhard Rost on protein structure  prediction (https://www.youtube.com/channel/UCU6j8BG4RbEtTgyIZJ6Vpow)\n\n---\n&quot;,&quot;.enlarge120[\n\n# ***De novo* Genome Assembly**\n\n]\n\n#### With thanks to T Seemann, D Bulach, I Cooke and Simon Gladman\n---\n.enlarge120[\n\n# ***De novo* assembly**\n\n]\n\n.pull-left[\n\n**The process of reconstructing the original DNA sequence from the fragment reads alone.**\n\n* Instinctively like a jigsaw puzzle\n\n  * Find reads which \&quot;fit together\&quot; (overlap)\n  * Could be missing pieces (sequencing bias)\n  * Some pieces will be dirty (sequencing errors)\n\n]\n\n.pull-right[ ![](../../images/Humpty.jpg) ]\n\n---\n\n# **Another View**\n\n![](../../images/newspaper.png)\n\n---\n\n# **Assembly: An Example**\n\n---\n\n# **A small \&quot;genome\&quot;**\n\n![](../../images/shakespear1.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear2.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear3.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear4.png)\n\n---\n\n# **So far, so good!**\n\n---\n\n# **The Awful Truth**\n\n![](../../images/notsimply.png)\n\n## \&quot;Genome assembly is impossible.\&quot; - A/Prof. Mihai Pop\n\n---\n.enlarge120[\n\n# **Why is it so hard?**\n\n]\n\n.pull-left[\n* Millions of pieces\n  * Much, much shorter than the genome\n  * Lots of them look similar\n* Missing pieces\n  * Some parts can&#39;t be sequenced easily\n* Dirty Pieces\n  * Lots of errors in reads\n]\n\n.pull-right[ ![](../../images/worlds_hardest.png) ]\n\n---\n\n# **Assembly recipe**\n\n* Find all overlaps between reads\n  * Hmm, sounds like a lot of work..\n* Build a graph\n  * A picture of the read connections\n* Simplify the graph\n  * Sequencing errors will mess it up a lot\n* Traverse the graph\n  * Trace a sensible path to produce a consensus\n\n---\n\n![](../../images/olc_pic.png)\n\n---\n\n# **A more realistic graph**\n\n![](../../images/real_graph.png)\n\n---\n\n# .image-15[![](../../images/nofun.png)] **What ruins the graph?**\n\n* Read errors\n  * Introduces false edges and nodes\n\n* Non haploid organisms\n  * Heterozygosity causes lots of detours\n\n* Repeats\n  * If they are longer than the read length\n  * Causes nodes to be shared, locality confusion.\n\n---\n\n# **Repeats**\n\n---\n.enlarge120[\n# **What is a repeat?**\n]\n\n.pull-left[\n\n#### ***A segment of DNA which occurs more than once in the genome sequence***\n\n* Very common\n  * Transposons (self replicating genes)\n  * Satellites (repetitive adjacent patterns)\n  * Gene duplications (paralogs)\n\n]\n\n.pull-right[\n\n![](../../images/triplets.png)\n\n]\n\n---\n\n# **Effect on Assembly**\n\n![](../../images/repeat_effect.png)\n\n---\n.enlarge120[\n# **The law of repeats** .image-15[![](../../images/repeatafterme.png)]\n]\n\n## **It is impossible to resolve repeats of length S unless you have reads longer than S**\n\n## **It is impossible to resolve repeats of length S unless you have reads longer than S**\n\n---\n\n# **Scaffolding**\n\n---\n.enlarge120[\n# **Beyond contigs**\n]\n\n.pull-left[\n\nContig sizes are limited by:\n\n* the length of the repeats in your genome\n  * Can&#39;t change this\n\n\n* the length (or \&quot;span\&quot;) of the reads\n  * Use long read technology\n  * Use tricks with other technology\n\n]\n\n---\n.enlarge120[\n# **Types of reads**\n]\n\n.pull-left[.enlarge120[**Example fragment**]]\n\n\n.remark-code[.enlarge120[atcgtatgatcttgagattctctcttcccttatagctgctata]]\n\n.pull-left[.enlarge120[**\&quot;Single-end\&quot; read**]]\n\n\n.remark-code[.enlarge120[**atcgtatg**atcttgagattctctcttcccttatagctgctata]]\n\nsequence *one* end of the fragment\n\n.pull-left[.enlarge120[**\&quot;Paired-end\&quot; read**]]\n\n\n.remark-code[.enlarge120[**atcgtatg**atcttgagattctctcttcccttatag**ctgctata**]]\n\nsequence both ends of the same fragment\n\n**We can exploit this information!**\n---\n\n.enlarge120[# **Scaffolding**]\n\n* **Paired end reads**\n  * Known sequences at each end of fragment\n  * Roughly known fragment length\n\n* **Most ends will occur in same contig**\n\n* **Some will occur in different contigs**\n  * ***evidence that these contigs are linked***\n---\n\n.enlarge120[# **Contigs to Scaffolds**]\n\n![](../../images/scaffolding.png)\n\n---\n\n.enlarge120[# **Assessing assemblies**]\n\n* We desire\n  * Total length similar to genome size\n  * Fewer, larger contigs\n  * Correct contigs\n\n* Metrics\n  * No generally useful measure. (No real prior information)\n  * Longest contigs, total base pairs in contigs, **N50**, ...\n\n---\n\n.enlarge120[# **The \&quot;N50\&quot;**]\n\n.enlarge120[***The length of that contig from which 50% of the bases are in it and shorter contigs***]\n\n* Imagine we have 7 contigs with lengths:\n  * 1, 1, 3, 5, 8, 12, 20\n\n* Total\n  * 1+1+3+5+8+12+20 = 50\n\n* N50 is the \&quot;halfway sum\&quot; = 25\n  * 1+1+3+5+8+**12** = 30 (&gt;25) so **N50 is 12**\n\n---\n\n.enlarge120[# **2 levels of assembly**]\n\n* Draft assembly\n  * Will contain a number of non-linked scaffolds with gaps of unknown sequence\n  * Fairly easy to get to\n\n* Closed (finished) assembly\n  * One sequence for each chromosome\n  * Takes a **lot** more work\n  * Small genomes are becoming easier with long read tech\n  * Large genomes are the province of big consortia (e.g. Human Genome Consortium)\n\n---\n.enlarge120[# **How do I do it?**]\n---\n.enlarge120[\n# **Example**\n\n* Culture your bacterium\n\n\n* Extract your genomic DNA\n\n\n* Send it to your sequencing centre for Illumina sequencing\n  * 250bp paired end\n\n\n* Get back 2 files\n  * .remark-code[MRSA_R1.fastq.gz]\n  * .remark-code[MRSA_R2.fastq.gz]\n\n\n* ***Now what?***\n]\n\n---\n.enlarge120[# **Assembly tools**\n\n* **Genome**\n  * **Velvet, Velvet Optimizer, Spades,** Abyss, MIRA, Newbler, SGA, AllPaths, Ray, SOAPdenovo, ...\n\n\n* Meta-genome\n  * Meta Velvet, SGA, custom scripts + above\n\n\n* Transcriptome\n  * Trinity, Oases, Trans-abyss\n\n***And many, many others...***\n\n]\n\n---\n.enlarge120[\n# **Assembly Exercise #1**\n\n* We will do a simple assembly using **Velvet** in **Galaxy**\n* We can do a number of different assemblies and compare some assembly metrics.\n\n]\n&quot;,&quot;# Genome assembly with Velvet: Background\n{:.no_toc}\n\nVelvet is one of a number of *de novo* assemblers that use short read sets as input (*e.g.* Illumina Reads). The assembly method is based on the manipulation of de Bruijn graphs, via the removal of errors and the simplication of repeated regions.\n\n&gt; ### {% icon comment %} Comment\n&gt;\n&gt; For information about Velvet, you can check its (nice) [Wikipedia page](https://en.wikipedia.org/wiki/Velvet_assembler).\n{: .comment}\n\nFor this tutorial, we have a set of reads from an imaginary *Staphylococcus aureus* bacterium with a miniature genome (197,394 bp). Our mutant strain read set was sequenced with the whole genome shotgun method, using an Illumina DNA sequencing instrument. From these reads, we would like to rebuild our imaginary *Staphylococcus aureus* bacterium via a *de novo* assembly of a short read set using the Velvet assembler.\n\n&gt; ### Agenda\n&gt;\n&gt; In this tutorial, we will deal with:\n&gt;\n&gt; 1. TOC\n&gt; {:toc}\n&gt;\n{: .agenda}\n\n# Get the data\n\nWe will now import the data that we will use for the tutorial.\n\n&gt; ### {% icon hands_on %} Hands-on: Getting the data\n&gt;\n&gt; 1. Create and name a new history for this tutorial.\n&gt; 2. Import from [Zenodo](https://doi.org/10.5281/zenodo.582600) or from the data library the files:\n&gt;    - [`mutant_R1.fastq`](https://zenodo.org/record/582600/files/mutant_R1.fastq)\n&gt;    - [`mutant_R2.fastq`](https://zenodo.org/record/582600/files/mutant_R2.fastq)\n&gt;\n&gt;    &gt; ### {% icon tip %} Tip: Importing data via links\n&gt;    &gt;\n&gt;    &gt; * Copy the link location (Right-click on the filename then \&quot;Copy Link Address\&quot;)\n&gt;    &gt; * Open the Galaxy Upload Manager\n&gt;    &gt; * Select **Paste/Fetch Data**\n&gt;    &gt; * Paste the link into the text field\n&gt;    &gt; * Change the data-type to **fastqsanger**\n&gt;    &gt; * Press **Start**\n&gt;    {: .tip}\n&gt;\n&gt; 3. Change the name of the files to `mutant_R1` and `mutant_R2`.\n&gt;\n&gt;    As a default, Galaxy uses the link as the name of the new dataset. It also does not link the dataset to a database or a reference genome.\n&gt;\n&gt;    {% include snippets/rename_dataset.md %}\n&gt;\n&gt; 4. Inspect the content of a file.\n&gt;\n&gt;    &gt; ### {% icon tip %} Tip: Inspecting the content of a dataset\n&gt;    &gt;\n&gt;    &gt; * Click on the {% icon galaxy-eye %} (eye) icon next to the relevant history entry\n&gt;    &gt; * View the content of the file in the central panel\n&gt;    {: .tip}\n&gt;\n&gt;    &gt; ### {% icon question %} Questions\n&gt;    &gt;\n&gt;    &gt; 1. What are four key features of a FASTQ file?\n&gt;    &gt; 2. What is the main difference between a FASTQ and a FASTA file?\n&gt;    &gt;\n&gt;    &gt; &gt; ### {% icon solution %} Solution\n&gt;    &gt; &gt; 1. Each sequence in a FASTQ file is represented by 4 lines: 1st line is the id, 2nd line is the sequence, 3rd line is not used, and 4th line is the quality of sequencing per nucleotide\n&gt;    &gt; &gt; 2. In a FASTQ file, not only are the sequences present, but information about the quality of sequencing is also included.\n&gt;    &gt; {: .solution }\n&gt;    {: .question}\n&gt;\n{: .hands_on}\n\nThe reads have been sequenced from an imaginary *Staphylococcus aureus* bacterium using an Illumina DNA sequencing instrument. We obtained the 2 files we imported (`mutant_R1` and `mutant_R2`)\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; Why do we have 2 files here if we only sequenced the bacteria once?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. The bacteria has been sequenced using paired-end sequencing. The first file corresponds to forward reads and the second file to reverse reads.\n&gt; {: .solution }\n{: .question}\n\n# Evaluate the input reads\n\nBefore doing any assembly, the first questions you should ask about your input reads include:\n\n- What is the coverage of my genome?\n- How good is my read set?\n- Do I need to ask for a new sequencing run?\n- Is it suitable for the analysis I need to do?\n\nWe will evaluate the input reads using the FastQC tool. This tool runs a standard series of tests on your read set and returns a relatively easy-to-interpret report. We will use it to evaluate the quality of our FASTQ files and combine the results with MultiQC.\n\n&gt; ### {% icon hands_on %} Hands-on: FastQC on a fastq file\n&gt;\n&gt; 1. **FastQC** {% icon tool %} with the following parameters\n&gt;    - \&quot;Short read data from your current history\&quot; to (**Multiple datasets**) `mutant_R1.fastq` and `mutant_R2.fastq`\n&gt;\n&gt; 2. **MultiQC** {% icon tool %} with the following parameters\n&gt;    - \&quot;Software name\&quot; to `FastQC`\n&gt;    - \&quot;Result file\&quot; to the raw data files generated by FastQC\n&gt;\n{: .hands_on}\n\nMultiQC generates a webpage combining reports for FastQC on both datasets. It includes these graphs and tables:\n\n- General statistics\n\n    This is important in setting maximum k-mer size for an assembly.\n\n    &gt; ### {% icon comment %} Getting the length of sequences\n    &gt;\n    &gt; * Click on **Configure Columns**\n    &gt; * Check **Length**\n    &gt; * Close the window\n    {: .comment}\n\n    &gt; ### {% icon question %} Questions\n    &gt;\n    &gt; 1. How long are the sequences?\n    &gt; 2. What is the average coverage of the genome, given our imaginary *Staphylococcus aureus* bacterium has a genome of 197,394 bp?\n    &gt;\n    &gt; &gt; ### {% icon solution %} Solution\n    &gt; &gt; 1. The sequences are 150 bp long\n    &gt; &gt; 2. We have 2 x 12,480 sequences of 150 bp, so the average genome coverage is: 2 * 12480 * 150 / 197394, or approximately 19 X coverage.\n    &gt; {: .solution }\n    {: .question}\n\n- Sequence Quality Histograms\n\n    Dips in quality near the beginning, middle or end of the reads may determine the trimming/cleanup methods and parameters to be used, or may indicate technical problems with the sequencing process/machine run.\n\n    &lt;figure id=\&quot;figure-1\&quot;&gt;&lt;img src=\&quot;../../images/fastqc_per_base_sequence_quality_plot.png\&quot; alt=\&quot;Sequence Quality Histograms with the mean quality value across each base position in the read\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 1:&lt;/span&gt; The mean quality value across each base position in the read&lt;/figcaption&gt;&lt;/figure&gt;\n\n    &gt; ### {% icon question %} Questions\n    &gt;\n    &gt; 1. What does the y-axis represent?\n    &gt; 2. Why is the quality score decreasing across the length of the reads?\n    &gt;\n    &gt; &gt; ### {% icon solution %} Solution\n    &gt; &gt; 1. The y-axis represents the quality score for each base (an estimate of the error during sequencing).\n    &gt; &gt; 2. The quality score is decreasing accross the length of the reads because the sequencing become less and less reliable at the end of the reads.\n    &gt; {: .solution }\n    {: .question}\n\n- Per Sequence GC Content\n\n    High GC organisms tend not to assemble well and may have an uneven read coverage distribution.\n\n- Per Base N Content\n\n    The presence of large numbers of Ns in reads may point to a poor quality sequencing run. You will need to trim these reads to remove Ns.\n\n- k-mer content\n\n    The presence of highly recurring k-mers may point to contamination of reads with barcodes or adapter sequences.\n\n\n&gt; ### {% icon comment %} Comment\n&gt;\n&gt; For a fuller discussion of FastQC outputs and warnings, see the [FastQC website link](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/), including the section on each of the output [reports](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/), and examples of [\&quot;good\&quot;](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html) and [\&quot;bad\&quot;](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html) Illumina data.\n{: .comment}\n\nWe won&#39;t be doing anything to these data to clean it up as there isn&#39;t much need. Therefore we will get on with the assembly!\n\n\n# Assemble reads with Velvet\n\nNow, we want to assemble our reads to find the sequence of our imaginary *Staphylococcus aureus* bacterium. We will perform a *de novo* assembly of the reads into long contiguous sequences using the Velvet short read assembler.\n\nThe first step of the assembler is to build a de Bruijn graph. For that, it will break our reads into k-mers, *i.e.* fragments of length *k*. Velvet requires the user to input a value of *k* (k-mer size) for the assembly process. Small k-mers will give greater connectivity, but large k-mers will give better specificity.\n\n&gt; ### {% icon hands_on %} Hands-on: Assemble the reads\n&gt;\n&gt; 1. **FASTQ interlacer** {% icon tool %} with the following parameters\n&gt;    - \&quot;Type of paired-end datasets\&quot; to `2 separate datasets`\n&gt;    - \&quot;Left-hand mates\&quot; to `mutant_R1.fastq`\n&gt;    - \&quot;Right-hand mates\&quot; to `mutant_R2.fastq`\n&gt;\n&gt;    Currently our paired-end reads are in 2 files (one with the forward reads and one with the reverse reads), but Velvet requires only one file, where each read is next to its mate read. In other words, if the reads are indexed from 0, then reads 0 and 1 are paired, 2 and 3, 4 and 5, etc. Before doing the assembly *per se*, we need to prepare the files by combining them.\n&gt;\n&gt; 2. **velveth** {% icon tool %} with the following parameters\n&gt;    - \&quot;Hash Length\&quot; to `29`\n&gt;    - \&quot;Input Files\&quot;: click on `Insert Input Files`\n&gt;    - \&quot;file format\&quot; to `fastq`\n&gt;    - \&quot;read type\&quot; to `shortPaired reads`\n&gt;    - \&quot;Dataset\&quot; to the pairs output of **FASTQ interlacer**\n&gt;\n&gt;    The tool takes our reads and break them into k-mers.\n&gt;\n&gt; 3. **velvetg** {% icon tool %} with the following parameters\n&gt;    - \&quot;Velvet Dataset\&quot; to the output of **velveth**\n&gt;    - \&quot;Using Paired Reads\&quot; to `Yes`\n&gt;\n&gt;    This last tool actually does the assembly.\n{: .hands_on}\n\nTwo files are generated:\n\n- A \&quot;Contigs\&quot; file\n\n    This file contains the sequences of the contigs longer than 2k. In the header of each contig, a bit of information is added:\n    - the k-mer length (called \&quot;length\&quot;): For the value of k chosen in the assembly, a measure of how many k-mers overlap (by 1 bp each overlap) to give this length\n    - the k-mer coverage (called \&quot;coverage\&quot;): For the value of k chosen in the assembly, a measure of how many k-mers overlap each base position (in the assembly).\n\n    ![Contigs output](../../images/image10.png)\n\n- A \&quot;Stats\&quot; file\n\n    This is a tabular file giving for each contig the k-mer lengths, k-mer coverages and other measures.\n\n    ![Contigs stats output](../../images/image11.png)\n\n# Collect some statistics on the contigs\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. How many contigs have been built?\n&gt; 2. What is the mean, min and max length of the contigs?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. 190\n&gt; &gt; 2. To compute this information, we can use the Datamash tool on the 2nd columns (length). Be careful with the first line, the header. As a result, we obtain: 597.82 as mean, 1 as min and 12904 as max. It would mean that the smallest contig has a length of 1 bp, even smaller than k. The length on the 2nd column corresponds to length of the contig in k-mers. This means that the smallest contig has a length of 1k = 29. So to obtain the real length, we need to add k-1 to the length. We then obtain a mean contig length of 625.82 bp, a min contig of 29 bp and a max contig of 12,932 bp.\n&gt; {: .solution }\n{: .question}\n\nThis table is limitted, but we will now collect more basic statistics on our assembly.\n\n&gt; ### {% icon hands_on %} Hands-on: Collect fasta statistics on our contigs\n&gt;\n&gt; 1. **Quast** {% icon tool %} with\n&gt;    - \&quot;Contigs/scaffolds output file\&quot; to the output of **velvetg**\n&gt;    - \&quot;Type of data\&quot; to `contig`\n&gt;    - \&quot;Reference File\&quot; to `wildtype.fna`\n&gt;    - \&quot;Type of organism\&quot; to `Prokaryotes`\n&gt;    - \&quot;Lower Threshold\&quot; to `500`\n&gt;    - \&quot;Thresholds\&quot; to `0,1000`\n{: .hands_on}\n\nThis tool generates 5 output files, but we will focus on the HTML report and the Icarus viewer.\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. What is represented in the Icarus viewer?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. Icarus is a novel genome visualizer for accurate assessment and analysis of genomic draft assemblies. It draws contigs ordered from longest to shortest, highlights N50, N75 (NG50, NG75) and long contigs larger than a user-specified threshold\n&gt; {: .solution }\n{: .question}\n\nThe HTML report reports many statistics computed by QUAST to assess the quality of the assembly:\n\n- Statistics about the quality of the assembly when compared to the reference (fraction of the genome, duplication ratio, etc)\n- Misassembly statistics, including the number of misassemblies\n\n    A misassembly is a position in the contigs (breakpoints) that satisfy one of the following criteria:\n    - the left flanking sequence aligns over 1 kbp away from the right flanking sequence on the reference;\n    - flanking sequences overlap on more than 1 kbp\n    - flanking sequences align to different strands or different chromosomes\n\n- Unaligned regions in the assembly\n- Mismatches compared to the reference genomes\n- Statistics about the assembly *per se*, such as the number of contigs and the length of the largest contig\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. How many contigs have been constructed?\n&gt; 2. Which proportion of the reference genome do they represent?\n&gt; 3. How many misassemblies have been found?\n&gt; 4. Has the assembly introduced mismatches and indels?\n&gt; 5. What are N50 and L50?\n&gt; 6. Is there a bias in GC percentage induced by the assembly?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. 190 contigs have been constructed, but only 47 have a length &gt; 500 bp.\n&gt; &gt; 2. The contigs represents 87.965% of the reference genome.\n&gt; &gt; 3. 1 misassembly has been found: it corresponds to a relocation, *i.e.* a misassembly event (breakpoint) where the left flanking sequence aligns over 1 kbp away from the right flanking sequence on the reference genome.\n&gt; &gt; 4. 8.06 mismatches per 100 kbp and 4.03 indels per 100 kbp are found.\n&gt; &gt; 5. N50 is the length for which the collection of all contigs of that length or longer covers at least half an assembly. In other words, if contigs were ordered from small to large, half of all the nucleotides will be in contigs this size or larger. And L50 is the number of contigs equal to or longer than N50: L50 is the minimal number of contigs that cover half the assembly.\n&gt; &gt; 6. The GC % in the assembly is 33.64%, really similar to the one of the reference genome (33.43%).\n&gt; {: .solution }\n{: .question}\n\n# Discussion\n\n&gt; ### {% icon hands_on %} (Optional) Hands-on: Rerun for values *k* ranging from 31 to 101\n&gt;\n&gt; 1. **velveth** {% icon tool %} with the same parameters as before except\n&gt;    - \&quot;Hash Length\&quot; to a value between 31 and 101\n&gt; 2. **velvetg** {% icon tool %} with the same parameters as before\n&gt; 3. **Quast** {% icon tool %} with the same parameters as before\n{: .hands_on}\n\nWe have completed an assembly on this data set for a number of k values ranging from 29 to 101. A few of the assembly metrics appear below.\n\n&lt;figure id=\&quot;figure-2\&quot;&gt;&lt;img src=\&quot;../../images/number_of_contigs.png\&quot; alt=\&quot;contigs\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 2:&lt;/span&gt; Number of contigs in the assembly for various k-mer sizes&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-3\&quot;&gt;&lt;img src=\&quot;../../images/largest_contig.png\&quot; alt=\&quot;largest_contig\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 3:&lt;/span&gt; Largest contig in each of the assemblies by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-4\&quot;&gt;&lt;img src=\&quot;../../images/total_bp.png\&quot; alt=\&quot;total_bp\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 4:&lt;/span&gt; Total number of base pairs in all the contigs for each assembly by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-5\&quot;&gt;&lt;img src=\&quot;../../images/n50.png\&quot; alt=\&quot;n50\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 5:&lt;/span&gt; N50 metric for each of the assemblies by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&gt; ### {% icon question %} Questions\n&gt;\n&gt; 1. Are there any distinct features in the charts?\n&gt; 2. Does it look like one assembly might be better than some of the others?\n&gt;\n{: .question}\n\nThe reasons for these patterns will be discussed in detail in the De Bruijn graph assembly slides and tutorial.\n&quot;,&quot;#### Create a project\n\nWhen you use qbase+ for the first time, you can&#39;t do anything unless you\ncreate a project to store your experiments in.\n\n| Create a new Project |\n| :----------------------------------- |\n| When you double click the qbase+ icon, the software starts up automatically opens the Start page where you can create a new project by clicking the Create new project button : This will create a new project with a default name like Project 1 . |\n\n#### Create an experiment\n\nTo open actual data (one/more runs) in qbase+, creating a project is not sufficient. You need to create an experiment in this project to hold the run data.\n\n| Create a new Experiment called GeneExpression in the new project. |\n| :----------------------------------- |\n| Select the Create a new qbase+ experiment option  in the Start page. Type a name for th new experiment . Click the Next button at the bottom of the page . This will create the experiment.\n\nWhen you leave the **Start page**, the **Import run** page is automatically opened allowing you to import the actual qPCR data into qbase+.\n\n#### Loading the data\n\nFirst a few quick words about the data set. We’ll be working with data coming from 3 runs (plates in the qPCR instrument): [Run1](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run1.xls), [Run2](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run2.xls) and [Run3](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run3.xls)\n\nThe data consist of Cq values for:\n\n  - 3 reference target genes: Stable, Nonregulated, and Flexible\n  - 3 target genes of interest: Duvel, Leffe, and Palm\n\neach measured twice (= technical replicates) in 16 different samples. Half of the samples have undergone a treatment, half of them are untreated control samples.\n\nThe data set also contains a series of standard samples consisting of a four-fold dilution series of cDNA for each target gene. These measurements allow to generate a standard curve from which target-specific amplification efficiencies can be calculated. Finally, negative controls (No Template Controls) have been measured. The goal of the analysis is to identify target genes of interest that have different expression levels in the treated samples compared to the untreated control samples.\n\n| In GeneExpression load CFX run files [Run1](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run1.xls), [Run2](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run2.xls) and [Run3](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run3.xls).\n| :---------------------------- |\n|  \n* Click the **Import runs** button  to open the **Import Run** window\n* Click the **Browse** button  to go to the directory that stores the files containing the qPCR data\n\nSelect the 3 run files simultaneously by holding the **Ctrl** key on your keyboard during the selection in Windows or the command button in MacOSX.\n\nClick the **Open** button \n\nNow you go back to the **Import Run** window, click the **Next** button (purple)\n\n* \nqbase+ tries to recognize the format of the selected import files. If only one format matches the files (as in our case CFX), it is selected and the quick import option is enabled. Click the **Finish** button.\n\nIn the Imported run names area on the **Import run** page you should now see the names of the 3 run files. If these are the correct files, click the **Next** button at the bottom of the page.\n\n#### Adding annotation to the data\n\nWhen you leave the **Import run** page, you are redirected to the **Sample target list** page, which gives you an overview of the targets (= genes) and samples qbase+ detected when reading in the datafiles.\nTake a look at the data. You see that the list of samples and targets matches the description of the qPCR experiment at the top of this page. The samples in this experiment are divided into two groups: samples that received some kind of treatment and untreated control samples. This information was not included in the run files so qbase+ does not know which sample belongs to which group. However, this is relevant information: in our analysis we are going to compare the expression of our genes of interest between treated and untreated samples. This means that qbase+ needs the grouping annotation to be able to perform the analysis we want to do. So we have to give qbase+ this annotation: we can do this by adding a custom sample property. To do this we need to create a sample properties file with a specific format that is described in [the tutorial](http://data.bits.vib.be/pub/trainingen/qbasePLUS/TutorialIII.pdf). You can find the file in the qbase+ folder on the BITS laptops or you can [download the file here](http://data.bits.vib.be/pub/trainingen/qbasePLUS/Sample_Properties_file.xlsx).\n\n| How to add the grouping annotation ?\n| :---------------------------- |\n|  To import the file containing to grouping annotation:\n\n* select **Add samples and targets** \n* click **Import sample list** \n* browse to the folder that contains the samples file\n* select the file and click **Open**\n* click **Next**\n\nIn the **Importing samples** window, you have to tell qbase+ which sample annotation you want to import from the sample properties file\n\nIn our case we could import Quantities (this annnotation is available in the sample properties file) but the quantities of the standard samples were included in the run files so qbase+ has already imported this annotation from the run files during data import.\nWe definitely need to import the Custom properties since they were not a part of the run files. The Treatment property will tell qbase+ which samples belong to the group of control samples and which samples belong to the group of treated samples.\nClick the **Next** button at the bottom of the page to finish the import.\n\nAt this point you don&#39;t see the custom annotation that you have imported, you will see it later in the analysis during scaling\nLeaving the **Sample target list** page takes you to the **Run annotation** page, where you have to confirm again that the sample and gene names are ok. If this is not the case you can adjust the annotation here.\n\nClick the **Next** button at the bottom of the page\n\nOur data file contains all required annotation:\n\n  - Cq values\n  - sample and target names\n  - sample types\n  - quantities for the standard samples\n  - grouping of the samples\n\n\n\nOnce runs are imported, you can start analyzing the data. Data consist\nof Cq values for all the wells.\n\n#### Specifying the aim of the experiment\n\nOn the **Aim** page you tell the software what type of analysis you want to do. Different types of analyses require different parameters, parameter settings and different calculations. By selecting the proper analysis type, qbase+ will only show the relevant parameters and parameter settings.\n\nSince we are doing a **gene expression analysis** in this exercise, this the option we should select. Click the **Next** button on the bottom of the page to go to the **Technical quality control** page.\n\n#### Checking the quality of technical replicates and controls\n\nThe **Technical quality control** page handles the settings of the requirements that the data have to meet to be considered high quality. For instance the maximum difference between technical replicates is defined on this page. If there are technical replicates in the data set, qbase+ will detect them automatically (they have the same sample and target name) and calculate the average Cq value. In theory, technical replicates should generate more or less identical signals.\n\n| How to set the maximum difference in Cq values for technical replicates ?\n| :---------------------------- |\n|  The quality criterium that the replicates must meet to be included for further analysis is one of the parameters in qbase+. You can set it on the **Technical quality control** page:\n\nThe default maximum allowed difference in Cq values between technical replicates is 0.5\n\nAdditionally, you can do quality checks based on the data of the positive and negative controls.\n| How to set quality requirements for the control samples ?\n| :---------------------------- |\n|  On the same **Technical quality control** page you can define the minimum requirements for a well to be included in the calculations:\n\n* **Negative control threshold** : minimum allowed difference in Cq value between the sample with the highest Cq value and the negative control with the lowest Cq value: the default is 5 which means that negative controls should be more than 5 cycles away from the sample of interest.\n* **Lower and upper boundary** : allowed range of Cq values for positive controls.\n\nExcluded means that the data are ignored in the calculations.\n\n| How to check if there are wells that do not meet these criteria ?\n| :---------------------------- |\n|  You can see flagged and excluded data by ticking the **Show details…** options  on the **Technical quality control** page and clicking the **Next** button (purple) at the bottom of the page.\n\nQbase+ will open the results of the quality checks for the replicates  and the controls  on two different tabs. These tabs show lists of samples that failed the quality control criteria. When you open the replicates tab  you can get an overview of the flagged  or the excluded (purple) wells. Select the **failing**  wells.\n\nWhen the difference in Cq between technical replicates exceeds 0.5, the wells end up in the flagged or failing list. They are included in calculations unless you exclude them by unticking them. You see that the two replicates of Palm in Sample05 have very different Cq values. All other bad replicates are coming from standard samples.\nIf you are finished checking the data quality, click **Next** to go to the **Amplification efficiencies** page.\n\n#### Taking into account amplification efficiencies\n\nQbase+ calculates an amplification efficiency (E) for each primer pair (= gene). Genes have different amplification efficiencies because:\n\n  - some primer pairs anneal better than others\n  - the presence of inhibitors in the reaction mix (salts, detergents…) decreases the amplification efficiency\n  - inaccurate pipetting\n\nQbase+ has a parameter that allows you to specify how you want to handle amplification efficiencies on the **Amplification efficiencies** page.\n\n| How to specify the amplification efficiencies strategy you want to use ?\n| :---------------------------- |\n|  Since we have included a dilution series for creating a standard curve in our qPCR experiment, we will select\n\n* **Use assay specific amplification efficiencies**\n* **Calculate efficiencies from included standard curves**\n\nAmplification efficiencies are calculated based on the Cq values of a serial dilution of representative template, preferably a mixture of cDNAs from all your samples. Since you know the quantity of the template in each dilution, you can plot Cq values against template quantities for each primer pair. Linear regression will fit a standard curve to the data of each gene, and the slope of this curve is used to calculate the amplification efficiency.\n\n| How to check the amplification efficiencies of the genes ?\n| :---------------------------- |\n|  Once you have made this selection, qbase+ starts calculating the efficiencies and the results are immediately shown in the **calculation efficiencies** table.\n\nIn this way, one amplification efficiency (E) for each gene is calculated and used to calculate **Relative Quantities (RQ)**:\n∆Cq is calculated for each well by subtracting the Cq of that well from the average Cq across all samples for the gene that is measured in the well. So ∆Cq is the difference between the Cq value of a gene in a given sample and the average Cq value of that gene across all samples. Cq is subtracted from the average because in this way high expression will result in a positive ∆Cq and low expression in a negative ∆Cq. \n**So at this point the data set contains one RQ value for each gene in each sample.**\n\nClick **Next** to go to the **Normalization** page.\n\n#### Normalization\n\nDifferences in amplification efficiency are not the only source of variability in a qPCR experiment. Several factors are responsible for noise in qPCR experiments e.g. differences in:\n\n  - amount of template cDNA between wells\n  - RNA integrity of samples\n  - efficiency of enzymes used in the PCR or in the reverse\n    transcription\n\nNormalization will eliminate this noise as much as possible. In this way it is possible to make a distinction between genes that are really upregulated and genes with high expression levels in one group of samples simply because higher cDNA concentrations were used in these samples.\nIn qPCR analysis, normalization is done based on housekeeping genes.\n\nHousekeeping genes are measured in all samples along with the genes of interest. In theory, a housekeeping gene should have identical RQ values in all samples. In reality, noise generates variation in the expression levels of the housekeeping genes. This variation is a direct measure of the noise and is used to calculate a normalization factor for each sample.\nThese normalization factors are used to adjust the RQ values of the genes of interest accordingly so that the variability is eliminated.\n\nThese adjusted RQ values are called **Normalized Relative Quantities (NRQs)**. In qbase+ housekeeping genes are called reference genes. In our data set there are three reference genes: Stable, Non-regulated and Flexible. On the **Normalization page** we can define the normalization strategy we are going to use, appoint the reference genes and check their stability of expression.\n\n| How to specify the normalization strategy you want to use ?\n| :---------------------------- |\n|  You can specify the normalization strategy you want to use on the Normalization method page:\n\n* **Reference genes** normalization is based on the RQ values of the housekeeping genes\n* **Global mean** normalization calculates normalization factors based on the RQ values of all genes instead of only using the reference genes. This strategy is recommended for experiments with more than 50 random genes. Random means that the genes are randomly distributed over all biological pathways.\n* **Custom value** normalization is used for specific study types. This strategy allows users to provide custom normalization factors such as for example the cell count.\n* **None** means that you choose to do no normalization at all. This option should only be used for single cell qPCR.\n\nWe have incorporated 3 housekeeping genes in our experiment so we select the **Reference genes** strategy.\n\n| How to appoint reference targets ?\n| :---------------------------- |\n|  You have to indicate which targets should be used as reference genes since qbase+ treats all genes as targets of interest unless you explicitly mark them as reference genes on the Normalization method page:\n\nWe have measured 3 housekeeping genes: Stable, Flexible and Non-regulated so we tick the boxes in front of their names.\n\nIt&#39;s not because you have appointed genes as reference genes that they necessarily are **good** reference genes. They should have stable expression values over all samples in your study. Fortunately, qbase+ checks the quality of the reference genes. For each appointed reference gene, qbase+ calculates two indicators of expression stability\n\n  - **M** (geNorm expression stability value): calculated based on the pairwise variations of the reference genes.\n  - **CV** (coefficient of variation): the ratio of the standard deviation of the NRQs of a reference gene over all samples to the mean NRQ of that reference gene.\n\nIt is considered that the higher these indicators the less stable the reference gene.\n\n| Are Flexible, Stable and Nonregulated good reference targets ?\n| :---------------------------- |\n|  M and CV values of the appointed reference genes are automatically calculated by qbase+ and shown on the Normalization method page:\n\nThe default limits for M and CV were determined by checking M-values and CVs for established reference genes in a pilot experiment that was done by Biogazelle. Based on the results of this pilot experiment, the threshold for CV and M was set to 0.2 and 0.5 respectively.\nIf a reference gene does not meet these criteria it is displayed in red. As you can see the M and CV values of all our reference exceed the limits and are displayed in red.\n\nIf the quality of the reference genes is not good enough, it is advised to remove the reference gene with the worst M and CV values and re-evaluate the remaining reference genes.\n\n| Which reference target are you going to remove ?                                                                                                                                    |\n| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |\n| Both the M-value and the CV are measures of variability. The higher these values the more variable the expression values are. So we will remove the gene with the highest M and CV. |\n\nYou can remove a reference gene simply by unticking the box in front of its name.\n\n| Are the two remaining reference genes good references ?\n| :---------------------------- |\n|  After removing Flexible as a reference gene the M and CV values of the two remaining reference genes decrease drastically to values that do meet the quality criteria. M and CV values that meet the criteria are displayed in green.\n\nThis exercise shows the importance of using a minimum of three reference genes. If one of the reference genes does not produce stable expression values as is the case for Flexible, you always have two remaining reference genes to do the normalization.\n\n[See how to select reference genes for your qPCR experiment](http://wiki.bits.vib.be/index.php/Using_GeneVestigator_to_select_candidate_reference_genes).\n\n**So after normalization you have one NRQ value for each gene in each sample.**\n\nClick **Next** to go to the **Scaling** page.\n\n#### Scaling\n\nRescaling means that you calculate NRQ values relative to a specified reference level.\n\nQbase+ allows you to rescale the NRQ values using one of the following as a reference:\n\n  - the sample with the minimal expression\n  - the average expression level of a gene across all samples\n  - the sample with the maximal expression\n  - a specific sample (e.g. untreated control)\n  - the average of a certain group (e.g. all control samples): this is\n    often how people want to visualize their results\n  - positive control: only to be used for copy number analysis\n\nAfter scaling, the expression values of the choice you make here will be set to 1 e.g. when you choose **average** the average expression level across all samples will be set to 1 and the expression levels of the individual samples will be scaled accordingly.\n\n| How to scale to the average of the untreated samples ?\n| :---------------------------- |\n|  You can specify the scaling strategy on the **Scaling** page. Select **Scale to group** and set the **Scaling group** to the **untreated** samples . This is one of the reasons why you need the grouping annotation.\n\nRescaling to the average of a group is typically used to compare results between 2 groups, e.g. treated samples against untreated controls. After rescaling, the average of the NRQs across all untreated samples is 1 and the NRQs of the treated samples are scaled accordingly.\n\nClick **Next** to go to the **Analysis** page.\n\n#### Visualization of the results\n\nOne of the things you can select to do on the **Analysis** page is viewing the relative expression levels (= scaled NRQs) of each of the genes in a bar chart per gene. It is recommended to visualize your results like this.\n\nIt is possible to view the relative expression levels of all genes of interest on the same bar chart. You can use this view to see if these genes show the same expression pattern but you cannot directly compare the heights of the different genes because each gene is independently rescaled\\!\n\n| How to visualize single gene expression bar charts ?\n| :---------------------------- |\n|  Select **Visually inspect results For individual targets** on the **Analysis** page and click **Finish**\n\n| How to visualize the expression levels of Palm in each sample ?\n| :---------------------------- |\n|  Select **Visually inspect results For individual targets** on the **Analysis** page and click **Finish**\n\nThe **Target** select box allows you to select the gene you want to view the expression levels of. Relative expression levels are shown for each sample. Error bars are shown and represent the technical variation in your experiment (variation generated by differences in amounts pipetted, efficiency of enzymes, purity of the samples...).\n\nYou see that Palm has a low expression level and a very large error bar in Sample05 because the two replicates of this sample had very different Cq values. You can group and colour the bars according to a property.\n\n| How to group the bars of Palm according to treatment (so treated at one side and untreated at the other side)\n| :---------------------------- |\n|  In the **Grouping** section you can specify the property you want to group by.\n\n| How to view average expression levels in each group ?\n| :---------------------------- |\n|  In the **Grouping** section you can choose to plot individual samples as shown above but you can also choose to **plot group average** expression levels.\n\nThe error bars that you see here represent biological variation and will be used later on in the statistical analysis. The error bars are 95% confidence intervals which means that they represent the range that will contain with 95% certainty the real average expression level in that group of samples.\nThe nice characteristic of 95% confidence intervals is the following:\n\n  - if they do not overlap you are sure that the expression levels in the two groups are significantly different, in other words the gene is differentially expressed\n  - if they do overlap you cannot say that you are sure that the expression levels are the same. You simply don’t know if the gene is differentially expressed or not.\n\n| Assess the effect of switching the Y-axis to a logarithmic scale for Palm.\n| :---------------------------- |\n|  In the **Y axis** section you can specify if you want a linear or logarithmic axis.\nAs you can see you do not change the expression values, you just change the scale of the Y axis. Switching the Y-axis to a logarithmic scale can be helpful if you have large differences in NRQs between different samples\n\n| Assess the effect of switching the Y-axis to a logarithmic scale for Flexible.\n| :---------------------------- |\n|  Switch to the bar charts of Flexible. By switching the Y-axis to logarithmic you can now see more clearly the differences between samples with small NRQs.\n\n#### Statistical analysis\n\nOnce you generate target bar charts you leave the **Analysis wizard** and you go to the regular qbase+ interface. Suppose that you want to perform a statistical test to prove that the difference in expression that you see in the target chart is significant.\nAt some point, qbase+ will ask you if your data is coming from a normal distribution. If you don&#39;t know, you can select **I don&#39;t know** and qbase+ will assume the data are not coming from a normal distribution and perform a stringent non-parametric test.\nHowever, when you have **7 or more replicates per group**, you can check if the data is normally distributed using a statistical test. If it is, qbase+ will perform a regular t-test. The upside is that the t-test is less stringent than the non-parametric tests and will find more DE genes. However, you may only perform it on normally distributed data. If you perform the t-test on data that is not normally distributed you will generate false positives i.e. qbase+ will say that genes are DE while in fact they are not. Performing a non-parametric test on normally distributed data will generate false negatives i.e. you will miss DE genes.\n\nChecking if the data is normally distributed can be easily done in GraphPad Prism. To this end you have to export the data.\n| How to export the data ?\n| :---------------------------- |\n|  To export the results click **the upward pointing arrow** in the qbase+ toolbar:\nYou want to export the normalized data so select **Export Result Table (CNRQ)**:\nYou will be given the choice to export results only (CNRQs) or to include the errors (standard error of the mean) as well . We don&#39;t need the errors in Prism so we do not select this option.\nThe scale of the Result table can be linear or logarithmic (base 10) . Without user intervention, qbase+ will automatically log10 transform the CNRQs prior to doing statistics. So we need to check in Prism if the log transformed data are normally distributed.\nAdditionally, you need to tell qbase+ where to store the file containing the exported data. Click the **Browse** button for this .\n\nExporting will generate an Excel file in the location that you specified. However, the file contains the results for all samples and we need to check the two groups (treated and untreated) separately. The sample properties show that the even samples belong to the treated group and the odd samples to the untreated group.\nThis means we have to generate two files:\n\n  - [a file containing the data of the untreated samples](http://data.bits.vib.be/pub/trainingen/qbasePLUS/resultslog.csv)\n  - [a file containing the data of the treated samples](http://data.bits.vib.be/pub/trainingen/qbasePLUS/resultslogTreated.csv)\n\nNow we can open these files in Prism to check if the data is normally distributed.\n\n| How to import the data of the untreated samples in Prism ?\n| :---------------------------- |\n|  \n* Open Prism\n* Expand **File** in the top menu\n* Select **New**\n* Click **New Project File**\n* In the left menu select to create a **Column** table. Data representing different groups (in our case measurements for different genes) should always be loaded into a column table.\n* Select **Enter replicate values, stacked into columns** (this is normally the default selection) since the replicates (measurements for the same gene) are stacked in the columns.\n* Click **Create**\n\nPrism has now created a table to hold the data of the untreated samples but at this point the table is still empty. To load the data:\n\n* Expand **File** in the top menu\n* Select **Import**\n* Browse to the resultslog.csv file, select it and click **Open**\n* In the **Source** tab select **Insert data only**\n* Since this is a European csv file commas are used as decimal separators so in contrast to what its name might imply, semicolons and not commas are used to separate the columns in the csv file (you can open the file in a text editor to take a look). In American csv files dots are used as decimal separator and the comma is used to separate the columns. Prism doesn&#39;t know the format of your csv file so you have to tell him the role of the comma in your file. Select **Separate decimals**\n* Go to the **Filter** tab and specify the rows you want to import (the last rows are these of the standard and the water samples, you don&#39;t want to include them)\n* Click **Import**\n\nAs the file is opened in Prism you see that the first column containing the sample names is treated as a data column. Right click the header of the first column and select **Delete**\n\n| How to check if the data of the untreated samples comes from a normal distribution ?\n| :---------------------------- |\n|  \n* Click the **Analyze** button in the top menu\n* Select to do the **Column statistics** analysis in the **Column analyses** section of the left menu\n* In the right menu, deselect **Flexible**. It&#39;s a bad reference gene so you will not include it in the qbase+ analysis so there&#39;s no point checking its normality (it is probably not normally distributed). In that respect you could also deselect the other two reference genes since you will do the DE test on the target genes and not on the reference genes.\n* Click **OK**\n* In the **Descriptive statistics** and the **Confidence intervals** section deselect everything except **Mean, SD, SEM**. These statistics is not what we are interested in: we want to know if the data comes from a normal distribution. The only reason we select Mean, SD, SEM is because if we make no selection here Prism throws an error.\n* In the **Test if the values come from a Gaussian distribution** section select the **D&#39;agostino-Pearson omnibus test** to test if the data are drawn from a normal distribution. Although Prism offers three tests for this, the D&#39;Agostino-Pearson test is the safest option.\n* Click **OK**\n\nPrism now generates a table to hold the results of the statistical analysis: As you can see, the data for Palm are not normally distributed.\n\nSince we found that there&#39;s one group of data that does not follow a normal distribution, it&#39;s no longer necessary to check if the treated data are normally distributed but you can do it if you want to. We will now proceed with the statistical analysis in qbase+. Statistical analyses can be performed via the **Statistics wizard**.\n\n| How to open the Statistics wizard ?\n| :---------------------------- |\n|  You can open it in the **Project Explorer** (window at the left):\n\n* expand **Project1** if it&#39;s not yet expanded\n* expand the **Experiments** folder in the project if it&#39;s not yet expanded\n* expand the **GeneExpression** experiment if it&#39;s not yet expanded\n* expand the **Analysis** section if it&#39;s not yet expanded\n* expand the **Statistics** section\n* double click **Stat wizard**\n\nThis opens the **Statistics wizard** that allows you to perform various kinds of statistical analyses.\n\n| Which kind of analysis are you going to do ?                                                                                                                                                                                                                                                      |\n| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |\n| On the **Goal** page: Select **Mean comparison** since you want to compare expression between two groups of samples so what you want to do is comparing the mean expression of each gene in the treated samples with its mean expression level in the untreated samples. Click **Next**. |\n\n| How to define the groups that you are going to compare ?\n| :---------------------------- |\n|  On the **Groups** page: specify how to define the two groups of samples that you want to compare. Select **Treatment** as the grouping variable to compare treated and untreated samples. Click **Next**.\n\n| How to define the genes that you want to analyze ?\n| :---------------------------- |\n|  On the **Targets** page: specify for which targets of interest you want to do the test. Deselect **Flexible** since you do not want to include it in the analysis. It&#39;s just a bad reference gene. Click **Next**.\n\nOn the **Settings** page you have to describe the characteristics of your data set, allowing qbase+ to choose the appropriate test for your data. \n\nThe first thing you need to tell qbase+ is whether the data was drawn from a normal or a non-normal distribution. Since we have 8 biological replicates per group we can do a test in Prism to check if the data are normally distributed.\n\n| Which gene(s) is/are differentially expressed ?\n| :---------------------------- |\n|  On the **Settings** page you describe the characteristics of your data set so that qbase+ can choose the ideal test for your data. For our data set we can use the default settings. Click **Next**. In the results **Table** you can see that the p-value for Palm is below 0.05 so Palm is differentially expressed.\n\n\n\nIn this example we will analyze data from another expression study with the following characteristics:\n\nAll samples fit in a single run: [Run7](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run7.xls)\nWe have the following samples:\n\n  - 5 control samples: control1, control2…\n  - 5 treated samples: treated1, treated2…\n  - 1 no template control: NTC\n\nThe expression of the following genes was measured:\n\n  - 2 reference genes: refgene1 and refgene2\n  - 2 genes of interest: gene1 and gene2\n\nThere are two technical replicates per reaction\n\n#### Creating a new experiment\n\n| Create a new Experiment called GeneExpression2 in Project1\n| :---------------------------- |\n| You can find the details on how to create a new experiment in Creating a project and an experiment\n\n#### Loading the data\n\n| Import [Run7](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run7.xls). This file is in qBase format.                    |\n| :-------------------------------------------------------------------------------------------------------------------------------------- |\n| You can find the details on how to import the data file in the **Loading the data into qbase+** section of Analyzing data from a geNorm pilot experiment in qbase+ |\n\n#### Adding sample annotation\n\nDownload the [the sample properties file](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Sample_Properties_GE2.xlsx).\n| Add a custom sample property called Treatment.\n| :---------------------------- |\n| You can find the details on how to add a custom sample property in the **Adding annotation to the data** section of Loading data into qbase+\n\n#### Analyzing the data\n\n| Choose the type of analysis you want to perform.\n| :---------------------------- |\n| \n\n| Check controls and replicates.\n| :---------------------------- |\n| First set the minimum requirements for controls and replicates  You see that 6 replicates do not meet these requirements . Select to **Show details and manually exclude bad replicates**\nAll negative controls pass the test . Positive controls were not included in this analysis. Qbase+ will now open the results for the failing replicates: as you can see the difference in Cq values between these replicates is not that big. They fail to meet the requirement just slightly.\n\n| Which amplification efficiencies strategy are you going to use ?\n| :---------------------------- |\n| You don&#39;t have data of serial dilutions of representative template to build standard curves so the only choice you have is to use the default amplification efficiency (E = 2) for all the genes.\n\n| Appoint the reference genes as reference targets.\n| :---------------------------- |\n| You can find the details on how to appoint reference targets in the **Normalization** section of Analyzing gene expression data in qbase+\n\n| Is the stability of the reference genes ok ?\n| :---------------------------- |\n| In the **Reference target stability window** the M and CV values of the reference genes are shown in green so the stability of the reference genes is ok. You can find the details on how to check reference target stability in the **Normalization** section of [](Analyzing_gene_expression_data_in_qbase+\&quot; title=\&quot;wikilink)Analyzing gene expression data in qbase+\n\n| Which scaling strategy are you going to use ?\n| :---------------------------- |\n| Since you have a treated and a **control** group, it seems logical to use the average of the control group for scaling. You can find the details on how to specify the scaling strategy in the **Scaling** section of Analyzing gene expression data in qbase+\n\nLook at the target bar charts.\n\n| In the target bar charts group the samples according to treatment.\n| :---------------------------- |\n| You can find the details on how to group the samples in the **Visualization of the results** section of Analyzing gene expression data in qbase+\n\nThe samples of each group are biological replicates so you might want to generate a plot that compares the average expression of the treated samples with the average expression of the untreated samples.\n\n| In the target bar charts plot the group averages instead of the individual samples.\n| :---------------------------- |\n| In the **Grouping** section at the bottom of the chart you can select **Plot group average**:\n\n| Are there any genes for which you see a clear difference in expression between the two groups ?\n| :---------------------------- |\n| For gene 1, the mean expression levels in the two groups are almost the same and the error bars completely overlap.\n\nWhen you look at the title of the Y-axis, you see that 95% confidence levels are used as error bars. In case of 95% confidence intervakls you can use the following rules:\n\n* if they do not overlap: you are certain that the difference between the means of the two groups is significant\n* if they do not overlap: you know nothing with certainty: the means can be different or they can be the same\n\nSo for gene 1 the means are very close but just based on the plot we may not make any conclusions with certainty. For gene 2, the mean expression levels in the two groups are very different and the error bars do not overlap. So the 95% confidence intervals do not overlap meaning that we can be certain that the difference between the means of the two groups is significant.\n\n| Use a statistical test to compare the expression levels between the two groups of samples ?\n| :---------------------------- |\n| You only have 5 replicates per group so you cannot test if the data comes from a normal distribution. Qbase+ will assume they&#39;re not normally distributed and perform a non-parametric Mann-Whitney test.\n\nThe p-value of gene2 is smaller than 0.05 so it has a statistically significant difference in expression levels in treated samples compared to untreated samples. For gene1 the p-value is 1 so we have no evidence to conclude that the expression of gene1 is different in treated compared to untreated samples. You can find the details on how to compare the means of the two groups in the **Statistical analysis** section of Analyzing gene expression data in qbase+\n&quot;,&quot;### Exercise 1: simple gene expression study\n\nIn my qPCR experiment I want to study the expression of 12 genes of interest in 8 samples of interest. I want to use 2 PCR replicates for each reaction.\n\n&gt; How many 96 well plates do I need for this experiment ?\n&gt; &gt; I have 12 genes in 8 samples which gives a total of 96 reactions (one plate). I want to perform each reaction twice (2 PCR replicates) so I need two plates. However, I need to include reference genes in my experiment, preferably more than one. I can put these reference genes on a separate plate, I do not have to include them on each plate.\nIdeally, you need to include 3 reference genes so having 8 samples and 2 replicates this gives an additional 48 reactions. Thus, I need three 96 well plates to perform this experiment.\n\n| Do I need to include IRCs (inter-run calibrators) ?                      |\n| :--------------------------------------------------- |\n| No, I can easily fit all samples of the same gene on the same plate so I don&#39;t need to include IRCs. |\n\n### Exercise 2: a large study\n\nIn my qPCR experiment I want to study the pattern of expression of 96 genes (genes of interest and reference genes) in 96 samples of interest, divided into a few groups. I want to use 2 PCR replicates for each reaction.\n\n| Do I need to include IRCs (inter-run calibrators) ?                    |\n| :------------------------------------------------------- |\n| No, I can fit all samples of the same gene on the same plate so I don&#39;t need to include IRCs. |\n\nI want to include PCR replicates.\n\n| Do I need to include IRCs when I work on a 96 well plate ?                                                                                    |\n| :-------------------------------------------------------------------------------------------------------------------------------------------- |\n| Yes, I have 192 reactions per gene so I cannot place them on the same plate. Remember that replicates have to be located on the same plate \\! |\n\n| Do I need to include IRCs when I work on a 384 well plate ?                        |\n| :--------------------------------------------------------------------------------- |\n| No, I have 192 reactions per gene so I can even place two genes on the same plate. |\n\nI want to include no template controls but I don&#39;t want to increase the\nnumber of plates.\n\n| What is the most elegant strategy to make room for including negative controls ?      |                                                      \n| :------------------------------------------------------------------------------------ |\n| This kind of study screen for expression patterns and requires statistical analysis. Since you have many samples divided over a few groups it means you have many biological replicates so you could easily do without the PCR replicates. By doing so you preserve the biological variability which is often far greater than the technical variation. |\n\n### Exercise 3: how to fill plates ?\n\nIn my qPCR experiment I want to study the pattern of expression of 5 genes (genes of interest and reference genes) in 38 samples (samples of interest and control samples). I want to use 2 PCR replicates for each reaction.\n\n| What is the minimum number of 96 well plates I need for this experiment ? |\n| :------------------------------ |\n| 5 genes * 38 samples * 2 replicates = 380 reactions.\nI need a minimum of 4 plates for this experiment.\n\n| If I use the minimum number of 96 well plates do I need to include IRCs ?                                                      |\n| :----------------------------------------------------------------------------------------------------------------------------- |\n| Yes, 5 genes spread over 4 plates with 72 reactions per gene means that at least one gene will be spread over multiple plates. |\n\n| What can I do to avoid inter-run variability ?                                                                                                                                                                                         |\n| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |\n| I can use 5 plates and fill them with one gene each. They will not be completely filled (72 reactions) but at least I do not have to use IRCs (which are additional reactions that also cost money) and I have no inter-run variation. |\n\nSuppose there&#39;s only one 96-well plate left in your lab. You have 10 samples (samples of interest + control samples) and you want to make the most of what you have.\n\n| How many genes of interest would you measure ? |\n| :------------------------------ |\n| Since you want to make most of what you have, let&#39;s assume you are omitting PCR replicates.\nTheoretically, you could fit 9 genes on your 96-well plate. However, to avoid pipetting mistakes I would measure only 8 genes so I can work with one row / gene. This is very handy for multichannel pipets.\n\n### Exercise 4: a growing study\n\nIn my qPCR experiment I want to study the pattern of expression of 24 genes (genes of interest and reference genes) in 48 samples (samples of interest and control samples). I want to use 2 PCR replicates for each reaction.\n\n| How many genes can I analyze on one 384 well plate ? |\n| :------------------------------ |\n| 48 samples * 2 replicates = 96 reactions per gene.\nI can analyze 4 genes on each 384 well plate.\n\nEach week I receive 2 additional samples to analyze.\n\n| Do I analyze them immediately after I get them ? |\n| :------------------------------ |\n| No. Since the samples are placed on different plates as in the previous experiment, you have to use IRCs. You typically need 3 IRCs and a no template control sample. It means that if you want to analyze these 2 samples you have to include 4 additional samples for each gene. This is a lot of overhead for just 2 samples !\nTry to avoid this: it&#39;s better to wait a few weeks until you have 6 or 8 or even more samples.\n\n### Exercise 5: a diagnostic copy number screen\n\nIn diagnostic screens all samples are important: you cannot leave out samples and all measurements need to be of the highest quality possible. In my qPCR experiment I want to study copy number variation of 16 genes\n(genes of interest and reference genes) and 2 calibrator samples (samples with known copy number). Since we need high quality data we will use 4 technical replicates.\n\n| Are we going to use sample maximization ?                                   |                   \n| :------------------------------------------------------------------------- |\n| No. In contrast to gene expression studies, where we want to compare expression levels of a gene between different groups of samples, copy number analyses do compare genes. It means that in this case the sample maximization approach (placing all samples of the same gene on the same plate) is not valid. Instead we use a gene maximization approach here (placing same sample for different genes on the same plate). |\n\n| How many samples can I fit on a 384 well plate ? |\n| :------------------------------ |\n| We have 16 (genes) * 4 (replicates) = 64 reactions per sample.\nThis means that we can fit 6 samples on a 384 well plate: 4 unknowns and 2 calibrators.\n\n### Exercise 6: fix experiments with bad or missing data\n\nIn my qPCR experiment I want to study gene expression of 6 genes (3 genes of interest and 3 reference genes) in 20 samples (samples of interest and control samples). I want to use 2 technical replicates. One of my genes of interest failed completely and I want to repeat the measurements for this gene in a new run.\n\n| Do I need to include IRCs ?                                                                                |\n| :--------------------------------------------------------------------------------------------------------- |\n| No. We can put the 20 samples of the gene that failed on a single plate so we do not have to include IRCs. |\n\n| Do I need to include reference genes ?                                                                 |\n| :----------------------------------------------------------------------------------------------------- |\n| No. We just repeat all samples for the gene that failed and replace the old data with the new results. |\n\nOne of the reference genes failed completely.\n\n| What should I do ?                                                                                                                                                                                                                                                                                     |\n| :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |\n| Depending on the quality of the two remaining reference genes, you should either do nothing or do the same as in the previous example where one of your genes of interest failed. If the two remaining reference genes are stable you can do the normalization with the two remaining reference genes. |\n\nThree samples failed completely.\n\n| What&#39;s the first thing I need to do ?                                                                                                                             |\n| :---------------------------------------------------------------------------------------------------------------------------------------------------------------- |\n| Since they failed completely, they are probably of low quality. Therefore, you have to prepare the samples again, check their quality and then use them for qPCR. |\n\n| Do I need to include IRCs ?                                                                                             |\n| :---------------------------------------------------------------------------------------------------------------------- |\n| Yes. If you want to compare these samples with the samples that didn&#39;t fail, you have to perform inter-run calibration. |\n\nThree samples failed for one of the genes of interest\n\n| What is the first question I need to ask ? |\n| :----------------------------------------- |\n| Is the gene expressed in these samples ?   |\n\n| Is it possible the RNA of these three samples was of low quality ?        |\n| :------------------------------------------------------------------------ |\n| Not likely, the measurements for the other genes in these samples are ok. |\n\nThree samples failed for one of the reference genes\n\n| Can I use the measurements of that reference gene in the non-failing samples for normalization ?                                         |\n| :--------------------------------------------------------------------------------------------------------------------------------------- |\n| No, qbasePLUS requires that you use the same reference genes for all samples so you have to discard all samples for that reference gene. |\n\n### Exercise 7: dilution series for calculating amplification efficiencies\n\nIn my qPCR experiment I want to study 8 new genes for which I had to design new primer pairs in 12 samples (samples of interest and control samples). I want to use 2 technical replicates and 96 well plates.\n\n| What is the first thing I need to do ?                                                        |\n| :-------------------------------------------------------------------------------------------- |\n| Perform a pilot experiment to determine the amplification efficiencies of these primer pairs. |\n\nFor this I need a dilution series of representative cDNA template.\n\n| How many dilutions would you include ?                                           |\n| :------------------------------------------------------------------------------- |\n| A dilution series with 6 dilutions for 8 genes nicely fits into a 96 well plate. |\n\nA few weeks after my initial qPCR experiment I want to test these 8 genes in a new set of samples.\n\n| Do I have to repeat the pilot experiment ?      |\n| :---------------------------------------------- |\n| No, dilution series do not need to be repeated. |&quot;,&quot;You need to do inter-run calibration if you want to compare samples from different runs e.g.:\n\n  - when it is not possible to get all samples for the same gene on the same plate\n  - when you do additional runs weeks or months after your initial experiment\n\nOf course there is a lot of variability between runs on a qPCR instrument:\n\n  - thermal block is not always heating uniformously\n  - quality of the lamp, the filters and the detector decreases over time\n  - data analysis settings on the qPCR instrument (baseline correction and threshold) can be slightly different\n  - efficiency of reagents (polymerase, fluorophores) is variable\n  - optical properties of the plastic plates vary\n\nFortunately, inter-run calibration allows you to eliminate most of this variability.\n\nIn this experiment we will analyze the data from the gene expression experiment (see Analyzing gene expression data in qbase+) together with data from 2 runs (Run4 and Run5) that were done weeks after the initial gene expression experiment.\n\nBecause the data comes from two different experiments spread over time, we have included three inter-run calibrators on the plates: Sample01, Sample02 and Sample03.\n\nThe principle of the IRCs is very similar to that of the reference genes:\nIn theory, the IRCs should have the same NRQ in each run. In practice, the difference in NRQ between two runs is a measure of the inter-run variation and can be used to adjust the NRQs to remove the inter-run variation.\n\n#### Creating a new Experiment\n| Import [Run1](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run1.xls), [Run2](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run2.xls), [Run3(all three in CFX format)](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run3.xls), [Run4](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run4.xls) and [Run5 (the latter two are in qBase format)](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run5.xls).\n| :----------------------------- |\n| Since the data is in files of two different format, you have to do a separate import for each format. So first import Run1, Run2 and Run3, then import Run4 and Run5. You can find the details on how to import CFX files in [](Loading_data_into_qbase+\&quot; title=\&quot;wikilink)Loading data into qbase+.\nThe details of importing qBase files are in [](Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase+\&quot; title=\&quot;wikilink)Analyzing data from a geNorm pilot experiment in qbase+\n\n#### Analyzing the data\n\n| Use assay specific amplification efficiencies.\n| :----------------------------- |\n| You can find the details on how to convert the targets in the **Taking into account amplification efficiencies** section of Analyzing gene expression data in qbase+\n\nIn Analyzing gene expression data in qbase+ we have already checked the stability of the reference genes (see **Normalization** section). We determined that Flexible did not show stable expression.\n\n| Convert Stable and Nonregulated to Reference targets.\n| :----------------------------- |\n| You can find the details on how to convert the targets in the **Normalization** section of Analyzing gene expression data in qbase+\n| Appoint Sample01, Sample02 and Sample03 as IRCs.\n| :----------------------------- |\n| Leave the Analysis wizard by clicking the **Close wizard** button in the top menu.\n\n - Expand **Intermediate results** (red) in the **Project Explorer**\n - Double click **Interrun calibration** (green)\n\nThis opens the **Interrun calibration window**:\n\n - Click the **New** button (blue) to create a IRC\n - Once the IRC is created you have to appoint samples to it: select **Sample01** in the list of **Other samples**\n - Click the **Add Sample** button (purple)\n - Remember that you cannot give IRCs the same name in different runs: the software would think that they are technical replicates spread over different plates (which is not allowed). Therefore, in Run4 and Run5 we have given Sample01 another name: Sample01_2. Select **Sample01_2** in the list of **Other samples**\n - Click the **Add Sample** button (purple)\n\nYou have appointed the first IRC (grey), now do the same for the other two IRCs.\n\nRemember that for each target the variability of the normalized\nexpression levels of the IRCs between different runs will be used to\nadjust the other normalized expression levels of that target gene. The\nadjustment is done by amplifying the normalized expression levels with a\ncalibration factor that is calculated based on the normalized expression\nlevels of the IRCs.\nSince variability between runs is the same for each IRC, you expect that\nall IRCs measure the variability between the runs to the same extent,\nhence leading to similar calibration factors.\n\n| Do these IRCs generate similar calibration factors ?\n| :----------------------------- |\n| Open the **Calibration Factors** tab (red) of the **Interrun calibration window** and look at the result for Duvel:\n\nYou see that IRC2 returns a substantially different calibration factor in Run5 (green) so the validity of this IRC should be interpreted with care.\nFor Leffe the IRCs also gives inconsistent results in Run5. Switch to the results for Leffe by selecting **Leffe** in the **Targets** list (blue)\n| Do you still see the same expression pattern for Palm as you did in the first three runs ?\n| :----------------------------- |\n| Open the target bar chart for Palm.\n\nYou see that the pattern Palm showed in the first three runs (sample01 to sample16): high expression in the odd and low expression in the even samples is reversed in the samples from Run4 and Run5 (sample17 to sample25). In the latter runs you see high expression in the even and low expression in the odd samples. However, without annotation for Run4 and Run5 (which samples are treated and which not) it&#39;s impossible to interpret the bar chart.\n\n1. [Link](http://youtu.be/OJFsuZqNUHs)&quot;,&quot;Since normalization of qPCR data is based on the assumption that the reference targets have the same expression level in all samples it is crucial that the expression of the chosen reference genes is stable.\nHowever, none of the so-called **housekeeping** genes is universally stably expressed.\n\n[Genevestigator](https://www.genevestigator.com/gv/), both the commercial and the free version, contains a tool, called RefGenes, that allows to identify candidate reference genes that display very stable expression in the context that you are working in, typically a certain tissue of a certain organism.\n\nGenevestigator is a platform that contains curated public microarray data from thousands of experiments/conditions.\n\nRefGenes allows you to select the conditions that are relevant for you, e.g. mouse liver, human fibroblasts, or Arabidopsis thaliana leaves. In a next step, RefGenes identifies the genes with the most stable expression in the selected conditions.\n\n## Starting the RefGenes tool\n\n| How to start the RefGenes tool ? |\n| :-------------------------------- |\n| - Open the [RefGenes page](http://www.refgenes.org/rg/).\n - Click **start GENEVESTIGATOR**\n - Click the **Install/Start** button\n - This will automatically open a Genevestigator startup page. Keep this page open during the analysis. Closing this page will close Genevestigator.\n - Login. Also for the free version you need to create an account (use your academic email for this since you will need your vib email to get access to the commercial version).\n - Genevestigator is opened automatically\n\n## The Genevestigator user interface\n\nThe Genevestigator consists of the following components:\n  - **Sample Selection** panel: to choose the experimental conditions you&#39;re interested in (green)\n  - **Gene Selection** panel: to choose the genes you&#39;re interested in (blue)\n  - Center panel shows an overview of all available tools (purple). Once you have selected a tool, the panel will show the results of the analysis that is done by the tool.\n  - **Home** button (red) allows to return to the overview of the tools at any time. The text next to the home button indicates the toolset that you have selected.\n\nClick the **RefGenes** tool at the bottom.\n\n## Using the RefGenes tool to find reference genes\n\n### STEP 1: Choose samples from a biological context similar to those in your qPCR expriment\n\n| How to choose the samples you want to analyze ? |\n| :-------------------------------- |\n|\n - Click the **New** button in the **Sample Selection** panel. The selection of samples defines which data are used for the analysis.\n - Select the organism you&#39;re interested in (in this example: human)\n - Select the array type you want to analyze (in this example: human 133_2).\nFor most organisms Genevestigator contains expression data from multiple types of microarrays, e.g. different generations of Affymetrix GeneChips®. On these arrays, genes are sometimes represented by different sets of probes. To keep the analysis results easily interpretable, data from different array types are not mixed.\n - Click the **Select particular conditions** button to select all samples with a certain annotation, e.g. all data from a certain tissue type.\n - Select the type of conditions (red) you want to base your selection on (in this example: Anatomy). For each type (anatomy, neoplasms, perturbations, development...) you can browse the corresponding ontologies and select the desired condition(s) (green) (in this example: cardiac muscle).\n - Click **OK**\n\nNote that you can select multiple tissues.\nWhen you select samples for use in the RefGenes tool, you have to focus on microarrays from samples that were collected in conditions similar to those in your qPCR experiment. Don&#39;t make a too general selection, e.g. all human samples: you might end up with genes that are stable in most conditions but not in yours. Don&#39;t make a very specific selection either, e.g. human heart samples from patients taking the same medication as yours. If you want to broaden your study later on with samples from other patients, your reference genes might not be valid anymore. It is recommended to select reference genes in the same organism and the same / a similar tissue type as the one that you used in your experiments.\n\n### STEP 2: Select the gene(s) you want to measure in your qPCR experiment\n\nThis step is not essential, but it helps you to see whether your target gene(s) is (are) strongly or weakly expressed in the conditions of interest selected in STEP1. This allows you to search for candidate reference genes in a similar range of expression.\n\n| How to choose the genes you want to analyze ? |\n| :-------------------------------- |\n|\n - Click the **New** button in the **Gene Selection** panel.\n - Enter the name of your target gene in the text area (in this example: GOT1) and click **OK**\n - Open the RefGenes tool (if you haven&#39;t done that already). A red box plot representing the distribution of the expression levels of GOT1 in the 68 selected human heart samples appears in the center panel. As you can see, this gene is highly expressed in heart.\n\n\n\n\n### STEP 3: Find candidate reference genes\n\nThe reference genes that are suggested by GeneVestigator have the\nfollowing characteristics:\n\n  - They have the most stable expression levels across all selected samples (a small boxplot)\n  - Their overall expression level is similar to that of the target gene(s) of your qPCR experiment\n| How to find the candidate reference genes ? |\n| :-------------------------------- |\n|Click the **Run** button in the RefGenes tool. RefGenes will show the top 20 most stable genes with similar expression levels:\n\n\n\n## Exercises\n\n### Finding candidate reference genes in the free version of Genevestigator\n\nNow we will make a more elaborate exercise on finding candidate reference genes. We will do the analysis in the free version of RefGenes but the analysis in the commercial version is very similar.\nSuppose we want to compare the expression stability of the 4 commonly used reference genes for qPCR on mouse liver samples (ACTB, GAPDH, HPRT and TUBB4B) to that of 4 reference genes that are suggested by Genevestigator.\nTo this end we open the RefGenes tool and select the liver samples of the mouse 430_2 arrays.\n\n| Check the expression stability of the 4 commonly used reference genes ? |\n| :-------------------------------- |\n|\n - Click the **New** button in the **Gene Selection** panel to create a new selection. The selection of samples defines which data are used for the analysis.\n - Enter the name of your target gene in the text area (for example: ACTB) and click **OK**\n\nWhen you are using the commercial version, you may enter multiple genes at the same time, in the free version you have to enter them one by one. This means that you have to add the first gene as described above and then add the next gene by clicking the **Add** button and so on...\n\nFinally you end up with an expandable list of the genes you asked for and you can tick or untick them to control the display of their expression data in the main window. When you tick the 4 commonly used reference genes you can see how stable they are expressed in the 651 mouse liver samples that are stored in Genevestigator:\n\nAs you can see, the expression levels of the commonly used reference genes in the selected mouse liver samples is pretty variable which is also confirmed by their relatively high SD values.\nOften there are multiple probe sets for the same gene. When you use the free version you may only choose one probe set per gene so you have to make a choice. How to make that choice ?\nAffymetrix probe set IDs have a certain meaning: what comes after the underscore tells you something about the quality of the probes:\n\n  - **_at** means that all the probes of the probe set hit one known transcript. This is what you want: probes specifically targeting one transcript of one gene\n  - **_a_at** means that all the probes in the probe set hit alternate transcripts from the same gene. This is still ok the probes bind to multiple transcripts but at least the transcripts come from the same gene (splice variants)\n  - **_x_at** means that some of the probes hit transcripts from different genes. This is still not what you want: the expression level is based on a combination of signals of all the probes in a probe set so also probes that cross-hybridize\n  - **_s_at** means that all the probes in the probe set hit transcripts from different genes. This is definitely not what you want: if the probes bind to multiple genes you have no idea whose expression you have measured on the array\n\nSo I always ignore probe sets with s or x. If you have two specific probe sets for a gene, they should more or less give similar signals. If this is not the case, I base my choice upon the expression level that I expect for that gene based on previous qPCR results.\n\nAs you can see, each of these 4 commonly used reference genes has a high expression level. Most genes do not have such high expression levels. In most qPCR experiments your genes of interest will have low or medium expression levels, so these reference genes will not be representative for the genes of interest.\n\nReference genes should ideally have similar expression levels as the genes of interest. Therefore, we will select the four most stably expressed genes with a medium expression level (between 8 and 12) according to the RefGenes tool.\n\n| Select the 4 most stably expressed candidate reference gene with medium expression levels. |\n| :-------------------------------- |\n|\n - Untick all target genes.\n - Click the **Run** button at the top of the main window and check if the range is set correctly\n\nSelect the 4 candidates with the lowest SD: Then, we performed qPCR on a representative set of 16 of our liver samples to measure the expression of these 8 candidate reference genes and analyzed the data ([See how to select the best reference genes using geNorm in qbase+](http://wiki.bits.vib.be/index.php/Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase%2B)).\n\n\n### Finding candidate reference genes in the commercial version of Genevestigator\n\nWe will do the same exercise as above in the commercial version of Genevestigator. The difference between the free and commercial version of RefGenes is the number of target genes you can select. In the free version you have to select one gene and then gradually add all other genes one at a time. The commercial version allows you to load as many target genes as you want simultaneously. As a consequence, you can select multiple probe sets for the same gene.\nAll VIB scientists have free access to the commercial version of Genevestigator via their VIB email address. If you don&#39;t know your VIB email address, check [the Who&#39;s Who of VIB](http://www.vib.be/en/whoiswho/Pages/default.aspx).\n\n  - Open a browser and go to the [Genevestigator website](https://www.genevestigator.com/)\n  - If it&#39;s your **first time to access Genevestigator**, create an account by clicking **join now** button. You will be redirected to a new window in which you will give some personal information including a valid VIB email address. Click **Register** and check your email to activate your new account. Go back to the [GeneVestigator website](https://www.genevestigator.com/)\n  - Choose the research field you want to investigate: **pharma/biomediacal** or **plant biology** by clicking the corresponding button\n  - Click **Start**\n  - Use your VIB email address and password to login to Genevestigator.\n  - This will automatically open a Genevestigator startup page in your browser. Keep this page open during the analysis. Closing this page will close Genevestigator.\n  - Genevestigator is opened automatically\n\nOpen the RefGenes tool by clicking its icon in the **Further tools** secion and select the liver samples of the mouse 430_2 arrays [as explained in the previous exercise](http://wiki.bits.vib.be/index.php/Using_GeneVestigator_to_select_candidate_reference_genes#STEP_1:_Choose_samples_from_a_biological_context_similar_to_those_in_your_qPCR_expriment).\n| Check the expression stability of the 4 commonly used reference genes ? |\n| :-------------------------------- |\n| - Click the **New** button in the **Gene Selection** panel to create a new selection. The selection of samples defines which data are used for the analysis.\n - Enter the names of the 4 commercial reference genes in the text area and click **OK**\n\nI still remove probe sets with an _s or _x since they do not specifically bind to one single gene:\nFinally you end up with an expandable list of the genes you asked for and you can tick or untick them to control the display of their expression data in the main window. By default all probe sets are ticked so you can see how stable the commonly used reference genes are expressed in the 651 mouse liver samples that are stored in Genevestigator:\nAs you can see, the expression levels of the commonly used reference genes in the selected mouse liver samples is pretty variable which is also confirmed by their relatively high SD values.\n\nThe next step of selecting the 4 most stable candidate reference genes with medium expression levels is exactly the same as described above for the free version of RefGenes.\n\n| Create a new gene selection with 20 found candidate reference genes and call it mouse_references. |\n| :-------------------------------- |\n|Click the **New** button at the top of the main window to create a new selection.\n\nTo change the name of the selection right click the name in the **Gene selection** panel and select **Rename**\n\n| Identify perturbations where the mouse_references genes show more than 1,5 fold differential expression using the Condition perturbations tool. |\n| :-------------------------------- |\n|Click the **Home** button at the top to go back to the tools overview page.\n\nClick the **Perturbations** tool in the **Condition Search tools** section\n\n\nMake a **New Sample selection** including all mouse 430_2 arrays.\nUntick all genes except for the first one and filter the long heatmap for at least 1.5 fold change differential expression:\n\n\nYou now get a list of mouse samples in which the gene is not stably expressed so you can check if any of these samples is related to the samples in your study. Hover your mouse over the name of a sample to see more details about the sample.\nYou can do this for each of the candidate reference genes and select the ones that best fit your needs\n\n[Exercise on selecting reference genes for metacaspases in Arabidopsis thaliana](http://wiki.bits.vib.be/index.php/GV_Exercise.1).\n\n\nIn a geNorm pilot experiment you analyze a set of candidate reference genes in a representative set of samples that you want to test in your final experiment. Based on the M-values and CVs that are calculated by qbase+, you can choose the genes that most satisfy the criteria for a good reference gene.\n\n### Exercise 1: reference genes for mouse liver\n\nWe come back on the 8 candidate reference genes that we selected for mouse liver:\n\n  - 4 commonly used reference genes: ACTB, TUBB4B, GAPDH and HPRT\n  - 4 candidate reference genes with very stable medium expression levels selected based on expression data coming from more than 600 microarrays of mouse liver samples using Genevestigator: Gm16845, MUSK, OTOP3, EDN3\n\nWe have measured their expression in a represetative set of 16 of our mouse liver samples, each in triplicate. We will now analyze the stability of these candidate reference genes in our samples.\n\n#### Creating a new Experiment\n\n| Create a new Experiment called GeNormMouse in Project1 |\n| :------------------------------------------- |\n| Open qbase+ or, if the software is already open, click the Launch Wizard button.\n\nYou can find the details on how to create a new experiment in Creating a project and an experiment\n\n#### Loading the data into qbase+\n\nThe data is stored in [the RefGenes folder](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/RefGenes.zip). It consists of 8 Excel files, one file for each candidate reference gene. If you are not working on a BITS laptop, download and unzip the folder.\n\n| Import the data. This files are in qBase format. |\n| :------------------------------------------- |\n| You can find the details on how to start the data import in Loading data into qbase+\n\nUnlike the previous exercise, qbase+ does not allow you to do a quick import this time. In the Import Run window Manual import is selected:\nMake sure that Upload file to Biogazelle support for further analysis is NOT selected and click Next\nMake sure the correct File type is selected (qBase) and click Finish.\nThis file contains the data of the geNorm pilot experiment. In the pilot experiment, 8 candidate reference genes were measured in 16 representative mouse liver samples.\n#### Analyzing the geNorm pilot data\n\n| Specify the aim of the experiment. |\n| :------------------------------------------- |\n| In this experiment we want to select the ideal reference genes for our next experiments so we choose selection of reference genes (geNorm)\n\n| Check the quality of the replicates (use default parameter settings). |\n| :------------------------------------------- |\n| You can find the details on how to check the quality of the replicates in the Checking the quality of technical replicates and controls section of Analyzing gene expression data in qbase+\n\nWe haven&#39;t included any positive or negative controls so you don&#39;t need to show their details.\n\n| Select the Amplification efficiencies strategy you want to use. |\n| :------------------------------------------- |\n| You can find the details on how to select the Amplification effciencies strategy in the Taking into account amplification efficiencies section of Analyzing gene expression data in qbase+\n\nWe haven&#39;t included dilution series nor do we have data from previous qPCR experiments regarding the amplification efficiencies so we choose to use the same efficiency for all genes.\nIt is of course better to include a dilution series for each gene to have an idea of the amplification efficiencies of each primer pair.\n\n| Convert all genes to Reference genes. |\n| :------------------------------------------- |\n| You can convert all the genes simultaneously by selecting Use all targets as candidate reference genes\n\nClick Finish.\n\n| Which genes are you going to use as reference targets in further experiments ? |\n| :------------------------------------------- |\n| Upon clicking Finish, the geNorm window containing the analysis results is automatically opened. The geNorm window consists of three tabs. The tabs are located at the bottom of the window: geNorm M, geNorm V and Interpretation.\nThe first tab, geNorm M, shows a ranking of candidate genes according to their stability, expressed in M values, from the most unstable genes at the left (highest M value) to the best reference genes at the right (lowest M value):\nThe second tab, geNorm V, shows a bar chart that helps determining the optimal number of reference genes to be used in subsequent analyses:\n\nThe number of reference genes is a trade-off between practical considerations and accuracy. It is a waste of resources to quantify more genes than necessary if all candidate reference genes are relatively stably expressed and if normalization factors do not significantly change when more genes are included. However, Biogazelle recommends the minimal use of 3 reference genes and stepwise inclusion of more reference genes until the next gene has no significant contribution to the normalization factors.\nTo determine the need of including more than 3 genes for normalization, pairwise variations Vn/n+1 are calculated between two sequential normalization factors. Simply stated: V is measure of the added value of adding a next reference gene to the analysis. A large variation means that the added gene has a significant effect and should be included.\nIn normal experiments like the Gene expression experiment (see Analyzing gene expression data in qbase+), we only have 3 reference genes so we will see only 1 bar here. But in this geNorm pilot experiment, we analyzed 8 candidate reference genes, so we see 6 bars.\nAll pairwise variations are very low, so even the inclusion of a third gene has no significant effect. Based on a preliminary experiment that was done by Biogazelle, 0.15 is taken as a cut-off value for V, below which the inclusion of an additional reference gene is not required. Normally this threshold is indicated by a green line on the geNorm V bar chart. However since all V-values fall below the threshold in this geNorm pilot experiment, you don’t see this line on the bar chart.\nSo, these results mean that for all subsequent experiments on these samples, two reference genes, EDN3 and MUSK, would be sufficient. However, as stated before, Biogazelle recommends to always include at least three reference genes in case something goes wrong with one of the reference genes (so also include Gm16845). |\nThese are artificial data. But when you read [the paper by Hruz et al., 2011](http://www.biomedcentral.com/1471-2164/12/156/abstract) you see that the genes that are selected by Genevestigator are often outperforming the commonly used reference genes.\n\n### Exercise 2: reference genes for human heart\n\n#### Creating a new Experiment\n\n| Create a new Experiment called GeNormHuman in Project1        |\n| :------------------------------------------------------------ |\n| You can find the details on how to create a new experiment in Creating a project and an experiment |\n\n#### Loading the data into qbase+\n| Import [Run6](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Run6.xls) . This file is in qBase format. |\n| :------------------------------------------- |\n| You can find the details on how to start the data import in Loading data into qbase+. Unlike the previous exercise, qbase+ does not allow you to do a quick import this time. In the Import Run window Manual import is selected:\n\nMake sure that Upload file to Biogazelle support for further analysis is NOT selected and click Next. Select the correct File type (qBase) and click Finish. This file contains the data of the geNorm pilot experiment. In the pilot experiment, 10 candidate reference genes were measured in 20 representative samples.\n\n#### Analyzing the geNorm pilot data\n\n| Specify the aim of the experiment.        |\n| :---------------------------------------- |\n| In this experiment we want to select the ideal reference genes for our next experiments so we choose selection of reference genes (geNorm) |\n\n| Check the quality of the replicates and the controls (use default parameter settings). |\n| :------------------------------------------- |\n| You can find the details on how to check the quality of the replicates in the Checking the quality of technical replicates and controls section of Analyzing gene expression data in qbase+\n\nAll replicates and controls have met the quality criteria so there&#39;s no need to inspect them further. |\n| Select the Amplification efficiencies strategy you want to use. |\n| :------------------------------------------- |\n| You can find the details on how to select the Amplification effciencies strategy in the Taking into account amplification efficiencies section of Analyzing gene expression data in qbase+. We haven&#39;t included dilution series nor do we have data from previous qPCR experiments regarding the amplification efficiencies so we choose to use the same efficiency (E=2) for all genes. |\n\nIt is of course better to include a dilution series for each gene to have an idea of the amplification efficiencies of each primer pair.\n| Convert all genes to Reference genes.                                                                         |\n| :------------------------------------------------------------------------------------------------------------ |\n| You can convert all the genes simultaneously by selecting Use all targets as candidate reference genes |\n\nClick Finish.\n\n| Which genes are you going to use as reference targets in further experiments ? |\n| :------------------------------------------- |\n| Upon clicking Finish, the geNorm window containing the analysis results is automatically opened. The geNorm window consists of three tabs. The tabs are located at the bottom of the window: geNorm M, geNorm V and Interpretation.\nThe first tab, geNorm M, shows a ranking of candidate genes according to their stability, expressed in M values, from the most unstable genes at the left (highest M value) to the best reference genes at the right (lowest M value):\nThe second tab, geNorm V, shows a bar chart that helps determining the optimal number of reference genes to be used in subsequent analyses:\n\nThe number of reference genes is a trade-off between practical considerations and accuracy. It is a waste of resources to quantify more genes than necessary if all candidate reference genes are relatively stably expressed and if normalization factors do not significantly change when more genes are included. However, Biogazelle recommends the minimal use of the 3 most stable candidate reference genes and stepwise inclusion of more reference genes until the next gene has no significant contribution to the normalization factors.\nTo determine the need of including more than 3 genes for normalization, pairwise variations Vn/n+1 are calculated between two sequential normalization factors. Simply stated: V is measure of the added value of adding a next reference gene to the analysis. A large variation means that the added gene has a significant effect and should be included.\nIn normal experiments like the Gene expression experiment, see Analyzing_gene_expression_data_in_qbase+, we only have 3 reference genes so we will see only 1 bar here. But in this geNorm pilot experiment, we analyzed 10 candidate reference genes, so we see 8 bars.\nAll pairwise variations are very low, so even the inclusion of a third gene has no significant effect. Based on a preliminary experiment that was done by Biogazelle, 0.15 is taken as a cut-off value for V, below which the inclusion of an additional reference gene is not required. Normally this threshold is indicated by a green line on the geNorm V bar chart. However since all V-values fall below the threshold in this geNorm pilot experiment, you don’t see this line on the bar chart.\nSo, these results mean that for all subsequent experiments on these samples, two reference genes, HPRT1 and GADP, would be sufficient. However, as stated before, Biogazelle recommends to always include at least three reference genes in case something goes wrong with one of the reference genes (so also include YHWAZ). \n\n\n\nIn this example we will analyze data from an artificial expression study containing the following samples:\n  - 6 treated samples: treated1, treated2, ... treated6\n  - 6 control samples: control1, control2, ... control6\n\nIn this study, the expression of the following genes was measured:\n  - 4 commonly used reference genes: ACTB, HPRT, GAPDH, and TUBB4. We have seen in [the previous exercise](http://wiki.bits.vib.be/index.php/Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase%2B#Exercise_1:_reference_genes_for_mouse_liver) that the expression of these reference genes in mouse liver samples is not as stable as generally thought.\n  - 3 genes of interest:\n      - Low: a gene with low expression levels\n      - Medium: a gene with moderate expression levels\n      - HighVar: a gene with low and very noisy expression\n\nIn general, the lower the expression level, the more noisy the qPCR results will become. For each of the genes of interest we have included a run in which a 2-fold difference in expression between control and treated samples was created (Low1, Medium1 and HighVar1) and a run with a 4-fold difference in expression (Low2, Medium2 and HighVar2).\nThere are three technical replicates per reaction. In a second experiment we used [the reference genes that were obtained via Genevestigator](http://wiki.bits.vib.be/index.php/Using_GeneVestigator_to_select_candidate_reference_genes#Finding_candidate_reference_genes_in_the_free_version_of_Genevestigator) and that proved to be [more stably expressed in mouse liver samples than the commonly used references](http://wiki.bits.vib.be/index.php/Analyzing_data_from_a_geNorm_pilot_experiment_in_qbase%2B#Exercise_1:_reference_genes_for_mouse_liver).\nThe data can be found in the NormGenes folder on the BITS laptops or can be downloaded: [from our website](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/NormGenes.zip).\n\n#### Creating a new experiment\n\n| Create a new Experiment called NormGenes1 in Project1 |\n| :---------------------------------------------------- |\n| You can find the details on how to create a new experiment in Creating a project and an experiment |\n\n#### Loading the data\n\n| Import Run1 to Run5. These files are in qBase format. |\n| :---------------------------------------------------- |\n| You can find the details on how to import the data file in the Loading the data into qbase+ section of Analyzing data from a geNorm pilot experiment in qbase+ |\n\nWe are going to compare expression in treated versus untreated samples so we need to tell qbase+ which samples are treated and which not. To this end, we have constructed [a sample properties file](http://data.bits.vib.be/pub/trainingen/qbasePLUS/DataTraining/Sample_Properties_Norm.xlsx) in Excel containing the grouping annotation as a custom property called Treatment.\n\n| Import the Sample Properties file. |\n| :---------------------------------------------------- |\n| You can find the details on how to import the data file in the Adding annotation to the data section of Loading data into qbase+.\n\n| Select to import the custom property. |\n| :---------------------------------------------------- |\n| So as you can see we have 6 treated and 6 untreated samples and we have measured the expression of the 4 commonly used reference genes and 6 genes of interest:\n\n#### Analyzing the data\n\n| Which amplification efficiencies strategy are you going to use ?     |\n| :------------------------------------------------------------------- |\n| You don&#39;t have data of serial dilutions of representative template to build standard curves so the only choice you have is to use the default amplification efficiency (E = 2) for all the genes. |\n\n| Appoint the reference genes. |ACTB, GAPDH, HPRT and TUBB4B are the reference genes:\nYou can find the details on how to appoint reference targets in the Normalization section of Analyzing gene expression data in qbase+ |\n\n| Is the stability of the reference genes ok ?                                                                 |\n| :----------------------------------------------------------------------------------------------------------- |\n| The M and CV values of the reference genes are shown in green so the stability of the reference genes is ok. |\n\n| Which scaling strategy are you going to use ? |Since you have a treated and a control group, it seems logical to use the average of the control group for scaling.\n\nYou can find the details on how to specify the scaling strategy in the Scaling section of Analyzing gene expression data in qbase+\nLook at the target bar charts.\n\n| In the target bar charts plot the average expression level of each group. |In the Grouping section at the bottom of the chart you can select Plot group average: Now do exactly the same for the second experiment with the same genes of interest but with other reference genes. This means that you have to return to the Analysis wizard. To this end, click the Launch wizard button a the top of the page:\n\n| Create a new Experiment called NormGenes2 in Project1 |\n| :---------------------------------------------------- |\n| You can find the details on how to create a new experiment in Creating a project and an experiment |\n\n| Import Run5 to Run9. These files are in qBase format. |\n| :---------------------------------------------------- |\n| You can find the details on how to import the data file in the Loading the data into qbase+ section of Analyzing data from a geNorm pilot experiment in qbase+ |\n\n| Import the Sample Properties file.                    |\n| :---------------------------------------------------- |\n| You can find the details on how to import the data file in the Adding annotation to the data section of Loading data into qbase+. Select to import the custom property. |\n\nSo as you can see we have 6 treated and 6 untreated samples and we have measured the expression of the 4 new reference genes and 6 genes of interest:\n| Appoint the reference genes. |EDN3, Gm16835, MUSK and OTOP3 are the reference genes:\n| :---------------------------------------------------- |\n| You can find the details on how to appoint reference targets in the Normalization section of Analyzing gene expression data in qbase+ |\n\n| Is the stability of the reference genes ok ?                                                                 |\n| :----------------------------------------------------------------------------------------------------------- |\n| The M and CV values of the reference genes are shown in green so the stability of the reference genes is ok. |\n\nAs you can see the M and CV values of these reference genes is much lower than these of the 4 commonly used reference genes pointing to the fact that genes are more stably expressed. It&#39;s not that the commonly used reference genes are bad references. Then qbase+ would not display them in green. It&#39;s just that the other reference genes are more stable. But this can have a big impact on the results of your analysis...\n\n| Use the average of the control group for scaling |You can find the details on how to specify the scaling strategy in the Scaling section of Analyzing gene expression data in qbase+\n\nPlot the average expression level of each group. Now we will compare the target bar charts of the second and the first experiment to assess the influence of the stability of the reference targets on the analysis results.\n\n| How to display the target bar charts of the second and the first experiment next to each other ? |You can display the bar charts next to each other by clicking the tab of the bar chart of the second experiment. Drag the tab to the right while you hold down the mouse button until you see and arrow at the right side of the qbase+ window and a dark grey box in the right half of qbase+ window. Release the mouse button when you see the arrow and the box. Now the two bar charts should be next to each other. Some laptop screens are too small to nicely display the two bar charts next to other. If this is the case switch to full screen mode by double clicking the tab of the first experiment. |\n\nNow you can compare the expression of each gene in the first and in the second experiment.\n\nWhen we do this for HighVar1 for instance, you see that the average expression levels of both groups are the same in the first and the second experiment (check the scales of the Y—axis\\!). Both experiments detect the two-fold difference in expression level between the groups. However, the error bars are much larger in the first experiment than in the second. The variability of the reference genes does have a strong influence on the errors and the size of the error bars will influence the outcome of the statistical test to determine if a gene is differentially expressed or not. The larger the error bars the smaller the less likely it is that the test will say that the groups differ.\n\nRemember that the error bars represent 95% confidence intervals:\n  - if the error bars of the two groups do not overlap: you are certain that the difference between the means of the two groups is significant\n  - if they do not overlap: you know nothing with certainty: the means can be different or they can be the same. Of course the more they overlap the smaller the chance that there is a significant difference between the groups.\n\nCheck out the results of HighVar2. Here, you clearly see the influence of the reference genes. Again, the fourfold difference in expression is detected by both experiments but:\n\n  - the least stable reference genes (experiment 1) give large overlapping error bars\n  - the most stable reference (experiment 2) give smaller, barely overlapping error bars\n\nThis means that in experiment 2, a statistical test will probably declare that HighVar2 is differentially expressed while in experiment 1 this will not be the case. We will test this assumption by performing a statistical test.\n\n#### Statistical analysis of differential expression\n\n| Use a non-parametric test to identify DE genes in experiment 1 ? |\n| :---------------------------------------------------- |\n| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+. In brief, you need to perform the following steps:\n\nOpen the Statistical wizard\n\nThe goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples\n\nUse the Treatment property to identify treated and untreated samples\n\nAnalyze all genes of interest\n\nUse the default settings to perform the non-parametric Mann-Whitney test\n\nAs you can see, none of the genes is considered DE by the very conservative non-parametric test. Additionally most genes have the same p-value. That&#39;s normal when you don&#39;t have many replicates. In our case, we have 6 replicates. Non-parametric tests are based on a ranking of the data values and there are not so many ways to rank 6 data points. This is why you see the same p-values for many genes.\nAs said before, the non-parametric test is very stringent. If the data do come from a normal distribution, the test will generate false positives. Some of the genes might have have been labeled not DE while in fact they are DE so you might have missed some differential expression. The choice of statistical test with 6 biological replicates depends on what you prefer: false negatives or false positives. Most people will choose false negatives since they don&#39;t want to invest time and money in research on a genes that was labeled DE while in fact it is not DE.\n\nSuppose I don&#39;t mind false positives but I don&#39;t want to miss any potential DE genes. In that case, it&#39;s better to go for a t-test. Let&#39;s repeat the test n ow choosing a parametric t-test.\n| Use a t-test to identify DE genes in experiment 1 ? |\n| :---------------------------------------------------- |\n| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+.\nIn brief, you need to perform the following steps:\nOpen the Statistical wizard\nThe goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples\nUse the Treatment property to identify treated and untreated samples\nAnalyze all genes of interest\nDescribe the data set as log-normally distributed\n\nStill none of the genes is considered DE but you do see that the p-values of the t-test are lower than these of the Mann-Whitney test.\n\n| Use a non parametric test to identify DE genes in experiment 2 ? |\n| :---------------------------------------------------- |\n| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+.\nIn brief, you need to perform the following steps:\n\nOpen the Statistical wizard\nThe goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples\nUse the Treatment property to identify treated and untreated samples\nAnalyze all genes of interest\nUse default settings\n\nNow you see that 4 out of the 6 genes are considered DE. This is also what we expected since 3 of our genes of interst have a 4-fold difference in expression level between the two groups. It&#39;s understandable that it&#39;s hard to detect 2-fold differences in expression especially when the expression of the gene is somewhat variable as is the case for Low1 and HighVar1 but a 4-fold difference is a difference that you would like to detect.\n| Use a t-test to identify DE genes in experiment 2 ? |\n| :---------------------------------------------------- |\n| You can find full details on statistical analyses in qbase+ in the statistical analysis section of analyzing gene expression data in qbase+.\nIn brief, you need to perform the following steps:\n\nOpen the Statistical wizard\nThe goal of this analysis is to compare the mean expression levels of our genes of interest in treated and untreated samples\nUse the Treatment property to identify treated and untreated samples\nAnalyze all genes of interest\nDescribe the data as log normally distributed\n\nAgain the t-test generates lower p-values than the Mann-Whitney test but realize that choosing the t-test when the data is not normally distributed will generate false positives \\!&quot;,&quot;The following exercise will make you familiar with the Primer3Plus software for designing primers for PCR. Primer3Plus is the user-friendly version of Primer3, the standard software for primer design.\n\n### Criteria for qPCR primers\n\nPrimers for qPCR have to follow all the gudelines for regular primers is and an additional set of rules specific for qPCR primers:\n\n  - qPCR products are small: 80-160 bp\n  - use intron or exon-exon junction spanning primers to detect genomic DNA contamination in the RNA samples. Primers of intron spanning primer pairs are located at both sides of an intron and will therefore generate a larger product on genomic DNA (containing the intron). Primer pairs containing an exon-exon junction spanning primer will not generate a PCR product on genomic DNA since the exon-exon junction only exist in the cDNA.\n  - primer length between 9 and 30 bp with an optimum at 20 bp\n  - melting temperature (Tm) of the primers between 58 and 60°C with an optimum at 59°C\n  - maximum Tm difference between the primers of a pair: 2°C\n  - GC content of the primers between 30 and 80% with an optimum at 50%\n  - the 5 nucleotides at the 3&#39; end of the primers should have no more than 2 G or C bases\n  - avoid runs of 4 or more identical nucleotides (especially Gs)\n  - primers must specifically target the region you want to amplify\n\nThere are many programs for designing primers, the most important ones:\n\n  - [Primer3](http://frodo.wi.mit.edu/) \\[1\\] or use it&#39;s user-friendly version: [Primer3Plus](http://primer3plus.com/cgi-bin/dev/primer3plus.cgi)\\[2\\]\n  - [PrimerBLAST](http://www.ncbi.nlm.nih.gov/tools/primer-blast/index.cgi?LINK_LOC=BlastHome)\\[3\\]\n\nThe major downside of Primer3 and Primer3Plus is the fact that you have to check the specificity of the primers yourself. Primer3 will suggest a number of primer pairs that fulfill all of the above requirements, but Primer3 will not check the specificity of the primers. So you have use BLAST to check the specificity of the suggested primer pairs. Very often, the selected primers are not specific and you have to repeat the entire Primer3 analysis.\nIf you use Primer3 and do the BLAST yourself, BLAST against Refseq sequences unless they are not available for the organism you work with or you have reasons to believe that they are not complete (i.e. they do not represent the full genome). For model organisms, you can BLASTagainst the Refseq database. Limit the database to sequences from the organism you work with.\nAdditionally, it is especially important to check that the primers are specific at the 3&#39; end because that&#39;s the site where the polymerase will attach nucleotides. So it is recommended to not use primers that contain long identical stretches (\\&gt; 15nt for primers of 20nt long) to other regions in the genome, and certainly not if these stretches comprise the last nucleotide at the 3&#39; end of the primer.\nFor these exercises we will use PrimerBLAST since [it uses the same algorithm to pick primers as Primer3](http://www.ncbi.nlm.nih.gov/tools/primer-blast/primerinfo.html) \\[4\\] and does the specificity check for you\\!\n\n## Designing qPCR primers for the fruit fly tap gene\n\n### Designing qPCR primers using PrimerBLAST\n\nThe RefSeq entry NM_079400 contains the sequence of the D. melanogaster mRNA coding for tap, the target of Poxn. Tap encodes a bHLH protein expressed in larval chemosensory organs and involved in the response to sugar and salt. We wish to amplify the region encoding the Helix-loop-helix domain. In the sequence of the RefSeq record, the domain is located between position +577 and +745.\nWe want to design qPCR primers for measuring the expression level of the hlh domain using SYBR green. Remember that it is advised to design intron/exon-exon junction spanning primers for qPCR experiments that are based on fluorescent labels to detect/avoid amplification of contaminating genomic DNA.\n\n| Check in NCBIs Gene database if the hlh domain contains any introns ? |\n| :------------------------------ |\n|To know the location of the introns, you need the genomic sequence instead of the mRNA sequence.\n\n - Go to [the NCBI RefSeq record](https://www.ncbi.nlm.nih.gov/nuccore/NM_079400).\n - In the right menu click the link to the **Gene** record\n - In the **Genomic regions, transcripts and products** secton you can see that the gene contains no introns: the transcript is not chopped up into pieces when aligned to the genome. Click [here](https://www.ncbi.nlm.nih.gov/gene/39934) for an example of a gene with introns.\n\nNext, we will design primers to measure the expression of the hlh domain.\n\n| Go to Primer BLAST by using the link in the Refseq record |\n| :------------------------------ |\n|Go back to the RefSeq mRNA record. There, you can go directly to PrimerBLAST by clicking the **Pick Primers** link in the **Analyze this sequence** section of the right menu.\n\nSince you want to measure the expression of the hlh domain you want\nprimers that are located inside the domain.\n\n| Define the range of the sequence in which you want to design primers. |\n| :------------------------------ |\n|You have to specify the range as follows:\n\n| Define the primer parameters to comply with the rules of qPCR primer design: product size and Tm. |\n| :------------------------------ |\n|To comply with the rules for qPCR primer design, you have to change the settings for PCR product size and melting temperature:\n\n| The PrimerBLAST automatically decides to check primer specificity in the Drosophila (organism ID: 7227) RefSeq mRNA database which is exactly what you want. For the qPCR you are going to use RNA samples from fruitfly. This means that the primers will only come into contact with Drosophila mRNAs so you only have to check their specifity in this database. Make sure the last 2 nucleotides are completely specific. |\n| :------------------------------ |\n|You want to ensure that the 3&#39; end of the primers really is specific:\n\nThe PrimerBLAST gives you a set of 9 primer pairs that are specific (according to the criteria that you have specified) and that fulfill all other requirements that you have defined. Look at the detailed report of the first primer pair:\nAll parameters are quite self-explanatory except for the Self complementary and Self 3&#39;complementarity scores.\n\n  - The first score represents the local alignment score when aligning a primer to itself. The scoring system gives 1.00 for a match, -1.00 for a mismatch. This means that the lower the score (the more mismatches), the less likely that the primer binds to itself.\n  - The second score represents the global alignment score when aligning a primer to itself. Here again, the lower the score, the better.\n\nThe scores are followed by information on the specificity of the primer: alignments of the two primers to all target sequences from the database that match the criteria that you specified. In these alignments dots represent matching nucleotides while letters represent mismatches. A specific primer pair will have two alignments (one for each primer): both perfect alignments (all dots) to the sequence you want to amplify.\n\n### Analyzing primer characteristics using OligoAnalyzer\n\n[OligoAnalyzer](https://eu.idtdna.com/calc/analyzer) is a tool implemented by ID\\&amp;T (who sell primers) to check the characteristics of your primers. Take the first primer that is suggested by Primer-BLAST, the pair resulting in a product of 100bp.\n\n| What&#39;s the Tm of the first primer ? |\n| :------------------------------ |\n|Copy the sequence of the first primer in the **Sequence** box, adjust the concentrations to these that are typically used in PCR (see slides) and click **Analyze**:\nAs you can see the predicted melting temperature is 63.9 ºC, which is slightly different from the prediction made by BLAST. There are many different methods to predict Tm and each method will give a different result. Assumed concentrations of primers and ions have an enormous impact on the Tm prediction. So don&#39;t worry about these differences: these are theoretical calculations anyway, the only way to determine Tm values is by doing actual PCR. As long as the difference in Tm between the two primers is not too large, everything is fine.\n\n| What&#39;s the Tm of the second primer ?                                                                                                                                             |\n| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |\n| Copy the sequence of the second primer in the **Sequence** box and click **Analyze**. The predicted melting temperature is also 63.9 ºC , the same Tm as the first primer. |\n\nRemember that the second primer had a large Self complementarity score according to PrimerBLAST.\n\n| Check the self-complementarity of the second primer in OligoAnalyzer ? |\n| :------------------------------ |\n|Click **Self-Dimer**:\n\nYou see that the highest scoring alignment indeed has 6 matches, giving a score of 6 as predicted by PrimerBLAST.\n\n| Do you expect this self-complementarity will give problems in the PCR ? |\n| :------------------------------ |\n|No, the complementarity is concentrated at the center of the primer, not at the 3&#39; end. Since polymerases add bases at the 3’ end of the primer, the primer duplex cannot be extended so it will not give rise to aspecific products. [ID&amp;amp;T](https://eu.idtdna.com/pages/docs/default-source/default-document-library/idt_self-dimer_and_hetero-dimer_results_guide.pdf) recommends to avoid complementary stretches of more than 2 bp at the 3’ end.\nHowever, even if the primer dimer cannot be extended, it could interfere when its formation competes with the annealing of primer and target. This is only the case when the stability of the dimer is similar to the stability of a perfectly matched primer-target duplex. The stability of the perfectly matched duplex is shown as a Maximum Delta G at the top of results. So non-extendable dimer structures that are much shorter than the intended duplex, as we have here, are not going to disrupt the PCR reaction.\nIt is advised to review all possible interactions between primers so both Self-Dimer (primers binding to themselves) and Hetero-Dimer (primers binding to each other) interactions between primers are examined.\n\n| Is it likely that the primers bind to each other ? |\n| :------------------------------ |\n|Click **Hetero-Dimer**:\n\nThis opens a text box to enter the second primer. Click **Analyze**. There is one structure (the fourth one) that looks problematic because there is a stretch of 3 matching nucleotides at the 3&#39;end of one of the primers.\n\nSo you might consider taking a look at the second pair of primers that PrimerBLAST suggests. On the other hand, this structure is has relatively high free energy (delta G). The structure with the lowest total free energy, the target-primer duplex, is most important because it will dominate in solution. Structures with higher free energy are less stable and will be present in smaller amounts in the reaction mixture.\n\nTake a look at the second primer pair that was suggested by PrimerBLAST.\n\n| Is it likely that these primers bind to each other ?                                 |\n| :----------------------------------------------------------------------------------- |\n| No these primers do not form duplex structures that could pose a problem during PCR. |\n\n## Designing qPCR primers for the human F9 gene\n\n### Designing qPCR primers using PrimerBLAST\n\nThe RefSeq entry NM_000133.3 contains the sequence of the human mRNA coding for coagulation factor F9. The gene contains 8 coding exons and gives rise to a transcript of 2780 bp encoding a protein of 461 amino acids.\nNext, we want to design primers to measure the expression of the F9 gene.\n\nGo to [the RefSeq record of this transcript](http://www.ncbi.nlm.nih.gov/nuccore/NM_000133.3) to study its structure. When you scroll down to the **features** section you see that the CDS is located from position 40 to position 1415. Since RNA degradation starts at the 5&#39;end of transcripts, we don&#39;t want to pick primers at the 5&#39;end. On the other hand, we don&#39;t want to pick primers in the long 3&#39;UTR either because it doesn&#39;t contain any introns (the exons are all coding) and we want to design exon-exon junction or intron spanning primers.\nLet&#39;s try to find exon-exon junction spanning primers between position 400 and 1600, with optimal anneal temperature = 60.\n\n| Find primers that fulfill the above defined criteria |\n| :------------------------------ |\n|Go to [PrimerBLAST](http://www.ncbi.nlm.nih.gov/tools/primer-blast/index.cgi?LINK_LOC=BlastHome) and fill in the form as follows:\n\nExclude predicted sequences in the database to search in .\n\n| Find primers that fulfill the above defined criteria |\n| :------------------------------ |\n|Go to [PrimerBLAST](http://www.ncbi.nlm.nih.gov/tools/primer-blast/index.cgi?LINK_LOC=BlastHome) and fill in the remainder of the form as follows:\n\nThe PrimerBLAST gives you a set of 10 primer pairs. Look at the detailed\nreport of the first primer pair:\n\nAs you can see the primers are not specific: they can bind to various other targets albeit with lower affinity because of the mismatches . The best option seems to be primer pair 7, which binds to both F9 transcript variants and potentially to one unintended target, but as you can see the last nucleotide at the 3&#39; end of both primers are specific.\n\n### In silico PCR in the UCSC Browser\n\nWe will proceed using the third primer pair Primer-BLAST suggests. You can visualize the PCR product (and additional annotation) in the UCSC Genome Browser using [UCSC&#39;s In Silico PCR tool](http://genome.ucsc.edu/cgi-bin/hgPcr).\nSelect the most recent version of the human genome and paste the sequences of forward and reverse primers in their respective boxes. Click **submit**\nNormally, this returns the location and the sequence of the PCR product but our primer pair doesn&#39;t return a match. When you think about this was to be expected since we are working with exon-exon junction spanning primers that are not able to match the genome sequence. So checking SNPs is not so straight-forward in the case of exon-exon junction spanning primers.\nWe will repeat the primer search now searching for intron-spanning primers to show you how to use the in silico PCR tool. Taking into account the fact that the results for the exon-exon junction spanning primers were so messy we will make the search more stringent this time:\n\n  - We will the minimum number of mismatches to 4\n  - and at least 3 mismatches in the last 3 bps at the 3&#39;end\n\n| Find intron spanning primers that fulfill the above defined criteria |\n| :------------------------------ |\n|Go back to the Primer-BLAST and fill in the form like in the previous exercise except that they should span an intron:\n\nPrimer-BLAST returns 10 primer pairs. Again the seventh primer pair is\nthe specific one.\n\n| Take the seventh suggested primer pair and check for SNPs in the UCSC Browser |\n| :------------------------------ |\n|Go to [PrimerBLAST](http://www.ncbi.nlm.nih.gov/tools/primer-blast/index.cgi?LINK_LOC=BlastHome) and paste the sequences of forward and reverse primers in their respective boxes.\nThis time the search finds a PCR product:\n\nClicking the location visualizes the PCR product in the UCSC genome browser. Remove unnecessary trancks by right clicking the box in front of them and selecting **hide**\n\nAdd tracks showing relevant annotation like position of SNPs...\n\nSetting the SNPs track from **hide** to **full** shows the SNPs in the browser. Center the forward primer by grabbing and dragging it to the center.\n\nZoom in to **base** display to see if the forward primer is matching any SNPs.\n\nAs you can see the forward primer does match two SNPs but none of them are located near the 3&#39;end of the primer.\n\n1.  &lt;http://frodo.wi.mit.edu/&gt;\n2.  &lt;http://primer3plus.com/cgi-bin/dev/primer3plus.cgi&gt;\n3.  &lt;http://www.ncbi.nlm.nih.gov/tools/primer-blast/index.cgi?LINK_LOC=BlastHome&gt;\n4.  &lt;http://www.ncbi.nlm.nih.gov/tools/primer-blast/primerinfo.html&gt;&quot;,&quot;qbase+ is software to visualize and analyze qPCR data. It allows you to perform various types of analyses:\n  - statistical analysis of gene expression\n  - advanced copy number analysis\n  - miRNA profiling\n  - ChIP-qPCR analysis\n# Installation and licensing\nYou can find the installation instructions on [VIB qbase+ support page](https://www.bits.vib.be/index.php/software-overview/qbaseplus)\nVIB only offers qbase+ to VIB scientists, you need a valid VIB email address to run the software. Biogazelle (the company who has developed the software) have written a manual with instructions on how to use the software. Download [Biogazelle&#39;s user manual](https://www.biogazelle.com/system/files/manuals/qbaseplus_manual_0.pdf). Before you can download the manual you have to log on to [the qbase+ website](https://www.qbaseplus.com/) using your qbase+ account. Use your VIB email address for setting up this account.\n# Training material\n  - [slides](http://data.bits.vib.be/pub/trainingen/qbasePLUS/qbase_2018.pdf)\n  \n  **Extra**\n  - [clean log10 transformed CNRQs](http://data.bits.vib.be/pub/trainingen/qbasePLUS/Log10CNRQsClean.xlsx) for checking normality in Prism\n  - [clean untransformed CNRQs](http://data.bits.vib.be/pub/trainingen/qbasePLUS/CNRQsClean.xlsx) for visualization in Prism\n  - [R script](http://data.bits.vib.be/pub/trainingen/qbasePLUS/qPCR.R) for analysis and visualization\n  - [log10 transformed CNRQs of control samples](http://data.bits.vib.be/pub/trainingen/qbasePLUS/resultslog.csv) for analysis and visualization in R\n  - [log10 transformed CNRQs of treated samples](http://data.bits.vib.be/pub/trainingen/qbasePLUS/resultslogTreated.csv) for analysis and visualization in R\n&quot;],&quot;related_posts&quot;:null,&quot;source&quot;:&quot;.&quot;,&quot;destination&quot;:&quot;_site/training-material&quot;,&quot;collections_dir&quot;:&quot;&quot;,&quot;plugins_dir&quot;:&quot;_plugins&quot;,&quot;layouts_dir&quot;:&quot;_layouts&quot;,&quot;data_dir&quot;:&quot;metadata&quot;,&quot;includes_dir&quot;:&quot;.&quot;,&quot;safe&quot;:false,&quot;include&quot;:[&quot;.nojekyll&quot;],&quot;exclude&quot;:[&quot;shared/font-awesome/src/&quot;,&quot;Gemfile&quot;,&quot;Gemfile.lock&quot;,&quot;package.json&quot;,&quot;package-lock.json&quot;,&quot;CONTRIBUTING.md&quot;,&quot;CONTRIBUTORS.yaml&quot;,&quot;LICENSE.md&quot;,&quot;README.md&quot;,&quot;Makefile&quot;,&quot;miniconda.sh&quot;,&quot;**/README.md&quot;,&quot;**/*.yaml&quot;,&quot;**/*.yml&quot;,&quot;**/*.sh&quot;,&quot;bin/&quot;,&quot;metadata/&quot;,&quot;templates/&quot;,&quot;vendor/&quot;,&quot;node_modules/&quot;],&quot;keep_files&quot;:[&quot;.git&quot;,&quot;.svn&quot;],&quot;encoding&quot;:&quot;utf-8&quot;,&quot;markdown_ext&quot;:&quot;markdown,mkdown,mkdn,mkd,md&quot;,&quot;strict_front_matter&quot;:true,&quot;show_drafts&quot;:null,&quot;limit_posts&quot;:0,&quot;future&quot;:false,&quot;unpublished&quot;:false,&quot;whitelist&quot;:[],&quot;plugins&quot;:[&quot;jekyll-feed&quot;,&quot;jekyll-environment-variables&quot;,&quot;jekyll-github-metadata&quot;,&quot;jekyll-scholar&quot;],&quot;markdown&quot;:&quot;kramdown&quot;,&quot;highlighter&quot;:&quot;rouge&quot;,&quot;lsi&quot;:false,&quot;excerpt_separator&quot;:&quot;\n\n&quot;,&quot;incremental&quot;:false,&quot;detach&quot;:false,&quot;port&quot;:&quot;4000&quot;,&quot;host&quot;:&quot;127.0.0.1&quot;,&quot;baseurl&quot;:&quot;/training-material&quot;,&quot;show_dir_listing&quot;:false,&quot;permalink&quot;:&quot;date&quot;,&quot;paginate_path&quot;:&quot;/page:num&quot;,&quot;timezone&quot;:null,&quot;quiet&quot;:false,&quot;verbose&quot;:true,&quot;defaults&quot;:[],&quot;liquid&quot;:{&quot;error_mode&quot;:&quot;warn&quot;,&quot;strict_filters&quot;:false,&quot;strict_variables&quot;:false},&quot;rdiscount&quot;:{&quot;extensions&quot;:[]},&quot;redcarpet&quot;:{&quot;extensions&quot;:[]},&quot;kramdown&quot;:{&quot;auto_ids&quot;:true,&quot;toc_levels&quot;:&quot;1..2&quot;,&quot;entity_output&quot;:&quot;as_char&quot;,&quot;smart_quotes&quot;:&quot;lsquo,rsquo,ldquo,rdquo&quot;,&quot;input&quot;:&quot;GFM&quot;,&quot;hard_wrap&quot;:false,&quot;footnote_nr&quot;:1,&quot;show_warnings&quot;:false,&quot;syntax_highlighter&quot;:&quot;rouge&quot;,&quot;syntax_highlighter_opts&quot;:{},&quot;coderay&quot;:{},&quot;auto_ids&quot;:true,&quot;toc_levels&quot;:&quot;1..2&quot;,&quot;entity_output&quot;:&quot;as_char&quot;,&quot;smart_quotes&quot;:&quot;lsquo,rsquo,ldquo,rdquo&quot;,&quot;input&quot;:&quot;GFM&quot;,&quot;hard_wrap&quot;:false,&quot;footnote_nr&quot;:1,&quot;show_warnings&quot;:false,&quot;syntax_highlighter&quot;:&quot;rouge&quot;,&quot;syntax_highlighter_opts&quot;:{},&quot;coderay&quot;:{}},&quot;title&quot;:&quot;VIB Bioinformatics Core&quot;,&quot;email&quot;:&quot;bits@vib.be&quot;,&quot;description&quot;:&quot;A collection of tutorials generated and maintained by VIB Bioinformatics Core&quot;,&quot;url&quot;:&quot;https://vibbits.github.io/&quot;,&quot;repository&quot;:&quot;vibbits/training-material&quot;,&quot;repository_branch&quot;:&quot;master&quot;,&quot;logo&quot;:&quot;assets/images/vib_notag_neg_rgb.jpg&quot;,&quot;small_logo&quot;:&quot;assets/images/bioinformatics_core_rgb_neg.png&quot;,&quot;help_url&quot;:&quot;https://www.bits.vib.be&quot;,&quot;other_languages&quot;:&quot;fr, ja, es, pt, ar&quot;,&quot;figurify&quot;:{&quot;skip_empty&quot;:true,&quot;skip_layouts&quot;:[&quot;introduction_slides&quot;,&quot;tutorial_slides&quot;,&quot;base_slides&quot;],&quot;skip_titles&quot;:[&quot;Example of an image with a caption&quot;]},&quot;scholar&quot;:{&quot;style&quot;:&quot;_layouts/g3.csl&quot;,&quot;locale&quot;:&quot;en&quot;,&quot;sort_by&quot;:&quot;year,month&quot;,&quot;order&quot;:&quot;ascending&quot;,&quot;group_by&quot;:&quot;none&quot;,&quot;group_order&quot;:&quot;ascending&quot;,&quot;bibliography_group_tag&quot;:&quot;h2,h3,h4,h5&quot;,&quot;bibliography_list_tag&quot;:&quot;ol&quot;,&quot;bibliography_item_tag&quot;:&quot;li&quot;,&quot;bibliography_list_attributes&quot;:{},&quot;bibliography_item_attributes&quot;:{},&quot;source&quot;:&quot;topics/&quot;,&quot;bibliography&quot;:&quot;**/*.bib&quot;,&quot;repository&quot;:null,&quot;repository_file_delimiter&quot;:&quot;.&quot;,&quot;bibtex_options&quot;:{&quot;strip&quot;:false,&quot;parse_months&quot;:true},&quot;bibtex_filters&quot;:[&quot;smallcaps&quot;,&quot;superscript&quot;,&quot;italics&quot;,&quot;latex&quot;],&quot;bibtex_skip_fields&quot;:[&quot;abstract&quot;,&quot;month_numeric&quot;],&quot;bibtex_quotes&quot;:[&quot;{&quot;,&quot;}&quot;],&quot;replace_strings&quot;:true,&quot;join_strings&quot;:true,&quot;details_dir&quot;:&quot;bibliography&quot;,&quot;details_layout&quot;:&quot;bibtex.html&quot;,&quot;details_link&quot;:&quot;Details&quot;,&quot;use_raw_bibtex_entry&quot;:true,&quot;bibliography_class&quot;:&quot;bibliography&quot;,&quot;bibliography_template&quot;:&quot;bibtemplate&quot;,&quot;reference_tagname&quot;:&quot;span&quot;,&quot;missing_reference&quot;:&quot;(missing reference)&quot;,&quot;details_link_class&quot;:&quot;details&quot;,&quot;query&quot;:&quot;@*&quot;,&quot;cite_class&quot;:&quot;citation&quot;,&quot;type_names&quot;:{&quot;article&quot;:&quot;Journal Articles&quot;,&quot;book&quot;:&quot;Books&quot;,&quot;incollection&quot;:&quot;Book Chapters&quot;,&quot;inproceedings&quot;:&quot;Conference Articles&quot;,&quot;thesis&quot;:&quot;Theses&quot;,&quot;mastersthesis&quot;:&quot;Master&#39;s Theses&quot;,&quot;phdthesis&quot;:&quot;PhD Theses&quot;,&quot;manual&quot;:&quot;Manuals&quot;,&quot;techreport&quot;:&quot;Technical Reports&quot;,&quot;misc&quot;:&quot;Miscellaneous&quot;,&quot;unpublished&quot;:&quot;Unpublished&quot;},&quot;type_aliases&quot;:{&quot;phdthesis&quot;:&quot;thesis&quot;,&quot;mastersthesis&quot;:&quot;thesis&quot;},&quot;type_order&quot;:[],&quot;month_names&quot;:null},&quot;icon-tag&quot;:{&quot;question&quot;:&quot;fa-question-circle&quot;,&quot;solution&quot;:&quot;fa-eye&quot;,&quot;hands_on&quot;:&quot;fa-pencil&quot;,&quot;comment&quot;:&quot;fa-commenting-o&quot;,&quot;tip&quot;:&quot;fa-lightbulb-o&quot;,&quot;objectives&quot;:&quot;fa-bullseye&quot;,&quot;requirements&quot;:&quot;fa-check-circle&quot;,&quot;time&quot;:&quot;fa-hourglass-end&quot;,&quot;keypoints&quot;:&quot;fa-key&quot;,&quot;tool&quot;:&quot;fa-wrench&quot;,&quot;workflow&quot;:&quot;fa-share-alt&quot;,&quot;feedback&quot;:&quot;fa-comments-o&quot;,&quot;congratulations&quot;:&quot;fa-thumbs-up&quot;,&quot;trophy&quot;:&quot;fa-trophy&quot;,&quot;warning&quot;:&quot;fa-warning&quot;,&quot;details&quot;:&quot;fa-info-circle&quot;,&quot;exchange&quot;:&quot;fa-exchange&quot;,&quot;param-file&quot;:&quot;fa-file-o&quot;,&quot;param-files&quot;:&quot;fa-files-o&quot;,&quot;param-collection&quot;:&quot;fa-folder-o&quot;,&quot;param-text&quot;:&quot;fa-pencil&quot;,&quot;param-check&quot;:&quot;fa-check-square-o&quot;,&quot;param-select&quot;:&quot;fa-filter&quot;,&quot;param-repeat&quot;:&quot;fa-plus-square-o&quot;,&quot;galaxy-eye&quot;:&quot;fa-eye&quot;,&quot;galaxy-gear&quot;:&quot;fa-cog&quot;,&quot;galaxy-history&quot;:&quot;fa-archive&quot;,&quot;galaxy-library&quot;:&quot;fa-folder&quot;,&quot;galaxy-pencil&quot;:&quot;fa-pencil&quot;,&quot;galaxy-refresh&quot;:&quot;fa-refresh&quot;,&quot;galaxy-barchart&quot;:&quot;fa-bar-chart&quot;,&quot;galaxy-cross&quot;:&quot;fa-times&quot;,&quot;galaxy-columns&quot;:&quot;fa-columns&quot;,&quot;galaxy-tags&quot;:&quot;fa-tags&quot;,&quot;galaxy-selector&quot;:&quot;fa-check-square-o&quot;,&quot;galaxy-upload&quot;:&quot;fa fa-upload&quot;,&quot;galaxy-chart-select-data&quot;:&quot;fa fa-database&quot;,&quot;galaxy-save&quot;:&quot;fa fa-save&quot;,&quot;galaxy-scratchbook&quot;:&quot;fa fa-th&quot;,&quot;galaxy-dropdown&quot;:&quot;fa-caret-down&quot;,&quot;zenodo_link&quot;:&quot;fa-files-o&quot;,&quot;tutorial&quot;:&quot;fa-laptop&quot;,&quot;slides&quot;:&quot;fa-slideshare&quot;,&quot;interactive_tour&quot;:&quot;fa-magic&quot;,&quot;topic&quot;:&quot;fa-folder-o&quot;,&quot;instances&quot;:&quot;fa-cog&quot;,&quot;docker_image&quot;:&quot;fa-ship&quot;,&quot;galaxy_instance&quot;:&quot;fa-external-link&quot;,&quot;references&quot;:&quot;fa-bookmark&quot;,&quot;gitter&quot;:&quot;fa-comments&quot;,&quot;help&quot;:&quot;fa-life-ring&quot;,&quot;github&quot;:&quot;fa-github&quot;,&quot;email&quot;:&quot;fa-envelope-o&quot;,&quot;twitter&quot;:&quot;fa-twitter&quot;,&quot;linkedin&quot;:&quot;fa-linkedin&quot;,&quot;orcid&quot;:&quot;ai-orcid&quot;,&quot;curriculum&quot;:&quot;fa-graduation-cap&quot;,&quot;level&quot;:&quot;fa-graduation-cap&quot;,&quot;hall-of-fame&quot;:&quot;fa-users&quot;},&quot;serving&quot;:false,&quot;github&quot;:{&quot;api_url&quot;:&quot;https://api.github.com&quot;,&quot;baseurl&quot;:&quot;/pages/vibbits/training-material&quot;,&quot;build_revision&quot;:&quot;455d1879e7d21f3825f6a86f621692698ee1a822&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/training-material.git&quot;,&quot;contributors&quot;:false,&quot;environment&quot;:&quot;development&quot;,&quot;help_url&quot;:&quot;https://help.github.com&quot;,&quot;hostname&quot;:&quot;github.com&quot;,&quot;is_project_page&quot;:true,&quot;is_user_page&quot;:false,&quot;issues_url&quot;:null,&quot;language&quot;:null,&quot;latest_release&quot;:false,&quot;license&quot;:null,&quot;organization_members&quot;:[],&quot;owner&quot;:{&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;bio&quot;:null,&quot;blog&quot;:&quot;www.bits.vib.be&quot;,&quot;collaborators&quot;:null,&quot;company&quot;:null,&quot;created_at&quot;:&quot;2016-10-18 09:13:54 UTC&quot;,&quot;description&quot;:null,&quot;email&quot;:null,&quot;followers&quot;:0,&quot;following&quot;:0,&quot;has_organization_projects&quot;:true,&quot;has_repository_projects&quot;:true,&quot;hireable&quot;:null,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;id&quot;:22908438,&quot;is_verified&quot;:false,&quot;location&quot;:&quot;Belgium&quot;,&quot;login&quot;:&quot;vibbits&quot;,&quot;name&quot;:&quot;VIB Bioinformatics Core&quot;,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;public_gists&quot;:0,&quot;public_repos&quot;:21,&quot;type&quot;:&quot;Organization&quot;,&quot;updated_at&quot;:&quot;2019-09-13 15:03:47 UTC&quot;},&quot;owner_gravatar_url&quot;:&quot;https://github.com/vibbits.png&quot;,&quot;owner_name&quot;:&quot;vibbits&quot;,&quot;owner_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;pages_env&quot;:&quot;development&quot;,&quot;pages_hostname&quot;:&quot;localhost:4000&quot;,&quot;private&quot;:null,&quot;project_tagline&quot;:null,&quot;project_title&quot;:&quot;training-material&quot;,&quot;public_repositories&quot;:[{&quot;id&quot;:160538640,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNjA1Mzg2NDA=&quot;,&quot;name&quot;:&quot;bigdatasurvey&quot;,&quot;full_name&quot;:&quot;vibbits/bigdatasurvey&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/bigdatasurvey&quot;,&quot;description&quot;:&quot;A survey of open-source big data platforms, centering around the Hadoop ecosystem.&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/deployments&quot;,&quot;created_at&quot;:&quot;2018-12-05 15:27:29 UTC&quot;,&quot;updated_at&quot;:&quot;2018-12-05 15:31:44 UTC&quot;,&quot;pushed_at&quot;:&quot;2015-10-28 20:33:50 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/bigdatasurvey.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/bigdatasurvey.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/bigdatasurvey.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/bigdatasurvey&quot;,&quot;homepage&quot;:null,&quot;size&quot;:524,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;JavaScript&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:false,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:false,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;gh-pages&quot;},{&quot;id&quot;:136517018,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzY1MTcwMTg=&quot;,&quot;name&quot;:&quot;bioconda-recipes&quot;,&quot;full_name&quot;:&quot;vibbits/bioconda-recipes&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/bioconda-recipes&quot;,&quot;description&quot;:&quot;Conda recipes for the bioconda channel.&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/deployments&quot;,&quot;created_at&quot;:&quot;2018-06-07 18:36:07 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:21 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-01-21 17:01:17 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/bioconda-recipes.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/bioconda-recipes.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/bioconda-recipes.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/bioconda-recipes&quot;,&quot;homepage&quot;:&quot;https://bioconda.github.io&quot;,&quot;size&quot;:167593,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Shell&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:170295072,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNzAyOTUwNzI=&quot;,&quot;name&quot;:&quot;CAMEL&quot;,&quot;full_name&quot;:&quot;vibbits/CAMEL&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/CAMEL&quot;,&quot;description&quot;:&quot; The CAMEL platform, an online compendium of Adaptive Microbial Experiments in the Lab.&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/deployments&quot;,&quot;created_at&quot;:&quot;2019-02-12 10:03:23 UTC&quot;,&quot;updated_at&quot;:&quot;2019-07-25 12:42:15 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-07-25 12:42:12 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/CAMEL.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/CAMEL.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/CAMEL.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/CAMEL&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:2307,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;HTML&quot;,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:5,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:5,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:173921762,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNzM5MjE3NjI=&quot;,&quot;name&quot;:&quot;cytokit&quot;,&quot;full_name&quot;:&quot;vibbits/cytokit&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/cytokit&quot;,&quot;description&quot;:&quot;Microscopy Image Cytometry Toolkit&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/deployments&quot;,&quot;created_at&quot;:&quot;2019-03-05 10:04:08 UTC&quot;,&quot;updated_at&quot;:&quot;2019-03-05 15:37:40 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-03-05 15:37:38 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/cytokit.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/cytokit.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/cytokit.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/cytokit&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:174062,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Jupyter Notebook&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;apache-2.0&quot;,&quot;name&quot;:&quot;Apache License 2.0&quot;,&quot;spdx_id&quot;:&quot;Apache-2.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/apache-2.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTI=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:135151492,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzUxNTE0OTI=&quot;,&quot;name&quot;:&quot;DataIntegrationFlows&quot;,&quot;full_name&quot;:&quot;vibbits/DataIntegrationFlows&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/DataIntegrationFlows&quot;,&quot;description&quot;:&quot;This is a docker image to set up the atmosphere for running several data integration workflows&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/deployments&quot;,&quot;created_at&quot;:&quot;2018-05-28 11:24:02 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:21 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-05-28 12:05:05 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/DataIntegrationFlows.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/DataIntegrationFlows.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/DataIntegrationFlows.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/DataIntegrationFlows&quot;,&quot;homepage&quot;:null,&quot;size&quot;:2,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:169587916,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNjk1ODc5MTY=&quot;,&quot;name&quot;:&quot;incubator-echarts&quot;,&quot;full_name&quot;:&quot;vibbits/incubator-echarts&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/incubator-echarts&quot;,&quot;description&quot;:&quot;A powerful, interactive charting and visualization library for browser&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/deployments&quot;,&quot;created_at&quot;:&quot;2019-02-07 14:50:26 UTC&quot;,&quot;updated_at&quot;:&quot;2019-02-07 14:50:32 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-02-07 13:03:45 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/incubator-echarts.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/incubator-echarts.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/incubator-echarts.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/incubator-echarts&quot;,&quot;homepage&quot;:&quot;http://echarts.apache.org/&quot;,&quot;size&quot;:151893,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;JavaScript&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;apache-2.0&quot;,&quot;name&quot;:&quot;Apache License 2.0&quot;,&quot;spdx_id&quot;:&quot;Apache-2.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/apache-2.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTI=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:118458037,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMTg0NTgwMzc=&quot;,&quot;name&quot;:&quot;modules-4-GenePattern&quot;,&quot;full_name&quot;:&quot;vibbits/modules-4-GenePattern&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/modules-4-GenePattern&quot;,&quot;description&quot;:&quot;modules for integrating under GenePattern some extra tools for NGS analysis&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/deployments&quot;,&quot;created_at&quot;:&quot;2018-01-22 13:04:16 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:21 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-07-03 14:14:58 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/modules-4-GenePattern.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/modules-4-GenePattern.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/modules-4-GenePattern.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/modules-4-GenePattern&quot;,&quot;homepage&quot;:null,&quot;size&quot;:1927,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:136658710,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzY2NTg3MTA=&quot;,&quot;name&quot;:&quot;MOFA&quot;,&quot;full_name&quot;:&quot;vibbits/MOFA&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/MOFA&quot;,&quot;description&quot;:&quot;Multi-Omics Factor Analysis&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/deployments&quot;,&quot;created_at&quot;:&quot;2018-06-08 19:31:44 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:21 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-06-23 20:50:58 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/MOFA.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/MOFA.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/MOFA.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/MOFA&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:110522,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;HTML&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:157213851,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNTcyMTM4NTE=&quot;,&quot;name&quot;:&quot;NextFlow_pipelines&quot;,&quot;full_name&quot;:&quot;vibbits/NextFlow_pipelines&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/NextFlow_pipelines&quot;,&quot;description&quot;:&quot;Tricks and tips for designing a NextFlow pipeline&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/deployments&quot;,&quot;created_at&quot;:&quot;2018-11-12 12:57:24 UTC&quot;,&quot;updated_at&quot;:&quot;2019-02-25 13:20:35 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-02-25 13:20:34 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/NextFlow_pipelines.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/NextFlow_pipelines.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/NextFlow_pipelines.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/NextFlow_pipelines&quot;,&quot;homepage&quot;:null,&quot;size&quot;:1578,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Nextflow&quot;,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:207801732,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkyMDc4MDE3MzI=&quot;,&quot;name&quot;:&quot;OMeta-Public&quot;,&quot;full_name&quot;:&quot;vibbits/OMeta-Public&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/OMeta-Public&quot;,&quot;description&quot;:&quot;OMeta-Public&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/OMeta-Public/deployments&quot;,&quot;created_at&quot;:&quot;2019-09-11 12:00:36 UTC&quot;,&quot;updated_at&quot;:&quot;2019-09-11 12:00:37 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-08-29 10:37:03 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/OMeta-Public.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/OMeta-Public.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/OMeta-Public.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/OMeta-Public&quot;,&quot;homepage&quot;:null,&quot;size&quot;:331121,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:186219236,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxODYyMTkyMzY=&quot;,&quot;name&quot;:&quot;OpenRefineTrainingData&quot;,&quot;full_name&quot;:&quot;vibbits/OpenRefineTrainingData&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/OpenRefineTrainingData&quot;,&quot;description&quot;:null,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/deployments&quot;,&quot;created_at&quot;:&quot;2019-05-12 06:19:18 UTC&quot;,&quot;updated_at&quot;:&quot;2019-05-18 14:15:38 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-05-18 14:15:36 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/OpenRefineTrainingData.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/OpenRefineTrainingData.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/OpenRefineTrainingData.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/OpenRefineTrainingData&quot;,&quot;homepage&quot;:null,&quot;size&quot;:7919,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;mit&quot;,&quot;name&quot;:&quot;MIT License&quot;,&quot;spdx_id&quot;:&quot;MIT&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/mit&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTEz&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:71230827,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnk3MTIzMDgyNw==&quot;,&quot;name&quot;:&quot;phyd3&quot;,&quot;full_name&quot;:&quot;vibbits/phyd3&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/phyd3&quot;,&quot;description&quot;:&quot;Phylogenetic tree viewer based on D3.js&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/deployments&quot;,&quot;created_at&quot;:&quot;2016-10-18 09:19:54 UTC&quot;,&quot;updated_at&quot;:&quot;2019-01-03 04:05:38 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-05-29 06:46:30 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/phyd3.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/phyd3.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/phyd3.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/phyd3&quot;,&quot;homepage&quot;:null,&quot;size&quot;:2992,&quot;stargazers_count&quot;:29,&quot;watchers_count&quot;:29,&quot;language&quot;:&quot;JavaScript&quot;,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:7,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:3,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-3.0&quot;,&quot;name&quot;:&quot;GNU General Public License v3.0&quot;,&quot;spdx_id&quot;:&quot;GPL-3.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-3.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTk=&quot;},&quot;forks&quot;:7,&quot;open_issues&quot;:3,&quot;watchers&quot;:29,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:138773305,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzg3NzMzMDU=&quot;,&quot;name&quot;:&quot;presentation&quot;,&quot;full_name&quot;:&quot;vibbits/presentation&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/presentation&quot;,&quot;description&quot;:&quot;20180907 presentation community meeting&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/presentation&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/deployments&quot;,&quot;created_at&quot;:&quot;2018-06-26 17:49:03 UTC&quot;,&quot;updated_at&quot;:&quot;2019-01-25 06:43:04 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-01-25 06:43:03 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/presentation.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/presentation.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/presentation.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/presentation&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:33,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:2,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:1,&quot;license&quot;:null,&quot;forks&quot;:2,&quot;open_issues&quot;:1,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:186428088,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxODY0MjgwODg=&quot;,&quot;name&quot;:&quot;qupath&quot;,&quot;full_name&quot;:&quot;vibbits/qupath&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/qupath&quot;,&quot;description&quot;:&quot;QuPath - Open Source Digital Pathology&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/qupath&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/deployments&quot;,&quot;created_at&quot;:&quot;2019-05-13 13:45:18 UTC&quot;,&quot;updated_at&quot;:&quot;2019-05-15 09:48:35 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-05-15 09:48:33 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/qupath.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/qupath.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/qupath.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/qupath&quot;,&quot;homepage&quot;:&quot;https://qupath.github.io&quot;,&quot;size&quot;:83357,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Java&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-3.0&quot;,&quot;name&quot;:&quot;GNU General Public License v3.0&quot;,&quot;spdx_id&quot;:&quot;GPL-3.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-3.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTk=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:133400916,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzM0MDA5MTY=&quot;,&quot;name&quot;:&quot;rocker_conda_data_integration&quot;,&quot;full_name&quot;:&quot;vibbits/rocker_conda_data_integration&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/rocker_conda_data_integration&quot;,&quot;description&quot;:&quot;Docker container based on rocker/tidyverse with miniconda3 to run various integration tools for Omics&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/deployments&quot;,&quot;created_at&quot;:&quot;2018-05-14 17:53:42 UTC&quot;,&quot;updated_at&quot;:&quot;2019-07-31 15:35:10 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-01-13 08:51:00 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/rocker_conda_data_integration.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/rocker_conda_data_integration.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/rocker_conda_data_integration.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/rocker_conda_data_integration&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:50406,&quot;stargazers_count&quot;:3,&quot;watchers_count&quot;:3,&quot;language&quot;:&quot;R&quot;,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:3,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-2.0&quot;,&quot;name&quot;:&quot;GNU General Public License v2.0&quot;,&quot;spdx_id&quot;:&quot;GPL-2.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-2.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTg=&quot;},&quot;forks&quot;:3,&quot;open_issues&quot;:0,&quot;watchers&quot;:3,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:131875427,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzE4NzU0Mjc=&quot;,&quot;name&quot;:&quot;rocker_conda_mofa&quot;,&quot;full_name&quot;:&quot;vibbits/rocker_conda_mofa&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/rocker_conda_mofa&quot;,&quot;description&quot;:&quot;Docker container on rocker/tidyverse with miniconda3 to run MOFA tool https://github.com/bioFAM/MOFA&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/deployments&quot;,&quot;created_at&quot;:&quot;2018-05-02 16:08:04 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:21 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-05-02 18:07:02 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/rocker_conda_mofa.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/rocker_conda_mofa.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/rocker_conda_mofa.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/rocker_conda_mofa&quot;,&quot;homepage&quot;:null,&quot;size&quot;:17,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-2.0&quot;,&quot;name&quot;:&quot;GNU General Public License v2.0&quot;,&quot;spdx_id&quot;:&quot;GPL-2.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-2.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTg=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:113286207,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMTMyODYyMDc=&quot;,&quot;name&quot;:&quot;scop3d&quot;,&quot;full_name&quot;:&quot;vibbits/scop3d&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/scop3d&quot;,&quot;description&quot;:&quot;Sequence conservation of protein on 3D structure.&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/deployments&quot;,&quot;created_at&quot;:&quot;2017-12-06 07:56:32 UTC&quot;,&quot;updated_at&quot;:&quot;2019-06-04 20:25:31 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-06-04 20:25:30 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/scop3d.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/scop3d.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/scop3d.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/scop3d&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:26190,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Python&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:false,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-3.0&quot;,&quot;name&quot;:&quot;GNU General Public License v3.0&quot;,&quot;spdx_id&quot;:&quot;GPL-3.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-3.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTk=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:104596713,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMDQ1OTY3MTM=&quot;,&quot;name&quot;:&quot;scRNA-Seq-TCC-prep&quot;,&quot;full_name&quot;:&quot;vibbits/scRNA-Seq-TCC-prep&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/scRNA-Seq-TCC-prep&quot;,&quot;description&quot;:&quot;Preprocessing of single-cell RNA-Seq data for input to kallisto&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/deployments&quot;,&quot;created_at&quot;:&quot;2017-09-23 20:22:54 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:22 UTC&quot;,&quot;pushed_at&quot;:&quot;2016-12-01 02:04:26 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/scRNA-Seq-TCC-prep.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/scRNA-Seq-TCC-prep.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/scRNA-Seq-TCC-prep.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/scRNA-Seq-TCC-prep&quot;,&quot;homepage&quot;:null,&quot;size&quot;:8740,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Jupyter Notebook&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-3.0&quot;,&quot;name&quot;:&quot;GNU General Public License v3.0&quot;,&quot;spdx_id&quot;:&quot;GPL-3.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-3.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTk=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:137523231,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzc1MjMyMzE=&quot;,&quot;name&quot;:&quot;scRNA-tools&quot;,&quot;full_name&quot;:&quot;vibbits/scRNA-tools&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/scRNA-tools&quot;,&quot;description&quot;:&quot;Table of software for the analysis of single-cell RNA-seq data.&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/deployments&quot;,&quot;created_at&quot;:&quot;2018-06-15 19:10:39 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:22 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-06-15 19:31:56 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/scRNA-tools.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/scRNA-tools.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/scRNA-tools.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/scRNA-tools&quot;,&quot;homepage&quot;:null,&quot;size&quot;:3835,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;R&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;mit&quot;,&quot;name&quot;:&quot;MIT License&quot;,&quot;spdx_id&quot;:&quot;MIT&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/mit&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTEz&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:100585992,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMDA1ODU5OTI=&quot;,&quot;name&quot;:&quot;sc_read_kallisto_wrapper&quot;,&quot;full_name&quot;:&quot;vibbits/sc_read_kallisto_wrapper&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/sc_read_kallisto_wrapper&quot;,&quot;description&quot;:&quot;wrapper for single cell transcriptomics pipeline with kallisto&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/deployments&quot;,&quot;created_at&quot;:&quot;2017-08-17 09:24:32 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:22 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-02-09 09:22:40 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/sc_read_kallisto_wrapper.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/sc_read_kallisto_wrapper.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/sc_read_kallisto_wrapper.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/sc_read_kallisto_wrapper&quot;,&quot;homepage&quot;:null,&quot;size&quot;:84,&quot;stargazers_count&quot;:6,&quot;watchers_count&quot;:6,&quot;language&quot;:&quot;HTML&quot;,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:2,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-3.0&quot;,&quot;name&quot;:&quot;GNU General Public License v3.0&quot;,&quot;spdx_id&quot;:&quot;GPL-3.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-3.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTk=&quot;},&quot;forks&quot;:2,&quot;open_issues&quot;:0,&quot;watchers&quot;:6,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:153169091,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNTMxNjkwOTE=&quot;,&quot;name&quot;:&quot;WaaS-Finnish-Cloud&quot;,&quot;full_name&quot;:&quot;vibbits/WaaS-Finnish-Cloud&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/WaaS-Finnish-Cloud&quot;,&quot;description&quot;:null,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/deployments&quot;,&quot;created_at&quot;:&quot;2018-10-15 19:20:17 UTC&quot;,&quot;updated_at&quot;:&quot;2019-03-27 05:45:56 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-03-27 05:44:56 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/WaaS-Finnish-Cloud.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/WaaS-Finnish-Cloud.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/WaaS-Finnish-Cloud.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/WaaS-Finnish-Cloud&quot;,&quot;homepage&quot;:null,&quot;size&quot;:3,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;}],&quot;releases&quot;:false,&quot;releases_url&quot;:&quot;https://github.com/vibbits/training-material/releases&quot;,&quot;repository_name&quot;:&quot;training-material&quot;,&quot;repository_nwo&quot;:&quot;vibbits/training-material&quot;,&quot;repository_url&quot;:&quot;https://github.com/vibbits/training-material&quot;,&quot;show_downloads&quot;:null,&quot;source&quot;:{&quot;branch&quot;:&quot;gh-pages&quot;,&quot;path&quot;:&quot;/&quot;},&quot;tar_url&quot;:&quot;https://github.com/vibbits/training-material/tarball/gh-pages&quot;,&quot;url&quot;:&quot;https://github.com/pages/vibbits/training-material&quot;,&quot;versions&quot;:{},&quot;wiki_url&quot;:null,&quot;zip_url&quot;:&quot;https://github.com/vibbits/training-material/zipball/gh-pages&quot;},&quot;env&quot;:{&quot;TERM&quot;:&quot;xterm-256color&quot;,&quot;SHELL&quot;:&quot;/bin/bash&quot;,&quot;MAKEFLAGS&quot;:&quot;&quot;,&quot;CONDA_SHLVL&quot;:&quot;2&quot;,&quot;CONDA_PROMPT_MODIFIER&quot;:&quot;(galaxy_training_material) &quot;,&quot;USER&quot;:&quot;albot&quot;,&quot;LS_COLORS&quot;:&quot;rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:&quot;,&quot;SUDO_USER&quot;:&quot;bitsfacility&quot;,&quot;SUDO_UID&quot;:&quot;1002&quot;,&quot;CONDA_EXE&quot;:&quot;/home/albot/miniconda3/bin/conda&quot;,&quot;MAKE_TERMOUT&quot;:&quot;/dev/pts/1&quot;,&quot;MAKELEVEL&quot;:&quot;1&quot;,&quot;USERNAME&quot;:&quot;albot&quot;,&quot;MFLAGS&quot;:&quot;&quot;,&quot;_CE_CONDA&quot;:&quot;&quot;,&quot;CONDA_PREFIX_1&quot;:&quot;/home/albot/miniconda3&quot;,&quot;MAIL&quot;:&quot;/var/mail/albot&quot;,&quot;PATH&quot;:&quot;/home/albot/bin:/home/albot/.local/bin:/home/albot/miniconda3/envs/galaxy_training_material/bin:/home/albot/miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin&quot;,&quot;CONDA_PREFIX&quot;:&quot;/home/albot/miniconda3/envs/galaxy_training_material&quot;,&quot;PWD&quot;:&quot;/home/albot/alex/training-material&quot;,&quot;LANG&quot;:&quot;en_US.UTF-8&quot;,&quot;_CE_M&quot;:&quot;&quot;,&quot;SHLVL&quot;:&quot;2&quot;,&quot;HOME&quot;:&quot;/home/albot&quot;,&quot;SUDO_COMMAND&quot;:&quot;/bin/bash&quot;,&quot;CONDA_PYTHON_EXE&quot;:&quot;/home/albot/miniconda3/bin/python&quot;,&quot;LOGNAME&quot;:&quot;albot&quot;,&quot;MAKE_TERMERR&quot;:&quot;/dev/pts/1&quot;,&quot;XDG_DATA_DIRS&quot;:&quot;/usr/local/share:/usr/share:/var/lib/snapd/desktop&quot;,&quot;CONDA_DEFAULT_ENV&quot;:&quot;galaxy_training_material&quot;,&quot;LESSOPEN&quot;:&quot;| /usr/bin/lesspipe %s&quot;,&quot;SUDO_GID&quot;:&quot;1003&quot;,&quot;LESSCLOSE&quot;:&quot;/usr/bin/lesspipe %s %s&quot;,&quot;_&quot;:&quot;/home/albot/miniconda3/envs/galaxy_training_material/bin/jekyll&quot;}}
              page: {&quot;layout&quot;:&quot;introduction_slides&quot;,&quot;logo&quot;:&quot;GTN&quot;,&quot;title&quot;:&quot;Introduction to Basic Bioinformatics Concepts, Databases and Tools&quot;,&quot;type&quot;:&quot;introduction&quot;,&quot;contributors&quot;:[&quot;abotzki&quot;],&quot;topic_name&quot;:&quot;basic-bioinformatics&quot;,&quot;tutorial_name&quot;:&quot;introduction&quot;,&quot;content&quot;:&quot;### How to fill the slide decks?\n\nPlease follow our\n[tutorial to learn how to fill the slides]({{ site.baseurl }}/topics/contributing/tutorials/create-new-tutorial-slides/slides.html)\n&quot;,&quot;dir&quot;:&quot;/topics/basic-bioinformatics/slides/&quot;,&quot;name&quot;:&quot;introduction.html&quot;,&quot;path&quot;:&quot;topics/basic-bioinformatics/slides/introduction.html&quot;,&quot;url&quot;:&quot;/topics/basic-bioinformatics/slides/introduction.html&quot;}
              topic: {&quot;name&quot;:&quot;basic-bioinformatics&quot;,&quot;type&quot;:&quot;basics&quot;,&quot;category&quot;:&quot;basics&quot;,&quot;title&quot;:&quot;Basic Bioinformatics Concepts, Databases and Tools&quot;,&quot;summary&quot;:&quot;Bioinformatics Introduction&quot;,&quot;requirements&quot;:null,&quot;maintainers&quot;:[&quot;abotzki&quot;]}
              material: {&quot;layout&quot;:&quot;introduction_slides&quot;,&quot;logo&quot;:&quot;GTN&quot;,&quot;title&quot;:&quot;Introduction to Basic Bioinformatics Concepts, Databases and Tools&quot;,&quot;type&quot;:&quot;introduction&quot;,&quot;contributors&quot;:[&quot;abotzki&quot;],&quot;topic_name&quot;:&quot;basic-bioinformatics&quot;,&quot;tutorial_name&quot;:&quot;introduction&quot;,&quot;content&quot;:&quot;### How to fill the slide decks?\n\nPlease follow our\n[tutorial to learn how to fill the slides]({{ site.baseurl }}/topics/contributing/tutorials/create-new-tutorial-slides/slides.html)\n&quot;,&quot;dir&quot;:&quot;/topics/basic-bioinformatics/slides/&quot;,&quot;name&quot;:&quot;introduction.html&quot;,&quot;path&quot;:&quot;topics/basic-bioinformatics/slides/introduction.html&quot;,&quot;url&quot;:&quot;/topics/basic-bioinformatics/slides/introduction.html&quot;}
        </pre>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-header"><span>
<a href="/training-material/topics/basic-bioinformatics" title="Return to topic page" ><i class="fa fa-level-up" aria-hidden="true"></i></a>

<a class="nav-link" href="https://github.com/vibbits/training-material/edit/master/topics/basic-bioinformatics/slides/introduction.html"><i class="fa fa-pencil" aria-hidden="true"></i></a>

</span></div>

<div class="my-footer"><span>

<img src="/training-material/assets/images/bioinformatics_core_rgb_neg.png" alt="VIB Bioinformatics Core" style="height: 40px;"/>

</span></div>

---

# Introduction to Basic Bioinformatics Concepts, Databases and Tools



---

### How to fill the slide decks?

Please follow our
[tutorial to learn how to fill the slides](/training-material/topics/contributing/tutorials/create-new-tutorial-slides/slides.html)


---




## Related tutorials


    

    
        
    



---

## Thank you!

This material is the result of a collaborative work. Thanks the colleagues of the VIB Bioinformatics Core and all the contributors!


<div class="contributors-line">


<a href="/training-material/hall-of-fame#abotzki" class="contributor-badge"><img src="https://avatars.githubusercontent.com/abotzki" alt="Alexander Botzki">Alexander Botzki</a>


</div>



<img src="/training-material/assets/images/vib_notag_neg_rgb.jpg" alt="Galaxy Training Network" style="height: 100px;"/>



    </textarea>
    <script src="/training-material/assets/js/remark-latest.min.js" type="text/javascript">
    </script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false,}});
      var hljs = remark.highlighter.engine;
    </script>
  </body>
</html>
