<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta name="google-site-verification" content="dw_1zBYRNxF7-vYNDXh3Ght_JuqgJt3BD1N_Pynqmv0" />
    <title>06 Visualization with deepTools</title>
    <script async="" src="https://www.google-analytics.com/analytics.js"></script>
    <script>
        (function(i, s, o, g, r, a, m) {
            i['GoogleAnalyticsObject'] = r;
            i[r] = i[r] || function() {
                (i[r].q = i[r].q || []).push(arguments)
            }, i[r].l = 1 * new Date();
            a = s.createElement(o),
                m = s.getElementsByTagName(o)[0];
            a.async = 1;
            a.src = g;
            m.parentNode.insertBefore(a, m)
        })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
        ga('create', 'UA-61194136-18', 'auto');
        ga('send', 'pageview');
    </script>
    <link rel="stylesheet" href="/assets/css/bootstrap.min.css?v=3">
    <link rel="stylesheet" href="/assets/css/bootstrap-toc.min.css">
    <link rel="stylesheet" href="/assets/css/font-awesome.css">
    <link rel="stylesheet" href="/assets/css/main.scss">
    <link rel="stylesheet" href="/assets/css/VIB.css">
    <link rel="stylesheet" href="/assets/css/academicons.css">
    <link rel="stylesheet" href="/assets/css/syntax_highlighting.css">
    <link rel="shortcut icon" href="/favicon.ico " type="image/x-icon" />     
    <meta name="description" content=" This training gives an introduction to ChIP-seq data ana..." />
    <meta property="og:title" content="VIB Bioinformatics Training: 06 Visualization with deepTools" />
    <meta property="og:description" content=" This training gives an introduction to ChIP-seq data ana..." />
    <meta property="og:image" content="/assets/images/bioinformatics_core_rgb_neg.png" />
</head>

<body data-spy="scroll" data-target="#toc">
    <div class="top-bar-vib">
 <a href="http://vib.be/en" target="_blank" style="font-size:14px">VIB Home </a>
</div>

<div class="container main-content">

<div class="container page-top__container">
          <div class="region region-top">
  <div id="logo" class="col-md-3 col-lg-3 text-left" style="padding-left:0px;margin-top:20px">
       <a href="/">
           <img src="/assets/images/logo.svg" height="60" alt="VIB logo" class="margin20">
       </a>
    </div>
    <div class="col-md-9 col-lg-9 text-right" style="margin-top:50px;">
    <nav role="navigation" aria-labelledby="block-mainnavigation-menu" id="block-mainnavigation" class="block block-menu navigation menu--main">
  


        <a href="/" class="topmenu">Home</a>


        <a href="/about.html" class="topmenu" >About</a>


        <a href="/hall-of-fame.html" class="topmenu">Contribute</a>


        <a href="/faq.html" class="topmenu">Faq</a>


        <a href="/search/index.html" class="topmenu">Search</a>


  </nav>

    </div>
  </div>
</div>














    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "headline": "06 Visualization with deepTools",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "provider": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "courseCode": "chip-seq / visualization-deeptools / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for '06 Visualization with deepTools' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "06 Visualization with deepTools",
    "description": "06 Visualization with deepTools",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "bits@vib.be",
      "name": "VIB Bioinformatics Core",
      "url": "https://www.bits.vib.be"
    }
  },
  "url": "https://material.bits.vib.be//topics/chip-seq/tutorials/visualization-deeptools/tutorial.html",
  "timeRequired": "PT15M",
  "description": "06 Visualization with deepTools\\nThe questions this  addresses are:\n - Download the data from GEO or ENA\n\n\\nThe objectives are:\n - Basic knowledge of downloading data from GEO or ENA\n\n",
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://data.bits.vib.be/pub/trainingen/PSA/",
      "name": "Training data for 06 Visualization with deepTools tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Janick Mathys"
    },
    {
      "@type": "Person",
      "name": "Morgane Thomas-Chollier"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Janick Mathys"
    },
    {
      "@type": "Person",
      "name": "Morgane Thomas-Chollier"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "ChIP-Seq analysis",
      "description": " This training gives an introduction to ChIP-seq data analysis, covering the processing steps starting from the reads to the peaks. Among all possible downstream analyses, the practical aspect will focus on motif analyses. A particular emphasis will be put on deciding which downstream analyses to perform depending on the biological question. This training does not cover all methods available today. It does not aim at bringing users to a professional NGS analyst level but provides enough information to allow biologists understand what DNA sequencing practically is and to communicate with NGS experts for more in-depth needs.\n- Have an understanding of the nature of ChIP-Seq data - Perform a complete analysis workflow including QC, read mapping, visualization in a genome browser and peak-calling - Use the GenePattern platform for each step of the workflow and feel the complexity of the task - Have an overview of possible downstream analyses - Perform a motif analysis with online web programs\nFor this training, we will use a dataset produced by Myers et al [1] involved in the regulation of gene expression under anaerobic conditions in bacteria. We will focus on one factor: FNR. The advantage of this dataset is its small size, allowing real time execution of all steps of the dataset. ",
      "url": "https://material.bits.vib.be//topics/chip-seq/"
    }
  ]
}
    </script>

    <div class="row tutorial">
      <div class="col-md-12" role="main">
        <h1 data-toc-skip>06 Visualization with deepTools</h1>
        <div class="contributors-line">By: 

<a href="/hall-of-fame#janick-bits" class="contributor-badge">Janick Mathys</a>
, <a href="/hall-of-fame#morganeTC" class="contributor-badge">Morgane Thomas-Chollier</a>


</div>
        <blockquote class="overview">

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li>Download the data from GEO or ENA</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li>Basic knowledge of downloading data from GEO or ENA</li>
            
            </ul>

            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 15 minutes</p>
            

            


            

        </blockquote>

        <div class="container" data-field="body">
            <div class="row">
                <div class="col-md-10" role="main">
                    <p>For visualization with deepTools we need a bam file in which the order of the reads is determined by genomic location. We have created such a bam file in the peak calling step using the SortSam tool from the Picard suite.
The bam file still contains duplicate reads (=reads that map to exactly the same position in the genome). Such reads represent technical duplicates often caused by biased PCR amplification during the library prep or by fragments coming from repetitive elements in the genome… Since we are going to quantify the reads (we look for regions that are enriched in the ChIP sample) these technical duplicates will distort the quantifications. So they should be removed from the .bam file
Additionally an index file should be created to allow for fast and easy access to the sorted and processed .bam file.</p>

<blockquote>
  <p>Which tool from the Picard suite can be used to mark/remove duplicates?
Picard MarkDuplicates can be used to remove duplicates.</p>
</blockquote>

<blockquote>
  <p>Remove duplicates and index the .bam files?</p>
  <ul>
    <li>Use the sorted .bam files as input files</li>
    <li>Indicate that the files are sorted according to coordinates</li>
    <li>Remove the sequencing duplicates, duplicates generated by PCR</li>
    <li>Create an index file</li>
  </ul>
</blockquote>

<p>MarkDuplicates generates an error but you can ignore the error. Open the metrics.txt file that is generated by MarkDuplicates.</p>

<blockquote>
  <p>How many duplicates were found in the ChIP sample?</p>

  <p><img src="../../images/DuplicationMatrics.png" alt="" /></p>
</blockquote>

<p>Now we will plot a Lorenz curve with DeepTools to assess the quality of the ChIP. It answers the question: “Did my ChIP work?” Did the antibody-treatment enrich sufficiently so that the ChIP signal can be differentiated from the background signal in the control sample? This is a valid question since around 90% of all fragments in a ChIP experiment will represent the genomic background.</p>

<p>For factors that enrich well-defined, narrow regions, the plot can be used to assess the strength of the ChIP, but the broader the enrichments, the less clear the plot will be. Vice versa, if you do not know what kind of signal to expect, the plot will give you an indication of how careful you have to be during downstream analyses to separate biological noise from meaningful signal.</p>

<p>The tool randomly samples genome regions (bins) of a specific legth in indexed BAM files, calculates the sum of all reads that map in a bin. These sums are sorted according to their rank and a profile of cumulative sums is plotted.</p>

<blockquote>
  <p>Which tool from the DeepTools toolset are you going to use for this? 
Run DeepTools <a href="http://deeptools.readthedocs.io/en/latest/content/tools/plotFingerprint.html">plotFingerprint</a> to draw the Lorenz curve.</p>
</blockquote>

<blockquote>
  <p>Create a Lorenz curve for the ChIP sample</p>
  <ul>
    <li>You have to provide both the .bam and the .bai file as input!</li>
    <li>The <strong>nsamples</strong> parameter represent the number of bins that is sampled from the genome. It has to be smaller than the genome size divided by the size of the bins (default 500nt). The size of the <em>E. coli</em> genome is 4639675 nt. So we will set this parameter to 9000.</li>
    <li>Other parameters can be kept at default settings</li>
  </ul>
</blockquote>

<p>An experiment with perfect uniform distribution of reads along the genome (without enrichment) and infinite sequencing coverage should generate a straight diagonal line. A very specific and strong ChIP enrichment will be indicated by a prominent and steep rise of the curve towards the highest rank. This means that a big chunk of reads from the ChIP sample is located in few bins.
Below you see a few examples on how to interpret this curve:</p>

<p><img src="../../images/DTLorenz.png" alt="" /></p>

<p><strong>What do you think about the fingerprint plot that was generated on the <em>E. coli</em> data?</strong></p>



                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>

                        <ul>
                            
                            <li>SRA ID</li>
                            
                            <li>GEO</li>
                            
                            <li>ENA</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/topics/chip-seq#references">here</a>.</p>
                    

                    

                </div>
                <!-- sidebar, which will be hidden on a small screen -->
                <nav class="col-md-2 sticky-toc bs-docs-sidebar hidden-print hidden-xs hidden-sm" role="complementary" id="toc" data-toggle="toc"></nav>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

</div>
</div>
</div>
<footer>
   <div class="well row">
       <div class="col-md-2 text-center" style="padding: 25px 20px 25px 20px;">				
         <a href="https://www.bits.vib.be" target="_blank"><img src="/assets/images/bioinformatics_core_rgb_neg.png" width="200"></a>
       </div>
       <div class="col-md-8 text-center" style="padding-top: 20px;">	
        <a href="https://github.com/vibbits/training-material" title="View on GitHub" class="margin20" target="_blank">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View on GitHub
                        </a>
       </div>
       <div class="col-md-2 text-center" style="padding-top: 30px;">
         <span class="slogan"><br>SCIENCE MEETS LIFE</span>
       </div>
   </div>
</footer>


</body>
<script type="text/javascript" src="/assets/js/jquery.slim.min.js"></script>
<script type="text/javascript" src="/assets/js/popper.min.js"></script>
<script type="text/javascript" src="/assets/js/bootstrap.min.js?v=3"></script>
<script type="text/javascript" src="/assets/js/details-element-polyfill.js"></script>
<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<script type="text/javascript" src="/assets/js/bootstrap-toc.min.js"></script>
<script type="text/javascript" src="/assets/js/main.js"></script>
<script type="text/javascript" src="/assets/js/clipboard.min.js"></script>
<script type="text/javascript">
    var snippets = document.querySelectorAll('div.highlight');
    [].forEach.call(snippets, function(snippet) {
        snippet.firstChild.insertAdjacentHTML('beforebegin', '<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets = new ClipboardJS('[data-clipboard-snippet]', {
        target: function(trigger) {
            return trigger.nextElementSibling;
        }
    });
</script>

</html>
