<!DOCTYPE html>
<html>









  <head>
    <meta charset="utf-8">
    <title>Protein Structure Analysis</title>
    
    <link rel="stylesheet" href="/training-material/assets/css/slides.css">
    <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css" id="theme">
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

    
    
    
    
    <meta name="description" content="Analyzing the protein structure of your protein-of-intere..." />
    <meta property="og:title" content="VIB Bioinformatics Training: Protein Structure Analysis" />
    <meta property="og:description" content="Analyzing the protein structure of your protein-of-intere..." />
    <meta property="og:image" content="/training-material/assets/images/bioinformatics_core_rgb_neg.png" />
    <script type="application/ld+json">
       


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "headline": "Protein Structure Analysis",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "provider": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "bits@vib.be",
    "name": "VIB Bioinformatics Core",
    "url": "https://www.bits.vib.be"
  },
  "courseCode": "protein-structure-analysis / introduction / introduction.html",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Protein Structure Analysis",
    "description": "Analyzing the protein structure of your protein-of-interest can be advantageous in multiple ways. It can help you discover regions which are good candidates to interact with other proteins. It can help you discover new domains. It can help with identifying differences with homologuous proteins and a lot more.",
    "url": "https://vibbits.github.io//training-material/topics/protein-structure-analysis/"
  },
  "learningResourceType": "slides",
  "name": "Introduction to 'Protein Structure Analysis'",
  "url": "https://vibbits.github.io//training-material/topics/protein-structure-analysis/slides/introduction.html",
  "description": "Protein Structure Analysis",
  "hasPart": [

  ],
  "mentions": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Alexander Botzki"
    },
    {
      "@type": "Person",
      "name": "Joost Van Durme"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Alexander Botzki"
    },
    {
      "@type": "Person",
      "name": "Joost Van Durme"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Protein Structure Analysis",
      "description": "Analyzing the protein structure of your protein-of-interest can be advantageous in multiple ways. It can help you discover regions which are good candidates to interact with other proteins. It can help you discover new domains. It can help with identifying differences with homologuous proteins and a lot more.",
      "url": "https://vibbits.github.io//training-material/topics/protein-structure-analysis/"
    }
  ]
}
    </script>
  </head>
  <body>
         <pre>
              site: {&quot;config&quot;:null,&quot;data&quot;:{&quot;basic-bioinformatics&quot;:{&quot;name&quot;:&quot;basic-bioinformatics&quot;,&quot;type&quot;:&quot;basics&quot;,&quot;category&quot;:&quot;basics&quot;,&quot;title&quot;:&quot;Basic Bioinformatics Concepts, Databases and Tools&quot;,&quot;summary&quot;:&quot;Bioinformatics Introduction&quot;,&quot;requirements&quot;:null,&quot;maintainers&quot;:[&quot;abotzki&quot;]},&quot;contributors&quot;:{&quot;chdeb&quot;:{&quot;name&quot;:&quot;Christof De Bo&quot;,&quot;email&quot;:&quot;christof.debo@vib.be&quot;},&quot;tmuylder&quot;:{&quot;name&quot;:&quot;Tuur Muyldermans&quot;,&quot;email&quot;:&quot;tuur.muyldermans@vib.be&quot;},&quot;janick-bits&quot;:{&quot;name&quot;:&quot;Janick Mathys&quot;,&quot;email&quot;:&quot;janick.mathys@vib.be&quot;},&quot;abotzki&quot;:{&quot;name&quot;:&quot;Alexander Botzki&quot;,&quot;email&quot;:&quot;alexander.botzki@vib.be&quot;},&quot;jvdurme&quot;:{&quot;name&quot;:&quot;Joost Van Durme&quot;,&quot;email&quot;:&quot;bits@vib.be&quot;}},&quot;protein-structure-analysis&quot;:{&quot;name&quot;:&quot;protein-structure-analysis&quot;,&quot;type&quot;:&quot;basics&quot;,&quot;category&quot;:&quot;basics&quot;,&quot;title&quot;:&quot;Protein Structure Analysis&quot;,&quot;summary&quot;:&quot;Analyzing the protein structure of your protein-of-interest can be advantageous in multiple ways. It can help you discover regions which are good candidates to interact with other proteins. It can help you discover new domains. It can help with identifying differences with homologuous proteins and a lot more.&quot;,&quot;extra&quot;:&quot;protein_analysis&quot;,&quot;requirements&quot;:null,&quot;maintainers&quot;:[&quot;abotzki&quot;],&quot;references&quot;:[{&quot;authors&quot;:&quot;Switchlab&quot;,&quot;title&quot;:&quot;Home of FoldX plugin&quot;,&quot;link&quot;:&quot;http://foldxyasara.switchlab.org/&quot;,&quot;summary&quot;:&quot;More information about FoldX plugin and troubleshooting&quot;},{&quot;authors&quot;:&quot;Wikipedia&quot;,&quot;title&quot;:&quot;Wiki page about PDB&quot;,&quot;link&quot;:&quot;https://en.wikipedia.org/wiki/Protein_Data_Bank&quot;,&quot;summary&quot;:&quot;wiki page explaining information on PDB&quot;},{&quot;authors&quot;:&quot;YASARA developers&quot;,&quot;title&quot;:&quot;Working with YASARA&quot;,&quot;link&quot;:&quot;http://www.yasara.org/movies.htm&quot;,&quot;summary&quot;:&quot;Movie tutorials on YASARA&quot;}]}},&quot;time&quot;:&quot;2019-08-09 12:47:31 +0000&quot;,&quot;tags&quot;:{},&quot;pages&quot;:[&quot;\n&quot;,&quot;---\nname: galaxy_training_material\nchannels:\n  - conda-forge\n  - bioconda\n  - defaults\ndependencies:\n  - gmp=6.1.2\n  - jemalloc=5.0.1\n  - libiconv=1.15\n  - nodejs=9.11.1\n  - openssl=1.0.2o\n  - pandas=0.22.0\n  - pip\n  - pip:\n    - pathspec==0.5.6\n    - oyaml\n  - planemo&gt;=0.55.0\n  - readline=7.0\n  - requests=2.18.4\n  - ruby=2.4.4\n  - yaml=0.1.7\n  - yamllint=1.11.0\n  - zlib=1.2.11\n  - libxml2\n  - ephemeris\n  - pkg-config\n&quot;,&quot;&lt;ol id=\&quot;markdown-toc\&quot;&gt;\n  &lt;li&gt;&lt;a href=\&quot;#overview-questions\&quot; id=\&quot;markdown-toc-overview-questions\&quot;&gt;Overview Questions&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-is-this-website\&quot; id=\&quot;markdown-toc-what-is-this-website\&quot;&gt;What is this website?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-is-galaxy\&quot; id=\&quot;markdown-toc-what-is-galaxy\&quot;&gt;What is Galaxy?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-are-the-tutorials-for\&quot; id=\&quot;markdown-toc-what-are-the-tutorials-for\&quot;&gt;What are the tutorials for?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-audiences-are-the-tutorials-for\&quot; id=\&quot;markdown-toc-what-audiences-are-the-tutorials-for\&quot;&gt;What audiences are the tutorials for?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-is-the-content-licensed\&quot; id=\&quot;markdown-toc-how-is-the-content-licensed\&quot;&gt;How is the content licensed?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-cite-this-project\&quot; id=\&quot;markdown-toc-how-can-i-cite-this-project\&quot;&gt;How can I cite this project?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-advertise-the-training-materials-on-my-posters\&quot; id=\&quot;markdown-toc-how-can-i-advertise-the-training-materials-on-my-posters\&quot;&gt;How can I advertise the training materials on my posters?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#for-individual-learners\&quot; id=\&quot;markdown-toc-for-individual-learners\&quot;&gt;For Individual Learners&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-do-i-start\&quot; id=\&quot;markdown-toc-where-do-i-start\&quot;&gt;Where do I start?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-do-i-use-this-material\&quot; id=\&quot;markdown-toc-how-do-i-use-this-material\&quot;&gt;How do I use this material?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-can-i-run-the-hands-on-tutorials\&quot; id=\&quot;markdown-toc-where-can-i-run-the-hands-on-tutorials\&quot;&gt;Where can I run the hands-on tutorials?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-get-help\&quot; id=\&quot;markdown-toc-how-can-i-get-help\&quot;&gt;How can I get help?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#for-instructors\&quot; id=\&quot;markdown-toc-for-instructors\&quot;&gt;For Instructors&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-do-i-start-1\&quot; id=\&quot;markdown-toc-where-do-i-start-1\&quot;&gt;Where do I start?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-galaxy-instance-should-i-use-for-my-training\&quot; id=\&quot;markdown-toc-what-galaxy-instance-should-i-use-for-my-training\&quot;&gt;What Galaxy instance should I use for my training?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-do-i-get-help\&quot; id=\&quot;markdown-toc-how-do-i-get-help\&quot;&gt;How do I get help?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#contributing\&quot; id=\&quot;markdown-toc-contributing\&quot;&gt;Contributing&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-get-started-with-contributing\&quot; id=\&quot;markdown-toc-how-can-i-get-started-with-contributing\&quot;&gt;How can I get started with contributing?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-give-feedback\&quot; id=\&quot;markdown-toc-how-can-i-give-feedback\&quot;&gt;How can I give feedback?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-test-an-interactive-tour\&quot; id=\&quot;markdown-toc-how-can-i-test-an-interactive-tour\&quot;&gt;How can I test an Interactive Tour?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot; id=\&quot;markdown-toc-how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot;&gt;How can I fix mistakes or expand an existing tutorial using the GitHub interface?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-contribute-in-advanced-mode\&quot; id=\&quot;markdown-toc-how-can-i-contribute-in-advanced-mode\&quot;&gt;How can I contribute in “advanced” mode?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#other-questions\&quot; id=\&quot;markdown-toc-other-questions\&quot;&gt;Other Questions&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#sustainability-of-the-training-material-and-metadata\&quot; id=\&quot;markdown-toc-sustainability-of-the-training-material-and-metadata\&quot;&gt;Sustainability of the training-material and metadata&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h1 id=\&quot;overview-questions\&quot;&gt;Overview Questions&lt;/h1&gt;\n\n&lt;h2 id=\&quot;what-is-this-website\&quot;&gt;What is this website?&lt;/h2&gt;\n\n&lt;p&gt;This website is a collection of hands-on tutorials that are designed to be interactive and are built around Galaxy:&lt;/p&gt;\n\n&lt;figure id=\&quot;figure-1\&quot;&gt;&lt;img src=\&quot;shared/images/interactive_training.png\&quot; alt=\&quot;Interactive training\&quot; /&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 1:&lt;/span&gt; The hands-on nature of our training material. It can be used with two web browser windows open side-by-side, one pointed at the current tutorial and the other at a Galaxy instance&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;p&gt;This material is developed and maintained by the &lt;a href=\&quot;https://galaxyproject.org/\&quot;&gt;worldwide Galaxy community&lt;/a&gt;. You can learn more about this effort by reading our &lt;a href=\&quot;https://doi.org/10.1016/j.cels.2018.05.012\&quot;&gt;article&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-is-galaxy\&quot;&gt;What is Galaxy?&lt;/h2&gt;\n\n&lt;p&gt;&lt;a href=\&quot;https://galaxyproject.org/\&quot;&gt;Galaxy&lt;/a&gt; is an open data integration and analysis platform for the life sciences, and it is particularly well-suited for data analysis training in life science research.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-are-the-tutorials-for\&quot;&gt;What are the tutorials for?&lt;/h2&gt;\n\n&lt;p&gt;These tutorials can be used for learning and teaching how to use Galaxy for general data analysis, and for learning/teaching specific domains such as assembly and differential gene expression analysis with RNA-Seq data.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-audiences-are-the-tutorials-for\&quot;&gt;What audiences are the tutorials for?&lt;/h2&gt;\n\n&lt;p&gt;There are two distinct audiences for these materials.&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;&lt;strong&gt;Self-paced individual learners.&lt;/strong&gt; These tutorials provide everything you need to learn a topic, from explanations of concepts to detailed hands-on exercises.&lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Instructors.&lt;/strong&gt; They are also designed to be used by instructors in teaching/training settings. Slides, and detailed tutorials are provided. Most tutorials also include computational support with the needed tools, data as well as Docker images that can be used to scale the lessons up to many participants.&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h2 id=\&quot;how-is-the-content-licensed\&quot;&gt;How is the content licensed?&lt;/h2&gt;\n\n&lt;p&gt;The content of this website is licensed under the &lt;a href=\&quot;https://creativecommons.org/licenses/by/4.0/\&quot;&gt;Creative Commons Attribution 4.0 License&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-cite-this-project\&quot;&gt;How can I cite this project?&lt;/h2&gt;\n\n&lt;p&gt;We wrote an &lt;a href=\&quot;https://doi.org/10.1016/j.cels.2018.05.012\&quot;&gt;article&lt;/a&gt; about our efforts.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-advertise-the-training-materials-on-my-posters\&quot;&gt;How can I advertise the training materials on my posters?&lt;/h2&gt;\n\n&lt;p&gt;We provide some QR codes and logos in the &lt;a href=\&quot;https://github.com/vibbits/training-material/tree/master/assets/images\&quot;&gt;images folder&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;for-individual-learners\&quot;&gt;For Individual Learners&lt;/h1&gt;\n\n&lt;p&gt;Learning Galaxy and data analysis on your own and at your own pace?  This material is for you.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-do-i-start\&quot;&gt;Where do I start?&lt;/h2&gt;\n\n&lt;p&gt;If you are new to Galaxy then start with one of the introductory topics.  These introduce you to concepts that are useful in Galaxy, no matter what domain you are doing analysis in.&lt;/p&gt;\n\n&lt;p&gt;If you are already familiar with Galaxy basics and want to learn how to use it in a particular domain (for example, ChIP-Seq), then start with one of those topics.&lt;/p&gt;\n\n&lt;p&gt;If you are already well informed about bioinformatics data analysis and you just want to get a feel for how it works in Galaxy, then many tutorials include &lt;em&gt;Instructions for the impatient&lt;/em&gt; sections.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-do-i-use-this-material\&quot;&gt;How do I use this material?&lt;/h2&gt;\n\n&lt;p&gt;Many topics include slide decks and if the topic you are interested in has slides then start there.  These will introduce the topic and important concepts.&lt;/p&gt;\n\n&lt;p&gt;Most of your learning will happen in the next step - the hands-on tutorials.  This is where you’ll become familiar with using the Galaxy interface and experiment with different ways to use Galaxy and the tools in Galaxy.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-can-i-run-the-hands-on-tutorials\&quot;&gt;Where can I run the hands-on tutorials?&lt;/h2&gt;\n\n&lt;p&gt;To run the hands-on tutorials you need a Galaxy server to run them on.&lt;/p&gt;\n\n&lt;p&gt;Each tutorial is annotated with information about which &lt;a href=\&quot;https://galaxyproject.org/public-galaxy-servers/\&quot;&gt;public Galaxy servers&lt;/a&gt; it can be run on. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial.&lt;/p&gt;\n\n&lt;p&gt;Some topics have a &lt;a href=\&quot;https://www.docker.com/\&quot;&gt;Docker&lt;/a&gt; image that can be installed and run on participants’ laptops.  These Docker images contain Galaxy instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant. You will need to &lt;a href=\&quot;https://docs.docker.com/install/\&quot;&gt;install Docker&lt;/a&gt;.&lt;/p&gt;\n\n&lt;p&gt;Finally, you can also run your tutorials on cloud-based infrastructures.  Galaxy is &lt;a href=\&quot;https://galaxyproject.org/galaxy-services/\&quot;&gt;available on many national research infrastructures&lt;/a&gt; such as &lt;a href=\&quot;https://galaxyproject.org/cloud/jetstream/\&quot;&gt;Jetstream&lt;/a&gt; (United States), &lt;a href=\&quot;https://www.genap.ca/\&quot;&gt;GenAP&lt;/a&gt; (Canada), &lt;a href=\&quot;https://www.gvl.org.au/\&quot;&gt;GVL&lt;/a&gt; (Australia), &lt;a href=\&quot;http://www.climb.ac.uk/\&quot;&gt;CLIMB&lt;/a&gt; (United Kingdom), and more.  These instances are typically easy to launch, and easy to shut down when you are done.&lt;/p&gt;\n\n&lt;p&gt;If you are already familiar with, and have an account on &lt;a href=\&quot;https://aws.amazon.com/\&quot;&gt;Amazon Web Services&lt;/a&gt; then you can also launch a Galaxy server there using &lt;a href=\&quot;https://launch.usegalaxy.org/\&quot;&gt;CloudLaunch&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-get-help\&quot;&gt;How can I get help?&lt;/h2&gt;\n\n&lt;p&gt;If you have questions about this training material, you can reach us using the &lt;a href=\&quot;\&quot;&gt;Gitter chat&lt;/a&gt;.  You’ll need a &lt;a href=\&quot;https://github.com/\&quot;&gt;GitHub&lt;/a&gt; or &lt;a href=\&quot;https://twitter.com/\&quot;&gt;Twitter&lt;/a&gt; account to post questions.  If you have questions about Galaxy outside the context of training, see the &lt;a href=\&quot;https://galaxyproject.org/support/\&quot;&gt;Galaxy Support page&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;for-instructors\&quot;&gt;For Instructors&lt;/h1&gt;\n\n&lt;p&gt;This material can also be used to teach Galaxy and data analysis in a group setting to students and researchers.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-do-i-start-1\&quot;&gt;Where do I start?&lt;/h2&gt;\n\n&lt;p&gt;Spend some time exploring the different tutorials and the different resources that are available. Become familiar with the structure of the tutorials and think about how you might use them in your teaching.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-galaxy-instance-should-i-use-for-my-training\&quot;&gt;What Galaxy instance should I use for my training?&lt;/h2&gt;\n\n&lt;p&gt;To teach the hands-on tutorials you need a Galaxy server to run the examples on.&lt;/p&gt;\n\n&lt;p&gt;Each tutorial is annotated with the information on which &lt;a href=\&quot;https://galaxyproject.org/public-galaxy-servers/\&quot;&gt;public Galaxy servers&lt;/a&gt; it can be run. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial. If you choose this option then you should work with that server’s admins to confirm that think the server can handle the workload for a workshop. For example, the &lt;a href=\&quot;https://usegalaxy.eu/\&quot;&gt;usegalaxy.eu&lt;/a&gt;&lt;/p&gt;\n\n&lt;p&gt;If your organization/consortia/community has its own Galaxy server, then you may  want to run tutorials on that. This can be ideal because then the instance you are teaching on is the same you your participants will be using after the training. They’ll also be able to revisit any analysis they did during the training. If you pursue this option you’ll need to work with your organization’s Galaxy Admins to confirm that&lt;/p&gt;\n\n&lt;ul&gt;\n  &lt;li&gt;the server can support a room full of people all doing the same analysis at the same time.&lt;/li&gt;\n  &lt;li&gt;all participants will be able to create/use accounts on the system.&lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;Some training topics have a Docker image that can be installed and run on all participants’ laptops.  These images contain Galaxy instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant.&lt;/p&gt;\n\n&lt;p&gt;Finally, you can also run your tutorials on cloud-based infrastructures.  Galaxy is &lt;a href=\&quot;https://galaxyproject.org/galaxy-services/\&quot;&gt;available on many national research infrastructures&lt;/a&gt; such as &lt;a href=\&quot;https://galaxyproject.org/cloud/jetstream/\&quot;&gt;Jetstream&lt;/a&gt; (United States), &lt;a href=\&quot;https://www.genap.ca/\&quot;&gt;GenAP&lt;/a&gt; (Canada), &lt;a href=\&quot;https://launch.genome.edu.au/launch\&quot;&gt;GVL&lt;/a&gt; (Australia), &lt;a href=\&quot;http://www.climb.ac.uk/\&quot;&gt;CLIMB&lt;/a&gt; (United Kingdom), and more.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-do-i-get-help\&quot;&gt;How do I get help?&lt;/h2&gt;\n\n&lt;p&gt;The support channel for instructors is the same as for individual learners.  We suggest you start by posting a question to the Galaxy Training Network &lt;a href=\&quot;\&quot;&gt;Gitter chat&lt;/a&gt;. Anyone can view the discussion, but you’ll need to login (using your GitHub or Twitter account) to add to the discussion.&lt;/p&gt;\n\n&lt;p&gt;If you have questions about Galaxy in general (that are not training-centric) then there are &lt;a href=\&quot;https://galaxyproject.org/support/\&quot;&gt;several support options&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;contributing\&quot;&gt;Contributing&lt;/h1&gt;\n\n&lt;p&gt;First off, thanks for your interest in contributing to the Galaxy training materials!&lt;/p&gt;\n\n&lt;p&gt;Individual learners and instructors can make these training more effective by contributing back to them. You can report mistakes and errors, create more content, etc. Whatever is your background, there is a way to contribute: via the GitHub website, via command-line or even without dealing with GitHub.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-get-started-with-contributing\&quot;&gt;How can I get started with contributing?&lt;/h2&gt;\n\n&lt;p&gt;If you would like to get involved in the project but are unsure where to start, there are some easy ways to contribute which will also help you familiarize yourself with the project.&lt;/p&gt;\n\n&lt;h3 id=\&quot;1-checking-existing-tutorials\&quot;&gt;1. Checking existing tutorials&lt;/h3&gt;\n\n&lt;p&gt;A great way to help out the project is to test/edit existing tutorials. Pick a tutorial and check the contents. Does everything work as expected? Are there things that could be improved?&lt;/p&gt;\n\n&lt;p&gt;Below is a checklist of things to look out for to help you get started. If you feel confident in making changes yourself, please open a pull request, otherwise please file an issue with any problems you run into or suggestions for improvements.&lt;/p&gt;\n\n&lt;p&gt;&lt;em&gt;Basic&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Test&lt;/strong&gt; the tutorial on a running Galaxy instance\n    &lt;ul&gt;\n      &lt;li&gt;For example &lt;a href=\&quot;https://usegalaxy.org\&quot;&gt;Galaxy Main&lt;/a&gt;, &lt;a href=\&quot;https://usegalaxy.eu\&quot;&gt;Galaxy Europe&lt;/a&gt;, or &lt;a href=\&quot;https://usegalaxy.org.au\&quot;&gt;Galaxy Australia&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;Report any issues you run into&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Language&lt;/strong&gt; editing\n    &lt;ul&gt;\n      &lt;li&gt;Fix spelling and grammar mistakes&lt;/li&gt;\n      &lt;li&gt;Simplify the English (to make it more accessible)&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;&lt;em&gt;Intermediate&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Metadata&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Are the objectives, keypoints and time estimate filled in?&lt;/li&gt;\n      &lt;li&gt;Do they fit with the contents of the tutorial?&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Content&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Is there enough background information provided in the introduction section and throughout the tutorial?&lt;/li&gt;\n      &lt;li&gt;&lt;strong&gt;Question boxes&lt;/strong&gt;\n        &lt;ul&gt;\n          &lt;li&gt;Add questions or question boxes where you think they might be useful (make people think about results they got, test their understanding, etc)&lt;/li&gt;\n          &lt;li&gt;Check that answers are still up-to-date&lt;/li&gt;\n        &lt;/ul&gt;\n      &lt;/li&gt;\n      &lt;li&gt;&lt;strong&gt;Screenshots and Videos&lt;/strong&gt;\n        &lt;ul&gt;\n          &lt;li&gt;Make sure there is also a textual description of the image/video contents&lt;/li&gt;\n          &lt;li&gt;Does the screenshot add value to the tutorial or can it be removed?&lt;/li&gt;\n        &lt;/ul&gt;\n      &lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;&lt;em&gt;Advanced&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Workflows&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Add a workflow definition file &lt;code class=\&quot;highlighter-rouge\&quot;&gt;.ga&lt;/code&gt; if none is present&lt;/li&gt;\n      &lt;li&gt;Check that the existing workflow is up-to-date with the tutorial contents&lt;/li&gt;\n      &lt;li&gt;Enable &lt;a href=\&quot;https://github.com/usegalaxy-eu/workflow-testing\&quot;&gt;workflow testing&lt;/a&gt;&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Tours&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Add a tour if none exists&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-test-an-interactive-tour\&quot;&gt;Run the existing tour&lt;/a&gt; and check that it is up-to-date with the tutorial contents&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Datasets&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Check that all datasets used in the tutorial are present in Zenodo&lt;/li&gt;\n      &lt;li&gt;Add a data-library.yaml file if none exists&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;h2 id=\&quot;how-can-i-give-feedback\&quot;&gt;How can I give feedback?&lt;/h2&gt;\n\n&lt;p&gt;At the end of each tutorial, there is a link to &lt;a href=\&quot;https://docs.google.com/forms/d/1BILuBp8bPv42odpZ55Kcf2rqzR-k0Czx46agVVSRgm4/edit\&quot;&gt;a feedback form&lt;/a&gt;. We use this information to improve our tutorials.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-test-an-interactive-tour\&quot;&gt;How can I test an Interactive Tour?&lt;/h2&gt;\n\n&lt;p&gt;Perhaps you’ve been asked to review an interactive tour, or maybe you just want to try one out. The easiest way to run an interactive tour is to use the &lt;a href=\&quot;https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/\&quot;&gt;Tour builder&lt;/a&gt; browser extension.&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;Install the Tour Builder extension to your browser (&lt;a href=\&quot;https://chrome.google.com/webstore/detail/galaxy-tour-builder/mdfbapknmcpnbmggahhaegehbbbmhmgg\&quot;&gt;Chrome Web Store&lt;/a&gt;, &lt;a href=\&quot;https://addons.mozilla.org/en-US/firefox/addon/galaxy-tour-builder/\&quot;&gt;Firefox add-on&lt;/a&gt;, &lt;a href=\&quot;https://addons.opera.com/en/extensions/details/galaxy-tour-builder/\&quot;&gt;Opera add-on&lt;/a&gt;).&lt;/li&gt;\n  &lt;li&gt;Navigate to a Galaxy instance supporting the tutorial. To find which Galaxy instances support each tutorial, please see the dropdown menu next to the tutorial on the training website. Using one of the usegalaxy.* instances (&lt;a href=\&quot;https://usegalaxy.org\&quot;&gt;Galaxy Main&lt;/a&gt;, &lt;a href=\&quot;https://usegalaxy.eu\&quot;&gt;Galaxy Europe&lt;/a&gt;, or &lt;a href=\&quot;https://usegalaxy.org.au\&quot;&gt;Galaxy Australia&lt;/a&gt;\n) is usually a good bet.&lt;/li&gt;\n  &lt;li&gt;Start the Tour Builder plugin by clicking on the icon in your browser menu bar&lt;/li&gt;\n  &lt;li&gt;Copy the contents of the &lt;code class=\&quot;highlighter-rouge\&quot;&gt;tour.yaml&lt;/code&gt; file into the Tour builder editor window&lt;/li&gt;\n  &lt;li&gt;Click &lt;code class=\&quot;highlighter-rouge\&quot;&gt;Save&lt;/code&gt; and then &lt;code class=\&quot;highlighter-rouge\&quot;&gt;Run&lt;/code&gt;&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h2 id=\&quot;how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot;&gt;How can I fix mistakes or expand an existing tutorial using the GitHub interface?&lt;/h2&gt;\n\n&lt;p&gt;Check our tutorial to learn how to use the GitHub interface (soon…)&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-contribute-in-advanced-mode\&quot;&gt;How can I contribute in “advanced” mode?&lt;/h2&gt;\n\n&lt;p&gt;Most of the content is written in &lt;a href=\&quot;https://guides.github.com/features/mastering-markdown/\&quot;&gt;GitHub Flavored Markdown&lt;/a&gt; with some metadata (or variables) found in &lt;a href=\&quot;http://yaml.org/\&quot;&gt;YAML&lt;/a&gt; files. Everything is stored on our &lt;a href=\&quot;https://github.com/vibbits/training-material\&quot;&gt;GitHub repository&lt;/a&gt;. Each training material is related to a topic. All training materials (slides, tutorials, etc) related to a topic are found in a dedicated directory (&lt;em&gt;e.g.&lt;/em&gt; &lt;code class=\&quot;highlighter-rouge\&quot;&gt;transcriptomics&lt;/code&gt; directory contains the material related to transcriptomic analysis). Each topic has the following &lt;a name=\&quot;structure\&quot;&gt;structure&lt;/a&gt;:&lt;/p&gt;\n\n&lt;p&gt;&lt;img src=\&quot;shared/images/repo_organization.png\&quot; alt=\&quot;Structure of the repository\&quot; width=\&quot;400px\&quot; /&gt;&lt;/p&gt;\n\n&lt;ul&gt;\n  &lt;li&gt;a metadata file in YAML format&lt;/li&gt;\n  &lt;li&gt;a directory with the topic introduction slide deck in Markdown with introductions to the topic&lt;/li&gt;\n  &lt;li&gt;\n    &lt;p&gt;a directory with the tutorials:&lt;/p&gt;\n\n    &lt;p&gt;Inside the tutorials directory, each tutorial related to the topic has its own subdirectory with several files:&lt;/p&gt;\n    &lt;ul&gt;\n      &lt;li&gt;a tutorial file written in Markdown with hands-on&lt;/li&gt;\n      &lt;li&gt;an optional slides file in Markdown with slides to support the tutorial&lt;/li&gt;\n      &lt;li&gt;a directory with Galaxy Interactive Tours to reproduce the tutorial&lt;/li&gt;\n      &lt;li&gt;a directory with workflows extracted from the tutoria&lt;/li&gt;\n      &lt;li&gt;a YAML file with the links to the input data needed for the tutorial&lt;/li&gt;\n      &lt;li&gt;a YAML file with the description of needed tools to run the tutorial&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;a directory with the Dockerfile describing the details to build a container for the topic (self-study environments).&lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;To manage changes, we use &lt;a href=\&quot;https://guides.github.com/introduction/flow/\&quot;&gt;GitHub flow&lt;/a&gt; based on Pull Requests:&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;&lt;a href=\&quot;https://help.github.com/articles/fork-a-repo/\&quot;&gt;Create a fork&lt;/a&gt; of this repository on GitHub&lt;/li&gt;\n  &lt;li&gt;Clone your fork of this repository to create a local copy on your computer and initialize the required submodules (&lt;code class=\&quot;highlighter-rouge\&quot;&gt;git submodule init&lt;/code&gt; and &lt;code class=\&quot;highlighter-rouge\&quot;&gt;git submodule update&lt;/code&gt;)&lt;/li&gt;\n  &lt;li&gt;Create a new branch in your local copy for each significant change&lt;/li&gt;\n  &lt;li&gt;Commit the changes in that branch&lt;/li&gt;\n  &lt;li&gt;Push that branch to your fork on GitHub&lt;/li&gt;\n  &lt;li&gt;Submit a pull request from that branch to the &lt;a href=\&quot;https://github.com/vibbits/training-material\&quot;&gt;original repository&lt;/a&gt;&lt;/li&gt;\n  &lt;li&gt;If you receive feedback, make changes in your local clone and push them to your branch on GitHub: the pull request will update automatically&lt;/li&gt;\n  &lt;li&gt;Pull requests will be merged by the training team members after at least one other person has reviewed the Pull request and approved it.&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;p&gt;Globally, the process of development of new content is open and transparent:&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;Creation of a branch derived from the master branch of the GitHub repository&lt;/li&gt;\n  &lt;li&gt;Initialization of a new directory for the tutorial&lt;/li&gt;\n  &lt;li&gt;Filling of the metadata with title, questions, learning objectives, etc&lt;/li&gt;\n  &lt;li&gt;Generation of the input dataset for the tutorial&lt;/li&gt;\n  &lt;li&gt;Filling of the tutorial content&lt;/li&gt;\n  &lt;li&gt;Extraction of the workflows of the tutorial&lt;/li&gt;\n  &lt;li&gt;Automatic extraction of the required tools to populate the tool file&lt;/li&gt;\n  &lt;li&gt;Automatic annotation of the public Galaxy servers&lt;/li&gt;\n  &lt;li&gt;Generation of an interactive tour for the tutorial with the &lt;a href=\&quot;https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/\&quot;&gt;Tourbuilder&lt;/a&gt; web-browser extension&lt;/li&gt;\n  &lt;li&gt;Upload of the datasets to Zenodo and addition of the links in the data library file.&lt;/li&gt;\n  &lt;li&gt;Once ready, opening a Pull Request&lt;/li&gt;\n  &lt;li&gt;Automatic checks of the changes are automatically checked for the right format and working links using continuous integration testing on Travis CI&lt;/li&gt;\n  &lt;li&gt;Review of the content by several other instructors via discussions&lt;/li&gt;\n  &lt;li&gt;After the review process, merge of the content into the main branch, starting a series of automatic steps triggered by Travis CI&lt;/li&gt;\n  &lt;li&gt;Regeneration of the website and publication on &lt;a href=\&quot;/training-material/\&quot;&gt;https://vibbits.github.io//training-material/&lt;/a&gt;&lt;/li&gt;\n  &lt;li&gt;Generation of PDF artifacts of the tutorials and slides and upload on the FTP server&lt;/li&gt;\n  &lt;li&gt;Population of &lt;a href=\&quot;https://tess.elixir-europe.org/\&quot;&gt;TeSS&lt;/a&gt;, the ELIXIR’s Training Portal, via the metadata&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;p&gt;&lt;img src=\&quot;shared/images/development_process.png\&quot; alt=\&quot;Development process\&quot; /&gt;&lt;/p&gt;\n\n&lt;ol&gt;\n\n\n &lt;/ol&gt;\n\n&lt;p&gt;We also strongly recommend you read and follow &lt;a href=\&quot;https://carpentries.org/\&quot;&gt;The Carpentries&lt;/a&gt; recommendations on &lt;a href=\&quot;https://carpentries.github.io/lesson-example/01-design/\&quot;&gt;lesson design&lt;/a&gt; and &lt;a href=\&quot;https://carpentries.github.io/instructor-training/15-lesson-study/\&quot;&gt;lesson writing&lt;/a&gt; if you plan to add or change some training materials, and also to check the &lt;a href=\&quot;#structure\&quot;&gt;structure of the training material&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;other-questions\&quot;&gt;Other Questions&lt;/h1&gt;\n\n&lt;p&gt;Yes, always! Have a look at the &lt;a href=\&quot;https://galaxyproject.org/events/\&quot;&gt;Galaxy Community Events Calendar&lt;/a&gt; for what coming up right now.&lt;/p&gt;\n\n&lt;h2 id=\&quot;sustainability-of-the-training-material-and-metadata\&quot;&gt;Sustainability of the training-material and metadata&lt;/h2&gt;\n\n&lt;p&gt;This repository is hosted on &lt;a href=\&quot;https://github.com/\&quot;&gt;GitHub&lt;/a&gt; using git as a &lt;a href=\&quot;https://en.wikipedia.org/wiki/Distributed_version_control\&quot;&gt;DVCS&lt;/a&gt;. Therefore the community is hosting backups of this\nrepository in a decentralised way. The repository is self-contained and contains all needed content and all metadata.\nIn addition we mirror snapshops of this repo on &lt;a href=\&quot;http://zenodo.org\&quot;&gt;Zenodo&lt;/a&gt;.&lt;/p&gt;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;### How to fill the slide decks?\n\nPlease follow our\n[tutorial to learn how to fill the slides](/training-material/topics/contributing/tutorials/create-new-tutorial-slides/slides.html)\n&quot;,&quot;## Protein Structure Analysis ##\n\n- Sequences, structures and databases\n- Experimental methods (X-rays, electrons and NMR)\n- Finding and visualising structures from the  Protein Data Bank\n- Comparing structures\n- Modelling mutations\n- Creating homology models\n\n---\n\n## The Structure-Function Connection ##\n\n.left[![Left-aligned image](/training-material/topics/protein-structure-analysis/images/hemoglobin.png)]\n.right[Folded proteins provide a well-  defined 3D  arrangement of functional groups, creating microenvironments and active sites.\nStructural changes  are often involved in functional  mechanisms  (motor proteins, ...)\n]\n\n---\n\n### Databases\n\n.pull-left[\n**Uniprot** approx. 1100,000 sequences\n\n- mainly determined by large-scale DNA sequencing of individual genes or whole genomes\n- increasingly automated annotation\n\n**Protein Data Bank** approx. 141,000 \nexperimentally determined structures\n\n- Mainly determined by X-ray crystallography and high-resolution NMR spectroscopy\n- Automation is increasing, but there are still significant limitations in the rate of solving new structures\n\n\n]\n.pull-right[ ![](/training-material/assets/images/GTN-60px.png) ]\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### The Protein Databank ###\n\n.pull-left[ ![](/training-material/assets/images/GTN-60px.png) ] \n\n\n.pull-right[\n[http://www.wwpdb.org](http://www.wwpdb.org)\n- contains structures of  proteins, nucleic acids  and complexes,  determined by X-ray  crystallography, NMR  spectroscopy\n- No purely theoretical  or ab initio models  (since 2006)\n- Also stores supporting  experimental data\n- Full deposition now  required by all peer-reviewed journals\n\n---\n\n### Exercise 1: Search the PDB ###\n\n- Use the UniProt site to search for “dnak”.  \n- Use the PDB site to search for “dnak”.\n- Compare the UniProt and PDB result lists.\n- Use the sequence search function to see if there are structures with sequences similar to that of the  DnaK C-terminal domain.\n- Look at the summary pages of a number of  structures and note some interesting properties.\n- Select a number of structures and create a report  page.\n\n---\n\n### PDB File Organisation ###\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- Nucleic Acid Database: DNA and RNA structures\n\n[http://ndbserver.rutgers.edu/](http://ndbserver.rutgers.edu/)\n\n- PDB-REDO: automatically re-refined deposited  structures, using the latest methods\n\n[http://www.cmbi.ru.nl/pdb_redo/](http://www.cmbi.ru.nl/pdb_redo)\n\n- EBI: many online tools for structure analysis\n\n[http://www.ebi.ac.uk/Tools/structure/](http://www.ebi.ac.uk/Tools/structure/)\n\n- Replaced Electron Density Server: convenient overview of  quality parameters for crystal structures\n\n[http://www.ebi.ac.uk/pdbe/litemol](http://www.ebi.ac.uk/pdbe/litemol)\n\n---\n\n### High-Resolution NMR Spectrometry ###\n\nMany atomic nuclei, including the ubiquitous hydrogen nuclei,  resonate at specific radio frequencies when placed in a  strong, uniform magnetic field. The chemical environment of each individual atom slightly modulates its exact resonance  frequency.\n\n![]()\n\nIn macromolecules with thousands of atoms, many different  effects combine to generate an extremely complicated pattern of chemical shifts, which therefore more or less uniquely  identify each atom. Multidimensional spectra allow these  frequencies to be assigned to specific atoms.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nWhen two atoms are near each other in 3D space, they can exchange magnetization, giving rise to crosspeaks at the  intersection of their respective frequencies.\n\n![]()\n\nThis nuclear Overhauser effect (NOE) is used to effectively measure the distances between pairs of atoms, at least  qualitatively.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nAfter identification of the atoms by means of their unique chemical shifts, distance restraints are derived from the  Overhauser crosspeaks. An extended model of the protein  is  generated,  and  then  condensed  into  a  shape  that  is consistent with as many of distance restraints as possible.\n\n![]()\n\n---\n\n### Other methods ###\n\nElectron microscopy (and especially Cryo-electron microscopy): Electron crystallography and single particle reconstruction\r\n\nSmall-angle X-ray and neutron scattering (SAXS and SANS)\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- BioMagResBank: experimental data for NMR-  derived structures (lists of distance restraints  and other experimentally derived properties)\n\n[http://www.bmrb.wisc.edu/](http://www.bmrb.wisc.edu/)\n\n- BioIsis: database of SAXS-derived structures\n\n[http://www.bioisis.net](http://www.bioisis.net)\n\n- EMBL database of SAXS-derived structures\n\n[http://www.sasbdb.org](http://www.sasbdb.org)\n\n- EM Databank for cryo-EM structures\n\n[http://www.emdatabank.org](http://www.emdatabank.org)\n\n---\n\n### Assessing Structure Quality ###\n\nGeneral geometric properties (bond lengths and angles,  Ramachandran distribution, …): MolProbity  [Link](http://molprobity.biochem.duke.edu/)\n\nCrystal Structures\n- Diffraction data resolution and completeness (PDB)\n- Final Rcryst and Rfree factors (PDB)\n- Local correspondence to electron density (EDS)\n\nNMR Structures\n- Number of distance restraints and additional experimental  data sources (BMRB)\n- Local restraint density (on-line NMR constraint analyser)\n[link](http://molsim.sci.univr.it/bioinfo/tools/constraint/index.html)\n\nOther techniques\n- Difficult to generalise: carefully read and consider the  published protocol\n\n---\n\n### Molecular Graphics Software ###\n\n- [PyMOL](http://www.pymol.org/): high-quality output,  good examples on [Wiki](http://www.pymolwiki.org/)\n- [Chimera](http://www.cgl.ucsf.edu/chimera/): good  documentation on website\n- [VMD](http://www.ks.uiuc.edu/Research/vmd/):  excellent for the analysis of MD trajectories\n- [Yasara](http://www.yasara.org)  \n- [SwissPDBViewer](http://spdbv.vital-it.ch/)\n\n---\n\n### YASARA ###\n\nYasara View is freely available and provides  basic visualisation functions.\n\nYasara Model, Dynamics and Structure provide  additional functionality.\n\nAn add-on module for NMR structures is  available.\n\nThe program can be combined with the WHAT-  IF program for structure validation, and with  FoldX for energy calculations.\n\n---\n\n### Exercise 2: show a structure ###\n\nLoad PDB entry 1TRZ using the File menu.\n\nCompare the default representations (F1-F8)  and use the various controls to change the view  of the protein.\n\nExplore the Edit and View menus to change  various aspects of the graphical representation  of the structure.\n\nExamine the hydrogen bonding patterns.  Display a molecular surface.\n\nCreate an interesting view of the molecule and  save it as a high-resolution image.\n\n---\n\n### Structure Superposition ###\n\nStructures can be superimposed to achieve the best  possible match between corresponding atoms. It is  possible to consider all atoms, or only a subset  (such as the Cα atoms).\n\nWhen the structures are very similar, determining  which atoms should be matched to each other is  trivial. When there are larger differences, it takes  more preparation.\n\nDifferent algorithms use different combinations of  sequence- and secondary of tertiary structure-based  information.\n\n---\n\n### Exercise 3: Compare Structures ###\n\nDownload the five provided PDB files and open  them in Yasara.\n\nUse the `Analyze|Align|Objects` with MUSTANG  function to align the four last objects with the first one.\n\nUse the space bar to open the text console and  see the reported root mean square deviations\n\n\nas well as the number of residues matched.\n\nColor all structures by B factor and compare the  distribution to the local variability of the structures.\n\n---\n\n### PDB File Surprises ###\n\n- Missing atoms, residues and loops  \n- Multiple molecules in the asymmetric unit\n- Incomplete oligomers due to coincident crystal and  oligomer symmetry\n- Cα-only models  lternate conformations\n- Multiple models, especially for NMR ensembles\n- Use of B factors to represent other properties  ther non-standard extensions (PDBQT, ...)\n\n---\n\n### Force Fields ###\n\n- Energy terms representing physical interactions\n\nCovalent bond lengths and angles\nDihedral angles and van der Waals forces (steric effects)\nElectrostatic interactions and hydrogen bonds\n…\n\n- Energy can be minimized, and forces can literally be derived from the potential function.\n\n- Consistency and careful consideration of the  properties to be simulated are essential.\n\n---\n\n### Force Field Terms ###\n\nEach energy term has a functional form, which includes one or more parameters:\n\n- Covalent bond energy term\n To do: add formula\n- Van der Waals contact energy term\n\nThe parameters are collectively optimized to  reproduce a chosen set of experimentally  observed parameters.\n\nA given force field should be used as a  consistent system, and can only be used to  predict properties that are covered by the  training set.\n\n---\n\n### FoldX ###\n\n\ns designed for quantitative modelling of the  contribution of structural interactions to the stability of  proteins and protein complexes. It also supports  protein/DNA complexes.\n\nThe force field describes the different interactions in a  protein structure or complex in analytical terms. It has  been calibrated using a set of experimentally determined  stabilities.\n\nApplications include the optimisation of structures, the calculation of the stability of complexes, and predicting  the effect of amino-acid or base-pair mutations on these  properties.\n\n---\n\n### The FoldX Plugin for Yasara\t###\n\nIn order to make FoldX more accessible and  integrate its functions into Yasara, dr. Joost van  Durme (SWITCH laboratory) made a Yasara plugin  module that can apply FoldX functions to structures  that are loaded as Yasara objects.\nThis greatly simplifies the use of FoldX, and allows  for a quick visual analysis of the resulting changes  in the structures.\nMore information can be found at [wiki](http://foldxyasara.switchlab.org/index.php/) [FoldX](http://foldxsuite.crg.eu/)\n\n---\n\n### Exercise 4a: Repair a PDB File ###\n\n- Load the 1CRN PDB file.\n- Use the “Repair object” option in the Analysis |  FoldX menu to activate the corresponding FoldX  function.\n- Select the object to repair.\n\nThis exports the object as a temporary PDB file,  starts FoldX with the appropriate options, and loads  the repaired PDB file as a new object in Yasara.\n\n- Compare the original and repaired objects.  \n- Describe the changes that were introduced.\n\n---\n\n### Exercise 4b: Model a Mutation ###\n\n- Load the 2AC0.sce Yasara scene file.\n- Set an appropriate structure representation.\n- Locate residue Ala159 using the sequence view,  and right-click to access the `FoldX|Mutate` residue function. Change it to a Trp residue.\n- Look at the effect of the substitution on the  structure, and use the space bar to open the text  console and read the output of the FoldX  calculation.\n- Mutate Arg273 to an alanine side chain. Discuss  the effects of this substitution.\n\n---\n\n### Homology Modelling ###\n\n- When a structure is available for a protein with a  similar sequences, it is possible to predict the  structure of a new sequence with varying degrees of  confidence.\n- All homology modelling procedures start from an  alignment of the template and target sequences.  The quality of this alignment will have a major  impact on the resulting model.\n- Available applications include stand-alone programs  (Modeller, FoldX, …) and web-based services (such as SwissModel).\n\n---\n\n### Exercise 5: Make a Homology  Model using Swiss Model ###\n\nsee []()\n\n---\n\n### Exercise 6: Study Protein-Ligand Interactions ###\n\nsee []()\n\n---\n\n### On-Line References ###\n\n- Crystallography 101 (Bernhard Rupp):  (http://www.ruppweb.org/Xray/101index.html)\n- Protein NMR, A Practical Guide (Vicky Higman) ( http://www.protein-nmr.org.uk/)\n- Model validation course  (http://xray.bmc.uu.se/gerard/embo2001/modval/index.html)\n- Assessing model quality (http://spdbv.vital-it.ch/TheMolecularLevel/ModQual/)\n\n- Lectures by Burkhard Rost on protein structure  prediction (https://www.youtube.com/channel/UCU6j8BG4RbEtTgyIZJ6Vpow)\n\n---\n&quot;,&quot;$font-size: 15px;\n$tutorial-box-spacing: 1rem;\n\n// palette\n$white: #fff;\n$gray: #777;\n$light-gray: #ddd;\n$text-color: #292b2c;\n\n$red: #f00000;\n$orange: #f06d00;\n$yellow: #be9900;\n$green: #339e3a;\n$blue: #009090;\n$dark-blue: #1b2944;\n$light-blue: #5daadd;\n$purple: #882d60;\n$mid-gray: #9d9d9d;\n\n// colors general\n$bg-color: $dark-blue;\n$menu-color: $dark-blue;\n\n// colors tutorial boxes\n$overview-color: #8A9AD0;\n$agenda-color: #86D486;\n$keypoints-color: #FFA1A1;\n$tip-color: #FFE19E;\n$warning-color: #de8875;\n$comment-color: #ffecc1;\n$handson-color: #dfe5f9;\n$question-color: #8A9AD0;\n$solution-color: #B8C3EA;\n$details-color: $light-gray;\n$feedback-color: #86D486;\n\n\n@mixin matrix-table (){\n  table {\n      width:unset;\n      margin-left:auto;\n      margin-right:auto;\n  }\n\n  table thead th {\n      border-bottom: 2px solid #777;\n      text-align: left;\n  }\n\n  table td:first-child, table th:first-child  {\n      border-right: 2px solid #777;\n      font-weight: bold;\n      text-align: left;\n  }\n\n  table td:not(first-child), table th:not(first-child) {\n      text-align: center;\n      border-right: 1px solid #ddd;\n      border-left: 1px solid #ddd;\n  }\n}\n\n@mixin tutorial-box ($bg-color, $color: rgba(255,255,255,.75)) {\n    margin-top: 3 * $tutorial-box-spacing;\n    padding: $tutorial-box-spacing $tutorial-box-spacing - 0.11rem $tutorial-box-spacing;\n    border: 1px solid $bg-color;\n    box-shadow: 3px 3px $bg-color;\n    &amp; &gt; h3 {\n        font-size: $font-size + 3;\n        margin-left: -$tutorial-box-spacing;\n        margin-right: -$tutorial-box-spacing;\n        margin-top: -3.90rem;\n        padding-top: $tutorial-box-spacing / 2;\n        padding-bottom: $tutorial-box-spacing / 2;\n        padding-left: $tutorial-box-spacing;\n        padding-right: $tutorial-box-spacing;\n        background-color: $bg-color;\n        color: $color;\n        vertical-align: 1em;\n        float: left;\n        border-top-left-radius: 8px;\n        border-top-right-radius: 8px;\n\n        .fa {\n            padding-right: $tutorial-box-spacing / 5;\n        }\n    }\n}\n\n@mixin matrix-table (){\n  table {\n      width:unset;\n      margin-left:auto;\n      margin-right:auto;\n      margin-top: 20pt;\n  }\n\n  table thead th {\n      border-bottom: 2px solid #777;\n      text-align: left;\n  }\n\n  table td:first-child, table th:first-child  {\n      border-right: 2px solid #777;\n      font-weight: bold;\n      text-align: left;\n  }\n\n  table td:not(first-child), table th:not(first-child) {\n      text-align: center;\n      border-right: 1px solid #ddd;\n      border-left: 1px solid #ddd;\n  }\n}\n\n\nbody {\n    font-size: $font-size;\n    word-wrap: break-word;\n    position: relative;\n}\n\nh1,\nh2,\nh3,\nh4 {\n    margin-top: 2rem;\n}\n\nimg {\n    max-width: 95%;\n    margin: 2.5%;\n}\n\npre {\n    padding: 1rem;\n    background-color: lighten($gray, 40%);\n}\n\nfigure {\n  text-align: center;\n  margin: 1rem 2rem;\n\n  &amp; &gt; img {\n      margin-bottom: 1rem;\n  }\n\n  .figcaption-prefix {\n      font-weight: 600;\n  }\n\n  figcaption {\n    text-align: justify;\n  }\n}\n\nfooter {\n    margin-top: 3em;\n    text-align: center;\n    color: $gray;\n    font-size: $font-size - 2;\n}\n\n.follow-up {\n    p {\n        font-size: 18px;\n    }\n}\n\n.navbar {\n    background-color: $bg-color!important;\n\n    .navbar-collapse {\n        justify-content: flex-end;\n        align-items: flex-end;\n        min-width: 0;\n    }\n}\n\n.main-content {\n    padding-top: 1rem;\n}\n\n.table {\n    td {\n        vertical-align: middle;\n    }\n\n    .fa {\n        font-size: $font-size + 5;\n    }\n\n}\n\n.topic-icon {\n    padding: .5rem .75rem;\n}\n\n.training-network-map {\n    width: 90%;\n    margin: 0 auto;\n}\n\n.tutorial {\n    &amp; &gt; h1:first-child {\n        margin-bottom: 1rem;\n    }\n\n    table {\n        display: block;\n        width: 100%;\n        overflow-x: auto;\n        max-width: 100%;\n        margin-bottom: 1rem;\n\n        th,\n        td {\n            padding: .75rem;\n            vertical-align: top;\n            border-top: 1px solid #eceeef;\n        }\n\n        thead th {\n            vertical-align: bottom;\n            border-bottom: 2px solid #eceeef;\n        }\n    }\n\n    // tutorial boxes\n    blockquote {\n        ul,\n        ol,\n        &amp; &gt; blockquote,\n        p {\n            margin-bottom: 0px;\n        }\n\n        &amp;.agenda {\n            @include tutorial-box($agenda-color, $text-color);\n        }\n\n        &amp;.tip {\n            @include tutorial-box($tip-color, $text-color);\n        }\n\n        &amp;.question {\n            @include tutorial-box($question-color, $text-color);\n        }\n\n        &amp;.solution {\n            @include tutorial-box($solution-color, $text-color);\n        }\n\n        &amp;.comment {\n            @include tutorial-box($comment-color, $text-color);\n        }\n\n        &amp;.hands_on {\n            @include tutorial-box($handson-color, $text-color);\n        }\n\n        &amp;.key_points {\n            @include tutorial-box($keypoints-color, $text-color);\n        }\n\n        &amp;.overview {\n            @include tutorial-box($overview-color, $text-color);\n        }\n\n        &amp;.warning {\n            @include tutorial-box($warning-color, $text-color);\n        }\n\n        &amp;.details {\n            @include tutorial-box($details-color, $text-color);\n        }\n\n        &amp;.quote {\n            border-left: 5px solid #eee;\n            padding-left: 1em;\n        }\n\n        &amp;.matrix {\n            @include matrix-table();\n        }\n\n        &amp;.feedback{\n            @include tutorial-box($feedback-color, $text-color);\n        }\n    }\n\n    .fold-unfold {\n        margin-left: $tutorial-box-spacing;\n    }\n\n    .google-form {\n        width: 100%;\n        height: 1100px;\n        border: 0;\n    }\n}\n\n.contributors {\n    padding: 0;\n\n    .carousel-item {\n        align-items: flex-start;\n        flex-wrap: wrap;\n        flex-direction: row;\n        justify-content: space-around;\n    }\n\n    .carousel-item.active,\n    .carousel-item-next,\n    .carousel-item-prev {\n        display: flex;\n    }\n\n    .hall-of-fame-hero {\n        margin: 0;\n        width: 20%;\n\n        .name {\n            font-size: 0.7em;\n        }\n    }\n}\n\n.contributors-line {\n    color: $gray;\n    font-size: 1.2em;\n    margin-bottom: 5rem;\n}\n\n.hall-of-fame {\n    display: flex;\n    flex-wrap: wrap;\n    flex-direction: row;\n    justify-content: space-around;\n}\n\n.hall-of-fame-hero {\n    width: 140px;\n    margin: 1rem;\n    text-align: center;\n\n    .name {\n        margin-top: 0.5rem;\n    }\n\n    .contact-items {\n        margin-top: 0.2rem;\n    }\n\n    .thumbnail img {\n        background-color: white;\n        border-radius: 20px;\n        border: 1px solid $light-gray;\n        padding: 5px;\n        width: 100%;\n        -webkit-transition: border 0.2s ease-in-out;\n        -o-transition: border 0.2s ease-in-out;\n        transition: border 0.2s ease-in-out;\n    }\n\n    a.thumbnail {\n        &amp;:hover,\n        &amp;:focus,\n        &amp;:active {\n            img {\n                border-color: $menu-color;\n            }\n        }\n    }\n}\n\n// This is a fix to get the hamburger button (menu/navbar) correctly placed on\n// small screens.\n@media (max-width: 575px) {\n    .navbar &gt; .container {\n        position: initial;\n        margin-left: initial;\n        margin-right: initial;\n    }\n}\n\n// This applies a \&quot;dark\&quot; style to the dropdown menus on small devices since\n// Bootstrap v4 does not do that by default.\n@media (max-width: 768px) {\n    .navbar {\n        .dropdown-menu {\n            background-color: transparent;\n            border-color: transparent;\n        }\n\n        .dropdown-item {\n            color: rgba($white, .5);\n\n            &amp;:hover,\n            &amp;:focus,\n            &amp;:active {\n                background-color: transparent;\n                color: rgba($white, .75);\n            }\n        }\n\n        .nav-item,\n        .dropdown-item {\n            overflow: hidden;\n            white-space: nowrap;\n            text-overflow: ellipsis;\n        }\n    }\n}\n\n@media (min-width: 768px) {\n    .tutorial table,\n    .table-responsive {\n        display: table;\n    }\n\n    .citation {\n        text-align: justify;\n    }\n}\n\n@media (max-width: 1200px) {\n    .navbar-nav i {\n        display: none;\n    }\n}\n\n.label {\n    display: inline;\n    padding: .2em .6em .3em;\n    font-size: 75%;\n    font-weight: bold;\n    line-height: 1;\n    color: #fff;\n    text-align: center;\n    white-space: nowrap;\n    vertical-align: baseline;\n    border-radius: .25em;\n}\n\n.label-default {\n    background-color: #999999;\n}\n\n.level {\n    color: lightgray;\n    margin-right: 0.5em;\n\n    .fa {\n        font-size: 0.9em;\n    }\n    &amp;.introductory i:first-of-type {\n        color: green;\n    }\n    &amp;.intermediate i:not(:last-of-type) {\n        color: orange;\n    }\n    &amp;.advanced {\n        color: red;\n    }\n}\n\n\n\n.visually-hidden {\n    border: 0;\n    clip: rect(0 0 0 0);\n    height: 1px;\n    margin: -1px;\n    overflow: hidden;\n    padding: 0;\n    position: absolute;\n    width: 1px;\n}\n\n\n/* WCAG Contrast. These aim to pass AA level for normal sized text. */\na {\n    color: #0070e8;\n\n    &amp;:hover,\n    &amp;:focus,\n    &amp;:active {\n        color: #004590;\n    }\n}\n\ncode {\n    color: #8b3035;\n}\n\n\nnav[data-toggle=&#39;toc&#39;] {\n    margin-top: 30px;\n    margin-bottom: 30px;\n    .nav {\n        li {\n            a {\n                padding-left: 0px;\n                &amp;:hover,\n                &amp;:focus,\n                &amp;:active{\n                    padding-left: 0px;\n                    color: #004590;\n                    border-left: none;\n                }\n            }\n\n            .active {\n                padding-left: 0px;\n                color: #004590;\n                border-left: none;\n            }\n        }\n\n        .nav li {\n            a,\n            .active {\n                padding-left: 10px;\n                &amp;:hover,\n                &amp;:focus,\n                &amp;:active{\n                    padding-left: 10px;\n                }\n            }\n\n            &amp;:hover,\n            &amp;:focus,\n            &amp;:active{\n                padding-left: 10px;\n            }\n        }\n    }\n\n    .nav-link.active:hover {\n        padding-left: 0px;\n        color: #004590;\n        border-left: none;\n    }\n}\n\n.col-sm-2 {\n    padding-left: 0px;\n}\n\n/* small screens */\n@media screen and (max-width: 768px) {\n    nav[data-toggle=&#39;toc&#39;] {\n        display: none !important;\n    }\n\n    .col-sm-10 {\n        max-width: 100%;\n    }\n}\n\n.contributor-badge {\n    /* prevent breaking across lines */\n    white-space: nowrap;\n\n    img {\n        height: 1.25em;\n        border-radius: 50%;\n        margin: 0.25em;\n    }\n}\n\n\ndiv.highlight {\n\tposition: relative;\n}\ndiv.highlight .btn{\n\t-webkit-transition:opacity .3s ease-in-out;\n\t-o-transition:opacity .3s ease-in-out;\n\ttransition:opacity .3s ease-in-out;\n\topacity:0;\n\tpadding:2px 6px;\n\tposition:absolute;\n\tright:4px;\n\ttop:4px;\n\tfont-family: -apple-system,BlinkMacSystemFont,\&quot;Segoe UI\&quot;,Roboto,\&quot;Helvetica Neue\&quot;,Arial,sans-serif;\n}\n\ndiv.highlight:hover .btn,div.highlight .btn:focus{\n\topacity:1\n}\n\n#supporting_materials ul li {\n    div {\n        display: flex;\n        flex-direction: row;\n        align-items: center;\n\n        div {\n            a.topic-icon {\n                padding: 0 .5rem 0 .5rem;\n            }\n\n            a.topic-icon.btn {\n                padding-top: 0;\n                padding-bottom: 0;\n            }\n        }\n\n        div:first-child {\n            a.topic-icon {\n                padding-left: 0;\n            }\n        }\n\n        div:not(:first-child):before {\n            content: \&quot; - \&quot;;\n        }\n    }\n}\n&quot;,&quot;.enlarge120[\n\n# ***De novo* Genome Assembly**\n\n]\n\n#### With thanks to T Seemann, D Bulach, I Cooke and Simon Gladman\n---\n.enlarge120[\n\n# ***De novo* assembly**\n\n]\n\n.pull-left[\n\n**The process of reconstructing the original DNA sequence from the fragment reads alone.**\n\n* Instinctively like a jigsaw puzzle\n\n  * Find reads which \&quot;fit together\&quot; (overlap)\n  * Could be missing pieces (sequencing bias)\n  * Some pieces will be dirty (sequencing errors)\n\n]\n\n.pull-right[ ![](../../images/Humpty.jpg) ]\n\n---\n\n# **Another View**\n\n![](../../images/newspaper.png)\n\n---\n\n# **Assembly: An Example**\n\n---\n\n# **A small \&quot;genome\&quot;**\n\n![](../../images/shakespear1.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear2.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear3.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear4.png)\n\n---\n\n# **So far, so good!**\n\n---\n\n# **The Awful Truth**\n\n![](../../images/notsimply.png)\n\n## \&quot;Genome assembly is impossible.\&quot; - A/Prof. Mihai Pop\n\n---\n.enlarge120[\n\n# **Why is it so hard?**\n\n]\n\n.pull-left[\n* Millions of pieces\n  * Much, much shorter than the genome\n  * Lots of them look similar\n* Missing pieces\n  * Some parts can&#39;t be sequenced easily\n* Dirty Pieces\n  * Lots of errors in reads\n]\n\n.pull-right[ ![](../../images/worlds_hardest.png) ]\n\n---\n\n# **Assembly recipe**\n\n* Find all overlaps between reads\n  * Hmm, sounds like a lot of work..\n* Build a graph\n  * A picture of the read connections\n* Simplify the graph\n  * Sequencing errors will mess it up a lot\n* Traverse the graph\n  * Trace a sensible path to produce a consensus\n\n---\n\n![](../../images/olc_pic.png)\n\n---\n\n# **A more realistic graph**\n\n![](../../images/real_graph.png)\n\n---\n\n# .image-15[![](../../images/nofun.png)] **What ruins the graph?**\n\n* Read errors\n  * Introduces false edges and nodes\n\n* Non haploid organisms\n  * Heterozygosity causes lots of detours\n\n* Repeats\n  * If they are longer than the read length\n  * Causes nodes to be shared, locality confusion.\n\n---\n\n# **Repeats**\n\n---\n.enlarge120[\n# **What is a repeat?**\n]\n\n.pull-left[\n\n#### ***A segment of DNA which occurs more than once in the genome sequence***\n\n* Very common\n  * Transposons (self replicating genes)\n  * Satellites (repetitive adjacent patterns)\n  * Gene duplications (paralogs)\n\n]\n\n.pull-right[\n\n![](../../images/triplets.png)\n\n]\n\n---\n\n# **Effect on Assembly**\n\n![](../../images/repeat_effect.png)\n\n---\n.enlarge120[\n# **The law of repeats** .image-15[![](../../images/repeatafterme.png)]\n]\n\n## **It is impossible to resolve repeats of length S unless you have reads longer than S**\n\n## **It is impossible to resolve repeats of length S unless you have reads longer than S**\n\n---\n\n# **Scaffolding**\n\n---\n.enlarge120[\n# **Beyond contigs**\n]\n\n.pull-left[\n\nContig sizes are limited by:\n\n* the length of the repeats in your genome\n  * Can&#39;t change this\n\n\n* the length (or \&quot;span\&quot;) of the reads\n  * Use long read technology\n  * Use tricks with other technology\n\n]\n\n---\n.enlarge120[\n# **Types of reads**\n]\n\n.pull-left[.enlarge120[**Example fragment**]]\n\n\n.remark-code[.enlarge120[atcgtatgatcttgagattctctcttcccttatagctgctata]]\n\n.pull-left[.enlarge120[**\&quot;Single-end\&quot; read**]]\n\n\n.remark-code[.enlarge120[**atcgtatg**atcttgagattctctcttcccttatagctgctata]]\n\nsequence *one* end of the fragment\n\n.pull-left[.enlarge120[**\&quot;Paired-end\&quot; read**]]\n\n\n.remark-code[.enlarge120[**atcgtatg**atcttgagattctctcttcccttatag**ctgctata**]]\n\nsequence both ends of the same fragment\n\n**We can exploit this information!**\n---\n\n.enlarge120[# **Scaffolding**]\n\n* **Paired end reads**\n  * Known sequences at each end of fragment\n  * Roughly known fragment length\n\n* **Most ends will occur in same contig**\n\n* **Some will occur in different contigs**\n  * ***evidence that these contigs are linked***\n---\n\n.enlarge120[# **Contigs to Scaffolds**]\n\n![](../../images/scaffolding.png)\n\n---\n\n.enlarge120[# **Assessing assemblies**]\n\n* We desire\n  * Total length similar to genome size\n  * Fewer, larger contigs\n  * Correct contigs\n\n* Metrics\n  * No generally useful measure. (No real prior information)\n  * Longest contigs, total base pairs in contigs, **N50**, ...\n\n---\n\n.enlarge120[# **The \&quot;N50\&quot;**]\n\n.enlarge120[***The length of that contig from which 50% of the bases are in it and shorter contigs***]\n\n* Imagine we have 7 contigs with lengths:\n  * 1, 1, 3, 5, 8, 12, 20\n\n* Total\n  * 1+1+3+5+8+12+20 = 50\n\n* N50 is the \&quot;halfway sum\&quot; = 25\n  * 1+1+3+5+8+**12** = 30 (&gt;25) so **N50 is 12**\n\n---\n\n.enlarge120[# **2 levels of assembly**]\n\n* Draft assembly\n  * Will contain a number of non-linked scaffolds with gaps of unknown sequence\n  * Fairly easy to get to\n\n* Closed (finished) assembly\n  * One sequence for each chromosome\n  * Takes a **lot** more work\n  * Small genomes are becoming easier with long read tech\n  * Large genomes are the province of big consortia (e.g. Human Genome Consortium)\n\n---\n.enlarge120[# **How do I do it?**]\n---\n.enlarge120[\n# **Example**\n\n* Culture your bacterium\n\n\n* Extract your genomic DNA\n\n\n* Send it to your sequencing centre for Illumina sequencing\n  * 250bp paired end\n\n\n* Get back 2 files\n  * .remark-code[MRSA_R1.fastq.gz]\n  * .remark-code[MRSA_R2.fastq.gz]\n\n\n* ***Now what?***\n]\n\n---\n.enlarge120[# **Assembly tools**\n\n* **Genome**\n  * **Velvet, Velvet Optimizer, Spades,** Abyss, MIRA, Newbler, SGA, AllPaths, Ray, SOAPdenovo, ...\n\n\n* Meta-genome\n  * Meta Velvet, SGA, custom scripts + above\n\n\n* Transcriptome\n  * Trinity, Oases, Trans-abyss\n\n***And many, many others...***\n\n]\n\n---\n.enlarge120[\n# **Assembly Exercise #1**\n\n* We will do a simple assembly using **Velvet** in **Galaxy**\n* We can do a number of different assemblies and compare some assembly metrics.\n\n]\n&quot;,&quot;# Genome assembly with Velvet: Background\n{:.no_toc}\n\nVelvet is one of a number of *de novo* assemblers that use short read sets as input (*e.g.* Illumina Reads). The assembly method is based on the manipulation of de Bruijn graphs, via the removal of errors and the simplication of repeated regions.\n\n&gt; ### {% icon comment %} Comment\n&gt;\n&gt; For information about Velvet, you can check its (nice) [Wikipedia page](https://en.wikipedia.org/wiki/Velvet_assembler).\n{: .comment}\n\nFor this tutorial, we have a set of reads from an imaginary *Staphylococcus aureus* bacterium with a miniature genome (197,394 bp). Our mutant strain read set was sequenced with the whole genome shotgun method, using an Illumina DNA sequencing instrument. From these reads, we would like to rebuild our imaginary *Staphylococcus aureus* bacterium via a *de novo* assembly of a short read set using the Velvet assembler.\n\n&gt; ### Agenda\n&gt;\n&gt; In this tutorial, we will deal with:\n&gt;\n&gt; 1. TOC\n&gt; {:toc}\n&gt;\n{: .agenda}\n\n# Get the data\n\nWe will now import the data that we will use for the tutorial.\n\n&gt; ### {% icon hands_on %} Hands-on: Getting the data\n&gt;\n&gt; 1. Create and name a new history for this tutorial.\n&gt; 2. Import from [Zenodo](https://doi.org/10.5281/zenodo.582600) or from the data library the files:\n&gt;    - [`mutant_R1.fastq`](https://zenodo.org/record/582600/files/mutant_R1.fastq)\n&gt;    - [`mutant_R2.fastq`](https://zenodo.org/record/582600/files/mutant_R2.fastq)\n&gt;\n&gt;    &gt; ### {% icon tip %} Tip: Importing data via links\n&gt;    &gt;\n&gt;    &gt; * Copy the link location (Right-click on the filename then \&quot;Copy Link Address\&quot;)\n&gt;    &gt; * Open the Galaxy Upload Manager\n&gt;    &gt; * Select **Paste/Fetch Data**\n&gt;    &gt; * Paste the link into the text field\n&gt;    &gt; * Change the data-type to **fastqsanger**\n&gt;    &gt; * Press **Start**\n&gt;    {: .tip}\n&gt;\n&gt; 3. Change the name of the files to `mutant_R1` and `mutant_R2`.\n&gt;\n&gt;    As a default, Galaxy uses the link as the name of the new dataset. It also does not link the dataset to a database or a reference genome.\n&gt;\n&gt;    {% include snippets/rename_dataset.md %}\n&gt;\n&gt; 4. Inspect the content of a file.\n&gt;\n&gt;    &gt; ### {% icon tip %} Tip: Inspecting the content of a dataset\n&gt;    &gt;\n&gt;    &gt; * Click on the {% icon galaxy-eye %} (eye) icon next to the relevant history entry\n&gt;    &gt; * View the content of the file in the central panel\n&gt;    {: .tip}\n&gt;\n&gt;    &gt; ### {% icon question %} Questions\n&gt;    &gt;\n&gt;    &gt; 1. What are four key features of a FASTQ file?\n&gt;    &gt; 2. What is the main difference between a FASTQ and a FASTA file?\n&gt;    &gt;\n&gt;    &gt; &gt; ### {% icon solution %} Solution\n&gt;    &gt; &gt; 1. Each sequence in a FASTQ file is represented by 4 lines: 1st line is the id, 2nd line is the sequence, 3rd line is not used, and 4th line is the quality of sequencing per nucleotide\n&gt;    &gt; &gt; 2. In a FASTQ file, not only are the sequences present, but information about the quality of sequencing is also included.\n&gt;    &gt; {: .solution }\n&gt;    {: .question}\n&gt;\n{: .hands_on}\n\nThe reads have been sequenced from an imaginary *Staphylococcus aureus* bacterium using an Illumina DNA sequencing instrument. We obtained the 2 files we imported (`mutant_R1` and `mutant_R2`)\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; Why do we have 2 files here if we only sequenced the bacteria once?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. The bacteria has been sequenced using paired-end sequencing. The first file corresponds to forward reads and the second file to reverse reads.\n&gt; {: .solution }\n{: .question}\n\n# Evaluate the input reads\n\nBefore doing any assembly, the first questions you should ask about your input reads include:\n\n- What is the coverage of my genome?\n- How good is my read set?\n- Do I need to ask for a new sequencing run?\n- Is it suitable for the analysis I need to do?\n\nWe will evaluate the input reads using the FastQC tool. This tool runs a standard series of tests on your read set and returns a relatively easy-to-interpret report. We will use it to evaluate the quality of our FASTQ files and combine the results with MultiQC.\n\n&gt; ### {% icon hands_on %} Hands-on: FastQC on a fastq file\n&gt;\n&gt; 1. **FastQC** {% icon tool %} with the following parameters\n&gt;    - \&quot;Short read data from your current history\&quot; to (**Multiple datasets**) `mutant_R1.fastq` and `mutant_R2.fastq`\n&gt;\n&gt; 2. **MultiQC** {% icon tool %} with the following parameters\n&gt;    - \&quot;Software name\&quot; to `FastQC`\n&gt;    - \&quot;Result file\&quot; to the raw data files generated by FastQC\n&gt;\n{: .hands_on}\n\nMultiQC generates a webpage combining reports for FastQC on both datasets. It includes these graphs and tables:\n\n- General statistics\n\n    This is important in setting maximum k-mer size for an assembly.\n\n    &gt; ### {% icon comment %} Getting the length of sequences\n    &gt;\n    &gt; * Click on **Configure Columns**\n    &gt; * Check **Length**\n    &gt; * Close the window\n    {: .comment}\n\n    &gt; ### {% icon question %} Questions\n    &gt;\n    &gt; 1. How long are the sequences?\n    &gt; 2. What is the average coverage of the genome, given our imaginary *Staphylococcus aureus* bacterium has a genome of 197,394 bp?\n    &gt;\n    &gt; &gt; ### {% icon solution %} Solution\n    &gt; &gt; 1. The sequences are 150 bp long\n    &gt; &gt; 2. We have 2 x 12,480 sequences of 150 bp, so the average genome coverage is: 2 * 12480 * 150 / 197394, or approximately 19 X coverage.\n    &gt; {: .solution }\n    {: .question}\n\n- Sequence Quality Histograms\n\n    Dips in quality near the beginning, middle or end of the reads may determine the trimming/cleanup methods and parameters to be used, or may indicate technical problems with the sequencing process/machine run.\n\n    &lt;figure id=\&quot;figure-1\&quot;&gt;&lt;img src=\&quot;../../images/fastqc_per_base_sequence_quality_plot.png\&quot; alt=\&quot;Sequence Quality Histograms with the mean quality value across each base position in the read\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 1:&lt;/span&gt; The mean quality value across each base position in the read&lt;/figcaption&gt;&lt;/figure&gt;\n\n    &gt; ### {% icon question %} Questions\n    &gt;\n    &gt; 1. What does the y-axis represent?\n    &gt; 2. Why is the quality score decreasing across the length of the reads?\n    &gt;\n    &gt; &gt; ### {% icon solution %} Solution\n    &gt; &gt; 1. The y-axis represents the quality score for each base (an estimate of the error during sequencing).\n    &gt; &gt; 2. The quality score is decreasing accross the length of the reads because the sequencing become less and less reliable at the end of the reads.\n    &gt; {: .solution }\n    {: .question}\n\n- Per Sequence GC Content\n\n    High GC organisms tend not to assemble well and may have an uneven read coverage distribution.\n\n- Per Base N Content\n\n    The presence of large numbers of Ns in reads may point to a poor quality sequencing run. You will need to trim these reads to remove Ns.\n\n- k-mer content\n\n    The presence of highly recurring k-mers may point to contamination of reads with barcodes or adapter sequences.\n\n\n&gt; ### {% icon comment %} Comment\n&gt;\n&gt; For a fuller discussion of FastQC outputs and warnings, see the [FastQC website link](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/), including the section on each of the output [reports](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/), and examples of [\&quot;good\&quot;](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html) and [\&quot;bad\&quot;](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html) Illumina data.\n{: .comment}\n\nWe won&#39;t be doing anything to these data to clean it up as there isn&#39;t much need. Therefore we will get on with the assembly!\n\n\n# Assemble reads with Velvet\n\nNow, we want to assemble our reads to find the sequence of our imaginary *Staphylococcus aureus* bacterium. We will perform a *de novo* assembly of the reads into long contiguous sequences using the Velvet short read assembler.\n\nThe first step of the assembler is to build a de Bruijn graph. For that, it will break our reads into k-mers, *i.e.* fragments of length *k*. Velvet requires the user to input a value of *k* (k-mer size) for the assembly process. Small k-mers will give greater connectivity, but large k-mers will give better specificity.\n\n&gt; ### {% icon hands_on %} Hands-on: Assemble the reads\n&gt;\n&gt; 1. **FASTQ interlacer** {% icon tool %} with the following parameters\n&gt;    - \&quot;Type of paired-end datasets\&quot; to `2 separate datasets`\n&gt;    - \&quot;Left-hand mates\&quot; to `mutant_R1.fastq`\n&gt;    - \&quot;Right-hand mates\&quot; to `mutant_R2.fastq`\n&gt;\n&gt;    Currently our paired-end reads are in 2 files (one with the forward reads and one with the reverse reads), but Velvet requires only one file, where each read is next to its mate read. In other words, if the reads are indexed from 0, then reads 0 and 1 are paired, 2 and 3, 4 and 5, etc. Before doing the assembly *per se*, we need to prepare the files by combining them.\n&gt;\n&gt; 2. **velveth** {% icon tool %} with the following parameters\n&gt;    - \&quot;Hash Length\&quot; to `29`\n&gt;    - \&quot;Input Files\&quot;: click on `Insert Input Files`\n&gt;    - \&quot;file format\&quot; to `fastq`\n&gt;    - \&quot;read type\&quot; to `shortPaired reads`\n&gt;    - \&quot;Dataset\&quot; to the pairs output of **FASTQ interlacer**\n&gt;\n&gt;    The tool takes our reads and break them into k-mers.\n&gt;\n&gt; 3. **velvetg** {% icon tool %} with the following parameters\n&gt;    - \&quot;Velvet Dataset\&quot; to the output of **velveth**\n&gt;    - \&quot;Using Paired Reads\&quot; to `Yes`\n&gt;\n&gt;    This last tool actually does the assembly.\n{: .hands_on}\n\nTwo files are generated:\n\n- A \&quot;Contigs\&quot; file\n\n    This file contains the sequences of the contigs longer than 2k. In the header of each contig, a bit of information is added:\n    - the k-mer length (called \&quot;length\&quot;): For the value of k chosen in the assembly, a measure of how many k-mers overlap (by 1 bp each overlap) to give this length\n    - the k-mer coverage (called \&quot;coverage\&quot;): For the value of k chosen in the assembly, a measure of how many k-mers overlap each base position (in the assembly).\n\n    ![Contigs output](../../images/image10.png)\n\n- A \&quot;Stats\&quot; file\n\n    This is a tabular file giving for each contig the k-mer lengths, k-mer coverages and other measures.\n\n    ![Contigs stats output](../../images/image11.png)\n\n# Collect some statistics on the contigs\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. How many contigs have been built?\n&gt; 2. What is the mean, min and max length of the contigs?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. 190\n&gt; &gt; 2. To compute this information, we can use the Datamash tool on the 2nd columns (length). Be careful with the first line, the header. As a result, we obtain: 597.82 as mean, 1 as min and 12904 as max. It would mean that the smallest contig has a length of 1 bp, even smaller than k. The length on the 2nd column corresponds to length of the contig in k-mers. This means that the smallest contig has a length of 1k = 29. So to obtain the real length, we need to add k-1 to the length. We then obtain a mean contig length of 625.82 bp, a min contig of 29 bp and a max contig of 12,932 bp.\n&gt; {: .solution }\n{: .question}\n\nThis table is limitted, but we will now collect more basic statistics on our assembly.\n\n&gt; ### {% icon hands_on %} Hands-on: Collect fasta statistics on our contigs\n&gt;\n&gt; 1. **Quast** {% icon tool %} with\n&gt;    - \&quot;Contigs/scaffolds output file\&quot; to the output of **velvetg**\n&gt;    - \&quot;Type of data\&quot; to `contig`\n&gt;    - \&quot;Reference File\&quot; to `wildtype.fna`\n&gt;    - \&quot;Type of organism\&quot; to `Prokaryotes`\n&gt;    - \&quot;Lower Threshold\&quot; to `500`\n&gt;    - \&quot;Thresholds\&quot; to `0,1000`\n{: .hands_on}\n\nThis tool generates 5 output files, but we will focus on the HTML report and the Icarus viewer.\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. What is represented in the Icarus viewer?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. Icarus is a novel genome visualizer for accurate assessment and analysis of genomic draft assemblies. It draws contigs ordered from longest to shortest, highlights N50, N75 (NG50, NG75) and long contigs larger than a user-specified threshold\n&gt; {: .solution }\n{: .question}\n\nThe HTML report reports many statistics computed by QUAST to assess the quality of the assembly:\n\n- Statistics about the quality of the assembly when compared to the reference (fraction of the genome, duplication ratio, etc)\n- Misassembly statistics, including the number of misassemblies\n\n    A misassembly is a position in the contigs (breakpoints) that satisfy one of the following criteria:\n    - the left flanking sequence aligns over 1 kbp away from the right flanking sequence on the reference;\n    - flanking sequences overlap on more than 1 kbp\n    - flanking sequences align to different strands or different chromosomes\n\n- Unaligned regions in the assembly\n- Mismatches compared to the reference genomes\n- Statistics about the assembly *per se*, such as the number of contigs and the length of the largest contig\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. How many contigs have been constructed?\n&gt; 2. Which proportion of the reference genome do they represent?\n&gt; 3. How many misassemblies have been found?\n&gt; 4. Has the assembly introduced mismatches and indels?\n&gt; 5. What are N50 and L50?\n&gt; 6. Is there a bias in GC percentage induced by the assembly?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. 190 contigs have been constructed, but only 47 have a length &gt; 500 bp.\n&gt; &gt; 2. The contigs represents 87.965% of the reference genome.\n&gt; &gt; 3. 1 misassembly has been found: it corresponds to a relocation, *i.e.* a misassembly event (breakpoint) where the left flanking sequence aligns over 1 kbp away from the right flanking sequence on the reference genome.\n&gt; &gt; 4. 8.06 mismatches per 100 kbp and 4.03 indels per 100 kbp are found.\n&gt; &gt; 5. N50 is the length for which the collection of all contigs of that length or longer covers at least half an assembly. In other words, if contigs were ordered from small to large, half of all the nucleotides will be in contigs this size or larger. And L50 is the number of contigs equal to or longer than N50: L50 is the minimal number of contigs that cover half the assembly.\n&gt; &gt; 6. The GC % in the assembly is 33.64%, really similar to the one of the reference genome (33.43%).\n&gt; {: .solution }\n{: .question}\n\n# Discussion\n\n&gt; ### {% icon hands_on %} (Optional) Hands-on: Rerun for values *k* ranging from 31 to 101\n&gt;\n&gt; 1. **velveth** {% icon tool %} with the same parameters as before except\n&gt;    - \&quot;Hash Length\&quot; to a value between 31 and 101\n&gt; 2. **velvetg** {% icon tool %} with the same parameters as before\n&gt; 3. **Quast** {% icon tool %} with the same parameters as before\n{: .hands_on}\n\nWe have completed an assembly on this data set for a number of k values ranging from 29 to 101. A few of the assembly metrics appear below.\n\n&lt;figure id=\&quot;figure-2\&quot;&gt;&lt;img src=\&quot;../../images/number_of_contigs.png\&quot; alt=\&quot;contigs\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 2:&lt;/span&gt; Number of contigs in the assembly for various k-mer sizes&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-3\&quot;&gt;&lt;img src=\&quot;../../images/largest_contig.png\&quot; alt=\&quot;largest_contig\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 3:&lt;/span&gt; Largest contig in each of the assemblies by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-4\&quot;&gt;&lt;img src=\&quot;../../images/total_bp.png\&quot; alt=\&quot;total_bp\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 4:&lt;/span&gt; Total number of base pairs in all the contigs for each assembly by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-5\&quot;&gt;&lt;img src=\&quot;../../images/n50.png\&quot; alt=\&quot;n50\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 5:&lt;/span&gt; N50 metric for each of the assemblies by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&gt; ### {% icon question %} Questions\n&gt;\n&gt; 1. Are there any distinct features in the charts?\n&gt; 2. Does it look like one assembly might be better than some of the others?\n&gt;\n{: .question}\n\nThe reasons for these patterns will be discussed in detail in the De Bruijn graph assembly slides and tutorial.\n&quot;,&quot;&lt;?xml version=\&quot;1.0\&quot; encoding=\&quot;utf-8\&quot;?&gt;{% if page.xsl %}&lt;?xml-stylesheet type=\&quot;text/xml\&quot; href=\&quot;{{ &#39;/feed.xslt.xml&#39; | absolute_url }}\&quot;?&gt;{% endif %}&lt;feed xmlns=\&quot;http://www.w3.org/2005/Atom\&quot; {% if site.lang %}xml:lang=\&quot;{{ site.lang }}\&quot;{% endif %}&gt;&lt;generator uri=\&quot;https://jekyllrb.com/\&quot; version=\&quot;{{ jekyll.version }}\&quot;&gt;Jekyll&lt;/generator&gt;&lt;link href=\&quot;{{ page.url | absolute_url }}\&quot; rel=\&quot;self\&quot; type=\&quot;application/atom+xml\&quot; /&gt;&lt;link href=\&quot;{{ &#39;/&#39; | absolute_url }}\&quot; rel=\&quot;alternate\&quot; type=\&quot;text/html\&quot; {% if site.lang %}hreflang=\&quot;{{ site.lang }}\&quot; {% endif %}/&gt;&lt;updated&gt;{{ site.time | date_to_xmlschema }}&lt;/updated&gt;&lt;id&gt;{{ page.url | absolute_url | xml_escape }}&lt;/id&gt;{% assign title = site.title | default: site.name %}{% if page.collection != \&quot;posts\&quot; %}{% assign collection = page.collection | capitalize %}{% assign title = title | append: \&quot; | \&quot; | append: collection %}{% endif %}{% if page.category %}{% assign category = page.category | capitalize %}{% assign title = title | append: \&quot; | \&quot; | append: category %}{% endif %}{% if title %}&lt;title type=\&quot;html\&quot;&gt;{{ title | smartify | xml_escape }}&lt;/title&gt;{% endif %}{% if site.description %}&lt;subtitle&gt;{{ site.description | xml_escape }}&lt;/subtitle&gt;{% endif %}{% if site.author %}&lt;author&gt;&lt;name&gt;{{ site.author.name | default: site.author | xml_escape }}&lt;/name&gt;{% if site.author.email %}&lt;email&gt;{{ site.author.email | xml_escape }}&lt;/email&gt;{% endif %}{% if site.author.uri %}&lt;uri&gt;{{ site.author.uri | xml_escape }}&lt;/uri&gt;{% endif %}&lt;/author&gt;{% endif %}{% assign posts = site[page.collection] | where_exp: \&quot;post\&quot;, \&quot;post.draft != true\&quot; | sort: \&quot;date\&quot; | reverse %}{% if page.category %}{% assign posts = posts | where: \&quot;category\&quot;,page.category %}{% endif %}{% for post in posts limit: 10 %}&lt;entry{% if post.lang %}{{\&quot; \&quot;}}xml:lang=\&quot;{{ post.lang }}\&quot;{% endif %}&gt;&lt;title type=\&quot;html\&quot;&gt;{{ post.title | smartify | strip_html | normalize_whitespace | xml_escape }}&lt;/title&gt;&lt;link href=\&quot;{{ post.url | absolute_url }}\&quot; rel=\&quot;alternate\&quot; type=\&quot;text/html\&quot; title=\&quot;{{ post.title | xml_escape }}\&quot; /&gt;&lt;published&gt;{{ post.date | date_to_xmlschema }}&lt;/published&gt;&lt;updated&gt;{{ post.last_modified_at | default: post.date | date_to_xmlschema }}&lt;/updated&gt;&lt;id&gt;{{ post.id | absolute_url | xml_escape }}&lt;/id&gt;&lt;content type=\&quot;html\&quot; xml:base=\&quot;{{ post.url | absolute_url | xml_escape }}\&quot;&gt;{{ post.content | strip | xml_escape }}&lt;/content&gt;{% assign post_author = post.author | default: post.authors[0] | default: site.author %}{% assign post_author = site.data.authors[post_author] | default: post_author %}{% assign post_author_email = post_author.email | default: nil %}{% assign post_author_uri = post_author.uri | default: nil %}{% assign post_author_name = post_author.name | default: post_author %}&lt;author&gt;&lt;name&gt;{{ post_author_name | default: \&quot;\&quot; | xml_escape }}&lt;/name&gt;{% if post_author_email %}&lt;email&gt;{{ post_author_email | xml_escape }}&lt;/email&gt;{% endif %}{% if post_author_uri %}&lt;uri&gt;{{ post_author_uri | xml_escape }}&lt;/uri&gt;{% endif %}&lt;/author&gt;{% if post.category %}&lt;category term=\&quot;{{ post.category | xml_escape }}\&quot; /&gt;{% endif %}{% for tag in post.tags %}&lt;category term=\&quot;{{ tag | xml_escape }}\&quot; /&gt;{% endfor %}{% if post.excerpt and post.excerpt != empty %}&lt;summary type=\&quot;html\&quot;&gt;{{ post.excerpt | strip_html | normalize_whitespace | xml_escape }}&lt;/summary&gt;{% endif %}{% assign post_image = post.image.path | default: post.image %}{% if post_image %}{% unless post_image contains \&quot;://\&quot; %}{% assign post_image = post_image | absolute_url %}{% endunless %}&lt;media:thumbnail xmlns:media=\&quot;http://search.yahoo.com/mrss/\&quot; url=\&quot;{{ post_image | xml_escape }}\&quot; /&gt;{% endif %}&lt;/entry&gt;{% endfor %}&lt;/feed&gt;&quot;],&quot;posts&quot;:[],&quot;static_files&quot;:[{&quot;name&quot;:&quot;CODEOWNERS&quot;,&quot;path&quot;:&quot;/CODEOWNERS&quot;,&quot;basename&quot;:&quot;CODEOWNERS&quot;,&quot;extname&quot;:&quot;&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;CODE_OF_CONDUCT.md&quot;,&quot;path&quot;:&quot;/CODE_OF_CONDUCT.md&quot;,&quot;basename&quot;:&quot;CODE_OF_CONDUCT&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Procfile&quot;,&quot;path&quot;:&quot;/Procfile&quot;,&quot;basename&quot;:&quot;Procfile&quot;,&quot;extname&quot;:&quot;&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;app.json&quot;,&quot;path&quot;:&quot;/app.json&quot;,&quot;basename&quot;:&quot;app&quot;,&quot;extname&quot;:&quot;.json&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;VIB.css&quot;,&quot;path&quot;:&quot;/assets/css/VIB.css&quot;,&quot;basename&quot;:&quot;VIB&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-08-09 12:47:05 +0000&quot;},{&quot;name&quot;:&quot;academicons.css&quot;,&quot;path&quot;:&quot;/assets/css/academicons.css&quot;,&quot;basename&quot;:&quot;academicons&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap-theme.css&quot;,&quot;path&quot;:&quot;/assets/css/bootstrap-theme.css&quot;,&quot;basename&quot;:&quot;bootstrap-theme&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap-theme.min.css&quot;,&quot;path&quot;:&quot;/assets/css/bootstrap-theme.min.css&quot;,&quot;basename&quot;:&quot;bootstrap-theme.min&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap-toc.min.css&quot;,&quot;path&quot;:&quot;/assets/css/bootstrap-toc.min.css&quot;,&quot;basename&quot;:&quot;bootstrap-toc.min&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap.css&quot;,&quot;path&quot;:&quot;/assets/css/bootstrap.css&quot;,&quot;basename&quot;:&quot;bootstrap&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap.min.css&quot;,&quot;path&quot;:&quot;/assets/css/bootstrap.min.css&quot;,&quot;basename&quot;:&quot;bootstrap.min&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;font-awesome.css&quot;,&quot;path&quot;:&quot;/assets/css/font-awesome.css&quot;,&quot;basename&quot;:&quot;font-awesome&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;slides.css&quot;,&quot;path&quot;:&quot;/assets/css/slides.css&quot;,&quot;basename&quot;:&quot;slides&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;syntax_highlighting.css&quot;,&quot;path&quot;:&quot;/assets/css/syntax_highlighting.css&quot;,&quot;basename&quot;:&quot;syntax_highlighting&quot;,&quot;extname&quot;:&quot;.css&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Dense-Regular.otf&quot;,&quot;path&quot;:&quot;/assets/fonts/Dense-Regular.otf&quot;,&quot;basename&quot;:&quot;Dense-Regular&quot;,&quot;extname&quot;:&quot;.otf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Dense-Regular.ttf&quot;,&quot;path&quot;:&quot;/assets/fonts/Dense-Regular.ttf&quot;,&quot;basename&quot;:&quot;Dense-Regular&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;FontAwesome.otf&quot;,&quot;path&quot;:&quot;/assets/fonts/FontAwesome.otf&quot;,&quot;basename&quot;:&quot;FontAwesome&quot;,&quot;extname&quot;:&quot;.otf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;MaterialIcons.woff&quot;,&quot;path&quot;:&quot;/assets/fonts/MaterialIcons.woff&quot;,&quot;basename&quot;:&quot;MaterialIcons&quot;,&quot;extname&quot;:&quot;.woff&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;MaterialIcons.woff2&quot;,&quot;path&quot;:&quot;/assets/fonts/MaterialIcons.woff2&quot;,&quot;basename&quot;:&quot;MaterialIcons&quot;,&quot;extname&quot;:&quot;.woff2&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;OpenSans.otf&quot;,&quot;path&quot;:&quot;/assets/fonts/OpenSans.otf&quot;,&quot;basename&quot;:&quot;OpenSans&quot;,&quot;extname&quot;:&quot;.otf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;OpenSans.woff&quot;,&quot;path&quot;:&quot;/assets/fonts/OpenSans.woff&quot;,&quot;basename&quot;:&quot;OpenSans&quot;,&quot;extname&quot;:&quot;.woff&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;OpenSans.woff2&quot;,&quot;path&quot;:&quot;/assets/fonts/OpenSans.woff2&quot;,&quot;basename&quot;:&quot;OpenSans&quot;,&quot;extname&quot;:&quot;.woff2&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;academicons.eot&quot;,&quot;path&quot;:&quot;/assets/fonts/academicons.eot&quot;,&quot;basename&quot;:&quot;academicons&quot;,&quot;extname&quot;:&quot;.eot&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;academicons.svg&quot;,&quot;path&quot;:&quot;/assets/fonts/academicons.svg&quot;,&quot;basename&quot;:&quot;academicons&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;academicons.ttf&quot;,&quot;path&quot;:&quot;/assets/fonts/academicons.ttf&quot;,&quot;basename&quot;:&quot;academicons&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;academicons.woff&quot;,&quot;path&quot;:&quot;/assets/fonts/academicons.woff&quot;,&quot;basename&quot;:&quot;academicons&quot;,&quot;extname&quot;:&quot;.woff&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;fontawesome-webfont.eot&quot;,&quot;path&quot;:&quot;/assets/fonts/fontawesome-webfont.eot&quot;,&quot;basename&quot;:&quot;fontawesome-webfont&quot;,&quot;extname&quot;:&quot;.eot&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;fontawesome-webfont.svg&quot;,&quot;path&quot;:&quot;/assets/fonts/fontawesome-webfont.svg&quot;,&quot;basename&quot;:&quot;fontawesome-webfont&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;fontawesome-webfont.ttf&quot;,&quot;path&quot;:&quot;/assets/fonts/fontawesome-webfont.ttf&quot;,&quot;basename&quot;:&quot;fontawesome-webfont&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;fontawesome-webfont.woff&quot;,&quot;path&quot;:&quot;/assets/fonts/fontawesome-webfont.woff&quot;,&quot;basename&quot;:&quot;fontawesome-webfont&quot;,&quot;extname&quot;:&quot;.woff&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;fontawesome-webfont.woff2&quot;,&quot;path&quot;:&quot;/assets/fonts/fontawesome-webfont.woff2&quot;,&quot;basename&quot;:&quot;fontawesome-webfont&quot;,&quot;extname&quot;:&quot;.woff2&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;glyphicons-halflings-regular.eot&quot;,&quot;path&quot;:&quot;/assets/fonts/glyphicons-halflings-regular.eot&quot;,&quot;basename&quot;:&quot;glyphicons-halflings-regular&quot;,&quot;extname&quot;:&quot;.eot&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;glyphicons-halflings-regular.svg&quot;,&quot;path&quot;:&quot;/assets/fonts/glyphicons-halflings-regular.svg&quot;,&quot;basename&quot;:&quot;glyphicons-halflings-regular&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;glyphicons-halflings-regular.ttf&quot;,&quot;path&quot;:&quot;/assets/fonts/glyphicons-halflings-regular.ttf&quot;,&quot;basename&quot;:&quot;glyphicons-halflings-regular&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;glyphicons-halflings-regular.woff&quot;,&quot;path&quot;:&quot;/assets/fonts/glyphicons-halflings-regular.woff&quot;,&quot;basename&quot;:&quot;glyphicons-halflings-regular&quot;,&quot;extname&quot;:&quot;.woff&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;glyphicons-halflings-regular.woff2&quot;,&quot;path&quot;:&quot;/assets/fonts/glyphicons-halflings-regular.woff2&quot;,&quot;basename&quot;:&quot;glyphicons-halflings-regular&quot;,&quot;extname&quot;:&quot;.woff2&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;GTN-60px.png&quot;,&quot;path&quot;:&quot;/assets/images/GTN-60px.png&quot;,&quot;basename&quot;:&quot;GTN-60px&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;GTN.png&quot;,&quot;path&quot;:&quot;/assets/images/GTN.png&quot;,&quot;basename&quot;:&quot;GTN&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;GTNLogo1000.png&quot;,&quot;path&quot;:&quot;/assets/images/GTNLogo1000.png&quot;,&quot;basename&quot;:&quot;GTNLogo1000&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bioinformatics_core_rgb_neg.png&quot;,&quot;path&quot;:&quot;/assets/images/bioinformatics_core_rgb_neg.png&quot;,&quot;basename&quot;:&quot;bioinformatics_core_rgb_neg&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-black-logo.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-black-logo.png&quot;,&quot;basename&quot;:&quot;qr-gtn-black-logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-black.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-black.png&quot;,&quot;basename&quot;:&quot;qr-gtn-black&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-darkblue-logo.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-darkblue-logo.png&quot;,&quot;basename&quot;:&quot;qr-gtn-darkblue-logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-darkblue.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-darkblue.png&quot;,&quot;basename&quot;:&quot;qr-gtn-darkblue&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-darkgrey-logo.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-darkgrey-logo.png&quot;,&quot;basename&quot;:&quot;qr-gtn-darkgrey-logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-darkgrey.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-darkgrey.png&quot;,&quot;basename&quot;:&quot;qr-gtn-darkgrey&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-logo.xcf&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-logo.xcf&quot;,&quot;basename&quot;:&quot;qr-gtn-logo&quot;,&quot;extname&quot;:&quot;.xcf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-white-logo.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-white-logo.png&quot;,&quot;basename&quot;:&quot;qr-gtn-white-logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn-white.png&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn-white.png&quot;,&quot;basename&quot;:&quot;qr-gtn-white&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;qr-gtn.xcf&quot;,&quot;path&quot;:&quot;/assets/images/qr-gtn.xcf&quot;,&quot;basename&quot;:&quot;qr-gtn&quot;,&quot;extname&quot;:&quot;.xcf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;vib_notag_neg_rgb.jpg&quot;,&quot;path&quot;:&quot;/assets/images/vib_notag_neg_rgb.jpg&quot;,&quot;basename&quot;:&quot;vib_notag_neg_rgb&quot;,&quot;extname&quot;:&quot;.jpg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap-toc.min.js&quot;,&quot;path&quot;:&quot;/assets/js/bootstrap-toc.min.js&quot;,&quot;basename&quot;:&quot;bootstrap-toc.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap.js&quot;,&quot;path&quot;:&quot;/assets/js/bootstrap.js&quot;,&quot;basename&quot;:&quot;bootstrap&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bootstrap.min.js&quot;,&quot;path&quot;:&quot;/assets/js/bootstrap.min.js&quot;,&quot;basename&quot;:&quot;bootstrap.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;clipboard.min.js&quot;,&quot;path&quot;:&quot;/assets/js/clipboard.min.js&quot;,&quot;basename&quot;:&quot;clipboard.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;details-element-polyfill.js&quot;,&quot;path&quot;:&quot;/assets/js/details-element-polyfill.js&quot;,&quot;basename&quot;:&quot;details-element-polyfill&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;jquery.slim.min.js&quot;,&quot;path&quot;:&quot;/assets/js/jquery.slim.min.js&quot;,&quot;basename&quot;:&quot;jquery.slim.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;main.js&quot;,&quot;path&quot;:&quot;/assets/js/main.js&quot;,&quot;basename&quot;:&quot;main&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;popper.min.js&quot;,&quot;path&quot;:&quot;/assets/js/popper.min.js&quot;,&quot;basename&quot;:&quot;popper.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;remark-latest.min.js&quot;,&quot;path&quot;:&quot;/assets/js/remark-latest.min.js&quot;,&quot;basename&quot;:&quot;remark-latest.min&quot;,&quot;extname&quot;:&quot;.js&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;favicon.ico&quot;,&quot;path&quot;:&quot;/favicon.ico&quot;,&quot;basename&quot;:&quot;favicon&quot;,&quot;extname&quot;:&quot;.ico&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;galaxy_instances.yaml&quot;,&quot;path&quot;:&quot;/galaxy_instances.yaml&quot;,&quot;basename&quot;:&quot;galaxy_instances&quot;,&quot;extname&quot;:&quot;.yaml&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;LICENSE.txt&quot;,&quot;path&quot;:&quot;/shared/fonts/NotoSans/LICENSE.txt&quot;,&quot;basename&quot;:&quot;LICENSE&quot;,&quot;extname&quot;:&quot;.txt&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;NotoSans-Bold.ttf&quot;,&quot;path&quot;:&quot;/shared/fonts/NotoSans/NotoSans-Bold.ttf&quot;,&quot;basename&quot;:&quot;NotoSans-Bold&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;NotoSans-BoldItalic.ttf&quot;,&quot;path&quot;:&quot;/shared/fonts/NotoSans/NotoSans-BoldItalic.ttf&quot;,&quot;basename&quot;:&quot;NotoSans-BoldItalic&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;NotoSans-Italic.ttf&quot;,&quot;path&quot;:&quot;/shared/fonts/NotoSans/NotoSans-Italic.ttf&quot;,&quot;basename&quot;:&quot;NotoSans-Italic&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;NotoSans-Regular.ttf&quot;,&quot;path&quot;:&quot;/shared/fonts/NotoSans/NotoSans-Regular.ttf&quot;,&quot;basename&quot;:&quot;NotoSans-Regular&quot;,&quot;extname&quot;:&quot;.ttf&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Excelerate_whitebackground.png&quot;,&quot;path&quot;:&quot;/shared/images/Excelerate_whitebackground.png&quot;,&quot;basename&quot;:&quot;Excelerate_whitebackground&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;FreeBayes_settings.png&quot;,&quot;path&quot;:&quot;/shared/images/FreeBayes_settings.png&quot;,&quot;basename&quot;:&quot;FreeBayes_settings&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;GTNLogo1000.png&quot;,&quot;path&quot;:&quot;/shared/images/GTNLogo1000.png&quot;,&quot;basename&quot;:&quot;GTNLogo1000&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;MA_plot.png&quot;,&quot;path&quot;:&quot;/shared/images/MA_plot.png&quot;,&quot;basename&quot;:&quot;MA_plot&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;RNAseq_histiry_image.key&quot;,&quot;path&quot;:&quot;/shared/images/RNAseq_histiry_image.key&quot;,&quot;basename&quot;:&quot;RNAseq_histiry_image&quot;,&quot;extname&quot;:&quot;.key&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Screen Shot 2016-04-21 at 3.55.19 PM.png&quot;,&quot;path&quot;:&quot;/shared/images/Screen Shot 2016-04-21 at 3.55.19 PM.png&quot;,&quot;basename&quot;:&quot;Screen Shot 2016-04-21 at 3.55.19 PM&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;Screen Shot 2016-12-06 at 18.14.25.png&quot;,&quot;path&quot;:&quot;/shared/images/Screen Shot 2016-12-06 at 18.14.25.png&quot;,&quot;basename&quot;:&quot;Screen Shot 2016-12-06 at 18.14.25&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;accepted_hits_1.png&quot;,&quot;path&quot;:&quot;/shared/images/accepted_hits_1.png&quot;,&quot;basename&quot;:&quot;accepted_hits_1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;accepted_hits_2.png&quot;,&quot;path&quot;:&quot;/shared/images/accepted_hits_2.png&quot;,&quot;basename&quot;:&quot;accepted_hits_2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;ansible_logo.png&quot;,&quot;path&quot;:&quot;/shared/images/ansible_logo.png&quot;,&quot;basename&quot;:&quot;ansible_logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bioconda_logo.png&quot;,&quot;path&quot;:&quot;/shared/images/bioconda_logo.png&quot;,&quot;basename&quot;:&quot;bioconda_logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bw.png&quot;,&quot;path&quot;:&quot;/shared/images/bw.png&quot;,&quot;basename&quot;:&quot;bw&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bw_b_rank.png&quot;,&quot;path&quot;:&quot;/shared/images/bw_b_rank.png&quot;,&quot;basename&quot;:&quot;bw_b_rank&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_q1.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_q1.png&quot;,&quot;basename&quot;:&quot;bwt_q1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_q2.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_q2.png&quot;,&quot;basename&quot;:&quot;bwt_q2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_q3.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_q3.png&quot;,&quot;basename&quot;:&quot;bwt_q3&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_q4.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_q4.png&quot;,&quot;basename&quot;:&quot;bwt_q4&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_q5.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_q5.png&quot;,&quot;basename&quot;:&quot;bwt_q5&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_rev.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_rev.png&quot;,&quot;basename&quot;:&quot;bwt_rev&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;bwt_rev2.png&quot;,&quot;path&quot;:&quot;/shared/images/bwt_rev2.png&quot;,&quot;basename&quot;:&quot;bwt_rev2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;cloudman_based_server_landing_page.png&quot;,&quot;path&quot;:&quot;/shared/images/cloudman_based_server_landing_page.png&quot;,&quot;basename&quot;:&quot;cloudman_based_server_landing_page&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;cloudman_management_console_showing_autoscaled_w_1_worker.png&quot;,&quot;path&quot;:&quot;/shared/images/cloudman_management_console_showing_autoscaled_w_1_worker.png&quot;,&quot;basename&quot;:&quot;cloudman_management_console_showing_autoscaled_w_1_worker&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;collections_video.png&quot;,&quot;path&quot;:&quot;/shared/images/collections_video.png&quot;,&quot;basename&quot;:&quot;collections_video&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;conda_logo.png&quot;,&quot;path&quot;:&quot;/shared/images/conda_logo.png&quot;,&quot;basename&quot;:&quot;conda_logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;create_index.png&quot;,&quot;path&quot;:&quot;/shared/images/create_index.png&quot;,&quot;basename&quot;:&quot;create_index&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;dT_random.png&quot;,&quot;path&quot;:&quot;/shared/images/dT_random.png&quot;,&quot;basename&quot;:&quot;dT_random&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;deNBI.png&quot;,&quot;path&quot;:&quot;/shared/images/deNBI.png&quot;,&quot;basename&quot;:&quot;deNBI&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;deseq2_interface.png&quot;,&quot;path&quot;:&quot;/shared/images/deseq2_interface.png&quot;,&quot;basename&quot;:&quot;deseq2_interface&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;deseq2_output.png&quot;,&quot;path&quot;:&quot;/shared/images/deseq2_output.png&quot;,&quot;basename&quot;:&quot;deseq2_output&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:48 +0000&quot;},{&quot;name&quot;:&quot;development_process.png&quot;,&quot;path&quot;:&quot;/shared/images/development_process.png&quot;,&quot;basename&quot;:&quot;development_process&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;diff.png&quot;,&quot;path&quot;:&quot;/shared/images/diff.png&quot;,&quot;basename&quot;:&quot;diff&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;dispersion.png&quot;,&quot;path&quot;:&quot;/shared/images/dispersion.png&quot;,&quot;basename&quot;:&quot;dispersion&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;dna_rna.png&quot;,&quot;path&quot;:&quot;/shared/images/dna_rna.png&quot;,&quot;basename&quot;:&quot;dna_rna&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;docker_logo.png&quot;,&quot;path&quot;:&quot;/shared/images/docker_logo.png&quot;,&quot;basename&quot;:&quot;docker_logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;dotplot.gif&quot;,&quot;path&quot;:&quot;/shared/images/dotplot.gif&quot;,&quot;basename&quot;:&quot;dotplot&quot;,&quot;extname&quot;:&quot;.gif&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;elixir.png&quot;,&quot;path&quot;:&quot;/shared/images/elixir.png&quot;,&quot;basename&quot;:&quot;elixir&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;em.png&quot;,&quot;path&quot;:&quot;/shared/images/em.png&quot;,&quot;basename&quot;:&quot;em&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;euc_dist.png&quot;,&quot;path&quot;:&quot;/shared/images/euc_dist.png&quot;,&quot;basename&quot;:&quot;euc_dist&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;everything_connected.png&quot;,&quot;path&quot;:&quot;/shared/images/everything_connected.png&quot;,&quot;basename&quot;:&quot;everything_connected&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;f_from_l.png&quot;,&quot;path&quot;:&quot;/shared/images/f_from_l.png&quot;,&quot;basename&quot;:&quot;f_from_l&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;filter_gtf.png&quot;,&quot;path&quot;:&quot;/shared/images/filter_gtf.png&quot;,&quot;basename&quot;:&quot;filter_gtf&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;forward_index.png&quot;,&quot;path&quot;:&quot;/shared/images/forward_index.png&quot;,&quot;basename&quot;:&quot;forward_index&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;freebayes.png&quot;,&quot;path&quot;:&quot;/shared/images/freebayes.png&quot;,&quot;basename&quot;:&quot;freebayes&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;freebayes_gq.png&quot;,&quot;path&quot;:&quot;/shared/images/freebayes_gq.png&quot;,&quot;basename&quot;:&quot;freebayes_gq&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;galaxy_command.png&quot;,&quot;path&quot;:&quot;/shared/images/galaxy_command.png&quot;,&quot;basename&quot;:&quot;galaxy_command&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;galaxy_logo.png&quot;,&quot;path&quot;:&quot;/shared/images/galaxy_logo.png&quot;,&quot;basename&quot;:&quot;galaxy_logo&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;galaxy_logo_25percent_transparent.png&quot;,&quot;path&quot;:&quot;/shared/images/galaxy_logo_25percent_transparent.png&quot;,&quot;basename&quot;:&quot;galaxy_logo_25percent_transparent&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;gemini_command.png&quot;,&quot;path&quot;:&quot;/shared/images/gemini_command.png&quot;,&quot;basename&quot;:&quot;gemini_command&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;gemini_db_info.png&quot;,&quot;path&quot;:&quot;/shared/images/gemini_db_info.png&quot;,&quot;basename&quot;:&quot;gemini_db_info&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;gemini_load.png&quot;,&quot;path&quot;:&quot;/shared/images/gemini_load.png&quot;,&quot;basename&quot;:&quot;gemini_load&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;gemini_query1.png&quot;,&quot;path&quot;:&quot;/shared/images/gemini_query1.png&quot;,&quot;basename&quot;:&quot;gemini_query1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;gemini_query2.png&quot;,&quot;path&quot;:&quot;/shared/images/gemini_query2.png&quot;,&quot;basename&quot;:&quot;gemini_query2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;goblet.png&quot;,&quot;path&quot;:&quot;/shared/images/goblet.png&quot;,&quot;basename&quot;:&quot;goblet&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;hash.png&quot;,&quot;path&quot;:&quot;/shared/images/hash.png&quot;,&quot;basename&quot;:&quot;hash&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;hisat.png&quot;,&quot;path&quot;:&quot;/shared/images/hisat.png&quot;,&quot;basename&quot;:&quot;hisat&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;history_menu_buttons2.png&quot;,&quot;path&quot;:&quot;/shared/images/history_menu_buttons2.png&quot;,&quot;basename&quot;:&quot;history_menu_buttons2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;history_menu_extract_workflow.png&quot;,&quot;path&quot;:&quot;/shared/images/history_menu_extract_workflow.png&quot;,&quot;basename&quot;:&quot;history_menu_extract_workflow&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;history_options_menu.png&quot;,&quot;path&quot;:&quot;/shared/images/history_options_menu.png&quot;,&quot;basename&quot;:&quot;history_options_menu&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;htseq_count.png&quot;,&quot;path&quot;:&quot;/shared/images/htseq_count.png&quot;,&quot;basename&quot;:&quot;htseq_count&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;htseq_count_interface.png&quot;,&quot;path&quot;:&quot;/shared/images/htseq_count_interface.png&quot;,&quot;basename&quot;:&quot;htseq_count_interface&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;igv_tophat.png&quot;,&quot;path&quot;:&quot;/shared/images/igv_tophat.png&quot;,&quot;basename&quot;:&quot;igv_tophat&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;import_history.png&quot;,&quot;path&quot;:&quot;/shared/images/import_history.png&quot;,&quot;basename&quot;:&quot;import_history&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;interactive_training.png&quot;,&quot;path&quot;:&quot;/shared/images/interactive_training.png&quot;,&quot;basename&quot;:&quot;interactive_training&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;inverted_index.png&quot;,&quot;path&quot;:&quot;/shared/images/inverted_index.png&quot;,&quot;basename&quot;:&quot;inverted_index&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;kallisto.png&quot;,&quot;path&quot;:&quot;/shared/images/kallisto.png&quot;,&quot;basename&quot;:&quot;kallisto&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;landing_page.png&quot;,&quot;path&quot;:&quot;/shared/images/landing_page.png&quot;,&quot;basename&quot;:&quot;landing_page&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;lcs.png&quot;,&quot;path&quot;:&quot;/shared/images/lcs.png&quot;,&quot;basename&quot;:&quot;lcs&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;lf_a.png&quot;,&quot;path&quot;:&quot;/shared/images/lf_a.png&quot;,&quot;basename&quot;:&quot;lf_a&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;lf_b.png&quot;,&quot;path&quot;:&quot;/shared/images/lf_b.png&quot;,&quot;basename&quot;:&quot;lf_b&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;lib_type.png&quot;,&quot;path&quot;:&quot;/shared/images/lib_type.png&quot;,&quot;basename&quot;:&quot;lib_type&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;library_import.png&quot;,&quot;path&quot;:&quot;/shared/images/library_import.png&quot;,&quot;basename&quot;:&quot;library_import&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;library_import_complete.png&quot;,&quot;path&quot;:&quot;/shared/images/library_import_complete.png&quot;,&quot;basename&quot;:&quot;library_import_complete&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;map.png&quot;,&quot;path&quot;:&quot;/shared/images/map.png&quot;,&quot;basename&quot;:&quot;map&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;mum.png&quot;,&quot;path&quot;:&quot;/shared/images/mum.png&quot;,&quot;basename&quot;:&quot;mum&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;nih.png&quot;,&quot;path&quot;:&quot;/shared/images/nih.png&quot;,&quot;basename&quot;:&quot;nih&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;nsf.gif&quot;,&quot;path&quot;:&quot;/shared/images/nsf.gif&quot;,&quot;basename&quot;:&quot;nsf&quot;,&quot;extname&quot;:&quot;.gif&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pA.png&quot;,&quot;path&quot;:&quot;/shared/images/pA.png&quot;,&quot;basename&quot;:&quot;pA&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pAB.png&quot;,&quot;path&quot;:&quot;/shared/images/pAB.png&quot;,&quot;basename&quot;:&quot;pAB&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pB.png&quot;,&quot;path&quot;:&quot;/shared/images/pB.png&quot;,&quot;basename&quot;:&quot;pB&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;p_val_hist.png&quot;,&quot;path&quot;:&quot;/shared/images/p_val_hist.png&quot;,&quot;basename&quot;:&quot;p_val_hist&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pca.png&quot;,&quot;path&quot;:&quot;/shared/images/pca.png&quot;,&quot;basename&quot;:&quot;pca&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pcr-duplicates.png&quot;,&quot;path&quot;:&quot;/shared/images/pcr-duplicates.png&quot;,&quot;basename&quot;:&quot;pcr-duplicates&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;phinch_overviewpage.png&quot;,&quot;path&quot;:&quot;/shared/images/phinch_overviewpage.png&quot;,&quot;basename&quot;:&quot;phinch_overviewpage&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;psu.png&quot;,&quot;path&quot;:&quot;/shared/images/psu.png&quot;,&quot;basename&quot;:&quot;psu&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;quasi_aln.png&quot;,&quot;path&quot;:&quot;/shared/images/quasi_aln.png&quot;,&quot;basename&quot;:&quot;quasi_aln&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rename_history.png&quot;,&quot;path&quot;:&quot;/shared/images/rename_history.png&quot;,&quot;basename&quot;:&quot;rename_history&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;repo_organization.png&quot;,&quot;path&quot;:&quot;/shared/images/repo_organization.png&quot;,&quot;basename&quot;:&quot;repo_organization&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;repo_organization.svg&quot;,&quot;path&quot;:&quot;/shared/images/repo_organization.svg&quot;,&quot;basename&quot;:&quot;repo_organization&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rnaseq_comparison.png&quot;,&quot;path&quot;:&quot;/shared/images/rnaseq_comparison.png&quot;,&quot;basename&quot;:&quot;rnaseq_comparison&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rnaseq_data_in_history.png&quot;,&quot;path&quot;:&quot;/shared/images/rnaseq_data_in_history.png&quot;,&quot;basename&quot;:&quot;rnaseq_data_in_history&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rnaseq_library.png&quot;,&quot;path&quot;:&quot;/shared/images/rnaseq_library.png&quot;,&quot;basename&quot;:&quot;rnaseq_library&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;sa.png&quot;,&quot;path&quot;:&quot;/shared/images/sa.png&quot;,&quot;basename&quot;:&quot;sa&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;sailfish.png&quot;,&quot;path&quot;:&quot;/shared/images/sailfish.png&quot;,&quot;basename&quot;:&quot;sailfish&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;salmon.png&quot;,&quot;path&quot;:&quot;/shared/images/salmon.png&quot;,&quot;basename&quot;:&quot;salmon&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;sampling-bias.png&quot;,&quot;path&quot;:&quot;/shared/images/sampling-bias.png&quot;,&quot;basename&quot;:&quot;sampling-bias&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;sashimi.png&quot;,&quot;path&quot;:&quot;/shared/images/sashimi.png&quot;,&quot;basename&quot;:&quot;sashimi&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;search_index.png&quot;,&quot;path&quot;:&quot;/shared/images/search_index.png&quot;,&quot;basename&quot;:&quot;search_index&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;side-by-side.png&quot;,&quot;path&quot;:&quot;/shared/images/side-by-side.png&quot;,&quot;basename&quot;:&quot;side-by-side&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;snpeff.png&quot;,&quot;path&quot;:&quot;/shared/images/snpeff.png&quot;,&quot;basename&quot;:&quot;snpeff&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;snpeff_chart.png&quot;,&quot;path&quot;:&quot;/shared/images/snpeff_chart.png&quot;,&quot;basename&quot;:&quot;snpeff_chart&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;snpeff_codons.png&quot;,&quot;path&quot;:&quot;/shared/images/snpeff_codons.png&quot;,&quot;basename&quot;:&quot;snpeff_codons&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;sorted_list.png&quot;,&quot;path&quot;:&quot;/shared/images/sorted_list.png&quot;,&quot;basename&quot;:&quot;sorted_list&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;specificity.png&quot;,&quot;path&quot;:&quot;/shared/images/specificity.png&quot;,&quot;basename&quot;:&quot;specificity&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;star.png&quot;,&quot;path&quot;:&quot;/shared/images/star.png&quot;,&quot;basename&quot;:&quot;star&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;stranded_protocols.png&quot;,&quot;path&quot;:&quot;/shared/images/stranded_protocols.png&quot;,&quot;basename&quot;:&quot;stranded_protocols&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;stranded_result.png&quot;,&quot;path&quot;:&quot;/shared/images/stranded_result.png&quot;,&quot;basename&quot;:&quot;stranded_result&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;stringtie1.png&quot;,&quot;path&quot;:&quot;/shared/images/stringtie1.png&quot;,&quot;basename&quot;:&quot;stringtie1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;stringtie2.png&quot;,&quot;path&quot;:&quot;/shared/images/stringtie2.png&quot;,&quot;basename&quot;:&quot;stringtie2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;substring_trie.png&quot;,&quot;path&quot;:&quot;/shared/images/substring_trie.png&quot;,&quot;basename&quot;:&quot;substring_trie&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_tree_1.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_tree_1.png&quot;,&quot;basename&quot;:&quot;suffix_tree_1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_tree_2.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_tree_2.png&quot;,&quot;basename&quot;:&quot;suffix_tree_2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_tree_3.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_tree_3.png&quot;,&quot;basename&quot;:&quot;suffix_tree_3&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_tree_4.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_tree_4.png&quot;,&quot;basename&quot;:&quot;suffix_tree_4&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_tree_5.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_tree_5.png&quot;,&quot;basename&quot;:&quot;suffix_tree_5&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_1.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_1.png&quot;,&quot;basename&quot;:&quot;suffix_trie_1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_2.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_2.png&quot;,&quot;basename&quot;:&quot;suffix_trie_2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_3.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_3.png&quot;,&quot;basename&quot;:&quot;suffix_trie_3&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_4.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_4.png&quot;,&quot;basename&quot;:&quot;suffix_trie_4&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_5.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_5.png&quot;,&quot;basename&quot;:&quot;suffix_trie_5&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_6.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_6.png&quot;,&quot;basename&quot;:&quot;suffix_trie_6&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_7.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_7.png&quot;,&quot;basename&quot;:&quot;suffix_trie_7&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_8.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_8.png&quot;,&quot;basename&quot;:&quot;suffix_trie_8&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;suffix_trie_9.png&quot;,&quot;path&quot;:&quot;/shared/images/suffix_trie_9.png&quot;,&quot;basename&quot;:&quot;suffix_trie_9&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tools_collection_input.png&quot;,&quot;path&quot;:&quot;/shared/images/tools_collection_input.png&quot;,&quot;basename&quot;:&quot;tools_collection_input&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tophat.png&quot;,&quot;path&quot;:&quot;/shared/images/tophat.png&quot;,&quot;basename&quot;:&quot;tophat&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tophat2.png&quot;,&quot;path&quot;:&quot;/shared/images/tophat2.png&quot;,&quot;basename&quot;:&quot;tophat2&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tophat_interface.png&quot;,&quot;path&quot;:&quot;/shared/images/tophat_interface.png&quot;,&quot;basename&quot;:&quot;tophat_interface&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tophat_output.png&quot;,&quot;path&quot;:&quot;/shared/images/tophat_output.png&quot;,&quot;basename&quot;:&quot;tophat_output&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;training_infra.svg&quot;,&quot;path&quot;:&quot;/shared/images/training_infra.svg&quot;,&quot;basename&quot;:&quot;training_infra&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;trie.png&quot;,&quot;path&quot;:&quot;/shared/images/trie.png&quot;,&quot;basename&quot;:&quot;trie&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;trie_no_end.png&quot;,&quot;path&quot;:&quot;/shared/images/trie_no_end.png&quot;,&quot;basename&quot;:&quot;trie_no_end&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_footer.png&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_footer.png&quot;,&quot;basename&quot;:&quot;tutorial_footer&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_footer.svg&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_footer.svg&quot;,&quot;basename&quot;:&quot;tutorial_footer&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_header.png&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_header.png&quot;,&quot;basename&quot;:&quot;tutorial_header&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_introduction.png&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_introduction.png&quot;,&quot;basename&quot;:&quot;tutorial_introduction&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_introduction.svg&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_introduction.svg&quot;,&quot;basename&quot;:&quot;tutorial_introduction&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_part.png&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_part.png&quot;,&quot;basename&quot;:&quot;tutorial_part&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;tutorial_part.svg&quot;,&quot;path&quot;:&quot;/shared/images/tutorial_part.svg&quot;,&quot;basename&quot;:&quot;tutorial_part&quot;,&quot;extname&quot;:&quot;.svg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;ucsc_dm3.png&quot;,&quot;path&quot;:&quot;/shared/images/ucsc_dm3.png&quot;,&quot;basename&quot;:&quot;ucsc_dm3&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;unification_scheme.graphml&quot;,&quot;path&quot;:&quot;/shared/images/unification_scheme.graphml&quot;,&quot;basename&quot;:&quot;unification_scheme&quot;,&quot;extname&quot;:&quot;.graphml&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;unification_scheme.png&quot;,&quot;path&quot;:&quot;/shared/images/unification_scheme.png&quot;,&quot;basename&quot;:&quot;unification_scheme&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;vcfallelicprimitives.png&quot;,&quot;path&quot;:&quot;/shared/images/vcfallelicprimitives.png&quot;,&quot;basename&quot;:&quot;vcfallelicprimitives&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;viewatphinch.png&quot;,&quot;path&quot;:&quot;/shared/images/viewatphinch.png&quot;,&quot;basename&quot;:&quot;viewatphinch&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;we_need_you.jpg&quot;,&quot;path&quot;:&quot;/shared/images/we_need_you.jpg&quot;,&quot;basename&quot;:&quot;we_need_you&quot;,&quot;extname&quot;:&quot;.jpg&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;within_norm.png&quot;,&quot;path&quot;:&quot;/shared/images/within_norm.png&quot;,&quot;basename&quot;:&quot;within_norm&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;workflow_editor_mark_output.png&quot;,&quot;path&quot;:&quot;/shared/images/workflow_editor_mark_output.png&quot;,&quot;basename&quot;:&quot;workflow_editor_mark_output&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;workflow_editor_save.png&quot;,&quot;path&quot;:&quot;/shared/images/workflow_editor_save.png&quot;,&quot;basename&quot;:&quot;workflow_editor_save&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;literature.md&quot;,&quot;path&quot;:&quot;/shared/literature.md&quot;,&quot;basename&quot;:&quot;literature&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;add_custom_build.md&quot;,&quot;path&quot;:&quot;/snippets/add_custom_build.md&quot;,&quot;basename&quot;:&quot;add_custom_build&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;add_tag.md&quot;,&quot;path&quot;:&quot;/snippets/add_tag.md&quot;,&quot;basename&quot;:&quot;add_tag&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;ansible_local.md&quot;,&quot;path&quot;:&quot;/snippets/ansible_local.md&quot;,&quot;basename&quot;:&quot;ansible_local&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;build_dataset_list.md&quot;,&quot;path&quot;:&quot;/snippets/build_dataset_list.md&quot;,&quot;basename&quot;:&quot;build_dataset_list&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;build_list_collection.md&quot;,&quot;path&quot;:&quot;/snippets/build_list_collection.md&quot;,&quot;basename&quot;:&quot;build_list_collection&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;change_datatype.md&quot;,&quot;path&quot;:&quot;/snippets/change_datatype.md&quot;,&quot;basename&quot;:&quot;change_datatype&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;change_dbkey.md&quot;,&quot;path&quot;:&quot;/snippets/change_dbkey.md&quot;,&quot;basename&quot;:&quot;change_dbkey&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;create_dataset_collection.md&quot;,&quot;path&quot;:&quot;/snippets/create_dataset_collection.md&quot;,&quot;basename&quot;:&quot;create_dataset_collection&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;create_new_file.md&quot;,&quot;path&quot;:&quot;/snippets/create_new_file.md&quot;,&quot;basename&quot;:&quot;create_new_file&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;create_new_history.md&quot;,&quot;path&quot;:&quot;/snippets/create_new_history.md&quot;,&quot;basename&quot;:&quot;create_new_history&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;display_extra_training.md&quot;,&quot;path&quot;:&quot;/snippets/display_extra_training.md&quot;,&quot;basename&quot;:&quot;display_extra_training&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;display_extra_training_slides.md&quot;,&quot;path&quot;:&quot;/snippets/display_extra_training_slides.md&quot;,&quot;basename&quot;:&quot;display_extra_training_slides&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;extra_protein.md&quot;,&quot;path&quot;:&quot;/snippets/extra_protein.md&quot;,&quot;basename&quot;:&quot;extra_protein&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;extract_workflow.md&quot;,&quot;path&quot;:&quot;/snippets/extract_workflow.md&quot;,&quot;basename&quot;:&quot;extract_workflow&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;history_create_new.md&quot;,&quot;path&quot;:&quot;/snippets/history_create_new.md&quot;,&quot;basename&quot;:&quot;history_create_new&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;import_from_data_library.md&quot;,&quot;path&quot;:&quot;/snippets/import_from_data_library.md&quot;,&quot;basename&quot;:&quot;import_from_data_library&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;import_via_link.md&quot;,&quot;path&quot;:&quot;/snippets/import_via_link.md&quot;,&quot;basename&quot;:&quot;import_via_link&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;import_workflow.md&quot;,&quot;path&quot;:&quot;/snippets/import_workflow.md&quot;,&quot;basename&quot;:&quot;import_workflow&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rename_dataset.md&quot;,&quot;path&quot;:&quot;/snippets/rename_dataset.md&quot;,&quot;basename&quot;:&quot;rename_dataset&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;rename_history.md&quot;,&quot;path&quot;:&quot;/snippets/rename_history.md&quot;,&quot;basename&quot;:&quot;rename_history&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;run_workflow.md&quot;,&quot;path&quot;:&quot;/snippets/run_workflow.md&quot;,&quot;basename&quot;:&quot;run_workflow&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;select_collection.md&quot;,&quot;path&quot;:&quot;/snippets/select_collection.md&quot;,&quot;basename&quot;:&quot;select_collection&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;select_multiple_datasets.md&quot;,&quot;path&quot;:&quot;/snippets/select_multiple_datasets.md&quot;,&quot;basename&quot;:&quot;select_multiple_datasets&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;use_scratchbook.md&quot;,&quot;path&quot;:&quot;/snippets/use_scratchbook.md&quot;,&quot;basename&quot;:&quot;use_scratchbook&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;warning_results_may_vary.md&quot;,&quot;path&quot;:&quot;/snippets/warning_results_may_vary.md&quot;,&quot;basename&quot;:&quot;warning_results_may_vary&quot;,&quot;extname&quot;:&quot;.md&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Dockerfile&quot;,&quot;path&quot;:&quot;/topics/basic-bioinformatics/docker/Dockerfile&quot;,&quot;basename&quot;:&quot;Dockerfile&quot;,&quot;extname&quot;:&quot;&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;assembly-general-introduction.ga&quot;,&quot;path&quot;:&quot;/topics/basic-bioinformatics/tutorials/general-introduction/workflows/assembly-general-introduction.ga&quot;,&quot;basename&quot;:&quot;assembly-general-introduction&quot;,&quot;extname&quot;:&quot;.ga&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Dockerfile&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/docker/Dockerfile&quot;,&quot;basename&quot;:&quot;Dockerfile&quot;,&quot;extname&quot;:&quot;&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Blastpdb.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Blastpdb.png&quot;,&quot;basename&quot;:&quot;Blastpdb&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Mol_desc_1DKX.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Mol_desc_1DKX.png&quot;,&quot;basename&quot;:&quot;Mol_desc_1DKX&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdb_expdetails_1dkx.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdb_expdetails_1dkx.png&quot;,&quot;basename&quot;:&quot;Pdb_expdetails_1dkx&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdb_firstresiduesmissing_1dkx.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdb_firstresiduesmissing_1dkx.png&quot;,&quot;basename&quot;:&quot;Pdb_firstresiduesmissing_1dkx&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdb_seqtab.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdb_seqtab.png&quot;,&quot;basename&quot;:&quot;Pdb_seqtab&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdbdownloadfile1.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdbdownloadfile1.png&quot;,&quot;basename&quot;:&quot;Pdbdownloadfile1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdbsearchbox_RCSB.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdbsearchbox_RCSB.png&quot;,&quot;basename&quot;:&quot;Pdbsearchbox_RCSB&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;Pdf_uniprotview_button.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/Pdf_uniprotview_button.png&quot;,&quot;basename&quot;:&quot;Pdf_uniprotview_button&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;blastpdb.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/blastpdb.png&quot;,&quot;basename&quot;:&quot;blastpdb&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pdbdownloadfile1.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/pdbdownloadfile1.png&quot;,&quot;basename&quot;:&quot;pdbdownloadfile1&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;pdbsearchbox_RCSB.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/pdbsearchbox_RCSB.png&quot;,&quot;basename&quot;:&quot;pdbsearchbox_RCSB&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;},{&quot;name&quot;:&quot;uniprotsearchbox.png&quot;,&quot;path&quot;:&quot;/topics/protein-structure-analysis/images/uniprotsearchbox.png&quot;,&quot;basename&quot;:&quot;uniprotsearchbox&quot;,&quot;extname&quot;:&quot;.png&quot;,&quot;collection&quot;:null,&quot;modified_time&quot;:&quot;2019-07-31 08:24:49 +0000&quot;}],&quot;documents&quot;:[],&quot;categories&quot;:{},&quot;collections&quot;:[{&quot;label&quot;:&quot;posts&quot;,&quot;files&quot;:[],&quot;directory&quot;:&quot;/home/albot/alex/training-material/_posts&quot;,&quot;output&quot;:true,&quot;docs&quot;:[],&quot;relative_directory&quot;:&quot;_posts&quot;,&quot;permalink&quot;:&quot;/:categories/:year/:month/:day/:title:output_ext&quot;}],&quot;html_pages&quot;:[&quot;\n&quot;,&quot;&lt;ol id=\&quot;markdown-toc\&quot;&gt;\n  &lt;li&gt;&lt;a href=\&quot;#overview-questions\&quot; id=\&quot;markdown-toc-overview-questions\&quot;&gt;Overview Questions&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-is-this-website\&quot; id=\&quot;markdown-toc-what-is-this-website\&quot;&gt;What is this website?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-is-galaxy\&quot; id=\&quot;markdown-toc-what-is-galaxy\&quot;&gt;What is Galaxy?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-are-the-tutorials-for\&quot; id=\&quot;markdown-toc-what-are-the-tutorials-for\&quot;&gt;What are the tutorials for?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-audiences-are-the-tutorials-for\&quot; id=\&quot;markdown-toc-what-audiences-are-the-tutorials-for\&quot;&gt;What audiences are the tutorials for?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-is-the-content-licensed\&quot; id=\&quot;markdown-toc-how-is-the-content-licensed\&quot;&gt;How is the content licensed?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-cite-this-project\&quot; id=\&quot;markdown-toc-how-can-i-cite-this-project\&quot;&gt;How can I cite this project?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-advertise-the-training-materials-on-my-posters\&quot; id=\&quot;markdown-toc-how-can-i-advertise-the-training-materials-on-my-posters\&quot;&gt;How can I advertise the training materials on my posters?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#for-individual-learners\&quot; id=\&quot;markdown-toc-for-individual-learners\&quot;&gt;For Individual Learners&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-do-i-start\&quot; id=\&quot;markdown-toc-where-do-i-start\&quot;&gt;Where do I start?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-do-i-use-this-material\&quot; id=\&quot;markdown-toc-how-do-i-use-this-material\&quot;&gt;How do I use this material?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-can-i-run-the-hands-on-tutorials\&quot; id=\&quot;markdown-toc-where-can-i-run-the-hands-on-tutorials\&quot;&gt;Where can I run the hands-on tutorials?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-get-help\&quot; id=\&quot;markdown-toc-how-can-i-get-help\&quot;&gt;How can I get help?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#for-instructors\&quot; id=\&quot;markdown-toc-for-instructors\&quot;&gt;For Instructors&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#where-do-i-start-1\&quot; id=\&quot;markdown-toc-where-do-i-start-1\&quot;&gt;Where do I start?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#what-galaxy-instance-should-i-use-for-my-training\&quot; id=\&quot;markdown-toc-what-galaxy-instance-should-i-use-for-my-training\&quot;&gt;What Galaxy instance should I use for my training?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-do-i-get-help\&quot; id=\&quot;markdown-toc-how-do-i-get-help\&quot;&gt;How do I get help?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#contributing\&quot; id=\&quot;markdown-toc-contributing\&quot;&gt;Contributing&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-get-started-with-contributing\&quot; id=\&quot;markdown-toc-how-can-i-get-started-with-contributing\&quot;&gt;How can I get started with contributing?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-give-feedback\&quot; id=\&quot;markdown-toc-how-can-i-give-feedback\&quot;&gt;How can I give feedback?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-test-an-interactive-tour\&quot; id=\&quot;markdown-toc-how-can-i-test-an-interactive-tour\&quot;&gt;How can I test an Interactive Tour?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot; id=\&quot;markdown-toc-how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot;&gt;How can I fix mistakes or expand an existing tutorial using the GitHub interface?&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-contribute-in-advanced-mode\&quot; id=\&quot;markdown-toc-how-can-i-contribute-in-advanced-mode\&quot;&gt;How can I contribute in “advanced” mode?&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;a href=\&quot;#other-questions\&quot; id=\&quot;markdown-toc-other-questions\&quot;&gt;Other Questions&lt;/a&gt;    &lt;ol&gt;\n      &lt;li&gt;&lt;a href=\&quot;#sustainability-of-the-training-material-and-metadata\&quot; id=\&quot;markdown-toc-sustainability-of-the-training-material-and-metadata\&quot;&gt;Sustainability of the training-material and metadata&lt;/a&gt;&lt;/li&gt;\n    &lt;/ol&gt;\n  &lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h1 id=\&quot;overview-questions\&quot;&gt;Overview Questions&lt;/h1&gt;\n\n&lt;h2 id=\&quot;what-is-this-website\&quot;&gt;What is this website?&lt;/h2&gt;\n\n&lt;p&gt;This website is a collection of hands-on tutorials that are designed to be interactive and are built around Galaxy:&lt;/p&gt;\n\n&lt;figure id=\&quot;figure-1\&quot;&gt;&lt;img src=\&quot;shared/images/interactive_training.png\&quot; alt=\&quot;Interactive training\&quot; /&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 1:&lt;/span&gt; The hands-on nature of our training material. It can be used with two web browser windows open side-by-side, one pointed at the current tutorial and the other at a Galaxy instance&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;p&gt;This material is developed and maintained by the &lt;a href=\&quot;https://galaxyproject.org/\&quot;&gt;worldwide Galaxy community&lt;/a&gt;. You can learn more about this effort by reading our &lt;a href=\&quot;https://doi.org/10.1016/j.cels.2018.05.012\&quot;&gt;article&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-is-galaxy\&quot;&gt;What is Galaxy?&lt;/h2&gt;\n\n&lt;p&gt;&lt;a href=\&quot;https://galaxyproject.org/\&quot;&gt;Galaxy&lt;/a&gt; is an open data integration and analysis platform for the life sciences, and it is particularly well-suited for data analysis training in life science research.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-are-the-tutorials-for\&quot;&gt;What are the tutorials for?&lt;/h2&gt;\n\n&lt;p&gt;These tutorials can be used for learning and teaching how to use Galaxy for general data analysis, and for learning/teaching specific domains such as assembly and differential gene expression analysis with RNA-Seq data.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-audiences-are-the-tutorials-for\&quot;&gt;What audiences are the tutorials for?&lt;/h2&gt;\n\n&lt;p&gt;There are two distinct audiences for these materials.&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;&lt;strong&gt;Self-paced individual learners.&lt;/strong&gt; These tutorials provide everything you need to learn a topic, from explanations of concepts to detailed hands-on exercises.&lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Instructors.&lt;/strong&gt; They are also designed to be used by instructors in teaching/training settings. Slides, and detailed tutorials are provided. Most tutorials also include computational support with the needed tools, data as well as Docker images that can be used to scale the lessons up to many participants.&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h2 id=\&quot;how-is-the-content-licensed\&quot;&gt;How is the content licensed?&lt;/h2&gt;\n\n&lt;p&gt;The content of this website is licensed under the &lt;a href=\&quot;https://creativecommons.org/licenses/by/4.0/\&quot;&gt;Creative Commons Attribution 4.0 License&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-cite-this-project\&quot;&gt;How can I cite this project?&lt;/h2&gt;\n\n&lt;p&gt;We wrote an &lt;a href=\&quot;https://doi.org/10.1016/j.cels.2018.05.012\&quot;&gt;article&lt;/a&gt; about our efforts.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-advertise-the-training-materials-on-my-posters\&quot;&gt;How can I advertise the training materials on my posters?&lt;/h2&gt;\n\n&lt;p&gt;We provide some QR codes and logos in the &lt;a href=\&quot;https://github.com/vibbits/training-material/tree/master/assets/images\&quot;&gt;images folder&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;for-individual-learners\&quot;&gt;For Individual Learners&lt;/h1&gt;\n\n&lt;p&gt;Learning Galaxy and data analysis on your own and at your own pace?  This material is for you.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-do-i-start\&quot;&gt;Where do I start?&lt;/h2&gt;\n\n&lt;p&gt;If you are new to Galaxy then start with one of the introductory topics.  These introduce you to concepts that are useful in Galaxy, no matter what domain you are doing analysis in.&lt;/p&gt;\n\n&lt;p&gt;If you are already familiar with Galaxy basics and want to learn how to use it in a particular domain (for example, ChIP-Seq), then start with one of those topics.&lt;/p&gt;\n\n&lt;p&gt;If you are already well informed about bioinformatics data analysis and you just want to get a feel for how it works in Galaxy, then many tutorials include &lt;em&gt;Instructions for the impatient&lt;/em&gt; sections.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-do-i-use-this-material\&quot;&gt;How do I use this material?&lt;/h2&gt;\n\n&lt;p&gt;Many topics include slide decks and if the topic you are interested in has slides then start there.  These will introduce the topic and important concepts.&lt;/p&gt;\n\n&lt;p&gt;Most of your learning will happen in the next step - the hands-on tutorials.  This is where you’ll become familiar with using the Galaxy interface and experiment with different ways to use Galaxy and the tools in Galaxy.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-can-i-run-the-hands-on-tutorials\&quot;&gt;Where can I run the hands-on tutorials?&lt;/h2&gt;\n\n&lt;p&gt;To run the hands-on tutorials you need a Galaxy server to run them on.&lt;/p&gt;\n\n&lt;p&gt;Each tutorial is annotated with information about which &lt;a href=\&quot;https://galaxyproject.org/public-galaxy-servers/\&quot;&gt;public Galaxy servers&lt;/a&gt; it can be run on. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial.&lt;/p&gt;\n\n&lt;p&gt;Some topics have a &lt;a href=\&quot;https://www.docker.com/\&quot;&gt;Docker&lt;/a&gt; image that can be installed and run on participants’ laptops.  These Docker images contain Galaxy instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant. You will need to &lt;a href=\&quot;https://docs.docker.com/install/\&quot;&gt;install Docker&lt;/a&gt;.&lt;/p&gt;\n\n&lt;p&gt;Finally, you can also run your tutorials on cloud-based infrastructures.  Galaxy is &lt;a href=\&quot;https://galaxyproject.org/galaxy-services/\&quot;&gt;available on many national research infrastructures&lt;/a&gt; such as &lt;a href=\&quot;https://galaxyproject.org/cloud/jetstream/\&quot;&gt;Jetstream&lt;/a&gt; (United States), &lt;a href=\&quot;https://www.genap.ca/\&quot;&gt;GenAP&lt;/a&gt; (Canada), &lt;a href=\&quot;https://www.gvl.org.au/\&quot;&gt;GVL&lt;/a&gt; (Australia), &lt;a href=\&quot;http://www.climb.ac.uk/\&quot;&gt;CLIMB&lt;/a&gt; (United Kingdom), and more.  These instances are typically easy to launch, and easy to shut down when you are done.&lt;/p&gt;\n\n&lt;p&gt;If you are already familiar with, and have an account on &lt;a href=\&quot;https://aws.amazon.com/\&quot;&gt;Amazon Web Services&lt;/a&gt; then you can also launch a Galaxy server there using &lt;a href=\&quot;https://launch.usegalaxy.org/\&quot;&gt;CloudLaunch&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-get-help\&quot;&gt;How can I get help?&lt;/h2&gt;\n\n&lt;p&gt;If you have questions about this training material, you can reach us using the &lt;a href=\&quot;\&quot;&gt;Gitter chat&lt;/a&gt;.  You’ll need a &lt;a href=\&quot;https://github.com/\&quot;&gt;GitHub&lt;/a&gt; or &lt;a href=\&quot;https://twitter.com/\&quot;&gt;Twitter&lt;/a&gt; account to post questions.  If you have questions about Galaxy outside the context of training, see the &lt;a href=\&quot;https://galaxyproject.org/support/\&quot;&gt;Galaxy Support page&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;for-instructors\&quot;&gt;For Instructors&lt;/h1&gt;\n\n&lt;p&gt;This material can also be used to teach Galaxy and data analysis in a group setting to students and researchers.&lt;/p&gt;\n\n&lt;h2 id=\&quot;where-do-i-start-1\&quot;&gt;Where do I start?&lt;/h2&gt;\n\n&lt;p&gt;Spend some time exploring the different tutorials and the different resources that are available. Become familiar with the structure of the tutorials and think about how you might use them in your teaching.&lt;/p&gt;\n\n&lt;h2 id=\&quot;what-galaxy-instance-should-i-use-for-my-training\&quot;&gt;What Galaxy instance should I use for my training?&lt;/h2&gt;\n\n&lt;p&gt;To teach the hands-on tutorials you need a Galaxy server to run the examples on.&lt;/p&gt;\n\n&lt;p&gt;Each tutorial is annotated with the information on which &lt;a href=\&quot;https://galaxyproject.org/public-galaxy-servers/\&quot;&gt;public Galaxy servers&lt;/a&gt; it can be run. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial. If you choose this option then you should work with that server’s admins to confirm that think the server can handle the workload for a workshop. For example, the &lt;a href=\&quot;https://usegalaxy.eu/\&quot;&gt;usegalaxy.eu&lt;/a&gt;&lt;/p&gt;\n\n&lt;p&gt;If your organization/consortia/community has its own Galaxy server, then you may  want to run tutorials on that. This can be ideal because then the instance you are teaching on is the same you your participants will be using after the training. They’ll also be able to revisit any analysis they did during the training. If you pursue this option you’ll need to work with your organization’s Galaxy Admins to confirm that&lt;/p&gt;\n\n&lt;ul&gt;\n  &lt;li&gt;the server can support a room full of people all doing the same analysis at the same time.&lt;/li&gt;\n  &lt;li&gt;all participants will be able to create/use accounts on the system.&lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;Some training topics have a Docker image that can be installed and run on all participants’ laptops.  These images contain Galaxy instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant.&lt;/p&gt;\n\n&lt;p&gt;Finally, you can also run your tutorials on cloud-based infrastructures.  Galaxy is &lt;a href=\&quot;https://galaxyproject.org/galaxy-services/\&quot;&gt;available on many national research infrastructures&lt;/a&gt; such as &lt;a href=\&quot;https://galaxyproject.org/cloud/jetstream/\&quot;&gt;Jetstream&lt;/a&gt; (United States), &lt;a href=\&quot;https://www.genap.ca/\&quot;&gt;GenAP&lt;/a&gt; (Canada), &lt;a href=\&quot;https://launch.genome.edu.au/launch\&quot;&gt;GVL&lt;/a&gt; (Australia), &lt;a href=\&quot;http://www.climb.ac.uk/\&quot;&gt;CLIMB&lt;/a&gt; (United Kingdom), and more.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-do-i-get-help\&quot;&gt;How do I get help?&lt;/h2&gt;\n\n&lt;p&gt;The support channel for instructors is the same as for individual learners.  We suggest you start by posting a question to the Galaxy Training Network &lt;a href=\&quot;\&quot;&gt;Gitter chat&lt;/a&gt;. Anyone can view the discussion, but you’ll need to login (using your GitHub or Twitter account) to add to the discussion.&lt;/p&gt;\n\n&lt;p&gt;If you have questions about Galaxy in general (that are not training-centric) then there are &lt;a href=\&quot;https://galaxyproject.org/support/\&quot;&gt;several support options&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;contributing\&quot;&gt;Contributing&lt;/h1&gt;\n\n&lt;p&gt;First off, thanks for your interest in contributing to the Galaxy training materials!&lt;/p&gt;\n\n&lt;p&gt;Individual learners and instructors can make these training more effective by contributing back to them. You can report mistakes and errors, create more content, etc. Whatever is your background, there is a way to contribute: via the GitHub website, via command-line or even without dealing with GitHub.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-get-started-with-contributing\&quot;&gt;How can I get started with contributing?&lt;/h2&gt;\n\n&lt;p&gt;If you would like to get involved in the project but are unsure where to start, there are some easy ways to contribute which will also help you familiarize yourself with the project.&lt;/p&gt;\n\n&lt;h3 id=\&quot;1-checking-existing-tutorials\&quot;&gt;1. Checking existing tutorials&lt;/h3&gt;\n\n&lt;p&gt;A great way to help out the project is to test/edit existing tutorials. Pick a tutorial and check the contents. Does everything work as expected? Are there things that could be improved?&lt;/p&gt;\n\n&lt;p&gt;Below is a checklist of things to look out for to help you get started. If you feel confident in making changes yourself, please open a pull request, otherwise please file an issue with any problems you run into or suggestions for improvements.&lt;/p&gt;\n\n&lt;p&gt;&lt;em&gt;Basic&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Test&lt;/strong&gt; the tutorial on a running Galaxy instance\n    &lt;ul&gt;\n      &lt;li&gt;For example &lt;a href=\&quot;https://usegalaxy.org\&quot;&gt;Galaxy Main&lt;/a&gt;, &lt;a href=\&quot;https://usegalaxy.eu\&quot;&gt;Galaxy Europe&lt;/a&gt;, or &lt;a href=\&quot;https://usegalaxy.org.au\&quot;&gt;Galaxy Australia&lt;/a&gt;&lt;/li&gt;\n      &lt;li&gt;Report any issues you run into&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Language&lt;/strong&gt; editing\n    &lt;ul&gt;\n      &lt;li&gt;Fix spelling and grammar mistakes&lt;/li&gt;\n      &lt;li&gt;Simplify the English (to make it more accessible)&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;&lt;em&gt;Intermediate&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Metadata&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Are the objectives, keypoints and time estimate filled in?&lt;/li&gt;\n      &lt;li&gt;Do they fit with the contents of the tutorial?&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Content&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Is there enough background information provided in the introduction section and throughout the tutorial?&lt;/li&gt;\n      &lt;li&gt;&lt;strong&gt;Question boxes&lt;/strong&gt;\n        &lt;ul&gt;\n          &lt;li&gt;Add questions or question boxes where you think they might be useful (make people think about results they got, test their understanding, etc)&lt;/li&gt;\n          &lt;li&gt;Check that answers are still up-to-date&lt;/li&gt;\n        &lt;/ul&gt;\n      &lt;/li&gt;\n      &lt;li&gt;&lt;strong&gt;Screenshots and Videos&lt;/strong&gt;\n        &lt;ul&gt;\n          &lt;li&gt;Make sure there is also a textual description of the image/video contents&lt;/li&gt;\n          &lt;li&gt;Does the screenshot add value to the tutorial or can it be removed?&lt;/li&gt;\n        &lt;/ul&gt;\n      &lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;&lt;em&gt;Advanced&lt;/em&gt;&lt;/p&gt;\n&lt;ul&gt;\n  &lt;li&gt;&lt;strong&gt;Workflows&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Add a workflow definition file &lt;code class=\&quot;highlighter-rouge\&quot;&gt;.ga&lt;/code&gt; if none is present&lt;/li&gt;\n      &lt;li&gt;Check that the existing workflow is up-to-date with the tutorial contents&lt;/li&gt;\n      &lt;li&gt;Enable &lt;a href=\&quot;https://github.com/usegalaxy-eu/workflow-testing\&quot;&gt;workflow testing&lt;/a&gt;&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Tours&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Add a tour if none exists&lt;/li&gt;\n      &lt;li&gt;&lt;a href=\&quot;#how-can-i-test-an-interactive-tour\&quot;&gt;Run the existing tour&lt;/a&gt; and check that it is up-to-date with the tutorial contents&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;&lt;strong&gt;Datasets&lt;/strong&gt;\n    &lt;ul&gt;\n      &lt;li&gt;Check that all datasets used in the tutorial are present in Zenodo&lt;/li&gt;\n      &lt;li&gt;Add a data-library.yaml file if none exists&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;h2 id=\&quot;how-can-i-give-feedback\&quot;&gt;How can I give feedback?&lt;/h2&gt;\n\n&lt;p&gt;At the end of each tutorial, there is a link to &lt;a href=\&quot;https://docs.google.com/forms/d/1BILuBp8bPv42odpZ55Kcf2rqzR-k0Czx46agVVSRgm4/edit\&quot;&gt;a feedback form&lt;/a&gt;. We use this information to improve our tutorials.&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-test-an-interactive-tour\&quot;&gt;How can I test an Interactive Tour?&lt;/h2&gt;\n\n&lt;p&gt;Perhaps you’ve been asked to review an interactive tour, or maybe you just want to try one out. The easiest way to run an interactive tour is to use the &lt;a href=\&quot;https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/\&quot;&gt;Tour builder&lt;/a&gt; browser extension.&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;Install the Tour Builder extension to your browser (&lt;a href=\&quot;https://chrome.google.com/webstore/detail/galaxy-tour-builder/mdfbapknmcpnbmggahhaegehbbbmhmgg\&quot;&gt;Chrome Web Store&lt;/a&gt;, &lt;a href=\&quot;https://addons.mozilla.org/en-US/firefox/addon/galaxy-tour-builder/\&quot;&gt;Firefox add-on&lt;/a&gt;, &lt;a href=\&quot;https://addons.opera.com/en/extensions/details/galaxy-tour-builder/\&quot;&gt;Opera add-on&lt;/a&gt;).&lt;/li&gt;\n  &lt;li&gt;Navigate to a Galaxy instance supporting the tutorial. To find which Galaxy instances support each tutorial, please see the dropdown menu next to the tutorial on the training website. Using one of the usegalaxy.* instances (&lt;a href=\&quot;https://usegalaxy.org\&quot;&gt;Galaxy Main&lt;/a&gt;, &lt;a href=\&quot;https://usegalaxy.eu\&quot;&gt;Galaxy Europe&lt;/a&gt;, or &lt;a href=\&quot;https://usegalaxy.org.au\&quot;&gt;Galaxy Australia&lt;/a&gt;\n) is usually a good bet.&lt;/li&gt;\n  &lt;li&gt;Start the Tour Builder plugin by clicking on the icon in your browser menu bar&lt;/li&gt;\n  &lt;li&gt;Copy the contents of the &lt;code class=\&quot;highlighter-rouge\&quot;&gt;tour.yaml&lt;/code&gt; file into the Tour builder editor window&lt;/li&gt;\n  &lt;li&gt;Click &lt;code class=\&quot;highlighter-rouge\&quot;&gt;Save&lt;/code&gt; and then &lt;code class=\&quot;highlighter-rouge\&quot;&gt;Run&lt;/code&gt;&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;h2 id=\&quot;how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface\&quot;&gt;How can I fix mistakes or expand an existing tutorial using the GitHub interface?&lt;/h2&gt;\n\n&lt;p&gt;Check our tutorial to learn how to use the GitHub interface (soon…)&lt;/p&gt;\n\n&lt;h2 id=\&quot;how-can-i-contribute-in-advanced-mode\&quot;&gt;How can I contribute in “advanced” mode?&lt;/h2&gt;\n\n&lt;p&gt;Most of the content is written in &lt;a href=\&quot;https://guides.github.com/features/mastering-markdown/\&quot;&gt;GitHub Flavored Markdown&lt;/a&gt; with some metadata (or variables) found in &lt;a href=\&quot;http://yaml.org/\&quot;&gt;YAML&lt;/a&gt; files. Everything is stored on our &lt;a href=\&quot;https://github.com/vibbits/training-material\&quot;&gt;GitHub repository&lt;/a&gt;. Each training material is related to a topic. All training materials (slides, tutorials, etc) related to a topic are found in a dedicated directory (&lt;em&gt;e.g.&lt;/em&gt; &lt;code class=\&quot;highlighter-rouge\&quot;&gt;transcriptomics&lt;/code&gt; directory contains the material related to transcriptomic analysis). Each topic has the following &lt;a name=\&quot;structure\&quot;&gt;structure&lt;/a&gt;:&lt;/p&gt;\n\n&lt;p&gt;&lt;img src=\&quot;shared/images/repo_organization.png\&quot; alt=\&quot;Structure of the repository\&quot; width=\&quot;400px\&quot; /&gt;&lt;/p&gt;\n\n&lt;ul&gt;\n  &lt;li&gt;a metadata file in YAML format&lt;/li&gt;\n  &lt;li&gt;a directory with the topic introduction slide deck in Markdown with introductions to the topic&lt;/li&gt;\n  &lt;li&gt;\n    &lt;p&gt;a directory with the tutorials:&lt;/p&gt;\n\n    &lt;p&gt;Inside the tutorials directory, each tutorial related to the topic has its own subdirectory with several files:&lt;/p&gt;\n    &lt;ul&gt;\n      &lt;li&gt;a tutorial file written in Markdown with hands-on&lt;/li&gt;\n      &lt;li&gt;an optional slides file in Markdown with slides to support the tutorial&lt;/li&gt;\n      &lt;li&gt;a directory with Galaxy Interactive Tours to reproduce the tutorial&lt;/li&gt;\n      &lt;li&gt;a directory with workflows extracted from the tutoria&lt;/li&gt;\n      &lt;li&gt;a YAML file with the links to the input data needed for the tutorial&lt;/li&gt;\n      &lt;li&gt;a YAML file with the description of needed tools to run the tutorial&lt;/li&gt;\n    &lt;/ul&gt;\n  &lt;/li&gt;\n  &lt;li&gt;a directory with the Dockerfile describing the details to build a container for the topic (self-study environments).&lt;/li&gt;\n&lt;/ul&gt;\n\n&lt;p&gt;To manage changes, we use &lt;a href=\&quot;https://guides.github.com/introduction/flow/\&quot;&gt;GitHub flow&lt;/a&gt; based on Pull Requests:&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;&lt;a href=\&quot;https://help.github.com/articles/fork-a-repo/\&quot;&gt;Create a fork&lt;/a&gt; of this repository on GitHub&lt;/li&gt;\n  &lt;li&gt;Clone your fork of this repository to create a local copy on your computer and initialize the required submodules (&lt;code class=\&quot;highlighter-rouge\&quot;&gt;git submodule init&lt;/code&gt; and &lt;code class=\&quot;highlighter-rouge\&quot;&gt;git submodule update&lt;/code&gt;)&lt;/li&gt;\n  &lt;li&gt;Create a new branch in your local copy for each significant change&lt;/li&gt;\n  &lt;li&gt;Commit the changes in that branch&lt;/li&gt;\n  &lt;li&gt;Push that branch to your fork on GitHub&lt;/li&gt;\n  &lt;li&gt;Submit a pull request from that branch to the &lt;a href=\&quot;https://github.com/vibbits/training-material\&quot;&gt;original repository&lt;/a&gt;&lt;/li&gt;\n  &lt;li&gt;If you receive feedback, make changes in your local clone and push them to your branch on GitHub: the pull request will update automatically&lt;/li&gt;\n  &lt;li&gt;Pull requests will be merged by the training team members after at least one other person has reviewed the Pull request and approved it.&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;p&gt;Globally, the process of development of new content is open and transparent:&lt;/p&gt;\n\n&lt;ol&gt;\n  &lt;li&gt;Creation of a branch derived from the master branch of the GitHub repository&lt;/li&gt;\n  &lt;li&gt;Initialization of a new directory for the tutorial&lt;/li&gt;\n  &lt;li&gt;Filling of the metadata with title, questions, learning objectives, etc&lt;/li&gt;\n  &lt;li&gt;Generation of the input dataset for the tutorial&lt;/li&gt;\n  &lt;li&gt;Filling of the tutorial content&lt;/li&gt;\n  &lt;li&gt;Extraction of the workflows of the tutorial&lt;/li&gt;\n  &lt;li&gt;Automatic extraction of the required tools to populate the tool file&lt;/li&gt;\n  &lt;li&gt;Automatic annotation of the public Galaxy servers&lt;/li&gt;\n  &lt;li&gt;Generation of an interactive tour for the tutorial with the &lt;a href=\&quot;https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/\&quot;&gt;Tourbuilder&lt;/a&gt; web-browser extension&lt;/li&gt;\n  &lt;li&gt;Upload of the datasets to Zenodo and addition of the links in the data library file.&lt;/li&gt;\n  &lt;li&gt;Once ready, opening a Pull Request&lt;/li&gt;\n  &lt;li&gt;Automatic checks of the changes are automatically checked for the right format and working links using continuous integration testing on Travis CI&lt;/li&gt;\n  &lt;li&gt;Review of the content by several other instructors via discussions&lt;/li&gt;\n  &lt;li&gt;After the review process, merge of the content into the main branch, starting a series of automatic steps triggered by Travis CI&lt;/li&gt;\n  &lt;li&gt;Regeneration of the website and publication on &lt;a href=\&quot;/training-material/\&quot;&gt;https://vibbits.github.io//training-material/&lt;/a&gt;&lt;/li&gt;\n  &lt;li&gt;Generation of PDF artifacts of the tutorials and slides and upload on the FTP server&lt;/li&gt;\n  &lt;li&gt;Population of &lt;a href=\&quot;https://tess.elixir-europe.org/\&quot;&gt;TeSS&lt;/a&gt;, the ELIXIR’s Training Portal, via the metadata&lt;/li&gt;\n&lt;/ol&gt;\n\n&lt;p&gt;&lt;img src=\&quot;shared/images/development_process.png\&quot; alt=\&quot;Development process\&quot; /&gt;&lt;/p&gt;\n\n&lt;ol&gt;\n\n\n &lt;/ol&gt;\n\n&lt;p&gt;We also strongly recommend you read and follow &lt;a href=\&quot;https://carpentries.org/\&quot;&gt;The Carpentries&lt;/a&gt; recommendations on &lt;a href=\&quot;https://carpentries.github.io/lesson-example/01-design/\&quot;&gt;lesson design&lt;/a&gt; and &lt;a href=\&quot;https://carpentries.github.io/instructor-training/15-lesson-study/\&quot;&gt;lesson writing&lt;/a&gt; if you plan to add or change some training materials, and also to check the &lt;a href=\&quot;#structure\&quot;&gt;structure of the training material&lt;/a&gt;.&lt;/p&gt;\n\n&lt;h1 id=\&quot;other-questions\&quot;&gt;Other Questions&lt;/h1&gt;\n\n&lt;p&gt;Yes, always! Have a look at the &lt;a href=\&quot;https://galaxyproject.org/events/\&quot;&gt;Galaxy Community Events Calendar&lt;/a&gt; for what coming up right now.&lt;/p&gt;\n\n&lt;h2 id=\&quot;sustainability-of-the-training-material-and-metadata\&quot;&gt;Sustainability of the training-material and metadata&lt;/h2&gt;\n\n&lt;p&gt;This repository is hosted on &lt;a href=\&quot;https://github.com/\&quot;&gt;GitHub&lt;/a&gt; using git as a &lt;a href=\&quot;https://en.wikipedia.org/wiki/Distributed_version_control\&quot;&gt;DVCS&lt;/a&gt;. Therefore the community is hosting backups of this\nrepository in a decentralised way. The repository is self-contained and contains all needed content and all metadata.\nIn addition we mirror snapshops of this repo on &lt;a href=\&quot;http://zenodo.org\&quot;&gt;Zenodo&lt;/a&gt;.&lt;/p&gt;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;\n&quot;,&quot;### How to fill the slide decks?\n\nPlease follow our\n[tutorial to learn how to fill the slides](/training-material/topics/contributing/tutorials/create-new-tutorial-slides/slides.html)\n&quot;,&quot;## Protein Structure Analysis ##\n\n- Sequences, structures and databases\n- Experimental methods (X-rays, electrons and NMR)\n- Finding and visualising structures from the  Protein Data Bank\n- Comparing structures\n- Modelling mutations\n- Creating homology models\n\n---\n\n## The Structure-Function Connection ##\n\n.left[![Left-aligned image](/training-material/topics/protein-structure-analysis/images/hemoglobin.png)]\n.right[Folded proteins provide a well-  defined 3D  arrangement of functional groups, creating microenvironments and active sites.\nStructural changes  are often involved in functional  mechanisms  (motor proteins, ...)\n]\n\n---\n\n### Databases\n\n.pull-left[\n**Uniprot** approx. 1100,000 sequences\n\n- mainly determined by large-scale DNA sequencing of individual genes or whole genomes\n- increasingly automated annotation\n\n**Protein Data Bank** approx. 141,000 \nexperimentally determined structures\n\n- Mainly determined by X-ray crystallography and high-resolution NMR spectroscopy\n- Automation is increasing, but there are still significant limitations in the rate of solving new structures\n\n\n]\n.pull-right[ ![](/training-material/assets/images/GTN-60px.png) ]\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### The Protein Databank ###\n\n.pull-left[ ![](/training-material/assets/images/GTN-60px.png) ] \n\n\n.pull-right[\n[http://www.wwpdb.org](http://www.wwpdb.org)\n- contains structures of  proteins, nucleic acids  and complexes,  determined by X-ray  crystallography, NMR  spectroscopy\n- No purely theoretical  or ab initio models  (since 2006)\n- Also stores supporting  experimental data\n- Full deposition now  required by all peer-reviewed journals\n\n---\n\n### Exercise 1: Search the PDB ###\n\n- Use the UniProt site to search for “dnak”.  \n- Use the PDB site to search for “dnak”.\n- Compare the UniProt and PDB result lists.\n- Use the sequence search function to see if there are structures with sequences similar to that of the  DnaK C-terminal domain.\n- Look at the summary pages of a number of  structures and note some interesting properties.\n- Select a number of structures and create a report  page.\n\n---\n\n### PDB File Organisation ###\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- Nucleic Acid Database: DNA and RNA structures\n\n[http://ndbserver.rutgers.edu/](http://ndbserver.rutgers.edu/)\n\n- PDB-REDO: automatically re-refined deposited  structures, using the latest methods\n\n[http://www.cmbi.ru.nl/pdb_redo/](http://www.cmbi.ru.nl/pdb_redo)\n\n- EBI: many online tools for structure analysis\n\n[http://www.ebi.ac.uk/Tools/structure/](http://www.ebi.ac.uk/Tools/structure/)\n\n- Replaced Electron Density Server: convenient overview of  quality parameters for crystal structures\n\n[http://www.ebi.ac.uk/pdbe/litemol](http://www.ebi.ac.uk/pdbe/litemol)\n\n---\n\n### High-Resolution NMR Spectrometry ###\n\nMany atomic nuclei, including the ubiquitous hydrogen nuclei,  resonate at specific radio frequencies when placed in a  strong, uniform magnetic field. The chemical environment of each individual atom slightly modulates its exact resonance  frequency.\n\n![]()\n\nIn macromolecules with thousands of atoms, many different  effects combine to generate an extremely complicated pattern of chemical shifts, which therefore more or less uniquely  identify each atom. Multidimensional spectra allow these  frequencies to be assigned to specific atoms.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nWhen two atoms are near each other in 3D space, they can exchange magnetization, giving rise to crosspeaks at the  intersection of their respective frequencies.\n\n![]()\n\nThis nuclear Overhauser effect (NOE) is used to effectively measure the distances between pairs of atoms, at least  qualitatively.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nAfter identification of the atoms by means of their unique chemical shifts, distance restraints are derived from the  Overhauser crosspeaks. An extended model of the protein  is  generated,  and  then  condensed  into  a  shape  that  is consistent with as many of distance restraints as possible.\n\n![]()\n\n---\n\n### Other methods ###\n\nElectron microscopy (and especially Cryo-electron microscopy): Electron crystallography and single particle reconstruction\r\n\nSmall-angle X-ray and neutron scattering (SAXS and SANS)\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- BioMagResBank: experimental data for NMR-  derived structures (lists of distance restraints  and other experimentally derived properties)\n\n[http://www.bmrb.wisc.edu/](http://www.bmrb.wisc.edu/)\n\n- BioIsis: database of SAXS-derived structures\n\n[http://www.bioisis.net](http://www.bioisis.net)\n\n- EMBL database of SAXS-derived structures\n\n[http://www.sasbdb.org](http://www.sasbdb.org)\n\n- EM Databank for cryo-EM structures\n\n[http://www.emdatabank.org](http://www.emdatabank.org)\n\n---\n\n### Assessing Structure Quality ###\n\nGeneral geometric properties (bond lengths and angles,  Ramachandran distribution, …): MolProbity  [Link](http://molprobity.biochem.duke.edu/)\n\nCrystal Structures\n- Diffraction data resolution and completeness (PDB)\n- Final Rcryst and Rfree factors (PDB)\n- Local correspondence to electron density (EDS)\n\nNMR Structures\n- Number of distance restraints and additional experimental  data sources (BMRB)\n- Local restraint density (on-line NMR constraint analyser)\n[link](http://molsim.sci.univr.it/bioinfo/tools/constraint/index.html)\n\nOther techniques\n- Difficult to generalise: carefully read and consider the  published protocol\n\n---\n\n### Molecular Graphics Software ###\n\n- [PyMOL](http://www.pymol.org/): high-quality output,  good examples on [Wiki](http://www.pymolwiki.org/)\n- [Chimera](http://www.cgl.ucsf.edu/chimera/): good  documentation on website\n- [VMD](http://www.ks.uiuc.edu/Research/vmd/):  excellent for the analysis of MD trajectories\n- [Yasara](http://www.yasara.org)  \n- [SwissPDBViewer](http://spdbv.vital-it.ch/)\n\n---\n\n### YASARA ###\n\nYasara View is freely available and provides  basic visualisation functions.\n\nYasara Model, Dynamics and Structure provide  additional functionality.\n\nAn add-on module for NMR structures is  available.\n\nThe program can be combined with the WHAT-  IF program for structure validation, and with  FoldX for energy calculations.\n\n---\n\n### Exercise 2: show a structure ###\n\nLoad PDB entry 1TRZ using the File menu.\n\nCompare the default representations (F1-F8)  and use the various controls to change the view  of the protein.\n\nExplore the Edit and View menus to change  various aspects of the graphical representation  of the structure.\n\nExamine the hydrogen bonding patterns.  Display a molecular surface.\n\nCreate an interesting view of the molecule and  save it as a high-resolution image.\n\n---\n\n### Structure Superposition ###\n\nStructures can be superimposed to achieve the best  possible match between corresponding atoms. It is  possible to consider all atoms, or only a subset  (such as the Cα atoms).\n\nWhen the structures are very similar, determining  which atoms should be matched to each other is  trivial. When there are larger differences, it takes  more preparation.\n\nDifferent algorithms use different combinations of  sequence- and secondary of tertiary structure-based  information.\n\n---\n\n### Exercise 3: Compare Structures ###\n\nDownload the five provided PDB files and open  them in Yasara.\n\nUse the `Analyze|Align|Objects` with MUSTANG  function to align the four last objects with the first one.\n\nUse the space bar to open the text console and  see the reported root mean square deviations\n\n\nas well as the number of residues matched.\n\nColor all structures by B factor and compare the  distribution to the local variability of the structures.\n\n---\n\n### PDB File Surprises ###\n\n- Missing atoms, residues and loops  \n- Multiple molecules in the asymmetric unit\n- Incomplete oligomers due to coincident crystal and  oligomer symmetry\n- Cα-only models  lternate conformations\n- Multiple models, especially for NMR ensembles\n- Use of B factors to represent other properties  ther non-standard extensions (PDBQT, ...)\n\n---\n\n### Force Fields ###\n\n- Energy terms representing physical interactions\n\nCovalent bond lengths and angles\nDihedral angles and van der Waals forces (steric effects)\nElectrostatic interactions and hydrogen bonds\n…\n\n- Energy can be minimized, and forces can literally be derived from the potential function.\n\n- Consistency and careful consideration of the  properties to be simulated are essential.\n\n---\n\n### Force Field Terms ###\n\nEach energy term has a functional form, which includes one or more parameters:\n\n- Covalent bond energy term\n To do: add formula\n- Van der Waals contact energy term\n\nThe parameters are collectively optimized to  reproduce a chosen set of experimentally  observed parameters.\n\nA given force field should be used as a  consistent system, and can only be used to  predict properties that are covered by the  training set.\n\n---\n\n### FoldX ###\n\n\ns designed for quantitative modelling of the  contribution of structural interactions to the stability of  proteins and protein complexes. It also supports  protein/DNA complexes.\n\nThe force field describes the different interactions in a  protein structure or complex in analytical terms. It has  been calibrated using a set of experimentally determined  stabilities.\n\nApplications include the optimisation of structures, the calculation of the stability of complexes, and predicting  the effect of amino-acid or base-pair mutations on these  properties.\n\n---\n\n### The FoldX Plugin for Yasara\t###\n\nIn order to make FoldX more accessible and  integrate its functions into Yasara, dr. Joost van  Durme (SWITCH laboratory) made a Yasara plugin  module that can apply FoldX functions to structures  that are loaded as Yasara objects.\nThis greatly simplifies the use of FoldX, and allows  for a quick visual analysis of the resulting changes  in the structures.\nMore information can be found at [wiki](http://foldxyasara.switchlab.org/index.php/) [FoldX](http://foldxsuite.crg.eu/)\n\n---\n\n### Exercise 4a: Repair a PDB File ###\n\n- Load the 1CRN PDB file.\n- Use the “Repair object” option in the Analysis |  FoldX menu to activate the corresponding FoldX  function.\n- Select the object to repair.\n\nThis exports the object as a temporary PDB file,  starts FoldX with the appropriate options, and loads  the repaired PDB file as a new object in Yasara.\n\n- Compare the original and repaired objects.  \n- Describe the changes that were introduced.\n\n---\n\n### Exercise 4b: Model a Mutation ###\n\n- Load the 2AC0.sce Yasara scene file.\n- Set an appropriate structure representation.\n- Locate residue Ala159 using the sequence view,  and right-click to access the `FoldX|Mutate` residue function. Change it to a Trp residue.\n- Look at the effect of the substitution on the  structure, and use the space bar to open the text  console and read the output of the FoldX  calculation.\n- Mutate Arg273 to an alanine side chain. Discuss  the effects of this substitution.\n\n---\n\n### Homology Modelling ###\n\n- When a structure is available for a protein with a  similar sequences, it is possible to predict the  structure of a new sequence with varying degrees of  confidence.\n- All homology modelling procedures start from an  alignment of the template and target sequences.  The quality of this alignment will have a major  impact on the resulting model.\n- Available applications include stand-alone programs  (Modeller, FoldX, …) and web-based services (such as SwissModel).\n\n---\n\n### Exercise 5: Make a Homology  Model using Swiss Model ###\n\nsee []()\n\n---\n\n### Exercise 6: Study Protein-Ligand Interactions ###\n\nsee []()\n\n---\n\n### On-Line References ###\n\n- Crystallography 101 (Bernhard Rupp):  (http://www.ruppweb.org/Xray/101index.html)\n- Protein NMR, A Practical Guide (Vicky Higman) ( http://www.protein-nmr.org.uk/)\n- Model validation course  (http://xray.bmc.uu.se/gerard/embo2001/modval/index.html)\n- Assessing model quality (http://spdbv.vital-it.ch/TheMolecularLevel/ModQual/)\n\n- Lectures by Burkhard Rost on protein structure  prediction (https://www.youtube.com/channel/UCU6j8BG4RbEtTgyIZJ6Vpow)\n\n---\n&quot;,&quot;.enlarge120[\n\n# ***De novo* Genome Assembly**\n\n]\n\n#### With thanks to T Seemann, D Bulach, I Cooke and Simon Gladman\n---\n.enlarge120[\n\n# ***De novo* assembly**\n\n]\n\n.pull-left[\n\n**The process of reconstructing the original DNA sequence from the fragment reads alone.**\n\n* Instinctively like a jigsaw puzzle\n\n  * Find reads which \&quot;fit together\&quot; (overlap)\n  * Could be missing pieces (sequencing bias)\n  * Some pieces will be dirty (sequencing errors)\n\n]\n\n.pull-right[ ![](../../images/Humpty.jpg) ]\n\n---\n\n# **Another View**\n\n![](../../images/newspaper.png)\n\n---\n\n# **Assembly: An Example**\n\n---\n\n# **A small \&quot;genome\&quot;**\n\n![](../../images/shakespear1.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear2.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear3.png)\n\n---\n\n# **Shakespearomics**\n\n![](../../images/shakespear4.png)\n\n---\n\n# **So far, so good!**\n\n---\n\n# **The Awful Truth**\n\n![](../../images/notsimply.png)\n\n## \&quot;Genome assembly is impossible.\&quot; - A/Prof. Mihai Pop\n\n---\n.enlarge120[\n\n# **Why is it so hard?**\n\n]\n\n.pull-left[\n* Millions of pieces\n  * Much, much shorter than the genome\n  * Lots of them look similar\n* Missing pieces\n  * Some parts can&#39;t be sequenced easily\n* Dirty Pieces\n  * Lots of errors in reads\n]\n\n.pull-right[ ![](../../images/worlds_hardest.png) ]\n\n---\n\n# **Assembly recipe**\n\n* Find all overlaps between reads\n  * Hmm, sounds like a lot of work..\n* Build a graph\n  * A picture of the read connections\n* Simplify the graph\n  * Sequencing errors will mess it up a lot\n* Traverse the graph\n  * Trace a sensible path to produce a consensus\n\n---\n\n![](../../images/olc_pic.png)\n\n---\n\n# **A more realistic graph**\n\n![](../../images/real_graph.png)\n\n---\n\n# .image-15[![](../../images/nofun.png)] **What ruins the graph?**\n\n* Read errors\n  * Introduces false edges and nodes\n\n* Non haploid organisms\n  * Heterozygosity causes lots of detours\n\n* Repeats\n  * If they are longer than the read length\n  * Causes nodes to be shared, locality confusion.\n\n---\n\n# **Repeats**\n\n---\n.enlarge120[\n# **What is a repeat?**\n]\n\n.pull-left[\n\n#### ***A segment of DNA which occurs more than once in the genome sequence***\n\n* Very common\n  * Transposons (self replicating genes)\n  * Satellites (repetitive adjacent patterns)\n  * Gene duplications (paralogs)\n\n]\n\n.pull-right[\n\n![](../../images/triplets.png)\n\n]\n\n---\n\n# **Effect on Assembly**\n\n![](../../images/repeat_effect.png)\n\n---\n.enlarge120[\n# **The law of repeats** .image-15[![](../../images/repeatafterme.png)]\n]\n\n## **It is impossible to resolve repeats of length S unless you have reads longer than S**\n\n## **It is impossible to resolve repeats of length S unless you have reads longer than S**\n\n---\n\n# **Scaffolding**\n\n---\n.enlarge120[\n# **Beyond contigs**\n]\n\n.pull-left[\n\nContig sizes are limited by:\n\n* the length of the repeats in your genome\n  * Can&#39;t change this\n\n\n* the length (or \&quot;span\&quot;) of the reads\n  * Use long read technology\n  * Use tricks with other technology\n\n]\n\n---\n.enlarge120[\n# **Types of reads**\n]\n\n.pull-left[.enlarge120[**Example fragment**]]\n\n\n.remark-code[.enlarge120[atcgtatgatcttgagattctctcttcccttatagctgctata]]\n\n.pull-left[.enlarge120[**\&quot;Single-end\&quot; read**]]\n\n\n.remark-code[.enlarge120[**atcgtatg**atcttgagattctctcttcccttatagctgctata]]\n\nsequence *one* end of the fragment\n\n.pull-left[.enlarge120[**\&quot;Paired-end\&quot; read**]]\n\n\n.remark-code[.enlarge120[**atcgtatg**atcttgagattctctcttcccttatag**ctgctata**]]\n\nsequence both ends of the same fragment\n\n**We can exploit this information!**\n---\n\n.enlarge120[# **Scaffolding**]\n\n* **Paired end reads**\n  * Known sequences at each end of fragment\n  * Roughly known fragment length\n\n* **Most ends will occur in same contig**\n\n* **Some will occur in different contigs**\n  * ***evidence that these contigs are linked***\n---\n\n.enlarge120[# **Contigs to Scaffolds**]\n\n![](../../images/scaffolding.png)\n\n---\n\n.enlarge120[# **Assessing assemblies**]\n\n* We desire\n  * Total length similar to genome size\n  * Fewer, larger contigs\n  * Correct contigs\n\n* Metrics\n  * No generally useful measure. (No real prior information)\n  * Longest contigs, total base pairs in contigs, **N50**, ...\n\n---\n\n.enlarge120[# **The \&quot;N50\&quot;**]\n\n.enlarge120[***The length of that contig from which 50% of the bases are in it and shorter contigs***]\n\n* Imagine we have 7 contigs with lengths:\n  * 1, 1, 3, 5, 8, 12, 20\n\n* Total\n  * 1+1+3+5+8+12+20 = 50\n\n* N50 is the \&quot;halfway sum\&quot; = 25\n  * 1+1+3+5+8+**12** = 30 (&gt;25) so **N50 is 12**\n\n---\n\n.enlarge120[# **2 levels of assembly**]\n\n* Draft assembly\n  * Will contain a number of non-linked scaffolds with gaps of unknown sequence\n  * Fairly easy to get to\n\n* Closed (finished) assembly\n  * One sequence for each chromosome\n  * Takes a **lot** more work\n  * Small genomes are becoming easier with long read tech\n  * Large genomes are the province of big consortia (e.g. Human Genome Consortium)\n\n---\n.enlarge120[# **How do I do it?**]\n---\n.enlarge120[\n# **Example**\n\n* Culture your bacterium\n\n\n* Extract your genomic DNA\n\n\n* Send it to your sequencing centre for Illumina sequencing\n  * 250bp paired end\n\n\n* Get back 2 files\n  * .remark-code[MRSA_R1.fastq.gz]\n  * .remark-code[MRSA_R2.fastq.gz]\n\n\n* ***Now what?***\n]\n\n---\n.enlarge120[# **Assembly tools**\n\n* **Genome**\n  * **Velvet, Velvet Optimizer, Spades,** Abyss, MIRA, Newbler, SGA, AllPaths, Ray, SOAPdenovo, ...\n\n\n* Meta-genome\n  * Meta Velvet, SGA, custom scripts + above\n\n\n* Transcriptome\n  * Trinity, Oases, Trans-abyss\n\n***And many, many others...***\n\n]\n\n---\n.enlarge120[\n# **Assembly Exercise #1**\n\n* We will do a simple assembly using **Velvet** in **Galaxy**\n* We can do a number of different assemblies and compare some assembly metrics.\n\n]\n&quot;,&quot;# Genome assembly with Velvet: Background\n{:.no_toc}\n\nVelvet is one of a number of *de novo* assemblers that use short read sets as input (*e.g.* Illumina Reads). The assembly method is based on the manipulation of de Bruijn graphs, via the removal of errors and the simplication of repeated regions.\n\n&gt; ### {% icon comment %} Comment\n&gt;\n&gt; For information about Velvet, you can check its (nice) [Wikipedia page](https://en.wikipedia.org/wiki/Velvet_assembler).\n{: .comment}\n\nFor this tutorial, we have a set of reads from an imaginary *Staphylococcus aureus* bacterium with a miniature genome (197,394 bp). Our mutant strain read set was sequenced with the whole genome shotgun method, using an Illumina DNA sequencing instrument. From these reads, we would like to rebuild our imaginary *Staphylococcus aureus* bacterium via a *de novo* assembly of a short read set using the Velvet assembler.\n\n&gt; ### Agenda\n&gt;\n&gt; In this tutorial, we will deal with:\n&gt;\n&gt; 1. TOC\n&gt; {:toc}\n&gt;\n{: .agenda}\n\n# Get the data\n\nWe will now import the data that we will use for the tutorial.\n\n&gt; ### {% icon hands_on %} Hands-on: Getting the data\n&gt;\n&gt; 1. Create and name a new history for this tutorial.\n&gt; 2. Import from [Zenodo](https://doi.org/10.5281/zenodo.582600) or from the data library the files:\n&gt;    - [`mutant_R1.fastq`](https://zenodo.org/record/582600/files/mutant_R1.fastq)\n&gt;    - [`mutant_R2.fastq`](https://zenodo.org/record/582600/files/mutant_R2.fastq)\n&gt;\n&gt;    &gt; ### {% icon tip %} Tip: Importing data via links\n&gt;    &gt;\n&gt;    &gt; * Copy the link location (Right-click on the filename then \&quot;Copy Link Address\&quot;)\n&gt;    &gt; * Open the Galaxy Upload Manager\n&gt;    &gt; * Select **Paste/Fetch Data**\n&gt;    &gt; * Paste the link into the text field\n&gt;    &gt; * Change the data-type to **fastqsanger**\n&gt;    &gt; * Press **Start**\n&gt;    {: .tip}\n&gt;\n&gt; 3. Change the name of the files to `mutant_R1` and `mutant_R2`.\n&gt;\n&gt;    As a default, Galaxy uses the link as the name of the new dataset. It also does not link the dataset to a database or a reference genome.\n&gt;\n&gt;    {% include snippets/rename_dataset.md %}\n&gt;\n&gt; 4. Inspect the content of a file.\n&gt;\n&gt;    &gt; ### {% icon tip %} Tip: Inspecting the content of a dataset\n&gt;    &gt;\n&gt;    &gt; * Click on the {% icon galaxy-eye %} (eye) icon next to the relevant history entry\n&gt;    &gt; * View the content of the file in the central panel\n&gt;    {: .tip}\n&gt;\n&gt;    &gt; ### {% icon question %} Questions\n&gt;    &gt;\n&gt;    &gt; 1. What are four key features of a FASTQ file?\n&gt;    &gt; 2. What is the main difference between a FASTQ and a FASTA file?\n&gt;    &gt;\n&gt;    &gt; &gt; ### {% icon solution %} Solution\n&gt;    &gt; &gt; 1. Each sequence in a FASTQ file is represented by 4 lines: 1st line is the id, 2nd line is the sequence, 3rd line is not used, and 4th line is the quality of sequencing per nucleotide\n&gt;    &gt; &gt; 2. In a FASTQ file, not only are the sequences present, but information about the quality of sequencing is also included.\n&gt;    &gt; {: .solution }\n&gt;    {: .question}\n&gt;\n{: .hands_on}\n\nThe reads have been sequenced from an imaginary *Staphylococcus aureus* bacterium using an Illumina DNA sequencing instrument. We obtained the 2 files we imported (`mutant_R1` and `mutant_R2`)\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; Why do we have 2 files here if we only sequenced the bacteria once?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. The bacteria has been sequenced using paired-end sequencing. The first file corresponds to forward reads and the second file to reverse reads.\n&gt; {: .solution }\n{: .question}\n\n# Evaluate the input reads\n\nBefore doing any assembly, the first questions you should ask about your input reads include:\n\n- What is the coverage of my genome?\n- How good is my read set?\n- Do I need to ask for a new sequencing run?\n- Is it suitable for the analysis I need to do?\n\nWe will evaluate the input reads using the FastQC tool. This tool runs a standard series of tests on your read set and returns a relatively easy-to-interpret report. We will use it to evaluate the quality of our FASTQ files and combine the results with MultiQC.\n\n&gt; ### {% icon hands_on %} Hands-on: FastQC on a fastq file\n&gt;\n&gt; 1. **FastQC** {% icon tool %} with the following parameters\n&gt;    - \&quot;Short read data from your current history\&quot; to (**Multiple datasets**) `mutant_R1.fastq` and `mutant_R2.fastq`\n&gt;\n&gt; 2. **MultiQC** {% icon tool %} with the following parameters\n&gt;    - \&quot;Software name\&quot; to `FastQC`\n&gt;    - \&quot;Result file\&quot; to the raw data files generated by FastQC\n&gt;\n{: .hands_on}\n\nMultiQC generates a webpage combining reports for FastQC on both datasets. It includes these graphs and tables:\n\n- General statistics\n\n    This is important in setting maximum k-mer size for an assembly.\n\n    &gt; ### {% icon comment %} Getting the length of sequences\n    &gt;\n    &gt; * Click on **Configure Columns**\n    &gt; * Check **Length**\n    &gt; * Close the window\n    {: .comment}\n\n    &gt; ### {% icon question %} Questions\n    &gt;\n    &gt; 1. How long are the sequences?\n    &gt; 2. What is the average coverage of the genome, given our imaginary *Staphylococcus aureus* bacterium has a genome of 197,394 bp?\n    &gt;\n    &gt; &gt; ### {% icon solution %} Solution\n    &gt; &gt; 1. The sequences are 150 bp long\n    &gt; &gt; 2. We have 2 x 12,480 sequences of 150 bp, so the average genome coverage is: 2 * 12480 * 150 / 197394, or approximately 19 X coverage.\n    &gt; {: .solution }\n    {: .question}\n\n- Sequence Quality Histograms\n\n    Dips in quality near the beginning, middle or end of the reads may determine the trimming/cleanup methods and parameters to be used, or may indicate technical problems with the sequencing process/machine run.\n\n    &lt;figure id=\&quot;figure-1\&quot;&gt;&lt;img src=\&quot;../../images/fastqc_per_base_sequence_quality_plot.png\&quot; alt=\&quot;Sequence Quality Histograms with the mean quality value across each base position in the read\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 1:&lt;/span&gt; The mean quality value across each base position in the read&lt;/figcaption&gt;&lt;/figure&gt;\n\n    &gt; ### {% icon question %} Questions\n    &gt;\n    &gt; 1. What does the y-axis represent?\n    &gt; 2. Why is the quality score decreasing across the length of the reads?\n    &gt;\n    &gt; &gt; ### {% icon solution %} Solution\n    &gt; &gt; 1. The y-axis represents the quality score for each base (an estimate of the error during sequencing).\n    &gt; &gt; 2. The quality score is decreasing accross the length of the reads because the sequencing become less and less reliable at the end of the reads.\n    &gt; {: .solution }\n    {: .question}\n\n- Per Sequence GC Content\n\n    High GC organisms tend not to assemble well and may have an uneven read coverage distribution.\n\n- Per Base N Content\n\n    The presence of large numbers of Ns in reads may point to a poor quality sequencing run. You will need to trim these reads to remove Ns.\n\n- k-mer content\n\n    The presence of highly recurring k-mers may point to contamination of reads with barcodes or adapter sequences.\n\n\n&gt; ### {% icon comment %} Comment\n&gt;\n&gt; For a fuller discussion of FastQC outputs and warnings, see the [FastQC website link](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/), including the section on each of the output [reports](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/), and examples of [\&quot;good\&quot;](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html) and [\&quot;bad\&quot;](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html) Illumina data.\n{: .comment}\n\nWe won&#39;t be doing anything to these data to clean it up as there isn&#39;t much need. Therefore we will get on with the assembly!\n\n\n# Assemble reads with Velvet\n\nNow, we want to assemble our reads to find the sequence of our imaginary *Staphylococcus aureus* bacterium. We will perform a *de novo* assembly of the reads into long contiguous sequences using the Velvet short read assembler.\n\nThe first step of the assembler is to build a de Bruijn graph. For that, it will break our reads into k-mers, *i.e.* fragments of length *k*. Velvet requires the user to input a value of *k* (k-mer size) for the assembly process. Small k-mers will give greater connectivity, but large k-mers will give better specificity.\n\n&gt; ### {% icon hands_on %} Hands-on: Assemble the reads\n&gt;\n&gt; 1. **FASTQ interlacer** {% icon tool %} with the following parameters\n&gt;    - \&quot;Type of paired-end datasets\&quot; to `2 separate datasets`\n&gt;    - \&quot;Left-hand mates\&quot; to `mutant_R1.fastq`\n&gt;    - \&quot;Right-hand mates\&quot; to `mutant_R2.fastq`\n&gt;\n&gt;    Currently our paired-end reads are in 2 files (one with the forward reads and one with the reverse reads), but Velvet requires only one file, where each read is next to its mate read. In other words, if the reads are indexed from 0, then reads 0 and 1 are paired, 2 and 3, 4 and 5, etc. Before doing the assembly *per se*, we need to prepare the files by combining them.\n&gt;\n&gt; 2. **velveth** {% icon tool %} with the following parameters\n&gt;    - \&quot;Hash Length\&quot; to `29`\n&gt;    - \&quot;Input Files\&quot;: click on `Insert Input Files`\n&gt;    - \&quot;file format\&quot; to `fastq`\n&gt;    - \&quot;read type\&quot; to `shortPaired reads`\n&gt;    - \&quot;Dataset\&quot; to the pairs output of **FASTQ interlacer**\n&gt;\n&gt;    The tool takes our reads and break them into k-mers.\n&gt;\n&gt; 3. **velvetg** {% icon tool %} with the following parameters\n&gt;    - \&quot;Velvet Dataset\&quot; to the output of **velveth**\n&gt;    - \&quot;Using Paired Reads\&quot; to `Yes`\n&gt;\n&gt;    This last tool actually does the assembly.\n{: .hands_on}\n\nTwo files are generated:\n\n- A \&quot;Contigs\&quot; file\n\n    This file contains the sequences of the contigs longer than 2k. In the header of each contig, a bit of information is added:\n    - the k-mer length (called \&quot;length\&quot;): For the value of k chosen in the assembly, a measure of how many k-mers overlap (by 1 bp each overlap) to give this length\n    - the k-mer coverage (called \&quot;coverage\&quot;): For the value of k chosen in the assembly, a measure of how many k-mers overlap each base position (in the assembly).\n\n    ![Contigs output](../../images/image10.png)\n\n- A \&quot;Stats\&quot; file\n\n    This is a tabular file giving for each contig the k-mer lengths, k-mer coverages and other measures.\n\n    ![Contigs stats output](../../images/image11.png)\n\n# Collect some statistics on the contigs\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. How many contigs have been built?\n&gt; 2. What is the mean, min and max length of the contigs?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. 190\n&gt; &gt; 2. To compute this information, we can use the Datamash tool on the 2nd columns (length). Be careful with the first line, the header. As a result, we obtain: 597.82 as mean, 1 as min and 12904 as max. It would mean that the smallest contig has a length of 1 bp, even smaller than k. The length on the 2nd column corresponds to length of the contig in k-mers. This means that the smallest contig has a length of 1k = 29. So to obtain the real length, we need to add k-1 to the length. We then obtain a mean contig length of 625.82 bp, a min contig of 29 bp and a max contig of 12,932 bp.\n&gt; {: .solution }\n{: .question}\n\nThis table is limitted, but we will now collect more basic statistics on our assembly.\n\n&gt; ### {% icon hands_on %} Hands-on: Collect fasta statistics on our contigs\n&gt;\n&gt; 1. **Quast** {% icon tool %} with\n&gt;    - \&quot;Contigs/scaffolds output file\&quot; to the output of **velvetg**\n&gt;    - \&quot;Type of data\&quot; to `contig`\n&gt;    - \&quot;Reference File\&quot; to `wildtype.fna`\n&gt;    - \&quot;Type of organism\&quot; to `Prokaryotes`\n&gt;    - \&quot;Lower Threshold\&quot; to `500`\n&gt;    - \&quot;Thresholds\&quot; to `0,1000`\n{: .hands_on}\n\nThis tool generates 5 output files, but we will focus on the HTML report and the Icarus viewer.\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. What is represented in the Icarus viewer?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. Icarus is a novel genome visualizer for accurate assessment and analysis of genomic draft assemblies. It draws contigs ordered from longest to shortest, highlights N50, N75 (NG50, NG75) and long contigs larger than a user-specified threshold\n&gt; {: .solution }\n{: .question}\n\nThe HTML report reports many statistics computed by QUAST to assess the quality of the assembly:\n\n- Statistics about the quality of the assembly when compared to the reference (fraction of the genome, duplication ratio, etc)\n- Misassembly statistics, including the number of misassemblies\n\n    A misassembly is a position in the contigs (breakpoints) that satisfy one of the following criteria:\n    - the left flanking sequence aligns over 1 kbp away from the right flanking sequence on the reference;\n    - flanking sequences overlap on more than 1 kbp\n    - flanking sequences align to different strands or different chromosomes\n\n- Unaligned regions in the assembly\n- Mismatches compared to the reference genomes\n- Statistics about the assembly *per se*, such as the number of contigs and the length of the largest contig\n\n&gt; ### {% icon question %} Question\n&gt;\n&gt; 1. How many contigs have been constructed?\n&gt; 2. Which proportion of the reference genome do they represent?\n&gt; 3. How many misassemblies have been found?\n&gt; 4. Has the assembly introduced mismatches and indels?\n&gt; 5. What are N50 and L50?\n&gt; 6. Is there a bias in GC percentage induced by the assembly?\n&gt;\n&gt; &gt; ### {% icon solution %} Solution\n&gt; &gt; 1. 190 contigs have been constructed, but only 47 have a length &gt; 500 bp.\n&gt; &gt; 2. The contigs represents 87.965% of the reference genome.\n&gt; &gt; 3. 1 misassembly has been found: it corresponds to a relocation, *i.e.* a misassembly event (breakpoint) where the left flanking sequence aligns over 1 kbp away from the right flanking sequence on the reference genome.\n&gt; &gt; 4. 8.06 mismatches per 100 kbp and 4.03 indels per 100 kbp are found.\n&gt; &gt; 5. N50 is the length for which the collection of all contigs of that length or longer covers at least half an assembly. In other words, if contigs were ordered from small to large, half of all the nucleotides will be in contigs this size or larger. And L50 is the number of contigs equal to or longer than N50: L50 is the minimal number of contigs that cover half the assembly.\n&gt; &gt; 6. The GC % in the assembly is 33.64%, really similar to the one of the reference genome (33.43%).\n&gt; {: .solution }\n{: .question}\n\n# Discussion\n\n&gt; ### {% icon hands_on %} (Optional) Hands-on: Rerun for values *k* ranging from 31 to 101\n&gt;\n&gt; 1. **velveth** {% icon tool %} with the same parameters as before except\n&gt;    - \&quot;Hash Length\&quot; to a value between 31 and 101\n&gt; 2. **velvetg** {% icon tool %} with the same parameters as before\n&gt; 3. **Quast** {% icon tool %} with the same parameters as before\n{: .hands_on}\n\nWe have completed an assembly on this data set for a number of k values ranging from 29 to 101. A few of the assembly metrics appear below.\n\n&lt;figure id=\&quot;figure-2\&quot;&gt;&lt;img src=\&quot;../../images/number_of_contigs.png\&quot; alt=\&quot;contigs\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 2:&lt;/span&gt; Number of contigs in the assembly for various k-mer sizes&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-3\&quot;&gt;&lt;img src=\&quot;../../images/largest_contig.png\&quot; alt=\&quot;largest_contig\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 3:&lt;/span&gt; Largest contig in each of the assemblies by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-4\&quot;&gt;&lt;img src=\&quot;../../images/total_bp.png\&quot; alt=\&quot;total_bp\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 4:&lt;/span&gt; Total number of base pairs in all the contigs for each assembly by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&lt;figure id=\&quot;figure-5\&quot;&gt;&lt;img src=\&quot;../../images/n50.png\&quot; alt=\&quot;n50\&quot;&gt;&lt;figcaption&gt;&lt;span class=\&quot;figcaption-prefix\&quot;&gt;Figure 5:&lt;/span&gt; N50 metric for each of the assemblies by k-mer size&lt;/figcaption&gt;&lt;/figure&gt;\n\n&gt; ### {% icon question %} Questions\n&gt;\n&gt; 1. Are there any distinct features in the charts?\n&gt; 2. Does it look like one assembly might be better than some of the others?\n&gt;\n{: .question}\n\nThe reasons for these patterns will be discussed in detail in the De Bruijn graph assembly slides and tutorial.\n&quot;],&quot;related_posts&quot;:null,&quot;source&quot;:&quot;.&quot;,&quot;destination&quot;:&quot;_site/training-material&quot;,&quot;collections_dir&quot;:&quot;&quot;,&quot;plugins_dir&quot;:&quot;_plugins&quot;,&quot;layouts_dir&quot;:&quot;_layouts&quot;,&quot;data_dir&quot;:&quot;metadata&quot;,&quot;includes_dir&quot;:&quot;.&quot;,&quot;safe&quot;:false,&quot;include&quot;:[&quot;.nojekyll&quot;],&quot;exclude&quot;:[&quot;shared/font-awesome/src/&quot;,&quot;Gemfile&quot;,&quot;Gemfile.lock&quot;,&quot;package.json&quot;,&quot;package-lock.json&quot;,&quot;CONTRIBUTING.md&quot;,&quot;CONTRIBUTORS.yaml&quot;,&quot;LICENSE.md&quot;,&quot;README.md&quot;,&quot;Makefile&quot;,&quot;miniconda.sh&quot;,&quot;**/README.md&quot;,&quot;**/*.yaml&quot;,&quot;**/*.yml&quot;,&quot;**/*.sh&quot;,&quot;bin/&quot;,&quot;metadata/&quot;,&quot;templates/&quot;,&quot;vendor/&quot;,&quot;node_modules/&quot;],&quot;keep_files&quot;:[&quot;.git&quot;,&quot;.svn&quot;],&quot;encoding&quot;:&quot;utf-8&quot;,&quot;markdown_ext&quot;:&quot;markdown,mkdown,mkdn,mkd,md&quot;,&quot;strict_front_matter&quot;:true,&quot;show_drafts&quot;:null,&quot;limit_posts&quot;:0,&quot;future&quot;:false,&quot;unpublished&quot;:false,&quot;whitelist&quot;:[],&quot;plugins&quot;:[&quot;jekyll-feed&quot;,&quot;jekyll-environment-variables&quot;,&quot;jekyll-github-metadata&quot;,&quot;jekyll-scholar&quot;],&quot;markdown&quot;:&quot;kramdown&quot;,&quot;highlighter&quot;:&quot;rouge&quot;,&quot;lsi&quot;:false,&quot;excerpt_separator&quot;:&quot;\n\n&quot;,&quot;incremental&quot;:false,&quot;detach&quot;:false,&quot;port&quot;:&quot;4000&quot;,&quot;host&quot;:&quot;127.0.0.1&quot;,&quot;baseurl&quot;:&quot;/training-material&quot;,&quot;show_dir_listing&quot;:false,&quot;permalink&quot;:&quot;date&quot;,&quot;paginate_path&quot;:&quot;/page:num&quot;,&quot;timezone&quot;:null,&quot;quiet&quot;:false,&quot;verbose&quot;:true,&quot;defaults&quot;:[],&quot;liquid&quot;:{&quot;error_mode&quot;:&quot;warn&quot;,&quot;strict_filters&quot;:false,&quot;strict_variables&quot;:false},&quot;rdiscount&quot;:{&quot;extensions&quot;:[]},&quot;redcarpet&quot;:{&quot;extensions&quot;:[]},&quot;kramdown&quot;:{&quot;auto_ids&quot;:true,&quot;toc_levels&quot;:&quot;1..2&quot;,&quot;entity_output&quot;:&quot;as_char&quot;,&quot;smart_quotes&quot;:&quot;lsquo,rsquo,ldquo,rdquo&quot;,&quot;input&quot;:&quot;GFM&quot;,&quot;hard_wrap&quot;:false,&quot;footnote_nr&quot;:1,&quot;show_warnings&quot;:false,&quot;syntax_highlighter&quot;:&quot;rouge&quot;,&quot;syntax_highlighter_opts&quot;:{},&quot;coderay&quot;:{},&quot;auto_ids&quot;:true,&quot;toc_levels&quot;:&quot;1..2&quot;,&quot;entity_output&quot;:&quot;as_char&quot;,&quot;smart_quotes&quot;:&quot;lsquo,rsquo,ldquo,rdquo&quot;,&quot;input&quot;:&quot;GFM&quot;,&quot;hard_wrap&quot;:false,&quot;footnote_nr&quot;:1,&quot;show_warnings&quot;:false,&quot;syntax_highlighter&quot;:&quot;rouge&quot;,&quot;syntax_highlighter_opts&quot;:{},&quot;coderay&quot;:{}},&quot;title&quot;:&quot;VIB Bioinformatics Core&quot;,&quot;email&quot;:&quot;bits@vib.be&quot;,&quot;description&quot;:&quot;A collection of tutorials generated and maintained by VIB Bioinformatics Core&quot;,&quot;url&quot;:&quot;https://vibbits.github.io/&quot;,&quot;repository&quot;:&quot;vibbits/training-material&quot;,&quot;repository_branch&quot;:&quot;master&quot;,&quot;logo&quot;:&quot;assets/images/vib_notag_neg_rgb.jpg&quot;,&quot;small_logo&quot;:&quot;assets/images/bioinformatics_core_rgb_neg.png&quot;,&quot;help_url&quot;:&quot;https://www.bits.vib.be&quot;,&quot;other_languages&quot;:&quot;fr, ja, es, pt, ar&quot;,&quot;figurify&quot;:{&quot;skip_empty&quot;:true,&quot;skip_layouts&quot;:[&quot;introduction_slides&quot;,&quot;tutorial_slides&quot;,&quot;base_slides&quot;],&quot;skip_titles&quot;:[&quot;Example of an image with a caption&quot;]},&quot;scholar&quot;:{&quot;style&quot;:&quot;_layouts/g3.csl&quot;,&quot;locale&quot;:&quot;en&quot;,&quot;sort_by&quot;:&quot;year,month&quot;,&quot;order&quot;:&quot;ascending&quot;,&quot;group_by&quot;:&quot;none&quot;,&quot;group_order&quot;:&quot;ascending&quot;,&quot;bibliography_group_tag&quot;:&quot;h2,h3,h4,h5&quot;,&quot;bibliography_list_tag&quot;:&quot;ol&quot;,&quot;bibliography_item_tag&quot;:&quot;li&quot;,&quot;bibliography_list_attributes&quot;:{},&quot;bibliography_item_attributes&quot;:{},&quot;source&quot;:&quot;topics/&quot;,&quot;bibliography&quot;:&quot;**/*.bib&quot;,&quot;repository&quot;:null,&quot;repository_file_delimiter&quot;:&quot;.&quot;,&quot;bibtex_options&quot;:{&quot;strip&quot;:false,&quot;parse_months&quot;:true},&quot;bibtex_filters&quot;:[&quot;smallcaps&quot;,&quot;superscript&quot;,&quot;italics&quot;,&quot;latex&quot;],&quot;bibtex_skip_fields&quot;:[&quot;abstract&quot;,&quot;month_numeric&quot;],&quot;bibtex_quotes&quot;:[&quot;{&quot;,&quot;}&quot;],&quot;replace_strings&quot;:true,&quot;join_strings&quot;:true,&quot;details_dir&quot;:&quot;bibliography&quot;,&quot;details_layout&quot;:&quot;bibtex.html&quot;,&quot;details_link&quot;:&quot;Details&quot;,&quot;use_raw_bibtex_entry&quot;:true,&quot;bibliography_class&quot;:&quot;bibliography&quot;,&quot;bibliography_template&quot;:&quot;bibtemplate&quot;,&quot;reference_tagname&quot;:&quot;span&quot;,&quot;missing_reference&quot;:&quot;(missing reference)&quot;,&quot;details_link_class&quot;:&quot;details&quot;,&quot;query&quot;:&quot;@*&quot;,&quot;cite_class&quot;:&quot;citation&quot;,&quot;type_names&quot;:{&quot;article&quot;:&quot;Journal Articles&quot;,&quot;book&quot;:&quot;Books&quot;,&quot;incollection&quot;:&quot;Book Chapters&quot;,&quot;inproceedings&quot;:&quot;Conference Articles&quot;,&quot;thesis&quot;:&quot;Theses&quot;,&quot;mastersthesis&quot;:&quot;Master&#39;s Theses&quot;,&quot;phdthesis&quot;:&quot;PhD Theses&quot;,&quot;manual&quot;:&quot;Manuals&quot;,&quot;techreport&quot;:&quot;Technical Reports&quot;,&quot;misc&quot;:&quot;Miscellaneous&quot;,&quot;unpublished&quot;:&quot;Unpublished&quot;},&quot;type_aliases&quot;:{&quot;phdthesis&quot;:&quot;thesis&quot;,&quot;mastersthesis&quot;:&quot;thesis&quot;},&quot;type_order&quot;:[],&quot;month_names&quot;:null},&quot;icon-tag&quot;:{&quot;question&quot;:&quot;fa-question-circle&quot;,&quot;solution&quot;:&quot;fa-eye&quot;,&quot;hands_on&quot;:&quot;fa-pencil&quot;,&quot;comment&quot;:&quot;fa-commenting-o&quot;,&quot;tip&quot;:&quot;fa-lightbulb-o&quot;,&quot;objectives&quot;:&quot;fa-bullseye&quot;,&quot;requirements&quot;:&quot;fa-check-circle&quot;,&quot;time&quot;:&quot;fa-hourglass-end&quot;,&quot;keypoints&quot;:&quot;fa-key&quot;,&quot;tool&quot;:&quot;fa-wrench&quot;,&quot;workflow&quot;:&quot;fa-share-alt&quot;,&quot;feedback&quot;:&quot;fa-comments-o&quot;,&quot;congratulations&quot;:&quot;fa-thumbs-up&quot;,&quot;trophy&quot;:&quot;fa-trophy&quot;,&quot;warning&quot;:&quot;fa-warning&quot;,&quot;details&quot;:&quot;fa-info-circle&quot;,&quot;exchange&quot;:&quot;fa-exchange&quot;,&quot;param-file&quot;:&quot;fa-file-o&quot;,&quot;param-files&quot;:&quot;fa-files-o&quot;,&quot;param-collection&quot;:&quot;fa-folder-o&quot;,&quot;param-text&quot;:&quot;fa-pencil&quot;,&quot;param-check&quot;:&quot;fa-check-square-o&quot;,&quot;param-select&quot;:&quot;fa-filter&quot;,&quot;param-repeat&quot;:&quot;fa-plus-square-o&quot;,&quot;galaxy-eye&quot;:&quot;fa-eye&quot;,&quot;galaxy-gear&quot;:&quot;fa-cog&quot;,&quot;galaxy-history&quot;:&quot;fa-archive&quot;,&quot;galaxy-library&quot;:&quot;fa-folder&quot;,&quot;galaxy-pencil&quot;:&quot;fa-pencil&quot;,&quot;galaxy-refresh&quot;:&quot;fa-refresh&quot;,&quot;galaxy-barchart&quot;:&quot;fa-bar-chart&quot;,&quot;galaxy-cross&quot;:&quot;fa-times&quot;,&quot;galaxy-columns&quot;:&quot;fa-columns&quot;,&quot;galaxy-tags&quot;:&quot;fa-tags&quot;,&quot;galaxy-selector&quot;:&quot;fa-check-square-o&quot;,&quot;galaxy-upload&quot;:&quot;fa fa-upload&quot;,&quot;galaxy-chart-select-data&quot;:&quot;fa fa-database&quot;,&quot;galaxy-save&quot;:&quot;fa fa-save&quot;,&quot;galaxy-scratchbook&quot;:&quot;fa fa-th&quot;,&quot;galaxy-dropdown&quot;:&quot;fa-caret-down&quot;,&quot;zenodo_link&quot;:&quot;fa-files-o&quot;,&quot;tutorial&quot;:&quot;fa-laptop&quot;,&quot;slides&quot;:&quot;fa-slideshare&quot;,&quot;interactive_tour&quot;:&quot;fa-magic&quot;,&quot;topic&quot;:&quot;fa-folder-o&quot;,&quot;instances&quot;:&quot;fa-cog&quot;,&quot;docker_image&quot;:&quot;fa-ship&quot;,&quot;galaxy_instance&quot;:&quot;fa-external-link&quot;,&quot;references&quot;:&quot;fa-bookmark&quot;,&quot;gitter&quot;:&quot;fa-comments&quot;,&quot;help&quot;:&quot;fa-life-ring&quot;,&quot;github&quot;:&quot;fa-github&quot;,&quot;email&quot;:&quot;fa-envelope-o&quot;,&quot;twitter&quot;:&quot;fa-twitter&quot;,&quot;linkedin&quot;:&quot;fa-linkedin&quot;,&quot;orcid&quot;:&quot;ai-orcid&quot;,&quot;curriculum&quot;:&quot;fa-graduation-cap&quot;,&quot;level&quot;:&quot;fa-graduation-cap&quot;,&quot;hall-of-fame&quot;:&quot;fa-users&quot;},&quot;serving&quot;:false,&quot;github&quot;:{&quot;api_url&quot;:&quot;https://api.github.com&quot;,&quot;baseurl&quot;:&quot;/pages/vibbits/training-material&quot;,&quot;build_revision&quot;:&quot;51b7d71ad1e7c1ba1e5f078218a844c13ffeaab3&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/training-material.git&quot;,&quot;contributors&quot;:false,&quot;environment&quot;:&quot;development&quot;,&quot;help_url&quot;:&quot;https://help.github.com&quot;,&quot;hostname&quot;:&quot;github.com&quot;,&quot;is_project_page&quot;:true,&quot;is_user_page&quot;:false,&quot;issues_url&quot;:null,&quot;language&quot;:null,&quot;latest_release&quot;:false,&quot;license&quot;:null,&quot;organization_members&quot;:[],&quot;owner&quot;:{&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;bio&quot;:null,&quot;blog&quot;:&quot;www.bits.vib.be&quot;,&quot;collaborators&quot;:null,&quot;company&quot;:null,&quot;created_at&quot;:&quot;2016-10-18 09:13:54 UTC&quot;,&quot;description&quot;:null,&quot;email&quot;:null,&quot;followers&quot;:0,&quot;following&quot;:0,&quot;has_organization_projects&quot;:true,&quot;has_repository_projects&quot;:true,&quot;hireable&quot;:null,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;id&quot;:22908438,&quot;is_verified&quot;:false,&quot;location&quot;:&quot;Belgium&quot;,&quot;login&quot;:&quot;vibbits&quot;,&quot;name&quot;:&quot;VIB Bioinformatics Core&quot;,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;public_gists&quot;:0,&quot;public_repos&quot;:20,&quot;type&quot;:&quot;Organization&quot;,&quot;updated_at&quot;:&quot;2019-07-17 16:48:30 UTC&quot;},&quot;owner_gravatar_url&quot;:&quot;https://github.com/vibbits.png&quot;,&quot;owner_name&quot;:&quot;vibbits&quot;,&quot;owner_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;pages_env&quot;:&quot;development&quot;,&quot;pages_hostname&quot;:&quot;localhost:4000&quot;,&quot;private&quot;:null,&quot;project_tagline&quot;:null,&quot;project_title&quot;:&quot;training-material&quot;,&quot;public_repositories&quot;:[{&quot;id&quot;:160538640,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNjA1Mzg2NDA=&quot;,&quot;name&quot;:&quot;bigdatasurvey&quot;,&quot;full_name&quot;:&quot;vibbits/bigdatasurvey&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/bigdatasurvey&quot;,&quot;description&quot;:&quot;A survey of open-source big data platforms, centering around the Hadoop ecosystem.&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/bigdatasurvey/deployments&quot;,&quot;created_at&quot;:&quot;2018-12-05 15:27:29 UTC&quot;,&quot;updated_at&quot;:&quot;2018-12-05 15:31:44 UTC&quot;,&quot;pushed_at&quot;:&quot;2015-10-28 20:33:50 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/bigdatasurvey.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/bigdatasurvey.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/bigdatasurvey.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/bigdatasurvey&quot;,&quot;homepage&quot;:null,&quot;size&quot;:524,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;JavaScript&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:false,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:false,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;gh-pages&quot;},{&quot;id&quot;:136517018,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzY1MTcwMTg=&quot;,&quot;name&quot;:&quot;bioconda-recipes&quot;,&quot;full_name&quot;:&quot;vibbits/bioconda-recipes&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/bioconda-recipes&quot;,&quot;description&quot;:&quot;Conda recipes for the bioconda channel.&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/bioconda-recipes/deployments&quot;,&quot;created_at&quot;:&quot;2018-06-07 18:36:07 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:21 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-01-21 17:01:17 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/bioconda-recipes.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/bioconda-recipes.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/bioconda-recipes.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/bioconda-recipes&quot;,&quot;homepage&quot;:&quot;https://bioconda.github.io&quot;,&quot;size&quot;:167593,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Shell&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:170295072,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNzAyOTUwNzI=&quot;,&quot;name&quot;:&quot;CAMEL&quot;,&quot;full_name&quot;:&quot;vibbits/CAMEL&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/CAMEL&quot;,&quot;description&quot;:&quot; The CAMEL platform, an online compendium of Adaptive Microbial Experiments in the Lab.&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/CAMEL/deployments&quot;,&quot;created_at&quot;:&quot;2019-02-12 10:03:23 UTC&quot;,&quot;updated_at&quot;:&quot;2019-07-25 12:42:15 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-07-25 12:42:12 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/CAMEL.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/CAMEL.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/CAMEL.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/CAMEL&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:2307,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;HTML&quot;,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:5,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:5,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:173921762,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNzM5MjE3NjI=&quot;,&quot;name&quot;:&quot;cytokit&quot;,&quot;full_name&quot;:&quot;vibbits/cytokit&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/cytokit&quot;,&quot;description&quot;:&quot;Microscopy Image Cytometry Toolkit&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/cytokit/deployments&quot;,&quot;created_at&quot;:&quot;2019-03-05 10:04:08 UTC&quot;,&quot;updated_at&quot;:&quot;2019-03-05 15:37:40 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-03-05 15:37:38 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/cytokit.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/cytokit.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/cytokit.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/cytokit&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:174062,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Jupyter Notebook&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;apache-2.0&quot;,&quot;name&quot;:&quot;Apache License 2.0&quot;,&quot;spdx_id&quot;:&quot;Apache-2.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/apache-2.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTI=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:135151492,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzUxNTE0OTI=&quot;,&quot;name&quot;:&quot;DataIntegrationFlows&quot;,&quot;full_name&quot;:&quot;vibbits/DataIntegrationFlows&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/DataIntegrationFlows&quot;,&quot;description&quot;:&quot;This is a docker image to set up the atmosphere for running several data integration workflows&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/DataIntegrationFlows/deployments&quot;,&quot;created_at&quot;:&quot;2018-05-28 11:24:02 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:21 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-05-28 12:05:05 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/DataIntegrationFlows.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/DataIntegrationFlows.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/DataIntegrationFlows.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/DataIntegrationFlows&quot;,&quot;homepage&quot;:null,&quot;size&quot;:2,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:169587916,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNjk1ODc5MTY=&quot;,&quot;name&quot;:&quot;incubator-echarts&quot;,&quot;full_name&quot;:&quot;vibbits/incubator-echarts&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/incubator-echarts&quot;,&quot;description&quot;:&quot;A powerful, interactive charting and visualization library for browser&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/incubator-echarts/deployments&quot;,&quot;created_at&quot;:&quot;2019-02-07 14:50:26 UTC&quot;,&quot;updated_at&quot;:&quot;2019-02-07 14:50:32 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-02-07 13:03:45 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/incubator-echarts.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/incubator-echarts.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/incubator-echarts.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/incubator-echarts&quot;,&quot;homepage&quot;:&quot;http://echarts.apache.org/&quot;,&quot;size&quot;:151893,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;JavaScript&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;apache-2.0&quot;,&quot;name&quot;:&quot;Apache License 2.0&quot;,&quot;spdx_id&quot;:&quot;Apache-2.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/apache-2.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTI=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:118458037,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMTg0NTgwMzc=&quot;,&quot;name&quot;:&quot;modules-4-GenePattern&quot;,&quot;full_name&quot;:&quot;vibbits/modules-4-GenePattern&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/modules-4-GenePattern&quot;,&quot;description&quot;:&quot;modules for integrating under GenePattern some extra tools for NGS analysis&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/modules-4-GenePattern/deployments&quot;,&quot;created_at&quot;:&quot;2018-01-22 13:04:16 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:21 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-07-03 14:14:58 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/modules-4-GenePattern.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/modules-4-GenePattern.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/modules-4-GenePattern.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/modules-4-GenePattern&quot;,&quot;homepage&quot;:null,&quot;size&quot;:1927,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:136658710,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzY2NTg3MTA=&quot;,&quot;name&quot;:&quot;MOFA&quot;,&quot;full_name&quot;:&quot;vibbits/MOFA&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/MOFA&quot;,&quot;description&quot;:&quot;Multi-Omics Factor Analysis&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/MOFA/deployments&quot;,&quot;created_at&quot;:&quot;2018-06-08 19:31:44 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:21 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-06-23 20:50:58 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/MOFA.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/MOFA.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/MOFA.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/MOFA&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:110522,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;HTML&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:157213851,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNTcyMTM4NTE=&quot;,&quot;name&quot;:&quot;NextFlow_pipelines&quot;,&quot;full_name&quot;:&quot;vibbits/NextFlow_pipelines&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/NextFlow_pipelines&quot;,&quot;description&quot;:&quot;Tricks and tips for designing a NextFlow pipeline&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/NextFlow_pipelines/deployments&quot;,&quot;created_at&quot;:&quot;2018-11-12 12:57:24 UTC&quot;,&quot;updated_at&quot;:&quot;2019-02-25 13:20:35 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-02-25 13:20:34 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/NextFlow_pipelines.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/NextFlow_pipelines.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/NextFlow_pipelines.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/NextFlow_pipelines&quot;,&quot;homepage&quot;:null,&quot;size&quot;:1578,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Nextflow&quot;,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:186219236,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxODYyMTkyMzY=&quot;,&quot;name&quot;:&quot;OpenRefineTrainingData&quot;,&quot;full_name&quot;:&quot;vibbits/OpenRefineTrainingData&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/OpenRefineTrainingData&quot;,&quot;description&quot;:null,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/OpenRefineTrainingData/deployments&quot;,&quot;created_at&quot;:&quot;2019-05-12 06:19:18 UTC&quot;,&quot;updated_at&quot;:&quot;2019-05-18 14:15:38 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-05-18 14:15:36 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/OpenRefineTrainingData.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/OpenRefineTrainingData.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/OpenRefineTrainingData.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/OpenRefineTrainingData&quot;,&quot;homepage&quot;:null,&quot;size&quot;:7919,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;mit&quot;,&quot;name&quot;:&quot;MIT License&quot;,&quot;spdx_id&quot;:&quot;MIT&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/mit&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTEz&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:71230827,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnk3MTIzMDgyNw==&quot;,&quot;name&quot;:&quot;phyd3&quot;,&quot;full_name&quot;:&quot;vibbits/phyd3&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/phyd3&quot;,&quot;description&quot;:&quot;Phylogenetic tree viewer based on D3.js&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/phyd3/deployments&quot;,&quot;created_at&quot;:&quot;2016-10-18 09:19:54 UTC&quot;,&quot;updated_at&quot;:&quot;2019-01-03 04:05:38 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-05-29 06:46:30 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/phyd3.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/phyd3.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/phyd3.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/phyd3&quot;,&quot;homepage&quot;:null,&quot;size&quot;:2992,&quot;stargazers_count&quot;:29,&quot;watchers_count&quot;:29,&quot;language&quot;:&quot;JavaScript&quot;,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:7,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:3,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-3.0&quot;,&quot;name&quot;:&quot;GNU General Public License v3.0&quot;,&quot;spdx_id&quot;:&quot;GPL-3.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-3.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTk=&quot;},&quot;forks&quot;:7,&quot;open_issues&quot;:3,&quot;watchers&quot;:29,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:138773305,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzg3NzMzMDU=&quot;,&quot;name&quot;:&quot;presentation&quot;,&quot;full_name&quot;:&quot;vibbits/presentation&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/presentation&quot;,&quot;description&quot;:&quot;20180907 presentation community meeting&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/presentation&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/presentation/deployments&quot;,&quot;created_at&quot;:&quot;2018-06-26 17:49:03 UTC&quot;,&quot;updated_at&quot;:&quot;2019-01-25 06:43:04 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-01-25 06:43:03 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/presentation.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/presentation.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/presentation.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/presentation&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:33,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:2,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:1,&quot;license&quot;:null,&quot;forks&quot;:2,&quot;open_issues&quot;:1,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:186428088,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxODY0MjgwODg=&quot;,&quot;name&quot;:&quot;qupath&quot;,&quot;full_name&quot;:&quot;vibbits/qupath&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/qupath&quot;,&quot;description&quot;:&quot;QuPath - Open Source Digital Pathology&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/qupath&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/qupath/deployments&quot;,&quot;created_at&quot;:&quot;2019-05-13 13:45:18 UTC&quot;,&quot;updated_at&quot;:&quot;2019-05-15 09:48:35 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-05-15 09:48:33 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/qupath.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/qupath.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/qupath.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/qupath&quot;,&quot;homepage&quot;:&quot;https://qupath.github.io&quot;,&quot;size&quot;:83357,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Java&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-3.0&quot;,&quot;name&quot;:&quot;GNU General Public License v3.0&quot;,&quot;spdx_id&quot;:&quot;GPL-3.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-3.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTk=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:133400916,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzM0MDA5MTY=&quot;,&quot;name&quot;:&quot;rocker_conda_data_integration&quot;,&quot;full_name&quot;:&quot;vibbits/rocker_conda_data_integration&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/rocker_conda_data_integration&quot;,&quot;description&quot;:&quot;Docker container based on rocker/tidyverse with miniconda3 to run various integration tools for Omics&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_data_integration/deployments&quot;,&quot;created_at&quot;:&quot;2018-05-14 17:53:42 UTC&quot;,&quot;updated_at&quot;:&quot;2019-07-31 15:35:10 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-01-13 08:51:00 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/rocker_conda_data_integration.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/rocker_conda_data_integration.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/rocker_conda_data_integration.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/rocker_conda_data_integration&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:50406,&quot;stargazers_count&quot;:3,&quot;watchers_count&quot;:3,&quot;language&quot;:&quot;R&quot;,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:3,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-2.0&quot;,&quot;name&quot;:&quot;GNU General Public License v2.0&quot;,&quot;spdx_id&quot;:&quot;GPL-2.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-2.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTg=&quot;},&quot;forks&quot;:3,&quot;open_issues&quot;:0,&quot;watchers&quot;:3,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:131875427,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzE4NzU0Mjc=&quot;,&quot;name&quot;:&quot;rocker_conda_mofa&quot;,&quot;full_name&quot;:&quot;vibbits/rocker_conda_mofa&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/rocker_conda_mofa&quot;,&quot;description&quot;:&quot;Docker container on rocker/tidyverse with miniconda3 to run MOFA tool https://github.com/bioFAM/MOFA&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/rocker_conda_mofa/deployments&quot;,&quot;created_at&quot;:&quot;2018-05-02 16:08:04 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:21 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-05-02 18:07:02 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/rocker_conda_mofa.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/rocker_conda_mofa.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/rocker_conda_mofa.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/rocker_conda_mofa&quot;,&quot;homepage&quot;:null,&quot;size&quot;:17,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-2.0&quot;,&quot;name&quot;:&quot;GNU General Public License v2.0&quot;,&quot;spdx_id&quot;:&quot;GPL-2.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-2.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTg=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:113286207,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMTMyODYyMDc=&quot;,&quot;name&quot;:&quot;scop3d&quot;,&quot;full_name&quot;:&quot;vibbits/scop3d&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/scop3d&quot;,&quot;description&quot;:&quot;Sequence conservation of protein on 3D structure.&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/scop3d/deployments&quot;,&quot;created_at&quot;:&quot;2017-12-06 07:56:32 UTC&quot;,&quot;updated_at&quot;:&quot;2019-06-04 20:25:31 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-06-04 20:25:30 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/scop3d.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/scop3d.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/scop3d.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/scop3d&quot;,&quot;homepage&quot;:&quot;&quot;,&quot;size&quot;:26190,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Python&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:false,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-3.0&quot;,&quot;name&quot;:&quot;GNU General Public License v3.0&quot;,&quot;spdx_id&quot;:&quot;GPL-3.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-3.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTk=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:104596713,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMDQ1OTY3MTM=&quot;,&quot;name&quot;:&quot;scRNA-Seq-TCC-prep&quot;,&quot;full_name&quot;:&quot;vibbits/scRNA-Seq-TCC-prep&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/scRNA-Seq-TCC-prep&quot;,&quot;description&quot;:&quot;Preprocessing of single-cell RNA-Seq data for input to kallisto&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-Seq-TCC-prep/deployments&quot;,&quot;created_at&quot;:&quot;2017-09-23 20:22:54 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:22 UTC&quot;,&quot;pushed_at&quot;:&quot;2016-12-01 02:04:26 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/scRNA-Seq-TCC-prep.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/scRNA-Seq-TCC-prep.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/scRNA-Seq-TCC-prep.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/scRNA-Seq-TCC-prep&quot;,&quot;homepage&quot;:null,&quot;size&quot;:8740,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;Jupyter Notebook&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-3.0&quot;,&quot;name&quot;:&quot;GNU General Public License v3.0&quot;,&quot;spdx_id&quot;:&quot;GPL-3.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-3.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTk=&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:137523231,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMzc1MjMyMzE=&quot;,&quot;name&quot;:&quot;scRNA-tools&quot;,&quot;full_name&quot;:&quot;vibbits/scRNA-tools&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/scRNA-tools&quot;,&quot;description&quot;:&quot;Table of software for the analysis of single-cell RNA-seq data.&quot;,&quot;fork&quot;:true,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/scRNA-tools/deployments&quot;,&quot;created_at&quot;:&quot;2018-06-15 19:10:39 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:22 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-06-15 19:31:56 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/scRNA-tools.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/scRNA-tools.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/scRNA-tools.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/scRNA-tools&quot;,&quot;homepage&quot;:null,&quot;size&quot;:3835,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:&quot;R&quot;,&quot;has_issues&quot;:false,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;mit&quot;,&quot;name&quot;:&quot;MIT License&quot;,&quot;spdx_id&quot;:&quot;MIT&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/mit&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTEz&quot;},&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:100585992,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxMDA1ODU5OTI=&quot;,&quot;name&quot;:&quot;sc_read_kallisto_wrapper&quot;,&quot;full_name&quot;:&quot;vibbits/sc_read_kallisto_wrapper&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/sc_read_kallisto_wrapper&quot;,&quot;description&quot;:&quot;wrapper for single cell transcriptomics pipeline with kallisto&quot;,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/sc_read_kallisto_wrapper/deployments&quot;,&quot;created_at&quot;:&quot;2017-08-17 09:24:32 UTC&quot;,&quot;updated_at&quot;:&quot;2018-10-11 09:37:22 UTC&quot;,&quot;pushed_at&quot;:&quot;2018-02-09 09:22:40 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/sc_read_kallisto_wrapper.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/sc_read_kallisto_wrapper.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/sc_read_kallisto_wrapper.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/sc_read_kallisto_wrapper&quot;,&quot;homepage&quot;:null,&quot;size&quot;:84,&quot;stargazers_count&quot;:6,&quot;watchers_count&quot;:6,&quot;language&quot;:&quot;HTML&quot;,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:2,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:{&quot;key&quot;:&quot;gpl-3.0&quot;,&quot;name&quot;:&quot;GNU General Public License v3.0&quot;,&quot;spdx_id&quot;:&quot;GPL-3.0&quot;,&quot;url&quot;:&quot;https://api.github.com/licenses/gpl-3.0&quot;,&quot;node_id&quot;:&quot;MDc6TGljZW5zZTk=&quot;},&quot;forks&quot;:2,&quot;open_issues&quot;:0,&quot;watchers&quot;:6,&quot;default_branch&quot;:&quot;master&quot;},{&quot;id&quot;:153169091,&quot;node_id&quot;:&quot;MDEwOlJlcG9zaXRvcnkxNTMxNjkwOTE=&quot;,&quot;name&quot;:&quot;WaaS-Finnish-Cloud&quot;,&quot;full_name&quot;:&quot;vibbits/WaaS-Finnish-Cloud&quot;,&quot;private&quot;:false,&quot;owner&quot;:{&quot;login&quot;:&quot;vibbits&quot;,&quot;id&quot;:22908438,&quot;node_id&quot;:&quot;MDEyOk9yZ2FuaXphdGlvbjIyOTA4NDM4&quot;,&quot;avatar_url&quot;:&quot;https://avatars1.githubusercontent.com/u/22908438?v=4&quot;,&quot;gravatar_id&quot;:&quot;&quot;,&quot;url&quot;:&quot;https://api.github.com/users/vibbits&quot;,&quot;html_url&quot;:&quot;https://github.com/vibbits&quot;,&quot;followers_url&quot;:&quot;https://api.github.com/users/vibbits/followers&quot;,&quot;following_url&quot;:&quot;https://api.github.com/users/vibbits/following{/other_user}&quot;,&quot;gists_url&quot;:&quot;https://api.github.com/users/vibbits/gists{/gist_id}&quot;,&quot;starred_url&quot;:&quot;https://api.github.com/users/vibbits/starred{/owner}{/repo}&quot;,&quot;subscriptions_url&quot;:&quot;https://api.github.com/users/vibbits/subscriptions&quot;,&quot;organizations_url&quot;:&quot;https://api.github.com/users/vibbits/orgs&quot;,&quot;repos_url&quot;:&quot;https://api.github.com/users/vibbits/repos&quot;,&quot;events_url&quot;:&quot;https://api.github.com/users/vibbits/events{/privacy}&quot;,&quot;received_events_url&quot;:&quot;https://api.github.com/users/vibbits/received_events&quot;,&quot;type&quot;:&quot;Organization&quot;,&quot;site_admin&quot;:false},&quot;html_url&quot;:&quot;https://github.com/vibbits/WaaS-Finnish-Cloud&quot;,&quot;description&quot;:null,&quot;fork&quot;:false,&quot;url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud&quot;,&quot;forks_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/forks&quot;,&quot;keys_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/keys{/key_id}&quot;,&quot;collaborators_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/collaborators{/collaborator}&quot;,&quot;teams_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/teams&quot;,&quot;hooks_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/hooks&quot;,&quot;issue_events_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/issues/events{/number}&quot;,&quot;events_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/events&quot;,&quot;assignees_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/assignees{/user}&quot;,&quot;branches_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/branches{/branch}&quot;,&quot;tags_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/tags&quot;,&quot;blobs_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/git/blobs{/sha}&quot;,&quot;git_tags_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/git/tags{/sha}&quot;,&quot;git_refs_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/git/refs{/sha}&quot;,&quot;trees_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/git/trees{/sha}&quot;,&quot;statuses_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/statuses/{sha}&quot;,&quot;languages_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/languages&quot;,&quot;stargazers_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/stargazers&quot;,&quot;contributors_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/contributors&quot;,&quot;subscribers_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/subscribers&quot;,&quot;subscription_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/subscription&quot;,&quot;commits_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/commits{/sha}&quot;,&quot;git_commits_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/git/commits{/sha}&quot;,&quot;comments_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/comments{/number}&quot;,&quot;issue_comment_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/issues/comments{/number}&quot;,&quot;contents_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/contents/{+path}&quot;,&quot;compare_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/compare/{base}...{head}&quot;,&quot;merges_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/merges&quot;,&quot;archive_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/{archive_format}{/ref}&quot;,&quot;downloads_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/downloads&quot;,&quot;issues_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/issues{/number}&quot;,&quot;pulls_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/pulls{/number}&quot;,&quot;milestones_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/milestones{/number}&quot;,&quot;notifications_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/notifications{?since,all,participating}&quot;,&quot;labels_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/labels{/name}&quot;,&quot;releases_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/releases{/id}&quot;,&quot;deployments_url&quot;:&quot;https://api.github.com/repos/vibbits/WaaS-Finnish-Cloud/deployments&quot;,&quot;created_at&quot;:&quot;2018-10-15 19:20:17 UTC&quot;,&quot;updated_at&quot;:&quot;2019-03-27 05:45:56 UTC&quot;,&quot;pushed_at&quot;:&quot;2019-03-27 05:44:56 UTC&quot;,&quot;git_url&quot;:&quot;git://github.com/vibbits/WaaS-Finnish-Cloud.git&quot;,&quot;ssh_url&quot;:&quot;git@github.com:vibbits/WaaS-Finnish-Cloud.git&quot;,&quot;clone_url&quot;:&quot;https://github.com/vibbits/WaaS-Finnish-Cloud.git&quot;,&quot;svn_url&quot;:&quot;https://github.com/vibbits/WaaS-Finnish-Cloud&quot;,&quot;homepage&quot;:null,&quot;size&quot;:3,&quot;stargazers_count&quot;:0,&quot;watchers_count&quot;:0,&quot;language&quot;:null,&quot;has_issues&quot;:true,&quot;has_projects&quot;:true,&quot;has_downloads&quot;:true,&quot;has_wiki&quot;:true,&quot;has_pages&quot;:false,&quot;forks_count&quot;:0,&quot;mirror_url&quot;:null,&quot;archived&quot;:false,&quot;disabled&quot;:false,&quot;open_issues_count&quot;:0,&quot;license&quot;:null,&quot;forks&quot;:0,&quot;open_issues&quot;:0,&quot;watchers&quot;:0,&quot;default_branch&quot;:&quot;master&quot;}],&quot;releases&quot;:false,&quot;releases_url&quot;:&quot;https://github.com/vibbits/training-material/releases&quot;,&quot;repository_name&quot;:&quot;training-material&quot;,&quot;repository_nwo&quot;:&quot;vibbits/training-material&quot;,&quot;repository_url&quot;:&quot;https://github.com/vibbits/training-material&quot;,&quot;show_downloads&quot;:null,&quot;source&quot;:{&quot;branch&quot;:&quot;gh-pages&quot;,&quot;path&quot;:&quot;/&quot;},&quot;tar_url&quot;:&quot;https://github.com/vibbits/training-material/tarball/gh-pages&quot;,&quot;url&quot;:&quot;https://github.com/pages/vibbits/training-material&quot;,&quot;versions&quot;:{},&quot;wiki_url&quot;:null,&quot;zip_url&quot;:&quot;https://github.com/vibbits/training-material/zipball/gh-pages&quot;},&quot;env&quot;:{&quot;XDG_SESSION_ID&quot;:&quot;130&quot;,&quot;TERM&quot;:&quot;xterm&quot;,&quot;SHELL&quot;:&quot;/bin/bash&quot;,&quot;MAKEFLAGS&quot;:&quot;&quot;,&quot;SSH_CLIENT&quot;:&quot;157.193.240.143 28737 22&quot;,&quot;CONDA_SHLVL&quot;:&quot;2&quot;,&quot;CONDA_PROMPT_MODIFIER&quot;:&quot;(galaxy_training_material) &quot;,&quot;SSH_TTY&quot;:&quot;/dev/pts/0&quot;,&quot;USER&quot;:&quot;albot&quot;,&quot;LS_COLORS&quot;:&quot;rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:&quot;,&quot;CONDA_EXE&quot;:&quot;/home/albot/miniconda3/bin/conda&quot;,&quot;MAKE_TERMOUT&quot;:&quot;/dev/pts/0&quot;,&quot;SSH_AUTH_SOCK&quot;:&quot;/tmp/ssh-QHESLfR2MM/agent.28921&quot;,&quot;MAKELEVEL&quot;:&quot;1&quot;,&quot;MFLAGS&quot;:&quot;&quot;,&quot;_CE_CONDA&quot;:&quot;&quot;,&quot;CONDA_PREFIX_1&quot;:&quot;/home/albot/miniconda3&quot;,&quot;MAIL&quot;:&quot;/var/mail/albot&quot;,&quot;PATH&quot;:&quot;/home/albot/bin:/home/albot/.local/bin:/home/albot/miniconda3/envs/galaxy_training_material/bin:/home/albot/miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin&quot;,&quot;CONDA_PREFIX&quot;:&quot;/home/albot/miniconda3/envs/galaxy_training_material&quot;,&quot;PWD&quot;:&quot;/home/albot/alex/training-material&quot;,&quot;LANG&quot;:&quot;en_US.UTF-8&quot;,&quot;_CE_M&quot;:&quot;&quot;,&quot;SHLVL&quot;:&quot;2&quot;,&quot;HOME&quot;:&quot;/home/albot&quot;,&quot;CONDA_PYTHON_EXE&quot;:&quot;/home/albot/miniconda3/bin/python&quot;,&quot;LOGNAME&quot;:&quot;albot&quot;,&quot;MAKE_TERMERR&quot;:&quot;/dev/pts/0&quot;,&quot;SSH_CONNECTION&quot;:&quot;157.193.240.143 28737 10.132.0.3 22&quot;,&quot;XDG_DATA_DIRS&quot;:&quot;/usr/local/share:/usr/share:/var/lib/snapd/desktop&quot;,&quot;CONDA_DEFAULT_ENV&quot;:&quot;galaxy_training_material&quot;,&quot;LESSOPEN&quot;:&quot;| /usr/bin/lesspipe %s&quot;,&quot;DISPLAY&quot;:&quot;localhost:10.0&quot;,&quot;XDG_RUNTIME_DIR&quot;:&quot;/run/user/1003&quot;,&quot;LESSCLOSE&quot;:&quot;/usr/bin/lesspipe %s %s&quot;,&quot;_&quot;:&quot;/home/albot/miniconda3/envs/galaxy_training_material/bin/jekyll&quot;}}
              page: {&quot;layout&quot;:&quot;introduction_slides&quot;,&quot;logo&quot;:&quot;GTN&quot;,&quot;title&quot;:&quot;Protein Structure Analysis&quot;,&quot;type&quot;:&quot;introduction&quot;,&quot;contributors&quot;:[&quot;abotzki&quot;,&quot;jvdurme&quot;],&quot;topic_name&quot;:&quot;protein-structure-analysis&quot;,&quot;tutorial_name&quot;:&quot;introduction&quot;,&quot;content&quot;:&quot;## Protein Structure Analysis ##\n\n- Sequences, structures and databases\n- Experimental methods (X-rays, electrons and NMR)\n- Finding and visualising structures from the  Protein Data Bank\n- Comparing structures\n- Modelling mutations\n- Creating homology models\n\n---\n\n## The Structure-Function Connection ##\n\n.left[![Left-aligned image](/training-material/topics/protein-structure-analysis/images/hemoglobin.png)]\n.right[Folded proteins provide a well-  defined 3D  arrangement of functional groups, creating microenvironments and active sites.\nStructural changes  are often involved in functional  mechanisms  (motor proteins, ...)\n]\n\n---\n\n### Databases\n\n.pull-left[\n**Uniprot** approx. 1100,000 sequences\n\n- mainly determined by large-scale DNA sequencing of individual genes or whole genomes\n- increasingly automated annotation\n\n**Protein Data Bank** approx. 141,000 \nexperimentally determined structures\n\n- Mainly determined by X-ray crystallography and high-resolution NMR spectroscopy\n- Automation is increasing, but there are still significant limitations in the rate of solving new structures\n\n\n]\n.pull-right[ ![](/training-material/assets/images/GTN-60px.png) ]\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### The Protein Databank ###\n\n.pull-left[ ![](/training-material/assets/images/GTN-60px.png) ] \n\n\n.pull-right[\n[http://www.wwpdb.org](http://www.wwpdb.org)\n- contains structures of  proteins, nucleic acids  and complexes,  determined by X-ray  crystallography, NMR  spectroscopy\n- No purely theoretical  or ab initio models  (since 2006)\n- Also stores supporting  experimental data\n- Full deposition now  required by all peer-reviewed journals\n\n---\n\n### Exercise 1: Search the PDB ###\n\n- Use the UniProt site to search for “dnak”.  \n- Use the PDB site to search for “dnak”.\n- Compare the UniProt and PDB result lists.\n- Use the sequence search function to see if there are structures with sequences similar to that of the  DnaK C-terminal domain.\n- Look at the summary pages of a number of  structures and note some interesting properties.\n- Select a number of structures and create a report  page.\n\n---\n\n### PDB File Organisation ###\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- Nucleic Acid Database: DNA and RNA structures\n\n[http://ndbserver.rutgers.edu/](http://ndbserver.rutgers.edu/)\n\n- PDB-REDO: automatically re-refined deposited  structures, using the latest methods\n\n[http://www.cmbi.ru.nl/pdb_redo/](http://www.cmbi.ru.nl/pdb_redo)\n\n- EBI: many online tools for structure analysis\n\n[http://www.ebi.ac.uk/Tools/structure/](http://www.ebi.ac.uk/Tools/structure/)\n\n- Replaced Electron Density Server: convenient overview of  quality parameters for crystal structures\n\n[http://www.ebi.ac.uk/pdbe/litemol](http://www.ebi.ac.uk/pdbe/litemol)\n\n---\n\n### High-Resolution NMR Spectrometry ###\n\nMany atomic nuclei, including the ubiquitous hydrogen nuclei,  resonate at specific radio frequencies when placed in a  strong, uniform magnetic field. The chemical environment of each individual atom slightly modulates its exact resonance  frequency.\n\n![]()\n\nIn macromolecules with thousands of atoms, many different  effects combine to generate an extremely complicated pattern of chemical shifts, which therefore more or less uniquely  identify each atom. Multidimensional spectra allow these  frequencies to be assigned to specific atoms.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nWhen two atoms are near each other in 3D space, they can exchange magnetization, giving rise to crosspeaks at the  intersection of their respective frequencies.\n\n![]()\n\nThis nuclear Overhauser effect (NOE) is used to effectively measure the distances between pairs of atoms, at least  qualitatively.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nAfter identification of the atoms by means of their unique chemical shifts, distance restraints are derived from the  Overhauser crosspeaks. An extended model of the protein  is  generated,  and  then  condensed  into  a  shape  that  is consistent with as many of distance restraints as possible.\n\n![]()\n\n---\n\n### Other methods ###\n\nElectron microscopy (and especially Cryo-electron microscopy): Electron crystallography and single particle reconstruction\r\n\nSmall-angle X-ray and neutron scattering (SAXS and SANS)\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- BioMagResBank: experimental data for NMR-  derived structures (lists of distance restraints  and other experimentally derived properties)\n\n[http://www.bmrb.wisc.edu/](http://www.bmrb.wisc.edu/)\n\n- BioIsis: database of SAXS-derived structures\n\n[http://www.bioisis.net](http://www.bioisis.net)\n\n- EMBL database of SAXS-derived structures\n\n[http://www.sasbdb.org](http://www.sasbdb.org)\n\n- EM Databank for cryo-EM structures\n\n[http://www.emdatabank.org](http://www.emdatabank.org)\n\n---\n\n### Assessing Structure Quality ###\n\nGeneral geometric properties (bond lengths and angles,  Ramachandran distribution, …): MolProbity  [Link](http://molprobity.biochem.duke.edu/)\n\nCrystal Structures\n- Diffraction data resolution and completeness (PDB)\n- Final Rcryst and Rfree factors (PDB)\n- Local correspondence to electron density (EDS)\n\nNMR Structures\n- Number of distance restraints and additional experimental  data sources (BMRB)\n- Local restraint density (on-line NMR constraint analyser)\n[link](http://molsim.sci.univr.it/bioinfo/tools/constraint/index.html)\n\nOther techniques\n- Difficult to generalise: carefully read and consider the  published protocol\n\n---\n\n### Molecular Graphics Software ###\n\n- [PyMOL](http://www.pymol.org/): high-quality output,  good examples on [Wiki](http://www.pymolwiki.org/)\n- [Chimera](http://www.cgl.ucsf.edu/chimera/): good  documentation on website\n- [VMD](http://www.ks.uiuc.edu/Research/vmd/):  excellent for the analysis of MD trajectories\n- [Yasara](http://www.yasara.org)  \n- [SwissPDBViewer](http://spdbv.vital-it.ch/)\n\n---\n\n### YASARA ###\n\nYasara View is freely available and provides  basic visualisation functions.\n\nYasara Model, Dynamics and Structure provide  additional functionality.\n\nAn add-on module for NMR structures is  available.\n\nThe program can be combined with the WHAT-  IF program for structure validation, and with  FoldX for energy calculations.\n\n---\n\n### Exercise 2: show a structure ###\n\nLoad PDB entry 1TRZ using the File menu.\n\nCompare the default representations (F1-F8)  and use the various controls to change the view  of the protein.\n\nExplore the Edit and View menus to change  various aspects of the graphical representation  of the structure.\n\nExamine the hydrogen bonding patterns.  Display a molecular surface.\n\nCreate an interesting view of the molecule and  save it as a high-resolution image.\n\n---\n\n### Structure Superposition ###\n\nStructures can be superimposed to achieve the best  possible match between corresponding atoms. It is  possible to consider all atoms, or only a subset  (such as the Cα atoms).\n\nWhen the structures are very similar, determining  which atoms should be matched to each other is  trivial. When there are larger differences, it takes  more preparation.\n\nDifferent algorithms use different combinations of  sequence- and secondary of tertiary structure-based  information.\n\n---\n\n### Exercise 3: Compare Structures ###\n\nDownload the five provided PDB files and open  them in Yasara.\n\nUse the `Analyze|Align|Objects` with MUSTANG  function to align the four last objects with the first one.\n\nUse the space bar to open the text console and  see the reported root mean square deviations\n\n\nas well as the number of residues matched.\n\nColor all structures by B factor and compare the  distribution to the local variability of the structures.\n\n---\n\n### PDB File Surprises ###\n\n- Missing atoms, residues and loops  \n- Multiple molecules in the asymmetric unit\n- Incomplete oligomers due to coincident crystal and  oligomer symmetry\n- Cα-only models  lternate conformations\n- Multiple models, especially for NMR ensembles\n- Use of B factors to represent other properties  ther non-standard extensions (PDBQT, ...)\n\n---\n\n### Force Fields ###\n\n- Energy terms representing physical interactions\n\nCovalent bond lengths and angles\nDihedral angles and van der Waals forces (steric effects)\nElectrostatic interactions and hydrogen bonds\n…\n\n- Energy can be minimized, and forces can literally be derived from the potential function.\n\n- Consistency and careful consideration of the  properties to be simulated are essential.\n\n---\n\n### Force Field Terms ###\n\nEach energy term has a functional form, which includes one or more parameters:\n\n- Covalent bond energy term\n To do: add formula\n- Van der Waals contact energy term\n\nThe parameters are collectively optimized to  reproduce a chosen set of experimentally  observed parameters.\n\nA given force field should be used as a  consistent system, and can only be used to  predict properties that are covered by the  training set.\n\n---\n\n### FoldX ###\n\n\ns designed for quantitative modelling of the  contribution of structural interactions to the stability of  proteins and protein complexes. It also supports  protein/DNA complexes.\n\nThe force field describes the different interactions in a  protein structure or complex in analytical terms. It has  been calibrated using a set of experimentally determined  stabilities.\n\nApplications include the optimisation of structures, the calculation of the stability of complexes, and predicting  the effect of amino-acid or base-pair mutations on these  properties.\n\n---\n\n### The FoldX Plugin for Yasara\t###\n\nIn order to make FoldX more accessible and  integrate its functions into Yasara, dr. Joost van  Durme (SWITCH laboratory) made a Yasara plugin  module that can apply FoldX functions to structures  that are loaded as Yasara objects.\nThis greatly simplifies the use of FoldX, and allows  for a quick visual analysis of the resulting changes  in the structures.\nMore information can be found at [wiki](http://foldxyasara.switchlab.org/index.php/) [FoldX](http://foldxsuite.crg.eu/)\n\n---\n\n### Exercise 4a: Repair a PDB File ###\n\n- Load the 1CRN PDB file.\n- Use the “Repair object” option in the Analysis |  FoldX menu to activate the corresponding FoldX  function.\n- Select the object to repair.\n\nThis exports the object as a temporary PDB file,  starts FoldX with the appropriate options, and loads  the repaired PDB file as a new object in Yasara.\n\n- Compare the original and repaired objects.  \n- Describe the changes that were introduced.\n\n---\n\n### Exercise 4b: Model a Mutation ###\n\n- Load the 2AC0.sce Yasara scene file.\n- Set an appropriate structure representation.\n- Locate residue Ala159 using the sequence view,  and right-click to access the `FoldX|Mutate` residue function. Change it to a Trp residue.\n- Look at the effect of the substitution on the  structure, and use the space bar to open the text  console and read the output of the FoldX  calculation.\n- Mutate Arg273 to an alanine side chain. Discuss  the effects of this substitution.\n\n---\n\n### Homology Modelling ###\n\n- When a structure is available for a protein with a  similar sequences, it is possible to predict the  structure of a new sequence with varying degrees of  confidence.\n- All homology modelling procedures start from an  alignment of the template and target sequences.  The quality of this alignment will have a major  impact on the resulting model.\n- Available applications include stand-alone programs  (Modeller, FoldX, …) and web-based services (such as SwissModel).\n\n---\n\n### Exercise 5: Make a Homology  Model using Swiss Model ###\n\nsee []()\n\n---\n\n### Exercise 6: Study Protein-Ligand Interactions ###\n\nsee []()\n\n---\n\n### On-Line References ###\n\n- Crystallography 101 (Bernhard Rupp):  (http://www.ruppweb.org/Xray/101index.html)\n- Protein NMR, A Practical Guide (Vicky Higman) ( http://www.protein-nmr.org.uk/)\n- Model validation course  (http://xray.bmc.uu.se/gerard/embo2001/modval/index.html)\n- Assessing model quality (http://spdbv.vital-it.ch/TheMolecularLevel/ModQual/)\n\n- Lectures by Burkhard Rost on protein structure  prediction (https://www.youtube.com/channel/UCU6j8BG4RbEtTgyIZJ6Vpow)\n\n---\n&quot;,&quot;dir&quot;:&quot;/topics/protein-structure-analysis/slides/&quot;,&quot;name&quot;:&quot;introduction.html&quot;,&quot;path&quot;:&quot;topics/protein-structure-analysis/slides/introduction.html&quot;,&quot;url&quot;:&quot;/topics/protein-structure-analysis/slides/introduction.html&quot;}
              topic: {&quot;name&quot;:&quot;protein-structure-analysis&quot;,&quot;type&quot;:&quot;basics&quot;,&quot;category&quot;:&quot;basics&quot;,&quot;title&quot;:&quot;Protein Structure Analysis&quot;,&quot;summary&quot;:&quot;Analyzing the protein structure of your protein-of-interest can be advantageous in multiple ways. It can help you discover regions which are good candidates to interact with other proteins. It can help you discover new domains. It can help with identifying differences with homologuous proteins and a lot more.&quot;,&quot;extra&quot;:&quot;protein_analysis&quot;,&quot;requirements&quot;:null,&quot;maintainers&quot;:[&quot;abotzki&quot;],&quot;references&quot;:[{&quot;authors&quot;:&quot;Switchlab&quot;,&quot;title&quot;:&quot;Home of FoldX plugin&quot;,&quot;link&quot;:&quot;http://foldxyasara.switchlab.org/&quot;,&quot;summary&quot;:&quot;More information about FoldX plugin and troubleshooting&quot;},{&quot;authors&quot;:&quot;Wikipedia&quot;,&quot;title&quot;:&quot;Wiki page about PDB&quot;,&quot;link&quot;:&quot;https://en.wikipedia.org/wiki/Protein_Data_Bank&quot;,&quot;summary&quot;:&quot;wiki page explaining information on PDB&quot;},{&quot;authors&quot;:&quot;YASARA developers&quot;,&quot;title&quot;:&quot;Working with YASARA&quot;,&quot;link&quot;:&quot;http://www.yasara.org/movies.htm&quot;,&quot;summary&quot;:&quot;Movie tutorials on YASARA&quot;}]}
              material: {&quot;layout&quot;:&quot;introduction_slides&quot;,&quot;logo&quot;:&quot;GTN&quot;,&quot;title&quot;:&quot;Protein Structure Analysis&quot;,&quot;type&quot;:&quot;introduction&quot;,&quot;contributors&quot;:[&quot;abotzki&quot;,&quot;jvdurme&quot;],&quot;topic_name&quot;:&quot;protein-structure-analysis&quot;,&quot;tutorial_name&quot;:&quot;introduction&quot;,&quot;content&quot;:&quot;## Protein Structure Analysis ##\n\n- Sequences, structures and databases\n- Experimental methods (X-rays, electrons and NMR)\n- Finding and visualising structures from the  Protein Data Bank\n- Comparing structures\n- Modelling mutations\n- Creating homology models\n\n---\n\n## The Structure-Function Connection ##\n\n.left[![Left-aligned image](/training-material/topics/protein-structure-analysis/images/hemoglobin.png)]\n.right[Folded proteins provide a well-  defined 3D  arrangement of functional groups, creating microenvironments and active sites.\nStructural changes  are often involved in functional  mechanisms  (motor proteins, ...)\n]\n\n---\n\n### Databases\n\n.pull-left[\n**Uniprot** approx. 1100,000 sequences\n\n- mainly determined by large-scale DNA sequencing of individual genes or whole genomes\n- increasingly automated annotation\n\n**Protein Data Bank** approx. 141,000 \nexperimentally determined structures\n\n- Mainly determined by X-ray crystallography and high-resolution NMR spectroscopy\n- Automation is increasing, but there are still significant limitations in the rate of solving new structures\n\n\n]\n.pull-right[ ![](/training-material/assets/images/GTN-60px.png) ]\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### X-Ray Crystallography ###\n\n\n---\n\n### The Protein Databank ###\n\n.pull-left[ ![](/training-material/assets/images/GTN-60px.png) ] \n\n\n.pull-right[\n[http://www.wwpdb.org](http://www.wwpdb.org)\n- contains structures of  proteins, nucleic acids  and complexes,  determined by X-ray  crystallography, NMR  spectroscopy\n- No purely theoretical  or ab initio models  (since 2006)\n- Also stores supporting  experimental data\n- Full deposition now  required by all peer-reviewed journals\n\n---\n\n### Exercise 1: Search the PDB ###\n\n- Use the UniProt site to search for “dnak”.  \n- Use the PDB site to search for “dnak”.\n- Compare the UniProt and PDB result lists.\n- Use the sequence search function to see if there are structures with sequences similar to that of the  DnaK C-terminal domain.\n- Look at the summary pages of a number of  structures and note some interesting properties.\n- Select a number of structures and create a report  page.\n\n---\n\n### PDB File Organisation ###\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- Nucleic Acid Database: DNA and RNA structures\n\n[http://ndbserver.rutgers.edu/](http://ndbserver.rutgers.edu/)\n\n- PDB-REDO: automatically re-refined deposited  structures, using the latest methods\n\n[http://www.cmbi.ru.nl/pdb_redo/](http://www.cmbi.ru.nl/pdb_redo)\n\n- EBI: many online tools for structure analysis\n\n[http://www.ebi.ac.uk/Tools/structure/](http://www.ebi.ac.uk/Tools/structure/)\n\n- Replaced Electron Density Server: convenient overview of  quality parameters for crystal structures\n\n[http://www.ebi.ac.uk/pdbe/litemol](http://www.ebi.ac.uk/pdbe/litemol)\n\n---\n\n### High-Resolution NMR Spectrometry ###\n\nMany atomic nuclei, including the ubiquitous hydrogen nuclei,  resonate at specific radio frequencies when placed in a  strong, uniform magnetic field. The chemical environment of each individual atom slightly modulates its exact resonance  frequency.\n\n![]()\n\nIn macromolecules with thousands of atoms, many different  effects combine to generate an extremely complicated pattern of chemical shifts, which therefore more or less uniquely  identify each atom. Multidimensional spectra allow these  frequencies to be assigned to specific atoms.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nWhen two atoms are near each other in 3D space, they can exchange magnetization, giving rise to crosspeaks at the  intersection of their respective frequencies.\n\n![]()\n\nThis nuclear Overhauser effect (NOE) is used to effectively measure the distances between pairs of atoms, at least  qualitatively.\n\n---\n\n### High-Resolution NMR Spectroscopy ###\n\nAfter identification of the atoms by means of their unique chemical shifts, distance restraints are derived from the  Overhauser crosspeaks. An extended model of the protein  is  generated,  and  then  condensed  into  a  shape  that  is consistent with as many of distance restraints as possible.\n\n![]()\n\n---\n\n### Other methods ###\n\nElectron microscopy (and especially Cryo-electron microscopy): Electron crystallography and single particle reconstruction\r\n\nSmall-angle X-ray and neutron scattering (SAXS and SANS)\n\n![]()\n\n---\n\n### Related Web sites ###\n\n- BioMagResBank: experimental data for NMR-  derived structures (lists of distance restraints  and other experimentally derived properties)\n\n[http://www.bmrb.wisc.edu/](http://www.bmrb.wisc.edu/)\n\n- BioIsis: database of SAXS-derived structures\n\n[http://www.bioisis.net](http://www.bioisis.net)\n\n- EMBL database of SAXS-derived structures\n\n[http://www.sasbdb.org](http://www.sasbdb.org)\n\n- EM Databank for cryo-EM structures\n\n[http://www.emdatabank.org](http://www.emdatabank.org)\n\n---\n\n### Assessing Structure Quality ###\n\nGeneral geometric properties (bond lengths and angles,  Ramachandran distribution, …): MolProbity  [Link](http://molprobity.biochem.duke.edu/)\n\nCrystal Structures\n- Diffraction data resolution and completeness (PDB)\n- Final Rcryst and Rfree factors (PDB)\n- Local correspondence to electron density (EDS)\n\nNMR Structures\n- Number of distance restraints and additional experimental  data sources (BMRB)\n- Local restraint density (on-line NMR constraint analyser)\n[link](http://molsim.sci.univr.it/bioinfo/tools/constraint/index.html)\n\nOther techniques\n- Difficult to generalise: carefully read and consider the  published protocol\n\n---\n\n### Molecular Graphics Software ###\n\n- [PyMOL](http://www.pymol.org/): high-quality output,  good examples on [Wiki](http://www.pymolwiki.org/)\n- [Chimera](http://www.cgl.ucsf.edu/chimera/): good  documentation on website\n- [VMD](http://www.ks.uiuc.edu/Research/vmd/):  excellent for the analysis of MD trajectories\n- [Yasara](http://www.yasara.org)  \n- [SwissPDBViewer](http://spdbv.vital-it.ch/)\n\n---\n\n### YASARA ###\n\nYasara View is freely available and provides  basic visualisation functions.\n\nYasara Model, Dynamics and Structure provide  additional functionality.\n\nAn add-on module for NMR structures is  available.\n\nThe program can be combined with the WHAT-  IF program for structure validation, and with  FoldX for energy calculations.\n\n---\n\n### Exercise 2: show a structure ###\n\nLoad PDB entry 1TRZ using the File menu.\n\nCompare the default representations (F1-F8)  and use the various controls to change the view  of the protein.\n\nExplore the Edit and View menus to change  various aspects of the graphical representation  of the structure.\n\nExamine the hydrogen bonding patterns.  Display a molecular surface.\n\nCreate an interesting view of the molecule and  save it as a high-resolution image.\n\n---\n\n### Structure Superposition ###\n\nStructures can be superimposed to achieve the best  possible match between corresponding atoms. It is  possible to consider all atoms, or only a subset  (such as the Cα atoms).\n\nWhen the structures are very similar, determining  which atoms should be matched to each other is  trivial. When there are larger differences, it takes  more preparation.\n\nDifferent algorithms use different combinations of  sequence- and secondary of tertiary structure-based  information.\n\n---\n\n### Exercise 3: Compare Structures ###\n\nDownload the five provided PDB files and open  them in Yasara.\n\nUse the `Analyze|Align|Objects` with MUSTANG  function to align the four last objects with the first one.\n\nUse the space bar to open the text console and  see the reported root mean square deviations\n\n\nas well as the number of residues matched.\n\nColor all structures by B factor and compare the  distribution to the local variability of the structures.\n\n---\n\n### PDB File Surprises ###\n\n- Missing atoms, residues and loops  \n- Multiple molecules in the asymmetric unit\n- Incomplete oligomers due to coincident crystal and  oligomer symmetry\n- Cα-only models  lternate conformations\n- Multiple models, especially for NMR ensembles\n- Use of B factors to represent other properties  ther non-standard extensions (PDBQT, ...)\n\n---\n\n### Force Fields ###\n\n- Energy terms representing physical interactions\n\nCovalent bond lengths and angles\nDihedral angles and van der Waals forces (steric effects)\nElectrostatic interactions and hydrogen bonds\n…\n\n- Energy can be minimized, and forces can literally be derived from the potential function.\n\n- Consistency and careful consideration of the  properties to be simulated are essential.\n\n---\n\n### Force Field Terms ###\n\nEach energy term has a functional form, which includes one or more parameters:\n\n- Covalent bond energy term\n To do: add formula\n- Van der Waals contact energy term\n\nThe parameters are collectively optimized to  reproduce a chosen set of experimentally  observed parameters.\n\nA given force field should be used as a  consistent system, and can only be used to  predict properties that are covered by the  training set.\n\n---\n\n### FoldX ###\n\n\ns designed for quantitative modelling of the  contribution of structural interactions to the stability of  proteins and protein complexes. It also supports  protein/DNA complexes.\n\nThe force field describes the different interactions in a  protein structure or complex in analytical terms. It has  been calibrated using a set of experimentally determined  stabilities.\n\nApplications include the optimisation of structures, the calculation of the stability of complexes, and predicting  the effect of amino-acid or base-pair mutations on these  properties.\n\n---\n\n### The FoldX Plugin for Yasara\t###\n\nIn order to make FoldX more accessible and  integrate its functions into Yasara, dr. Joost van  Durme (SWITCH laboratory) made a Yasara plugin  module that can apply FoldX functions to structures  that are loaded as Yasara objects.\nThis greatly simplifies the use of FoldX, and allows  for a quick visual analysis of the resulting changes  in the structures.\nMore information can be found at [wiki](http://foldxyasara.switchlab.org/index.php/) [FoldX](http://foldxsuite.crg.eu/)\n\n---\n\n### Exercise 4a: Repair a PDB File ###\n\n- Load the 1CRN PDB file.\n- Use the “Repair object” option in the Analysis |  FoldX menu to activate the corresponding FoldX  function.\n- Select the object to repair.\n\nThis exports the object as a temporary PDB file,  starts FoldX with the appropriate options, and loads  the repaired PDB file as a new object in Yasara.\n\n- Compare the original and repaired objects.  \n- Describe the changes that were introduced.\n\n---\n\n### Exercise 4b: Model a Mutation ###\n\n- Load the 2AC0.sce Yasara scene file.\n- Set an appropriate structure representation.\n- Locate residue Ala159 using the sequence view,  and right-click to access the `FoldX|Mutate` residue function. Change it to a Trp residue.\n- Look at the effect of the substitution on the  structure, and use the space bar to open the text  console and read the output of the FoldX  calculation.\n- Mutate Arg273 to an alanine side chain. Discuss  the effects of this substitution.\n\n---\n\n### Homology Modelling ###\n\n- When a structure is available for a protein with a  similar sequences, it is possible to predict the  structure of a new sequence with varying degrees of  confidence.\n- All homology modelling procedures start from an  alignment of the template and target sequences.  The quality of this alignment will have a major  impact on the resulting model.\n- Available applications include stand-alone programs  (Modeller, FoldX, …) and web-based services (such as SwissModel).\n\n---\n\n### Exercise 5: Make a Homology  Model using Swiss Model ###\n\nsee []()\n\n---\n\n### Exercise 6: Study Protein-Ligand Interactions ###\n\nsee []()\n\n---\n\n### On-Line References ###\n\n- Crystallography 101 (Bernhard Rupp):  (http://www.ruppweb.org/Xray/101index.html)\n- Protein NMR, A Practical Guide (Vicky Higman) ( http://www.protein-nmr.org.uk/)\n- Model validation course  (http://xray.bmc.uu.se/gerard/embo2001/modval/index.html)\n- Assessing model quality (http://spdbv.vital-it.ch/TheMolecularLevel/ModQual/)\n\n- Lectures by Burkhard Rost on protein structure  prediction (https://www.youtube.com/channel/UCU6j8BG4RbEtTgyIZJ6Vpow)\n\n---\n&quot;,&quot;dir&quot;:&quot;/topics/protein-structure-analysis/slides/&quot;,&quot;name&quot;:&quot;introduction.html&quot;,&quot;path&quot;:&quot;topics/protein-structure-analysis/slides/introduction.html&quot;,&quot;url&quot;:&quot;/topics/protein-structure-analysis/slides/introduction.html&quot;}
        </pre>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-header"><span>
<a href="/training-material/topics/protein-structure-analysis" title="Return to topic page" ><i class="fa fa-level-up" aria-hidden="true"></i></a>

<a class="nav-link" href="https://github.com/vibbits/training-material/edit/master/topics/protein-structure-analysis/slides/introduction.html"><i class="fa fa-pencil" aria-hidden="true"></i></a>

</span></div>

<div class="my-footer"><span>

<img src="/training-material/assets/images/bioinformatics_core_rgb_neg.png" alt="VIB Bioinformatics Core" style="height: 40px;"/>

</span></div>

---

# Protein Structure Analysis



---

## Protein Structure Analysis ##

- Sequences, structures and databases
- Experimental methods (X-rays, electrons and NMR)
- Finding and visualising structures from the  Protein Data Bank
- Comparing structures
- Modelling mutations
- Creating homology models

---

## The Structure-Function Connection ##

.left[![Left-aligned image](/training-material/topics/protein-structure-analysis/images/hemoglobin.png)]
.right[Folded proteins provide a well-  defined 3D  arrangement of functional groups, creating microenvironments and active sites.
Structural changes  are often involved in functional  mechanisms  (motor proteins, ...)
]

---

### Databases

.pull-left[
**Uniprot** approx. 1100,000 sequences

- mainly determined by large-scale DNA sequencing of individual genes or whole genomes
- increasingly automated annotation

**Protein Data Bank** approx. 141,000 
experimentally determined structures

- Mainly determined by X-ray crystallography and high-resolution NMR spectroscopy
- Automation is increasing, but there are still significant limitations in the rate of solving new structures


]
.pull-right[ ![](/training-material/assets/images/GTN-60px.png) ]

---

### X-Ray Crystallography ###


---

### X-Ray Crystallography ###


---

### The Protein Databank ###

.pull-left[ ![](/training-material/assets/images/GTN-60px.png) ] 


.pull-right[
[http://www.wwpdb.org](http://www.wwpdb.org)
- contains structures of  proteins, nucleic acids  and complexes,  determined by X-ray  crystallography, NMR  spectroscopy
- No purely theoretical  or ab initio models  (since 2006)
- Also stores supporting  experimental data
- Full deposition now  required by all peer-reviewed journals

---

### Exercise 1: Search the PDB ###

- Use the UniProt site to search for “dnak”.  
- Use the PDB site to search for “dnak”.
- Compare the UniProt and PDB result lists.
- Use the sequence search function to see if there are structures with sequences similar to that of the  DnaK C-terminal domain.
- Look at the summary pages of a number of  structures and note some interesting properties.
- Select a number of structures and create a report  page.

---

### PDB File Organisation ###

![]()

---

### Related Web sites ###

- Nucleic Acid Database: DNA and RNA structures

[http://ndbserver.rutgers.edu/](http://ndbserver.rutgers.edu/)

- PDB-REDO: automatically re-refined deposited  structures, using the latest methods

[http://www.cmbi.ru.nl/pdb_redo/](http://www.cmbi.ru.nl/pdb_redo)

- EBI: many online tools for structure analysis

[http://www.ebi.ac.uk/Tools/structure/](http://www.ebi.ac.uk/Tools/structure/)

- Replaced Electron Density Server: convenient overview of  quality parameters for crystal structures

[http://www.ebi.ac.uk/pdbe/litemol](http://www.ebi.ac.uk/pdbe/litemol)

---

### High-Resolution NMR Spectrometry ###

Many atomic nuclei, including the ubiquitous hydrogen nuclei,  resonate at specific radio frequencies when placed in a  strong, uniform magnetic field. The chemical environment of each individual atom slightly modulates its exact resonance  frequency.

![]()

In macromolecules with thousands of atoms, many different  effects combine to generate an extremely complicated pattern of chemical shifts, which therefore more or less uniquely  identify each atom. Multidimensional spectra allow these  frequencies to be assigned to specific atoms.

---

### High-Resolution NMR Spectroscopy ###

When two atoms are near each other in 3D space, they can exchange magnetization, giving rise to crosspeaks at the  intersection of their respective frequencies.

![]()

This nuclear Overhauser effect (NOE) is used to effectively measure the distances between pairs of atoms, at least  qualitatively.

---

### High-Resolution NMR Spectroscopy ###

After identification of the atoms by means of their unique chemical shifts, distance restraints are derived from the  Overhauser crosspeaks. An extended model of the protein  is  generated,  and  then  condensed  into  a  shape  that  is consistent with as many of distance restraints as possible.

![]()

---

### Other methods ###

Electron microscopy (and especially Cryo-electron microscopy): Electron crystallography and single particle reconstruction

Small-angle X-ray and neutron scattering (SAXS and SANS)

![]()

---

### Related Web sites ###

- BioMagResBank: experimental data for NMR-  derived structures (lists of distance restraints  and other experimentally derived properties)

[http://www.bmrb.wisc.edu/](http://www.bmrb.wisc.edu/)

- BioIsis: database of SAXS-derived structures

[http://www.bioisis.net](http://www.bioisis.net)

- EMBL database of SAXS-derived structures

[http://www.sasbdb.org](http://www.sasbdb.org)

- EM Databank for cryo-EM structures

[http://www.emdatabank.org](http://www.emdatabank.org)

---

### Assessing Structure Quality ###

General geometric properties (bond lengths and angles,  Ramachandran distribution, …): MolProbity  [Link](http://molprobity.biochem.duke.edu/)

Crystal Structures
- Diffraction data resolution and completeness (PDB)
- Final Rcryst and Rfree factors (PDB)
- Local correspondence to electron density (EDS)

NMR Structures
- Number of distance restraints and additional experimental  data sources (BMRB)
- Local restraint density (on-line NMR constraint analyser)
[link](http://molsim.sci.univr.it/bioinfo/tools/constraint/index.html)

Other techniques
- Difficult to generalise: carefully read and consider the  published protocol

---

### Molecular Graphics Software ###

- [PyMOL](http://www.pymol.org/): high-quality output,  good examples on [Wiki](http://www.pymolwiki.org/)
- [Chimera](http://www.cgl.ucsf.edu/chimera/): good  documentation on website
- [VMD](http://www.ks.uiuc.edu/Research/vmd/):  excellent for the analysis of MD trajectories
- [Yasara](http://www.yasara.org)  
- [SwissPDBViewer](http://spdbv.vital-it.ch/)

---

### YASARA ###

Yasara View is freely available and provides  basic visualisation functions.

Yasara Model, Dynamics and Structure provide  additional functionality.

An add-on module for NMR structures is  available.

The program can be combined with the WHAT-  IF program for structure validation, and with  FoldX for energy calculations.

---

### Exercise 2: show a structure ###

Load PDB entry 1TRZ using the File menu.

Compare the default representations (F1-F8)  and use the various controls to change the view  of the protein.

Explore the Edit and View menus to change  various aspects of the graphical representation  of the structure.

Examine the hydrogen bonding patterns.  Display a molecular surface.

Create an interesting view of the molecule and  save it as a high-resolution image.

---

### Structure Superposition ###

Structures can be superimposed to achieve the best  possible match between corresponding atoms. It is  possible to consider all atoms, or only a subset  (such as the Cα atoms).

When the structures are very similar, determining  which atoms should be matched to each other is  trivial. When there are larger differences, it takes  more preparation.

Different algorithms use different combinations of  sequence- and secondary of tertiary structure-based  information.

---

### Exercise 3: Compare Structures ###

Download the five provided PDB files and open  them in Yasara.

Use the `Analyze|Align|Objects` with MUSTANG  function to align the four last objects with the first one.

Use the space bar to open the text console and  see the reported root mean square deviations


as well as the number of residues matched.

Color all structures by B factor and compare the  distribution to the local variability of the structures.

---

### PDB File Surprises ###

- Missing atoms, residues and loops  
- Multiple molecules in the asymmetric unit
- Incomplete oligomers due to coincident crystal and  oligomer symmetry
- Cα-only models  lternate conformations
- Multiple models, especially for NMR ensembles
- Use of B factors to represent other properties  ther non-standard extensions (PDBQT, ...)

---

### Force Fields ###

- Energy terms representing physical interactions

Covalent bond lengths and angles
Dihedral angles and van der Waals forces (steric effects)
Electrostatic interactions and hydrogen bonds
…

- Energy can be minimized, and forces can literally be derived from the potential function.

- Consistency and careful consideration of the  properties to be simulated are essential.

---

### Force Field Terms ###

Each energy term has a functional form, which includes one or more parameters:

- Covalent bond energy term
 To do: add formula
- Van der Waals contact energy term

The parameters are collectively optimized to  reproduce a chosen set of experimentally  observed parameters.

A given force field should be used as a  consistent system, and can only be used to  predict properties that are covered by the  training set.

---

### FoldX ###


s designed for quantitative modelling of the  contribution of structural interactions to the stability of  proteins and protein complexes. It also supports  protein/DNA complexes.

The force field describes the different interactions in a  protein structure or complex in analytical terms. It has  been calibrated using a set of experimentally determined  stabilities.

Applications include the optimisation of structures, the calculation of the stability of complexes, and predicting  the effect of amino-acid or base-pair mutations on these  properties.

---

### The FoldX Plugin for Yasara	###

In order to make FoldX more accessible and  integrate its functions into Yasara, dr. Joost van  Durme (SWITCH laboratory) made a Yasara plugin  module that can apply FoldX functions to structures  that are loaded as Yasara objects.
This greatly simplifies the use of FoldX, and allows  for a quick visual analysis of the resulting changes  in the structures.
More information can be found at [wiki](http://foldxyasara.switchlab.org/index.php/) [FoldX](http://foldxsuite.crg.eu/)

---

### Exercise 4a: Repair a PDB File ###

- Load the 1CRN PDB file.
- Use the “Repair object” option in the Analysis |  FoldX menu to activate the corresponding FoldX  function.
- Select the object to repair.

This exports the object as a temporary PDB file,  starts FoldX with the appropriate options, and loads  the repaired PDB file as a new object in Yasara.

- Compare the original and repaired objects.  
- Describe the changes that were introduced.

---

### Exercise 4b: Model a Mutation ###

- Load the 2AC0.sce Yasara scene file.
- Set an appropriate structure representation.
- Locate residue Ala159 using the sequence view,  and right-click to access the `FoldX|Mutate` residue function. Change it to a Trp residue.
- Look at the effect of the substitution on the  structure, and use the space bar to open the text  console and read the output of the FoldX  calculation.
- Mutate Arg273 to an alanine side chain. Discuss  the effects of this substitution.

---

### Homology Modelling ###

- When a structure is available for a protein with a  similar sequences, it is possible to predict the  structure of a new sequence with varying degrees of  confidence.
- All homology modelling procedures start from an  alignment of the template and target sequences.  The quality of this alignment will have a major  impact on the resulting model.
- Available applications include stand-alone programs  (Modeller, FoldX, …) and web-based services (such as SwissModel).

---

### Exercise 5: Make a Homology  Model using Swiss Model ###

see []()

---

### Exercise 6: Study Protein-Ligand Interactions ###

see []()

---

### On-Line References ###

- Crystallography 101 (Bernhard Rupp):  (http://www.ruppweb.org/Xray/101index.html)
- Protein NMR, A Practical Guide (Vicky Higman) ( http://www.protein-nmr.org.uk/)
- Model validation course  (http://xray.bmc.uu.se/gerard/embo2001/modval/index.html)
- Assessing model quality (http://spdbv.vital-it.ch/TheMolecularLevel/ModQual/)

- Lectures by Burkhard Rost on protein structure  prediction (https://www.youtube.com/channel/UCU6j8BG4RbEtTgyIZJ6Vpow)

---


---




## Related tutorials


    



---

## Thank you!

This material is the result of a collaborative work. Thanks the colleagues of the VIB Bioinformatics Core and all the contributors!


<div class="contributors-line">


<a href="/training-material/hall-of-fame#abotzki" class="contributor-badge"><img src="https://avatars.githubusercontent.com/abotzki" alt="Alexander Botzki">Alexander Botzki</a>, <a href="/training-material/hall-of-fame#jvdurme" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jvdurme" alt="Joost Van Durme">Joost Van Durme</a>


</div>



<img src="/training-material/assets/images/vib_notag_neg_rgb.jpg" alt="Galaxy Training Network" style="height: 100px;"/>



    </textarea>
    <script src="/training-material/assets/js/remark-latest.min.js" type="text/javascript">
    </script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false,}});
      var hljs = remark.highlighter.engine;
    </script>
  </body>
</html>
